// rdf-namespaces/dist/arg-baaa5c75.js
'use strict';

/**
 *
 *
 *
 *
 * http://www.w3.org/ns/pim/arg#Conjunction
 */
var Conjunction = 'http://www.w3.org/ns/pim/arg#Conjunction';
/**
 * position
 *
 *
 *
 * http://www.w3.org/ns/pim/arg#Position
 */
var Position = 'http://www.w3.org/ns/pim/arg#Position';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/pim/arg#Resolution
 */
var Resolution = 'http://www.w3.org/ns/pim/arg#Resolution';
/**
 * opposition
 *
 *
 *
 * http://www.w3.org/ns/pim/arg#opposition
 */
var opposition = 'http://www.w3.org/ns/pim/arg#opposition';
/**
 * requires
 *
 *
 *
 * http://www.w3.org/ns/pim/arg#requires
 */
var requires = 'http://www.w3.org/ns/pim/arg#requires';
/**
 * support
 *
 *
 *
 * http://www.w3.org/ns/pim/arg#support
 */
var support = 'http://www.w3.org/ns/pim/arg#support';

var argImport = /*#__PURE__*/Object.freeze({
	Conjunction: Conjunction,
	Position: Position,
	Resolution: Resolution,
	opposition: opposition,
	requires: requires,
	support: support
});

exports.Conjunction = Conjunction;
exports.Position = Position;
exports.Resolution = Resolution;
exports.argImport = argImport;
exports.opposition = opposition;
exports.requires = requires;
exports.support = support;
