// rdf-namespaces/dist/as-b1b4cee5.js
'use strict';

/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#actor
 */
var actor = 'http://www.w3.org/ns/activitystreams#actor';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#attributedTo
 */
var attributedTo = 'http://www.w3.org/ns/activitystreams#attributedTo';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#attachment
 */
var attachment = 'http://www.w3.org/ns/activitystreams#attachment';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#attachments
 */
var attachments = 'http://www.w3.org/ns/activitystreams#attachments';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#author
 */
var author = 'http://www.w3.org/ns/activitystreams#author';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#bcc
 */
var bcc = 'http://www.w3.org/ns/activitystreams#bcc';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#bto
 */
var bto = 'http://www.w3.org/ns/activitystreams#bto';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#cc
 */
var cc = 'http://www.w3.org/ns/activitystreams#cc';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#context
 */
var context = 'http://www.w3.org/ns/activitystreams#context';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#generator
 */
var generator = 'http://www.w3.org/ns/activitystreams#generator';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#icon
 */
var icon = 'http://www.w3.org/ns/activitystreams#icon';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#image
 */
var image = 'http://www.w3.org/ns/activitystreams#image';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#inReplyTo
 */
var inReplyTo = 'http://www.w3.org/ns/activitystreams#inReplyTo';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#items
 */
var items = 'http://www.w3.org/ns/activitystreams#items';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#location
 */
var location = 'http://www.w3.org/ns/activitystreams#location';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#object
 */
var object = 'http://www.w3.org/ns/activitystreams#object';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#oneOf
 */
var oneOf = 'http://www.w3.org/ns/activitystreams#oneOf';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#anyOf
 */
var anyOf = 'http://www.w3.org/ns/activitystreams#anyOf';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#preview
 */
var preview = 'http://www.w3.org/ns/activitystreams#preview';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#provider
 */
var provider = 'http://www.w3.org/ns/activitystreams#provider';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#replies
 */
var replies = 'http://www.w3.org/ns/activitystreams#replies';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#result
 */
var result = 'http://www.w3.org/ns/activitystreams#result';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#audience
 */
var audience = 'http://www.w3.org/ns/activitystreams#audience';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#tag
 */
var tag = 'http://www.w3.org/ns/activitystreams#tag';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#tags
 */
var tags = 'http://www.w3.org/ns/activitystreams#tags';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#target
 */
var target = 'http://www.w3.org/ns/activitystreams#target';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#origin
 */
var origin = 'http://www.w3.org/ns/activitystreams#origin';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#instrument
 */
var instrument = 'http://www.w3.org/ns/activitystreams#instrument';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#to
 */
var to = 'http://www.w3.org/ns/activitystreams#to';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#url
 */
var url = 'http://www.w3.org/ns/activitystreams#url';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#relationship
 */
var relationship = 'http://www.w3.org/ns/activitystreams#relationship';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#describes
 */
var describes = 'http://www.w3.org/ns/activitystreams#describes';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#formerType
 */
var formerType = 'http://www.w3.org/ns/activitystreams#formerType';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#accuracy
 */
var accuracy = 'http://www.w3.org/ns/activitystreams#accuracy';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#altitude
 */
var altitude = 'http://www.w3.org/ns/activitystreams#altitude';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#content
 */
var content = 'http://www.w3.org/ns/activitystreams#content';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#name
 */
var name = 'http://www.w3.org/ns/activitystreams#name';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#downstreamDuplicates
 */
var downstreamDuplicates = 'http://www.w3.org/ns/activitystreams#downstreamDuplicates';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#duration
 */
var duration = 'http://www.w3.org/ns/activitystreams#duration';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#endTime
 */
var endTime = 'http://www.w3.org/ns/activitystreams#endTime';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#height
 */
var height = 'http://www.w3.org/ns/activitystreams#height';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#href
 */
var href = 'http://www.w3.org/ns/activitystreams#href';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#hreflang
 */
var hreflang = 'http://www.w3.org/ns/activitystreams#hreflang';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#id
 */
var id = 'http://www.w3.org/ns/activitystreams#id';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#latitude
 */
var latitude = 'http://www.w3.org/ns/activitystreams#latitude';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#longitude
 */
var longitude = 'http://www.w3.org/ns/activitystreams#longitude';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#mediaType
 */
var mediaType = 'http://www.w3.org/ns/activitystreams#mediaType';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#objectType
 */
var objectType = 'http://www.w3.org/ns/activitystreams#objectType';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#published
 */
var published = 'http://www.w3.org/ns/activitystreams#published';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#radius
 */
var radius = 'http://www.w3.org/ns/activitystreams#radius';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#rating
 */
var rating = 'http://www.w3.org/ns/activitystreams#rating';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#rel
 */
var rel = 'http://www.w3.org/ns/activitystreams#rel';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#startIndex
 */
var startIndex = 'http://www.w3.org/ns/activitystreams#startIndex';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#startTime
 */
var startTime = 'http://www.w3.org/ns/activitystreams#startTime';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#summary
 */
var summary = 'http://www.w3.org/ns/activitystreams#summary';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#totalItems
 */
var totalItems = 'http://www.w3.org/ns/activitystreams#totalItems';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#units
 */
var units = 'http://www.w3.org/ns/activitystreams#units';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#updated
 */
var updated = 'http://www.w3.org/ns/activitystreams#updated';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#upstreamDuplicates
 */
var upstreamDuplicates = 'http://www.w3.org/ns/activitystreams#upstreamDuplicates';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#verb
 */
var verb = 'http://www.w3.org/ns/activitystreams#verb';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#width
 */
var width = 'http://www.w3.org/ns/activitystreams#width';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#deleted
 */
var deleted = 'http://www.w3.org/ns/activitystreams#deleted';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Accept
 */
var Accept = 'http://www.w3.org/ns/activitystreams#Accept';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Activity
 */
var Activity = 'http://www.w3.org/ns/activitystreams#Activity';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Block
 */
var Block = 'http://www.w3.org/ns/activitystreams#Block';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#IntransitiveActivity
 */
var IntransitiveActivity = 'http://www.w3.org/ns/activitystreams#IntransitiveActivity';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Add
 */
var Add = 'http://www.w3.org/ns/activitystreams#Add';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Announce
 */
var Announce = 'http://www.w3.org/ns/activitystreams#Announce';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Application
 */
var Application = 'http://www.w3.org/ns/activitystreams#Application';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Arrive
 */
var Arrive = 'http://www.w3.org/ns/activitystreams#Arrive';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Article
 */
var Article = 'http://www.w3.org/ns/activitystreams#Article';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Audio
 */
var Audio = 'http://www.w3.org/ns/activitystreams#Audio';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Collection
 */
var Collection = 'http://www.w3.org/ns/activitystreams#Collection';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#CollectionPage
 */
var CollectionPage = 'http://www.w3.org/ns/activitystreams#CollectionPage';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#OrderedCollectionPage
 */
var OrderedCollectionPage = 'http://www.w3.org/ns/activitystreams#OrderedCollectionPage';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Relationship
 */
var Relationship = 'http://www.w3.org/ns/activitystreams#Relationship';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Create
 */
var Create = 'http://www.w3.org/ns/activitystreams#Create';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Delete
 */
var Delete = 'http://www.w3.org/ns/activitystreams#Delete';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Dislike
 */
var Dislike = 'http://www.w3.org/ns/activitystreams#Dislike';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Document
 */
var Document = 'http://www.w3.org/ns/activitystreams#Document';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Event
 */
var Event = 'http://www.w3.org/ns/activitystreams#Event';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Flag
 */
var Flag = 'http://www.w3.org/ns/activitystreams#Flag';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Follow
 */
var Follow = 'http://www.w3.org/ns/activitystreams#Follow';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Group
 */
var Group = 'http://www.w3.org/ns/activitystreams#Group';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Ignore
 */
var Ignore = 'http://www.w3.org/ns/activitystreams#Ignore';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Image
 */
var Image = 'http://www.w3.org/ns/activitystreams#Image';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Invite
 */
var Invite = 'http://www.w3.org/ns/activitystreams#Invite';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Join
 */
var Join = 'http://www.w3.org/ns/activitystreams#Join';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Leave
 */
var Leave = 'http://www.w3.org/ns/activitystreams#Leave';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Like
 */
var Like = 'http://www.w3.org/ns/activitystreams#Like';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#View
 */
var View = 'http://www.w3.org/ns/activitystreams#View';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Listen
 */
var Listen = 'http://www.w3.org/ns/activitystreams#Listen';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Read
 */
var Read = 'http://www.w3.org/ns/activitystreams#Read';
/**
 *
 *
 * The actor is moving the object. The target specifies where the object is moving to. The origin specifies where the object is moving from.
 *
 * http://www.w3.org/ns/activitystreams#Move
 */
var Move = 'http://www.w3.org/ns/activitystreams#Move';
/**
 *
 *
 * The actor is traveling to the target. The origin specifies where the actor is traveling from.
 *
 * http://www.w3.org/ns/activitystreams#Travel
 */
var Travel = 'http://www.w3.org/ns/activitystreams#Travel';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Link
 */
var Link = 'http://www.w3.org/ns/activitystreams#Link';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Mention
 */
var Mention = 'http://www.w3.org/ns/activitystreams#Mention';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Note
 */
var Note = 'http://www.w3.org/ns/activitystreams#Note';
/**
 *
 *
 * null
 * Note that `Object` is a reserved Javascript keyword, and is therefore suffixed by `__workaround`.
 * For a list of reserved keywords, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Keywords.
 *
 * http://www.w3.org/ns/activitystreams#Object
 */
var Object__workaround = 'http://www.w3.org/ns/activitystreams#Object';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Offer
 */
var Offer = 'http://www.w3.org/ns/activitystreams#Offer';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#OrderedCollection
 */
var OrderedCollection = 'http://www.w3.org/ns/activitystreams#OrderedCollection';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#OrderedItems
 */
var OrderedItems = 'http://www.w3.org/ns/activitystreams#OrderedItems';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Page
 */
var Page = 'http://www.w3.org/ns/activitystreams#Page';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Person
 */
var Person = 'http://www.w3.org/ns/activitystreams#Person';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Organization
 */
var Organization = 'http://www.w3.org/ns/activitystreams#Organization';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Profile
 */
var Profile = 'http://www.w3.org/ns/activitystreams#Profile';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Place
 */
var Place = 'http://www.w3.org/ns/activitystreams#Place';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Question
 */
var Question = 'http://www.w3.org/ns/activitystreams#Question';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Reject
 */
var Reject = 'http://www.w3.org/ns/activitystreams#Reject';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Remove
 */
var Remove = 'http://www.w3.org/ns/activitystreams#Remove';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Service
 */
var Service = 'http://www.w3.org/ns/activitystreams#Service';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#TentativeAccept
 */
var TentativeAccept = 'http://www.w3.org/ns/activitystreams#TentativeAccept';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#TentativeReject
 */
var TentativeReject = 'http://www.w3.org/ns/activitystreams#TentativeReject';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Tombstone
 */
var Tombstone = 'http://www.w3.org/ns/activitystreams#Tombstone';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Undo
 */
var Undo = 'http://www.w3.org/ns/activitystreams#Undo';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Update
 */
var Update = 'http://www.w3.org/ns/activitystreams#Update';
/**
 *
 *
 *
 *
 * http://www.w3.org/ns/activitystreams#Video
 */
var Video = 'http://www.w3.org/ns/activitystreams#Video';

var asImport = /*#__PURE__*/Object.freeze({
	actor: actor,
	attributedTo: attributedTo,
	attachment: attachment,
	attachments: attachments,
	author: author,
	bcc: bcc,
	bto: bto,
	cc: cc,
	context: context,
	generator: generator,
	icon: icon,
	image: image,
	inReplyTo: inReplyTo,
	items: items,
	location: location,
	object: object,
	oneOf: oneOf,
	anyOf: anyOf,
	preview: preview,
	provider: provider,
	replies: replies,
	result: result,
	audience: audience,
	tag: tag,
	tags: tags,
	target: target,
	origin: origin,
	instrument: instrument,
	to: to,
	url: url,
	relationship: relationship,
	describes: describes,
	formerType: formerType,
	accuracy: accuracy,
	altitude: altitude,
	content: content,
	name: name,
	downstreamDuplicates: downstreamDuplicates,
	duration: duration,
	endTime: endTime,
	height: height,
	href: href,
	hreflang: hreflang,
	id: id,
	latitude: latitude,
	longitude: longitude,
	mediaType: mediaType,
	objectType: objectType,
	published: published,
	radius: radius,
	rating: rating,
	rel: rel,
	startIndex: startIndex,
	startTime: startTime,
	summary: summary,
	totalItems: totalItems,
	units: units,
	updated: updated,
	upstreamDuplicates: upstreamDuplicates,
	verb: verb,
	width: width,
	deleted: deleted,
	Accept: Accept,
	Activity: Activity,
	Block: Block,
	IntransitiveActivity: IntransitiveActivity,
	Add: Add,
	Announce: Announce,
	Application: Application,
	Arrive: Arrive,
	Article: Article,
	Audio: Audio,
	Collection: Collection,
	CollectionPage: CollectionPage,
	OrderedCollectionPage: OrderedCollectionPage,
	Relationship: Relationship,
	Create: Create,
	Delete: Delete,
	Dislike: Dislike,
	Document: Document,
	Event: Event,
	Flag: Flag,
	Follow: Follow,
	Group: Group,
	Ignore: Ignore,
	Image: Image,
	Invite: Invite,
	Join: Join,
	Leave: Leave,
	Like: Like,
	View: View,
	Listen: Listen,
	Read: Read,
	Move: Move,
	Travel: Travel,
	Link: Link,
	Mention: Mention,
	Note: Note,
	Object__workaround: Object__workaround,
	Offer: Offer,
	OrderedCollection: OrderedCollection,
	OrderedItems: OrderedItems,
	Page: Page,
	Person: Person,
	Organization: Organization,
	Profile: Profile,
	Place: Place,
	Question: Question,
	Reject: Reject,
	Remove: Remove,
	Service: Service,
	TentativeAccept: TentativeAccept,
	TentativeReject: TentativeReject,
	Tombstone: Tombstone,
	Undo: Undo,
	Update: Update,
	Video: Video
});

exports.Accept = Accept;
exports.Activity = Activity;
exports.Add = Add;
exports.Announce = Announce;
exports.Application = Application;
exports.Arrive = Arrive;
exports.Article = Article;
exports.Audio = Audio;
exports.Block = Block;
exports.Collection = Collection;
exports.CollectionPage = CollectionPage;
exports.Create = Create;
exports.Delete = Delete;
exports.Dislike = Dislike;
exports.Document = Document;
exports.Event = Event;
exports.Flag = Flag;
exports.Follow = Follow;
exports.Group = Group;
exports.Ignore = Ignore;
exports.Image = Image;
exports.IntransitiveActivity = IntransitiveActivity;
exports.Invite = Invite;
exports.Join = Join;
exports.Leave = Leave;
exports.Like = Like;
exports.Link = Link;
exports.Listen = Listen;
exports.Mention = Mention;
exports.Move = Move;
exports.Note = Note;
exports.Object__workaround = Object__workaround;
exports.Offer = Offer;
exports.OrderedCollection = OrderedCollection;
exports.OrderedCollectionPage = OrderedCollectionPage;
exports.OrderedItems = OrderedItems;
exports.Organization = Organization;
exports.Page = Page;
exports.Person = Person;
exports.Place = Place;
exports.Profile = Profile;
exports.Question = Question;
exports.Read = Read;
exports.Reject = Reject;
exports.Relationship = Relationship;
exports.Remove = Remove;
exports.Service = Service;
exports.TentativeAccept = TentativeAccept;
exports.TentativeReject = TentativeReject;
exports.Tombstone = Tombstone;
exports.Travel = Travel;
exports.Undo = Undo;
exports.Update = Update;
exports.Video = Video;
exports.View = View;
exports.accuracy = accuracy;
exports.actor = actor;
exports.altitude = altitude;
exports.anyOf = anyOf;
exports.asImport = asImport;
exports.attachment = attachment;
exports.attachments = attachments;
exports.attributedTo = attributedTo;
exports.audience = audience;
exports.author = author;
exports.bcc = bcc;
exports.bto = bto;
exports.cc = cc;
exports.content = content;
exports.context = context;
exports.deleted = deleted;
exports.describes = describes;
exports.downstreamDuplicates = downstreamDuplicates;
exports.duration = duration;
exports.endTime = endTime;
exports.formerType = formerType;
exports.generator = generator;
exports.height = height;
exports.href = href;
exports.hreflang = hreflang;
exports.icon = icon;
exports.id = id;
exports.image = image;
exports.inReplyTo = inReplyTo;
exports.instrument = instrument;
exports.items = items;
exports.latitude = latitude;
exports.location = location;
exports.longitude = longitude;
exports.mediaType = mediaType;
exports.name = name;
exports.object = object;
exports.objectType = objectType;
exports.oneOf = oneOf;
exports.origin = origin;
exports.preview = preview;
exports.provider = provider;
exports.published = published;
exports.radius = radius;
exports.rating = rating;
exports.rel = rel;
exports.relationship = relationship;
exports.replies = replies;
exports.result = result;
exports.startIndex = startIndex;
exports.startTime = startTime;
exports.summary = summary;
exports.tag = tag;
exports.tags = tags;
exports.target = target;
exports.to = to;
exports.totalItems = totalItems;
exports.units = units;
exports.updated = updated;
exports.upstreamDuplicates = upstreamDuplicates;
exports.url = url;
exports.verb = verb;
exports.width = width;
