// rdf-namespaces/dist/as.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var as$1 = require('./as-b1b4cee5.js');



exports.Accept = as$1.Accept;
exports.Activity = as$1.Activity;
exports.Add = as$1.Add;
exports.Announce = as$1.Announce;
exports.Application = as$1.Application;
exports.Arrive = as$1.Arrive;
exports.Article = as$1.Article;
exports.Audio = as$1.Audio;
exports.Block = as$1.Block;
exports.Collection = as$1.Collection;
exports.CollectionPage = as$1.CollectionPage;
exports.Create = as$1.Create;
exports.Delete = as$1.Delete;
exports.Dislike = as$1.Dislike;
exports.Document = as$1.Document;
exports.Event = as$1.Event;
exports.Flag = as$1.Flag;
exports.Follow = as$1.Follow;
exports.Group = as$1.Group;
exports.Ignore = as$1.Ignore;
exports.Image = as$1.Image;
exports.IntransitiveActivity = as$1.IntransitiveActivity;
exports.Invite = as$1.Invite;
exports.Join = as$1.Join;
exports.Leave = as$1.Leave;
exports.Like = as$1.Like;
exports.Link = as$1.Link;
exports.Listen = as$1.Listen;
exports.Mention = as$1.Mention;
exports.Move = as$1.Move;
exports.Note = as$1.Note;
exports.Object__workaround = as$1.Object__workaround;
exports.Offer = as$1.Offer;
exports.OrderedCollection = as$1.OrderedCollection;
exports.OrderedCollectionPage = as$1.OrderedCollectionPage;
exports.OrderedItems = as$1.OrderedItems;
exports.Organization = as$1.Organization;
exports.Page = as$1.Page;
exports.Person = as$1.Person;
exports.Place = as$1.Place;
exports.Profile = as$1.Profile;
exports.Question = as$1.Question;
exports.Read = as$1.Read;
exports.Reject = as$1.Reject;
exports.Relationship = as$1.Relationship;
exports.Remove = as$1.Remove;
exports.Service = as$1.Service;
exports.TentativeAccept = as$1.TentativeAccept;
exports.TentativeReject = as$1.TentativeReject;
exports.Tombstone = as$1.Tombstone;
exports.Travel = as$1.Travel;
exports.Undo = as$1.Undo;
exports.Update = as$1.Update;
exports.Video = as$1.Video;
exports.View = as$1.View;
exports.accuracy = as$1.accuracy;
exports.actor = as$1.actor;
exports.altitude = as$1.altitude;
exports.anyOf = as$1.anyOf;
exports.attachment = as$1.attachment;
exports.attachments = as$1.attachments;
exports.attributedTo = as$1.attributedTo;
exports.audience = as$1.audience;
exports.author = as$1.author;
exports.bcc = as$1.bcc;
exports.bto = as$1.bto;
exports.cc = as$1.cc;
exports.content = as$1.content;
exports.context = as$1.context;
exports.deleted = as$1.deleted;
exports.describes = as$1.describes;
exports.downstreamDuplicates = as$1.downstreamDuplicates;
exports.duration = as$1.duration;
exports.endTime = as$1.endTime;
exports.formerType = as$1.formerType;
exports.generator = as$1.generator;
exports.height = as$1.height;
exports.href = as$1.href;
exports.hreflang = as$1.hreflang;
exports.icon = as$1.icon;
exports.id = as$1.id;
exports.image = as$1.image;
exports.inReplyTo = as$1.inReplyTo;
exports.instrument = as$1.instrument;
exports.items = as$1.items;
exports.latitude = as$1.latitude;
exports.location = as$1.location;
exports.longitude = as$1.longitude;
exports.mediaType = as$1.mediaType;
exports.name = as$1.name;
exports.object = as$1.object;
exports.objectType = as$1.objectType;
exports.oneOf = as$1.oneOf;
exports.origin = as$1.origin;
exports.preview = as$1.preview;
exports.provider = as$1.provider;
exports.published = as$1.published;
exports.radius = as$1.radius;
exports.rating = as$1.rating;
exports.rel = as$1.rel;
exports.relationship = as$1.relationship;
exports.replies = as$1.replies;
exports.result = as$1.result;
exports.startIndex = as$1.startIndex;
exports.startTime = as$1.startTime;
exports.summary = as$1.summary;
exports.tag = as$1.tag;
exports.tags = as$1.tags;
exports.target = as$1.target;
exports.to = as$1.to;
exports.totalItems = as$1.totalItems;
exports.units = as$1.units;
exports.updated = as$1.updated;
exports.upstreamDuplicates = as$1.upstreamDuplicates;
exports.url = as$1.url;
exports.verb = as$1.verb;
exports.width = as$1.width;
