// rdf-namespaces/dist/as.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#actor
 */
export const actor: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#actor';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#attributedTo
 */
export const attributedTo: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#attributedTo';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#attachment
 */
export const attachment: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#attachment';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#attachments
 */
export const attachments: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#attachments';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#author
 */
export const author: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#author';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#bcc
 */
export const bcc: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#bcc';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#bto
 */
export const bto: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#bto';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#cc
 */
export const cc: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#cc';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#context
 */
export const context: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#context';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#generator
 */
export const generator: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#generator';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#icon
 */
export const icon: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#icon';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#image
 */
export const image: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#image';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#inReplyTo
 */
export const inReplyTo: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#inReplyTo';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#items
 */
export const items: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#items';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#location
 */
export const location: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#location';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#object
 */
export const object: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#object';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#oneOf
 */
export const oneOf: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#oneOf';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#anyOf
 */
export const anyOf: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#anyOf';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#preview
 */
export const preview: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#preview';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#provider
 */
export const provider: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#provider';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#replies
 */
export const replies: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#replies';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#result
 */
export const result: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#result';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#audience
 */
export const audience: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#audience';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#tag
 */
export const tag: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#tag';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#tags
 */
export const tags: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#tags';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#target
 */
export const target: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#target';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#origin
 */
export const origin: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#origin';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#instrument
 */
export const instrument: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#instrument';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#to
 */
export const to: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#to';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#url
 */
export const url: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#url';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#relationship
 */
export const relationship: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#relationship';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#describes
 */
export const describes: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#describes';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#formerType
 */
export const formerType: OwlObjectProperty = 'http://www.w3.org/ns/activitystreams#formerType';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#accuracy
 */
export const accuracy: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#accuracy';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#altitude
 */
export const altitude: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#altitude';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#content
 */
export const content: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#content';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#name
 */
export const name: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#name';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#downstreamDuplicates
 */
export const downstreamDuplicates: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#downstreamDuplicates';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#duration
 */
export const duration: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#duration';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#endTime
 */
export const endTime: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#endTime';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#height
 */
export const height: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#height';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#href
 */
export const href: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#href';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#hreflang
 */
export const hreflang: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#hreflang';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#id
 */
export const id: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#id';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#latitude
 */
export const latitude: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#latitude';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#longitude
 */
export const longitude: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#longitude';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#mediaType
 */
export const mediaType: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#mediaType';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#objectType
 */
export const objectType: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#objectType';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#published
 */
export const published: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#published';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#radius
 */
export const radius: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#radius';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#rating
 */
export const rating: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#rating';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#rel
 */
export const rel: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#rel';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#startIndex
 */
export const startIndex: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#startIndex';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#startTime
 */
export const startTime: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#startTime';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#summary
 */
export const summary: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#summary';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#totalItems
 */
export const totalItems: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#totalItems';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#units
 */
export const units: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#units';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#updated
 */
export const updated: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#updated';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#upstreamDuplicates
 */
export const upstreamDuplicates: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#upstreamDuplicates';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#verb
 */
export const verb: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#verb';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#width
 */
export const width: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#width';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#deleted
 */
export const deleted: OwlDatatypeProperty = 'http://www.w3.org/ns/activitystreams#deleted';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Accept
 */
export const Accept: OwlClass = 'http://www.w3.org/ns/activitystreams#Accept';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Activity
 */
export const Activity: OwlClass = 'http://www.w3.org/ns/activitystreams#Activity';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Block
 */
export const Block: OwlClass = 'http://www.w3.org/ns/activitystreams#Block';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#IntransitiveActivity
 */
export const IntransitiveActivity: OwlClass = 'http://www.w3.org/ns/activitystreams#IntransitiveActivity';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Add
 */
export const Add: OwlClass = 'http://www.w3.org/ns/activitystreams#Add';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Announce
 */
export const Announce: OwlClass = 'http://www.w3.org/ns/activitystreams#Announce';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Application
 */
export const Application: OwlClass = 'http://www.w3.org/ns/activitystreams#Application';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Arrive
 */
export const Arrive: OwlClass = 'http://www.w3.org/ns/activitystreams#Arrive';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Article
 */
export const Article: OwlClass = 'http://www.w3.org/ns/activitystreams#Article';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Audio
 */
export const Audio: OwlClass = 'http://www.w3.org/ns/activitystreams#Audio';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Collection
 */
export const Collection: OwlClass = 'http://www.w3.org/ns/activitystreams#Collection';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#CollectionPage
 */
export const CollectionPage: OwlClass = 'http://www.w3.org/ns/activitystreams#CollectionPage';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#OrderedCollectionPage
 */
export const OrderedCollectionPage: OwlClass = 'http://www.w3.org/ns/activitystreams#OrderedCollectionPage';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Relationship
 */
export const Relationship: OwlClass = 'http://www.w3.org/ns/activitystreams#Relationship';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Create
 */
export const Create: OwlClass = 'http://www.w3.org/ns/activitystreams#Create';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Delete
 */
export const Delete: OwlClass = 'http://www.w3.org/ns/activitystreams#Delete';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Dislike
 */
export const Dislike: OwlClass = 'http://www.w3.org/ns/activitystreams#Dislike';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Document
 */
export const Document: OwlClass = 'http://www.w3.org/ns/activitystreams#Document';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Event
 */
export const Event: OwlClass = 'http://www.w3.org/ns/activitystreams#Event';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Flag
 */
export const Flag: OwlClass = 'http://www.w3.org/ns/activitystreams#Flag';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Follow
 */
export const Follow: OwlClass = 'http://www.w3.org/ns/activitystreams#Follow';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Group
 */
export const Group: OwlClass = 'http://www.w3.org/ns/activitystreams#Group';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Ignore
 */
export const Ignore: OwlClass = 'http://www.w3.org/ns/activitystreams#Ignore';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Image
 */
export const Image: OwlClass = 'http://www.w3.org/ns/activitystreams#Image';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Invite
 */
export const Invite: OwlClass = 'http://www.w3.org/ns/activitystreams#Invite';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Join
 */
export const Join: OwlClass = 'http://www.w3.org/ns/activitystreams#Join';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Leave
 */
export const Leave: OwlClass = 'http://www.w3.org/ns/activitystreams#Leave';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Like
 */
export const Like: OwlClass = 'http://www.w3.org/ns/activitystreams#Like';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#View
 */
export const View: OwlClass = 'http://www.w3.org/ns/activitystreams#View';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Listen
 */
export const Listen: OwlClass = 'http://www.w3.org/ns/activitystreams#Listen';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Read
 */
export const Read: OwlClass = 'http://www.w3.org/ns/activitystreams#Read';

/**
 * 
 * 
 * The actor is moving the object. The target specifies where the object is moving to. The origin specifies where the object is moving from.
 *
 * http://www.w3.org/ns/activitystreams#Move
 */
export const Move: OwlClass = 'http://www.w3.org/ns/activitystreams#Move';

/**
 * 
 * 
 * The actor is traveling to the target. The origin specifies where the actor is traveling from.
 *
 * http://www.w3.org/ns/activitystreams#Travel
 */
export const Travel: OwlClass = 'http://www.w3.org/ns/activitystreams#Travel';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Link
 */
export const Link: OwlClass = 'http://www.w3.org/ns/activitystreams#Link';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Mention
 */
export const Mention: OwlClass = 'http://www.w3.org/ns/activitystreams#Mention';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Note
 */
export const Note: OwlClass = 'http://www.w3.org/ns/activitystreams#Note';

/**
 * 
 * 
 * null
 * Note that `Object` is a reserved Javascript keyword, and is therefore suffixed by `__workaround`.
 * For a list of reserved keywords, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Keywords.
 *
 * http://www.w3.org/ns/activitystreams#Object
 */
export const Object__workaround: OwlClass = 'http://www.w3.org/ns/activitystreams#Object';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Offer
 */
export const Offer: OwlClass = 'http://www.w3.org/ns/activitystreams#Offer';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#OrderedCollection
 */
export const OrderedCollection: OwlClass = 'http://www.w3.org/ns/activitystreams#OrderedCollection';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#OrderedItems
 */
export const OrderedItems: OwlClass = 'http://www.w3.org/ns/activitystreams#OrderedItems';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Page
 */
export const Page: OwlClass = 'http://www.w3.org/ns/activitystreams#Page';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Person
 */
export const Person: OwlClass = 'http://www.w3.org/ns/activitystreams#Person';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Organization
 */
export const Organization: OwlClass = 'http://www.w3.org/ns/activitystreams#Organization';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Profile
 */
export const Profile: OwlClass = 'http://www.w3.org/ns/activitystreams#Profile';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Place
 */
export const Place: OwlClass = 'http://www.w3.org/ns/activitystreams#Place';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Question
 */
export const Question: OwlClass = 'http://www.w3.org/ns/activitystreams#Question';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Reject
 */
export const Reject: OwlClass = 'http://www.w3.org/ns/activitystreams#Reject';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Remove
 */
export const Remove: OwlClass = 'http://www.w3.org/ns/activitystreams#Remove';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Service
 */
export const Service: OwlClass = 'http://www.w3.org/ns/activitystreams#Service';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#TentativeAccept
 */
export const TentativeAccept: OwlClass = 'http://www.w3.org/ns/activitystreams#TentativeAccept';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#TentativeReject
 */
export const TentativeReject: OwlClass = 'http://www.w3.org/ns/activitystreams#TentativeReject';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Tombstone
 */
export const Tombstone: OwlClass = 'http://www.w3.org/ns/activitystreams#Tombstone';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Undo
 */
export const Undo: OwlClass = 'http://www.w3.org/ns/activitystreams#Undo';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Update
 */
export const Update: OwlClass = 'http://www.w3.org/ns/activitystreams#Update';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/activitystreams#Video
 */
export const Video: OwlClass = 'http://www.w3.org/ns/activitystreams#Video';
