// rdf-namespaces/dist/cal-07be56c3.js
'use strict';

/**
 * VEVENT
 *
 * Provide a grouping of component properties that describe an event.
 *
 * http://www.w3.org/2002/12/cal/ical#Vevent
 */
var Vevent = 'http://www.w3.org/2002/12/cal/ical#Vevent';
/**
 * VTODO
 *
 * Provide a grouping of calendar properties that describe a to-do.
 *
 * http://www.w3.org/2002/12/cal/ical#Vtodo
 */
var Vtodo = 'http://www.w3.org/2002/12/cal/ical#Vtodo';
/**
 * VJOURNAL
 *
 * Provide a grouping of component properties that describe a journal entry.
 *
 * http://www.w3.org/2002/12/cal/ical#Vjournal
 */
var Vjournal = 'http://www.w3.org/2002/12/cal/ical#Vjournal';
/**
 * VFREEBUSY
 *
 * Provide a grouping of component properties that describe either a request for free/busy time, describe a response to a request for free/busy time or describe a published set of busy time.
 *
 * http://www.w3.org/2002/12/cal/ical#Vfreebusy
 */
var Vfreebusy = 'http://www.w3.org/2002/12/cal/ical#Vfreebusy';
/**
 * VTIMEZONE
 *
 * Provide a grouping of component properties that defines a time zone.
 *
 * http://www.w3.org/2002/12/cal/ical#Vtimezone
 */
var Vtimezone = 'http://www.w3.org/2002/12/cal/ical#Vtimezone';
/**
 * VALARM
 *
 * Provide a grouping of component properties that define an alarm.
 *
 * http://www.w3.org/2002/12/cal/ical#Valarm
 */
var Valarm = 'http://www.w3.org/2002/12/cal/ical#Valarm';
/**
 * CALSCALE
 *
 * This property defines the calendar scale used for the calendar information specified in the iCalendar object.
 *
 * http://www.w3.org/2002/12/cal/ical#calscale
 */
var calscale = 'http://www.w3.org/2002/12/cal/ical#calscale';
/**
 * METHOD
 *
 * This property defines the iCalendar object method associated with the calendar object.
 *
 * http://www.w3.org/2002/12/cal/ical#method
 */
var method = 'http://www.w3.org/2002/12/cal/ical#method';
/**
 * PRODID
 *
 * This property specifies the identifier for the product that created the iCalendar object.
 *
 * http://www.w3.org/2002/12/cal/ical#prodid
 */
var prodid = 'http://www.w3.org/2002/12/cal/ical#prodid';
/**
 * VERSION
 *
 * This property specifies the identifier corresponding to the highest version number or the minimum and maximum range of the iCalendar specification that is required in order to interpret the iCalendar object.
 *
 * http://www.w3.org/2002/12/cal/ical#version
 */
var version = 'http://www.w3.org/2002/12/cal/ical#version';
/**
 * ATTACH
 *
 * The property provides the capability to associate a document object with a calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#attach
 */
var attach = 'http://www.w3.org/2002/12/cal/ical#attach';
/**
 * CATEGORIES
 *
 * This property defines the categories for a calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#categories
 */
var categories = 'http://www.w3.org/2002/12/cal/ical#categories';
/**
 * CLASS
 *
 * This property defines the access classification for a calendar component.
 * Note that `class` is a reserved Javascript keyword, and is therefore suffixed by `__workaround`.
 * For a list of reserved keywords, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Keywords.
 *
 * http://www.w3.org/2002/12/cal/ical#class
 */
var class__workaround = 'http://www.w3.org/2002/12/cal/ical#class';
/**
 * COMMENT
 *
 * This property specifies non-processing information intended to provide a comment to the calendar user.
 *
 * http://www.w3.org/2002/12/cal/ical#comment
 */
var comment = 'http://www.w3.org/2002/12/cal/ical#comment';
/**
 * DESCRIPTION
 *
 * This property provides a more complete description of the calendar component, than that provided by the "SUMMARY" property.
 *
 * http://www.w3.org/2002/12/cal/ical#description
 */
var description = 'http://www.w3.org/2002/12/cal/ical#description';
/**
 * GEO
 *
 * This property specifies information related to the global position for the activity specified by a calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#geo
 */
var geo = 'http://www.w3.org/2002/12/cal/ical#geo';
/**
 *
 *
 *
 *
 * http://www.w3.org/2002/12/cal/ical#List_of_Float
 */
var List_of_Float = 'http://www.w3.org/2002/12/cal/ical#List_of_Float';
/**
 * LOCATION
 *
 * The property defines the intended venue for the activity defined by a calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#location
 */
var location = 'http://www.w3.org/2002/12/cal/ical#location';
/**
 * PERCENT-COMPLETE
 *
 * This property is used by an assignee or delegatee of a to-do to convey the percent completion of a to-do to the Organizer.
 *
 * http://www.w3.org/2002/12/cal/ical#percentComplete
 */
var percentComplete = 'http://www.w3.org/2002/12/cal/ical#percentComplete';
/**
 * PRIORITY
 *
 * The property defines the relative priority for a calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#priority
 */
var priority = 'http://www.w3.org/2002/12/cal/ical#priority';
/**
 * RESOURCES
 *
 * This property defines the equipment or resources anticipated for an activity specified by a calendar entity..
 *
 * http://www.w3.org/2002/12/cal/ical#resources
 */
var resources = 'http://www.w3.org/2002/12/cal/ical#resources';
/**
 * STATUS
 *
 * This property defines the overall status or confirmation for the calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#status
 */
var status = 'http://www.w3.org/2002/12/cal/ical#status';
/**
 * SUMMARY
 *
 * This property defines a short summary or subject for the calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#summary
 */
var summary = 'http://www.w3.org/2002/12/cal/ical#summary';
/**
 * COMPLETED
 *
 * This property defines the date and time that a to-do was actually completed.
 *
 * http://www.w3.org/2002/12/cal/ical#completed
 */
var completed = 'http://www.w3.org/2002/12/cal/ical#completed';
/**
 * DTEND
 *
 * This property specifies the date and time that a calendar component ends.
 *
 * http://www.w3.org/2002/12/cal/ical#dtend
 */
var dtend = 'http://www.w3.org/2002/12/cal/ical#dtend';
/**
 *
 *
 *
 *
 * http://www.w3.org/2002/12/cal/ical#Value_DATE
 */
var Value_DATE = 'http://www.w3.org/2002/12/cal/ical#Value_DATE';
/**
 * DUE
 *
 * This property defines the date and time that a to-do is expected to be completed.
 *
 * http://www.w3.org/2002/12/cal/ical#due
 */
var due = 'http://www.w3.org/2002/12/cal/ical#due';
/**
 * DTSTART
 *
 * This property specifies when the calendar component begins.
 *
 * http://www.w3.org/2002/12/cal/ical#dtstart
 */
var dtstart = 'http://www.w3.org/2002/12/cal/ical#dtstart';
/**
 * DURATION
 *
 * The property specifies a positive duration of time.
 *
 * http://www.w3.org/2002/12/cal/ical#duration
 */
var duration = 'http://www.w3.org/2002/12/cal/ical#duration';
/**
 *
 *
 *
 *
 * http://www.w3.org/2002/12/cal/ical#Value_DURATION
 */
var Value_DURATION = 'http://www.w3.org/2002/12/cal/ical#Value_DURATION';
/**
 * FREEBUSY
 *
 * The property defines one or more free or busy time intervals.
 *
 * http://www.w3.org/2002/12/cal/ical#freebusy
 */
var freebusy = 'http://www.w3.org/2002/12/cal/ical#freebusy';
/**
 *
 *
 *
 *
 * http://www.w3.org/2002/12/cal/ical#Value_PERIOD
 */
var Value_PERIOD = 'http://www.w3.org/2002/12/cal/ical#Value_PERIOD';
/**
 * TRANSP
 *
 * This property defines whether an event is transparent or not to busy time searches.
 *
 * http://www.w3.org/2002/12/cal/ical#transp
 */
var transp = 'http://www.w3.org/2002/12/cal/ical#transp';
/**
 * TZID
 *
 * This property specifies the text value that uniquely identifies the "VTIMEZONE" calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#tzid
 */
var tzid = 'http://www.w3.org/2002/12/cal/ical#tzid';
/**
 * TZNAME
 *
 * This property specifies the customary designation for a time zone description.
 *
 * http://www.w3.org/2002/12/cal/ical#tzname
 */
var tzname = 'http://www.w3.org/2002/12/cal/ical#tzname';
/**
 * TZOFFSETFROM
 *
 * This property specifies the offset which is in use prior to this time zone observance.
 *
 * http://www.w3.org/2002/12/cal/ical#tzoffsetfrom
 */
var tzoffsetfrom = 'http://www.w3.org/2002/12/cal/ical#tzoffsetfrom';
/**
 * TZOFFSETTO
 *
 * This property specifies the offset which is in use in this time zone observance.
 *
 * http://www.w3.org/2002/12/cal/ical#tzoffsetto
 */
var tzoffsetto = 'http://www.w3.org/2002/12/cal/ical#tzoffsetto';
/**
 * TZURL
 *
 * The TZURL provides a means for a VTIMEZONE component to point to a network location that can be used to retrieve an up-to- date version of itself.
 *
 * http://www.w3.org/2002/12/cal/ical#tzurl
 */
var tzurl = 'http://www.w3.org/2002/12/cal/ical#tzurl';
/**
 * ATTENDEE
 *
 * The property defines an "Attendee" within a calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#attendee
 */
var attendee = 'http://www.w3.org/2002/12/cal/ical#attendee';
/**
 * CONTACT
 *
 * The property is used to represent contact information or alternately a reference to contact information associated with the calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#contact
 */
var contact = 'http://www.w3.org/2002/12/cal/ical#contact';
/**
 * ORGANIZER
 *
 * The property defines the organizer for a calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#organizer
 */
var organizer = 'http://www.w3.org/2002/12/cal/ical#organizer';
/**
 * RECURRENCE-ID
 *
 * This property is used in conjunction with the "UID" and "SEQUENCE" property to identify a specific instance of a recurring "VEVENT", "VTODO" or "VJOURNAL" calendar component. The property value is the effective value of the "DTSTART" property of the recurrence instance.
 *
 * http://www.w3.org/2002/12/cal/ical#recurrenceId
 */
var recurrenceId = 'http://www.w3.org/2002/12/cal/ical#recurrenceId';
/**
 *
 *
 *
 *
 * http://www.w3.org/2002/12/cal/ical#DomainOf_rrule
 */
var DomainOf_rrule = 'http://www.w3.org/2002/12/cal/ical#DomainOf_rrule';
/**
 * RELATED-TO
 *
 * The property is used to represent a relationship or reference between one calendar component and another.
 *
 * http://www.w3.org/2002/12/cal/ical#relatedTo
 */
var relatedTo = 'http://www.w3.org/2002/12/cal/ical#relatedTo';
/**
 * URL
 *
 * This property defines a Uniform Resource Locator (URL) associated with the iCalendar object.
 *
 * http://www.w3.org/2002/12/cal/ical#url
 */
var url = 'http://www.w3.org/2002/12/cal/ical#url';
/**
 * UID
 *
 * This property defines the persistent, globally unique identifier for the calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#uid
 */
var uid = 'http://www.w3.org/2002/12/cal/ical#uid';
/**
 * EXDATE
 *
 * This property defines the list of date/time exceptions for a recurring calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#exdate
 */
var exdate = 'http://www.w3.org/2002/12/cal/ical#exdate';
/**
 * EXRULE
 *
 * This property defines a rule or repeating pattern for an exception to a recurrence set.
 *
 * http://www.w3.org/2002/12/cal/ical#exrule
 */
var exrule = 'http://www.w3.org/2002/12/cal/ical#exrule';
/**
 *
 *
 *
 *
 * http://www.w3.org/2002/12/cal/ical#Value_RECUR
 */
var Value_RECUR = 'http://www.w3.org/2002/12/cal/ical#Value_RECUR';
/**
 * RDATE
 *
 * This property defines the list of date/times for a recurrence set.
 *
 * http://www.w3.org/2002/12/cal/ical#rdate
 */
var rdate = 'http://www.w3.org/2002/12/cal/ical#rdate';
/**
 * RRULE
 *
 * This property defines a rule or repeating pattern for recurring events, to-dos, or time zone definitions.
 *
 * http://www.w3.org/2002/12/cal/ical#rrule
 */
var rrule = 'http://www.w3.org/2002/12/cal/ical#rrule';
/**
 * ACTION
 *
 * This property defines the action to be invoked when an alarm is triggered.
 *
 * http://www.w3.org/2002/12/cal/ical#action
 */
var action = 'http://www.w3.org/2002/12/cal/ical#action';
/**
 * REPEAT
 *
 * This property defines the number of time the alarm should be repeated, after the initial trigger.
 *
 * http://www.w3.org/2002/12/cal/ical#repeat
 */
var repeat = 'http://www.w3.org/2002/12/cal/ical#repeat';
/**
 * TRIGGER
 *
 * This property specifies when an alarm will trigger.
 *
 * http://www.w3.org/2002/12/cal/ical#trigger
 */
var trigger = 'http://www.w3.org/2002/12/cal/ical#trigger';
/**
 * CREATED
 *
 * This property specifies the date and time that the calendar information was created by the calendar user agent in the calendar store. Note: This is analogous to the creation date and time for a file in the file system.
 *
 * http://www.w3.org/2002/12/cal/ical#created
 */
var created = 'http://www.w3.org/2002/12/cal/ical#created';
/**
 * DTSTAMP
 *
 * The property indicates the date/time that the instance of the iCalendar object was created.
 *
 * http://www.w3.org/2002/12/cal/ical#dtstamp
 */
var dtstamp = 'http://www.w3.org/2002/12/cal/ical#dtstamp';
/**
 * LAST-MODIFIED
 *
 * The property specifies the date and time that the information associated with the calendar component was last revised in the calendar store. Note: This is analogous to the modification date and time for a file in the file system.
 *
 * http://www.w3.org/2002/12/cal/ical#lastModified
 */
var lastModified = 'http://www.w3.org/2002/12/cal/ical#lastModified';
/**
 * SEQUENCE
 *
 * This property defines the revision sequence number of the calendar component within a sequence of revisions.
 *
 * http://www.w3.org/2002/12/cal/ical#sequence
 */
var sequence = 'http://www.w3.org/2002/12/cal/ical#sequence';
/**
 * REQUEST-STATUS
 *
 * This property defines the status code returned for a scheduling request.
 *
 * http://www.w3.org/2002/12/cal/ical#requestStatus
 */
var requestStatus = 'http://www.w3.org/2002/12/cal/ical#requestStatus';

var calImport = /*#__PURE__*/Object.freeze({
	Vevent: Vevent,
	Vtodo: Vtodo,
	Vjournal: Vjournal,
	Vfreebusy: Vfreebusy,
	Vtimezone: Vtimezone,
	Valarm: Valarm,
	calscale: calscale,
	method: method,
	prodid: prodid,
	version: version,
	attach: attach,
	categories: categories,
	class__workaround: class__workaround,
	comment: comment,
	description: description,
	geo: geo,
	List_of_Float: List_of_Float,
	location: location,
	percentComplete: percentComplete,
	priority: priority,
	resources: resources,
	status: status,
	summary: summary,
	completed: completed,
	dtend: dtend,
	Value_DATE: Value_DATE,
	due: due,
	dtstart: dtstart,
	duration: duration,
	Value_DURATION: Value_DURATION,
	freebusy: freebusy,
	Value_PERIOD: Value_PERIOD,
	transp: transp,
	tzid: tzid,
	tzname: tzname,
	tzoffsetfrom: tzoffsetfrom,
	tzoffsetto: tzoffsetto,
	tzurl: tzurl,
	attendee: attendee,
	contact: contact,
	organizer: organizer,
	recurrenceId: recurrenceId,
	DomainOf_rrule: DomainOf_rrule,
	relatedTo: relatedTo,
	url: url,
	uid: uid,
	exdate: exdate,
	exrule: exrule,
	Value_RECUR: Value_RECUR,
	rdate: rdate,
	rrule: rrule,
	action: action,
	repeat: repeat,
	trigger: trigger,
	created: created,
	dtstamp: dtstamp,
	lastModified: lastModified,
	sequence: sequence,
	requestStatus: requestStatus
});

exports.DomainOf_rrule = DomainOf_rrule;
exports.List_of_Float = List_of_Float;
exports.Valarm = Valarm;
exports.Value_DATE = Value_DATE;
exports.Value_DURATION = Value_DURATION;
exports.Value_PERIOD = Value_PERIOD;
exports.Value_RECUR = Value_RECUR;
exports.Vevent = Vevent;
exports.Vfreebusy = Vfreebusy;
exports.Vjournal = Vjournal;
exports.Vtimezone = Vtimezone;
exports.Vtodo = Vtodo;
exports.action = action;
exports.attach = attach;
exports.attendee = attendee;
exports.calImport = calImport;
exports.calscale = calscale;
exports.categories = categories;
exports.class__workaround = class__workaround;
exports.comment = comment;
exports.completed = completed;
exports.contact = contact;
exports.created = created;
exports.description = description;
exports.dtend = dtend;
exports.dtstamp = dtstamp;
exports.dtstart = dtstart;
exports.due = due;
exports.duration = duration;
exports.exdate = exdate;
exports.exrule = exrule;
exports.freebusy = freebusy;
exports.geo = geo;
exports.lastModified = lastModified;
exports.location = location;
exports.method = method;
exports.organizer = organizer;
exports.percentComplete = percentComplete;
exports.priority = priority;
exports.prodid = prodid;
exports.rdate = rdate;
exports.recurrenceId = recurrenceId;
exports.relatedTo = relatedTo;
exports.repeat = repeat;
exports.requestStatus = requestStatus;
exports.resources = resources;
exports.rrule = rrule;
exports.sequence = sequence;
exports.status = status;
exports.summary = summary;
exports.transp = transp;
exports.trigger = trigger;
exports.tzid = tzid;
exports.tzname = tzname;
exports.tzoffsetfrom = tzoffsetfrom;
exports.tzoffsetto = tzoffsetto;
exports.tzurl = tzurl;
exports.uid = uid;
exports.url = url;
exports.version = version;
