// rdf-namespaces/dist/cal.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * VEVENT
 * 
 * Provide a grouping of component properties that describe an event.
 *
 * http://www.w3.org/2002/12/cal/ical#Vevent
 */
export const Vevent: OwlClass = 'http://www.w3.org/2002/12/cal/ical#Vevent';

/**
 * VTODO
 * 
 * Provide a grouping of calendar properties that describe a to-do.
 *
 * http://www.w3.org/2002/12/cal/ical#Vtodo
 */
export const Vtodo: OwlClass = 'http://www.w3.org/2002/12/cal/ical#Vtodo';

/**
 * VJOURNAL
 * 
 * Provide a grouping of component properties that describe a journal entry.
 *
 * http://www.w3.org/2002/12/cal/ical#Vjournal
 */
export const Vjournal: OwlClass = 'http://www.w3.org/2002/12/cal/ical#Vjournal';

/**
 * VFREEBUSY
 * 
 * Provide a grouping of component properties that describe either a request for free/busy time, describe a response to a request for free/busy time or describe a published set of busy time.
 *
 * http://www.w3.org/2002/12/cal/ical#Vfreebusy
 */
export const Vfreebusy: OwlClass = 'http://www.w3.org/2002/12/cal/ical#Vfreebusy';

/**
 * VTIMEZONE
 * 
 * Provide a grouping of component properties that defines a time zone.
 *
 * http://www.w3.org/2002/12/cal/ical#Vtimezone
 */
export const Vtimezone: OwlClass = 'http://www.w3.org/2002/12/cal/ical#Vtimezone';

/**
 * VALARM
 * 
 * Provide a grouping of component properties that define an alarm.
 *
 * http://www.w3.org/2002/12/cal/ical#Valarm
 */
export const Valarm: OwlClass = 'http://www.w3.org/2002/12/cal/ical#Valarm';

/**
 * CALSCALE
 * 
 * This property defines the calendar scale used for the calendar information specified in the iCalendar object.
 *
 * http://www.w3.org/2002/12/cal/ical#calscale
 */
export const calscale: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#calscale';

/**
 * METHOD
 * 
 * This property defines the iCalendar object method associated with the calendar object.
 *
 * http://www.w3.org/2002/12/cal/ical#method
 */
export const method: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#method';

/**
 * PRODID
 * 
 * This property specifies the identifier for the product that created the iCalendar object.
 *
 * http://www.w3.org/2002/12/cal/ical#prodid
 */
export const prodid: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#prodid';

/**
 * VERSION
 * 
 * This property specifies the identifier corresponding to the highest version number or the minimum and maximum range of the iCalendar specification that is required in order to interpret the iCalendar object.
 *
 * http://www.w3.org/2002/12/cal/ical#version
 */
export const version: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#version';

/**
 * ATTACH
 * 
 * The property provides the capability to associate a document object with a calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#attach
 */
export const attach: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#attach';

/**
 * CATEGORIES
 * 
 * This property defines the categories for a calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#categories
 */
export const categories: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#categories';

/**
 * CLASS
 * 
 * This property defines the access classification for a calendar component.
 * Note that `class` is a reserved Javascript keyword, and is therefore suffixed by `__workaround`.
 * For a list of reserved keywords, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Keywords.
 *
 * http://www.w3.org/2002/12/cal/ical#class
 */
export const class__workaround: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#class';

/**
 * COMMENT
 * 
 * This property specifies non-processing information intended to provide a comment to the calendar user.
 *
 * http://www.w3.org/2002/12/cal/ical#comment
 */
export const comment: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#comment';

/**
 * DESCRIPTION
 * 
 * This property provides a more complete description of the calendar component, than that provided by the "SUMMARY" property.
 *
 * http://www.w3.org/2002/12/cal/ical#description
 */
export const description: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#description';

/**
 * GEO
 * 
 * This property specifies information related to the global position for the activity specified by a calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#geo
 */
export const geo: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#geo';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2002/12/cal/ical#List_of_Float
 */
export const List_of_Float: OwlClass = 'http://www.w3.org/2002/12/cal/ical#List_of_Float';

/**
 * LOCATION
 * 
 * The property defines the intended venue for the activity defined by a calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#location
 */
export const location: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#location';

/**
 * PERCENT-COMPLETE
 * 
 * This property is used by an assignee or delegatee of a to-do to convey the percent completion of a to-do to the Organizer.
 *
 * http://www.w3.org/2002/12/cal/ical#percentComplete
 */
export const percentComplete: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#percentComplete';

/**
 * PRIORITY
 * 
 * The property defines the relative priority for a calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#priority
 */
export const priority: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#priority';

/**
 * RESOURCES
 * 
 * This property defines the equipment or resources anticipated for an activity specified by a calendar entity..
 *
 * http://www.w3.org/2002/12/cal/ical#resources
 */
export const resources: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#resources';

/**
 * STATUS
 * 
 * This property defines the overall status or confirmation for the calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#status
 */
export const status: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#status';

/**
 * SUMMARY
 * 
 * This property defines a short summary or subject for the calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#summary
 */
export const summary: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#summary';

/**
 * COMPLETED
 * 
 * This property defines the date and time that a to-do was actually completed.
 *
 * http://www.w3.org/2002/12/cal/ical#completed
 */
export const completed: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#completed';

/**
 * DTEND
 * 
 * This property specifies the date and time that a calendar component ends.
 *
 * http://www.w3.org/2002/12/cal/ical#dtend
 */
export const dtend: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#dtend';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2002/12/cal/ical#Value_DATE
 */
export const Value_DATE: OwlClass = 'http://www.w3.org/2002/12/cal/ical#Value_DATE';

/**
 * DUE
 * 
 * This property defines the date and time that a to-do is expected to be completed.
 *
 * http://www.w3.org/2002/12/cal/ical#due
 */
export const due: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#due';

/**
 * DTSTART
 * 
 * This property specifies when the calendar component begins.
 *
 * http://www.w3.org/2002/12/cal/ical#dtstart
 */
export const dtstart: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#dtstart';

/**
 * DURATION
 * 
 * The property specifies a positive duration of time.
 *
 * http://www.w3.org/2002/12/cal/ical#duration
 */
export const duration: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#duration';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2002/12/cal/ical#Value_DURATION
 */
export const Value_DURATION: OwlClass = 'http://www.w3.org/2002/12/cal/ical#Value_DURATION';

/**
 * FREEBUSY
 * 
 * The property defines one or more free or busy time intervals.
 *
 * http://www.w3.org/2002/12/cal/ical#freebusy
 */
export const freebusy: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#freebusy';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2002/12/cal/ical#Value_PERIOD
 */
export const Value_PERIOD: OwlClass = 'http://www.w3.org/2002/12/cal/ical#Value_PERIOD';

/**
 * TRANSP
 * 
 * This property defines whether an event is transparent or not to busy time searches.
 *
 * http://www.w3.org/2002/12/cal/ical#transp
 */
export const transp: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#transp';

/**
 * TZID
 * 
 * This property specifies the text value that uniquely identifies the "VTIMEZONE" calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#tzid
 */
export const tzid: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#tzid';

/**
 * TZNAME
 * 
 * This property specifies the customary designation for a time zone description.
 *
 * http://www.w3.org/2002/12/cal/ical#tzname
 */
export const tzname: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#tzname';

/**
 * TZOFFSETFROM
 * 
 * This property specifies the offset which is in use prior to this time zone observance.
 *
 * http://www.w3.org/2002/12/cal/ical#tzoffsetfrom
 */
export const tzoffsetfrom: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#tzoffsetfrom';

/**
 * TZOFFSETTO
 * 
 * This property specifies the offset which is in use in this time zone observance.
 *
 * http://www.w3.org/2002/12/cal/ical#tzoffsetto
 */
export const tzoffsetto: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#tzoffsetto';

/**
 * TZURL
 * 
 * The TZURL provides a means for a VTIMEZONE component to point to a network location that can be used to retrieve an up-to- date version of itself.
 *
 * http://www.w3.org/2002/12/cal/ical#tzurl
 */
export const tzurl: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#tzurl';

/**
 * ATTENDEE
 * 
 * The property defines an "Attendee" within a calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#attendee
 */
export const attendee: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#attendee';

/**
 * CONTACT
 * 
 * The property is used to represent contact information or alternately a reference to contact information associated with the calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#contact
 */
export const contact: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#contact';

/**
 * ORGANIZER
 * 
 * The property defines the organizer for a calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#organizer
 */
export const organizer: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#organizer';

/**
 * RECURRENCE-ID
 * 
 * This property is used in conjunction with the "UID" and "SEQUENCE" property to identify a specific instance of a recurring "VEVENT", "VTODO" or "VJOURNAL" calendar component. The property value is the effective value of the "DTSTART" property of the recurrence instance.
 *
 * http://www.w3.org/2002/12/cal/ical#recurrenceId
 */
export const recurrenceId: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#recurrenceId';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2002/12/cal/ical#DomainOf_rrule
 */
export const DomainOf_rrule: OwlClass = 'http://www.w3.org/2002/12/cal/ical#DomainOf_rrule';

/**
 * RELATED-TO
 * 
 * The property is used to represent a relationship or reference between one calendar component and another.
 *
 * http://www.w3.org/2002/12/cal/ical#relatedTo
 */
export const relatedTo: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#relatedTo';

/**
 * URL
 * 
 * This property defines a Uniform Resource Locator (URL) associated with the iCalendar object.
 *
 * http://www.w3.org/2002/12/cal/ical#url
 */
export const url: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#url';

/**
 * UID
 * 
 * This property defines the persistent, globally unique identifier for the calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#uid
 */
export const uid: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#uid';

/**
 * EXDATE
 * 
 * This property defines the list of date/time exceptions for a recurring calendar component.
 *
 * http://www.w3.org/2002/12/cal/ical#exdate
 */
export const exdate: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#exdate';

/**
 * EXRULE
 * 
 * This property defines a rule or repeating pattern for an exception to a recurrence set.
 *
 * http://www.w3.org/2002/12/cal/ical#exrule
 */
export const exrule: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#exrule';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2002/12/cal/ical#Value_RECUR
 */
export const Value_RECUR: OwlClass = 'http://www.w3.org/2002/12/cal/ical#Value_RECUR';

/**
 * RDATE
 * 
 * This property defines the list of date/times for a recurrence set.
 *
 * http://www.w3.org/2002/12/cal/ical#rdate
 */
export const rdate: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#rdate';

/**
 * RRULE
 * 
 * This property defines a rule or repeating pattern for recurring events, to-dos, or time zone definitions.
 *
 * http://www.w3.org/2002/12/cal/ical#rrule
 */
export const rrule: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#rrule';

/**
 * ACTION
 * 
 * This property defines the action to be invoked when an alarm is triggered.
 *
 * http://www.w3.org/2002/12/cal/ical#action
 */
export const action: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#action';

/**
 * REPEAT
 * 
 * This property defines the number of time the alarm should be repeated, after the initial trigger.
 *
 * http://www.w3.org/2002/12/cal/ical#repeat
 */
export const repeat: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#repeat';

/**
 * TRIGGER
 * 
 * This property specifies when an alarm will trigger.
 *
 * http://www.w3.org/2002/12/cal/ical#trigger
 */
export const trigger: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#trigger';

/**
 * CREATED
 * 
 * This property specifies the date and time that the calendar information was created by the calendar user agent in the calendar store. Note: This is analogous to the creation date and time for a file in the file system.
 *
 * http://www.w3.org/2002/12/cal/ical#created
 */
export const created: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#created';

/**
 * DTSTAMP
 * 
 * The property indicates the date/time that the instance of the iCalendar object was created.
 *
 * http://www.w3.org/2002/12/cal/ical#dtstamp
 */
export const dtstamp: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#dtstamp';

/**
 * LAST-MODIFIED
 * 
 * The property specifies the date and time that the information associated with the calendar component was last revised in the calendar store. Note: This is analogous to the modification date and time for a file in the file system.
 *
 * http://www.w3.org/2002/12/cal/ical#lastModified
 */
export const lastModified: OwlObjectProperty = 'http://www.w3.org/2002/12/cal/ical#lastModified';

/**
 * SEQUENCE
 * 
 * This property defines the revision sequence number of the calendar component within a sequence of revisions.
 *
 * http://www.w3.org/2002/12/cal/ical#sequence
 */
export const sequence: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#sequence';

/**
 * REQUEST-STATUS
 * 
 * This property defines the status code returned for a scheduling request.
 *
 * http://www.w3.org/2002/12/cal/ical#requestStatus
 */
export const requestStatus: OwlDatatypeProperty = 'http://www.w3.org/2002/12/cal/ical#requestStatus';
