// rdf-namespaces/dist/dc-b082ffe2.js
'use strict';

/**
 *
 *
 *
 *
 * http://purl.org/dc/elements/1.1/contributor
 */
var contributor = 'http://purl.org/dc/elements/1.1/contributor';
/**
 *
 *
 *
 *
 * http://purl.org/dc/elements/1.1/coverage
 */
var coverage = 'http://purl.org/dc/elements/1.1/coverage';
/**
 *
 *
 *
 *
 * http://purl.org/dc/elements/1.1/creator
 */
var creator = 'http://purl.org/dc/elements/1.1/creator';
/**
 *
 *
 *
 *
 * http://purl.org/dc/elements/1.1/date
 */
var date = 'http://purl.org/dc/elements/1.1/date';
/**
 *
 *
 *
 *
 * http://purl.org/dc/elements/1.1/description
 */
var description = 'http://purl.org/dc/elements/1.1/description';
/**
 *
 *
 *
 *
 * http://purl.org/dc/elements/1.1/format
 */
var format = 'http://purl.org/dc/elements/1.1/format';
/**
 *
 *
 *
 *
 * http://purl.org/dc/elements/1.1/identifier
 */
var identifier = 'http://purl.org/dc/elements/1.1/identifier';
/**
 *
 *
 *
 *
 * http://purl.org/dc/elements/1.1/language
 */
var language = 'http://purl.org/dc/elements/1.1/language';
/**
 *
 *
 *
 *
 * http://purl.org/dc/elements/1.1/publisher
 */
var publisher = 'http://purl.org/dc/elements/1.1/publisher';
/**
 *
 *
 *
 *
 * http://purl.org/dc/elements/1.1/relation
 */
var relation = 'http://purl.org/dc/elements/1.1/relation';
/**
 *
 *
 *
 *
 * http://purl.org/dc/elements/1.1/rights
 */
var rights = 'http://purl.org/dc/elements/1.1/rights';
/**
 *
 *
 *
 *
 * http://purl.org/dc/elements/1.1/source
 */
var source = 'http://purl.org/dc/elements/1.1/source';
/**
 *
 *
 *
 *
 * http://purl.org/dc/elements/1.1/subject
 */
var subject = 'http://purl.org/dc/elements/1.1/subject';
/**
 *
 *
 *
 *
 * http://purl.org/dc/elements/1.1/title
 */
var title = 'http://purl.org/dc/elements/1.1/title';
/**
 *
 *
 *
 *
 * http://purl.org/dc/elements/1.1/type
 */
var type = 'http://purl.org/dc/elements/1.1/type';

var dcImport = /*#__PURE__*/Object.freeze({
	contributor: contributor,
	coverage: coverage,
	creator: creator,
	date: date,
	description: description,
	format: format,
	identifier: identifier,
	language: language,
	publisher: publisher,
	relation: relation,
	rights: rights,
	source: source,
	subject: subject,
	title: title,
	type: type
});

exports.contributor = contributor;
exports.coverage = coverage;
exports.creator = creator;
exports.date = date;
exports.dcImport = dcImport;
exports.description = description;
exports.format = format;
exports.identifier = identifier;
exports.language = language;
exports.publisher = publisher;
exports.relation = relation;
exports.rights = rights;
exports.source = source;
exports.subject = subject;
exports.title = title;
exports.type = type;
