// rdf-namespaces/dist/dc.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/elements/1.1/contributor
 */
export const contributor: Property = 'http://purl.org/dc/elements/1.1/contributor';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/elements/1.1/coverage
 */
export const coverage: Property = 'http://purl.org/dc/elements/1.1/coverage';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/elements/1.1/creator
 */
export const creator: Property = 'http://purl.org/dc/elements/1.1/creator';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/elements/1.1/date
 */
export const date: Property = 'http://purl.org/dc/elements/1.1/date';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/elements/1.1/description
 */
export const description: Property = 'http://purl.org/dc/elements/1.1/description';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/elements/1.1/format
 */
export const format: Property = 'http://purl.org/dc/elements/1.1/format';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/elements/1.1/identifier
 */
export const identifier: Property = 'http://purl.org/dc/elements/1.1/identifier';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/elements/1.1/language
 */
export const language: Property = 'http://purl.org/dc/elements/1.1/language';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/elements/1.1/publisher
 */
export const publisher: Property = 'http://purl.org/dc/elements/1.1/publisher';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/elements/1.1/relation
 */
export const relation: Property = 'http://purl.org/dc/elements/1.1/relation';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/elements/1.1/rights
 */
export const rights: Property = 'http://purl.org/dc/elements/1.1/rights';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/elements/1.1/source
 */
export const source: Property = 'http://purl.org/dc/elements/1.1/source';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/elements/1.1/subject
 */
export const subject: Property = 'http://purl.org/dc/elements/1.1/subject';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/elements/1.1/title
 */
export const title: Property = 'http://purl.org/dc/elements/1.1/title';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/elements/1.1/type
 */
export const type: Property = 'http://purl.org/dc/elements/1.1/type';
