// rdf-namespaces/dist/dct.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/AgentClass
 */
export const AgentClass: Class = 'http://purl.org/dc/terms/AgentClass';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/BibliographicResource
 */
export const BibliographicResource: Class = 'http://purl.org/dc/terms/BibliographicResource';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/Box
 */
export const Box: Datatype = 'http://purl.org/dc/terms/Box';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/FileFormat
 */
export const FileFormat: Class = 'http://purl.org/dc/terms/FileFormat';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/Frequency
 */
export const Frequency: Class = 'http://purl.org/dc/terms/Frequency';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/ISO3166
 */
export const ISO3166: Datatype = 'http://purl.org/dc/terms/ISO3166';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/Jurisdiction
 */
export const Jurisdiction: Class = 'http://purl.org/dc/terms/Jurisdiction';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/LicenseDocument
 */
export const LicenseDocument: Class = 'http://purl.org/dc/terms/LicenseDocument';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/LinguisticSystem
 */
export const LinguisticSystem: Class = 'http://purl.org/dc/terms/LinguisticSystem';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/Location
 */
export const Location: Class = 'http://purl.org/dc/terms/Location';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/LocationPeriodOrJurisdiction
 */
export const LocationPeriodOrJurisdiction: Class = 'http://purl.org/dc/terms/LocationPeriodOrJurisdiction';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/MediaType
 */
export const MediaType: Class = 'http://purl.org/dc/terms/MediaType';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/MediaTypeOrExtent
 */
export const MediaTypeOrExtent: Class = 'http://purl.org/dc/terms/MediaTypeOrExtent';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/MethodOfAccrual
 */
export const MethodOfAccrual: Class = 'http://purl.org/dc/terms/MethodOfAccrual';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/MethodOfInstruction
 */
export const MethodOfInstruction: Class = 'http://purl.org/dc/terms/MethodOfInstruction';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/Period
 */
export const Period: Datatype = 'http://purl.org/dc/terms/Period';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/PeriodOfTime
 */
export const PeriodOfTime: Class = 'http://purl.org/dc/terms/PeriodOfTime';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/PhysicalMedium
 */
export const PhysicalMedium: Class = 'http://purl.org/dc/terms/PhysicalMedium';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/PhysicalResource
 */
export const PhysicalResource: Class = 'http://purl.org/dc/terms/PhysicalResource';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/Point
 */
export const Point: Datatype = 'http://purl.org/dc/terms/Point';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/Policy
 */
export const Policy: Class = 'http://purl.org/dc/terms/Policy';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/ProvenanceStatement
 */
export const ProvenanceStatement: Class = 'http://purl.org/dc/terms/ProvenanceStatement';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/RFC1766
 */
export const RFC1766: Datatype = 'http://purl.org/dc/terms/RFC1766';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/RFC3066
 */
export const RFC3066: Datatype = 'http://purl.org/dc/terms/RFC3066';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/RFC4646
 */
export const RFC4646: Datatype = 'http://purl.org/dc/terms/RFC4646';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/RFC5646
 */
export const RFC5646: Datatype = 'http://purl.org/dc/terms/RFC5646';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/RightsStatement
 */
export const RightsStatement: Class = 'http://purl.org/dc/terms/RightsStatement';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/SizeOrDuration
 */
export const SizeOrDuration: Class = 'http://purl.org/dc/terms/SizeOrDuration';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/Standard
 */
export const Standard: Class = 'http://purl.org/dc/terms/Standard';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/URI
 */
export const URI: Datatype = 'http://purl.org/dc/terms/URI';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/W3CDTF
 */
export const W3CDTF: Datatype = 'http://purl.org/dc/terms/W3CDTF';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/abstract
 */
export const abstract: Property = 'http://purl.org/dc/terms/abstract';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/accessRights
 */
export const accessRights: Property = 'http://purl.org/dc/terms/accessRights';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/accrualMethod
 */
export const accrualMethod: Property = 'http://purl.org/dc/terms/accrualMethod';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/accrualPeriodicity
 */
export const accrualPeriodicity: Property = 'http://purl.org/dc/terms/accrualPeriodicity';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/accrualPolicy
 */
export const accrualPolicy: Property = 'http://purl.org/dc/terms/accrualPolicy';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/alternative
 */
export const alternative: Property = 'http://purl.org/dc/terms/alternative';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/audience
 */
export const audience: Property = 'http://purl.org/dc/terms/audience';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/available
 */
export const available: Property = 'http://purl.org/dc/terms/available';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/bibliographicCitation
 */
export const bibliographicCitation: Property = 'http://purl.org/dc/terms/bibliographicCitation';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/conformsTo
 */
export const conformsTo: Property = 'http://purl.org/dc/terms/conformsTo';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/contributor
 */
export const contributor: Property = 'http://purl.org/dc/terms/contributor';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/coverage
 */
export const coverage: Property = 'http://purl.org/dc/terms/coverage';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/created
 */
export const created: Property = 'http://purl.org/dc/terms/created';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/creator
 */
export const creator: Property = 'http://purl.org/dc/terms/creator';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/date
 */
export const date: Property = 'http://purl.org/dc/terms/date';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/dateAccepted
 */
export const dateAccepted: Property = 'http://purl.org/dc/terms/dateAccepted';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/dateCopyrighted
 */
export const dateCopyrighted: Property = 'http://purl.org/dc/terms/dateCopyrighted';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/dateSubmitted
 */
export const dateSubmitted: Property = 'http://purl.org/dc/terms/dateSubmitted';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/description
 */
export const description: Property = 'http://purl.org/dc/terms/description';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/educationLevel
 */
export const educationLevel: Property = 'http://purl.org/dc/terms/educationLevel';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/extent
 */
export const extent: Property = 'http://purl.org/dc/terms/extent';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/format
 */
export const format: Property = 'http://purl.org/dc/terms/format';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/hasFormat
 */
export const hasFormat: Property = 'http://purl.org/dc/terms/hasFormat';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/hasPart
 */
export const hasPart: Property = 'http://purl.org/dc/terms/hasPart';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/hasVersion
 */
export const hasVersion: Property = 'http://purl.org/dc/terms/hasVersion';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/identifier
 */
export const identifier: Property = 'http://purl.org/dc/terms/identifier';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/instructionalMethod
 */
export const instructionalMethod: Property = 'http://purl.org/dc/terms/instructionalMethod';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/isFormatOf
 */
export const isFormatOf: Property = 'http://purl.org/dc/terms/isFormatOf';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/isPartOf
 */
export const isPartOf: Property = 'http://purl.org/dc/terms/isPartOf';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/isReferencedBy
 */
export const isReferencedBy: Property = 'http://purl.org/dc/terms/isReferencedBy';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/isReplacedBy
 */
export const isReplacedBy: Property = 'http://purl.org/dc/terms/isReplacedBy';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/isRequiredBy
 */
export const isRequiredBy: Property = 'http://purl.org/dc/terms/isRequiredBy';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/isVersionOf
 */
export const isVersionOf: Property = 'http://purl.org/dc/terms/isVersionOf';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/issued
 */
export const issued: Property = 'http://purl.org/dc/terms/issued';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/language
 */
export const language: Property = 'http://purl.org/dc/terms/language';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/license
 */
export const license: Property = 'http://purl.org/dc/terms/license';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/mediator
 */
export const mediator: Property = 'http://purl.org/dc/terms/mediator';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/medium
 */
export const medium: Property = 'http://purl.org/dc/terms/medium';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/modified
 */
export const modified: Property = 'http://purl.org/dc/terms/modified';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/provenance
 */
export const provenance: Property = 'http://purl.org/dc/terms/provenance';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/publisher
 */
export const publisher: Property = 'http://purl.org/dc/terms/publisher';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/references
 */
export const references: Property = 'http://purl.org/dc/terms/references';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/relation
 */
export const relation: Property = 'http://purl.org/dc/terms/relation';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/replaces
 */
export const replaces: Property = 'http://purl.org/dc/terms/replaces';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/requires
 */
export const requires: Property = 'http://purl.org/dc/terms/requires';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/rights
 */
export const rights: Property = 'http://purl.org/dc/terms/rights';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/rightsHolder
 */
export const rightsHolder: Property = 'http://purl.org/dc/terms/rightsHolder';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/source
 */
export const source: Property = 'http://purl.org/dc/terms/source';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/spatial
 */
export const spatial: Property = 'http://purl.org/dc/terms/spatial';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/subject
 */
export const subject: Property = 'http://purl.org/dc/terms/subject';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/tableOfContents
 */
export const tableOfContents: Property = 'http://purl.org/dc/terms/tableOfContents';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/temporal
 */
export const temporal: Property = 'http://purl.org/dc/terms/temporal';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/title
 */
export const title: Property = 'http://purl.org/dc/terms/title';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/type
 */
export const type: Property = 'http://purl.org/dc/terms/type';

/**
 * 
 * 
 * 
 *
 * http://purl.org/dc/terms/valid
 */
export const valid: Property = 'http://purl.org/dc/terms/valid';
