// rdf-namespaces/dist/hydra-a772aef4.js
/**
 * ApiDocumentation
 *
 * The Hydra API documentation class
 *
 * http://www.w3.org/ns/hydra/core#ApiDocumentation
 */
var ApiDocumentation = 'http://www.w3.org/ns/hydra/core#ApiDocumentation';
/**
 * Base Uri source
 *
 * Provides a base abstract for base Uri source for Iri template resolution.
 *
 * http://www.w3.org/ns/hydra/core#BaseUriSource
 */
var BaseUriSource = 'http://www.w3.org/ns/hydra/core#BaseUriSource';
/**
 * BasicRepresentation
 *
 * A representation that serializes just the lexical form of a variable value, but omits language and type information.
 *
 * http://www.w3.org/ns/hydra/core#BasicRepresentation
 */
var BasicRepresentation = 'http://www.w3.org/ns/hydra/core#BasicRepresentation';
/**
 * Hydra Class
 *
 * The class of Hydra classes.
 *
 * http://www.w3.org/ns/hydra/core#Class
 */
var Class = 'http://www.w3.org/ns/hydra/core#Class';
/**
 * Collection
 *
 * A collection holding references to a number of related resources.
 *
 * http://www.w3.org/ns/hydra/core#Collection
 */
var Collection = 'http://www.w3.org/ns/hydra/core#Collection';
/**
 * Error
 *
 * A runtime error, used to report information beyond the returned status code.
 * Note that `Error` is a reserved Javascript keyword, and is therefore suffixed by `__workaround`.
 * For a list of reserved keywords, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Keywords.
 *
 * http://www.w3.org/ns/hydra/core#Error
 */
var Error__workaround = 'http://www.w3.org/ns/hydra/core#Error';
/**
 * ExplicitRepresentation
 *
 * A representation that serializes a variable value including its language and type information and thus differentiating between IRIs and literals.
 *
 * http://www.w3.org/ns/hydra/core#ExplicitRepresentation
 */
var ExplicitRepresentation = 'http://www.w3.org/ns/hydra/core#ExplicitRepresentation';
/**
 * Header specification
 *
 * Specifies a possible either expected or returned header values
 *
 * http://www.w3.org/ns/hydra/core#HeaderSpecification
 */
var HeaderSpecification = 'http://www.w3.org/ns/hydra/core#HeaderSpecification';
/**
 * IRI Template
 *
 * The class of IRI templates.
 *
 * http://www.w3.org/ns/hydra/core#IriTemplate
 */
var IriTemplate = 'http://www.w3.org/ns/hydra/core#IriTemplate';
/**
 * IriTemplateMapping
 *
 * A mapping from an IRI template variable to a property.
 *
 * http://www.w3.org/ns/hydra/core#IriTemplateMapping
 */
var IriTemplateMapping = 'http://www.w3.org/ns/hydra/core#IriTemplateMapping';
/**
 * Link
 *
 * The class of properties representing links.
 *
 * http://www.w3.org/ns/hydra/core#Link
 */
var Link = 'http://www.w3.org/ns/hydra/core#Link';
/**
 * Operation
 *
 * An operation.
 *
 * http://www.w3.org/ns/hydra/core#Operation
 */
var Operation = 'http://www.w3.org/ns/hydra/core#Operation';
/**
 * PartialCollectionView
 *
 * A PartialCollectionView describes a partial view of a Collection. Multiple PartialCollectionViews can be connected with the the next/previous properties to allow a client to retrieve all members of the collection.
 *
 * http://www.w3.org/ns/hydra/core#PartialCollectionView
 */
var PartialCollectionView = 'http://www.w3.org/ns/hydra/core#PartialCollectionView';
/**
 * Hydra Resource
 *
 * The class of dereferenceable resources by means a client can attempt to dereference; however, the received responses should still be verified.
 *
 * http://www.w3.org/ns/hydra/core#Resource
 */
var Resource = 'http://www.w3.org/ns/hydra/core#Resource';
/**
 * RFC6570 IRI template
 *
 * An IRI template as defined by RFC6570.
 *
 * http://www.w3.org/ns/hydra/core#Rfc6570Template
 */
var Rfc6570Template = 'http://www.w3.org/ns/hydra/core#Rfc6570Template';
/**
 * Status code description
 *
 * Additional information about a status code that might be returned.
 *
 * http://www.w3.org/ns/hydra/core#Status
 */
var Status = 'http://www.w3.org/ns/hydra/core#Status';
/**
 * Supported Property
 *
 * A property known to be supported by a Hydra class.
 *
 * http://www.w3.org/ns/hydra/core#SupportedProperty
 */
var SupportedProperty = 'http://www.w3.org/ns/hydra/core#SupportedProperty';
/**
 * Templated Link
 *
 * A templated link.
 *
 * http://www.w3.org/ns/hydra/core#TemplatedLink
 */
var TemplatedLink = 'http://www.w3.org/ns/hydra/core#TemplatedLink';
/**
 * VariableRepresentation
 *
 * A representation specifies how to serialize variable values into strings.
 *
 * http://www.w3.org/ns/hydra/core#VariableRepresentation
 */
var VariableRepresentation = 'http://www.w3.org/ns/hydra/core#VariableRepresentation';
/**
 * apiDocumentation
 *
 * A link to the API documentation
 *
 * http://www.w3.org/ns/hydra/core#apiDocumentation
 */
var apiDocumentation = 'http://www.w3.org/ns/hydra/core#apiDocumentation';
/**
 * closed set
 *
 * Determines whether the provided set of header values is closed or not.
 *
 * http://www.w3.org/ns/hydra/core#closedSet
 */
var closedSet = 'http://www.w3.org/ns/hydra/core#closedSet';
/**
 * collection
 *
 * Collections somehow related to this resource.
 *
 * http://www.w3.org/ns/hydra/core#collection
 */
var collection = 'http://www.w3.org/ns/hydra/core#collection';
/**
 * description
 *
 * A description.
 *
 * http://www.w3.org/ns/hydra/core#description
 */
var description = 'http://www.w3.org/ns/hydra/core#description';
/**
 * entrypoint
 *
 * A link to main entry point of the Web API
 *
 * http://www.w3.org/ns/hydra/core#entrypoint
 */
var entrypoint = 'http://www.w3.org/ns/hydra/core#entrypoint';
/**
 * expects
 *
 * The information expected by the Web API.
 *
 * http://www.w3.org/ns/hydra/core#expects
 */
var expects = 'http://www.w3.org/ns/hydra/core#expects';
/**
 * expects header
 *
 * Specification of the header expected by the operation.
 *
 * http://www.w3.org/ns/hydra/core#expectsHeader
 */
var expectsHeader = 'http://www.w3.org/ns/hydra/core#expectsHeader';
/**
 * extension
 *
 * Hint on what kind of extensions are in use.
 *
 * http://www.w3.org/ns/hydra/core#extension
 */
var extension = 'http://www.w3.org/ns/hydra/core#extension';
/**
 * first
 *
 * The first resource of an interlinked set of resources.
 *
 * http://www.w3.org/ns/hydra/core#first
 */
var first = 'http://www.w3.org/ns/hydra/core#first';
/**
 * freetext query
 *
 * A property representing a freetext query.
 *
 * http://www.w3.org/ns/hydra/core#freetextQuery
 */
var freetextQuery = 'http://www.w3.org/ns/hydra/core#freetextQuery';
/**
 * header name
 *
 * Name of the header.
 *
 * http://www.w3.org/ns/hydra/core#headerName
 */
var headerName = 'http://www.w3.org/ns/hydra/core#headerName';
/**
 * last
 *
 * The last resource of an interlinked set of resources.
 *
 * http://www.w3.org/ns/hydra/core#last
 */
var last = 'http://www.w3.org/ns/hydra/core#last';
/**
 * take
 *
 * Instructs to limit set only to N elements.
 *
 * http://www.w3.org/ns/hydra/core#limit
 */
var limit = 'http://www.w3.org/ns/hydra/core#limit';
/**
 * mapping
 *
 * A variable-to-property mapping of the IRI template.
 *
 * http://www.w3.org/ns/hydra/core#mapping
 */
var mapping = 'http://www.w3.org/ns/hydra/core#mapping';
/**
 * member
 *
 * A member of the collection
 *
 * http://www.w3.org/ns/hydra/core#member
 */
var member = 'http://www.w3.org/ns/hydra/core#member';
/**
 * method
 *
 * The HTTP method.
 *
 * http://www.w3.org/ns/hydra/core#method
 */
var method = 'http://www.w3.org/ns/hydra/core#method';
/**
 * next
 *
 * The resource following the current instance in an interlinked set of resources.
 *
 * http://www.w3.org/ns/hydra/core#next
 */
var next = 'http://www.w3.org/ns/hydra/core#next';
/**
 * skip
 *
 * Instructs to skip N elements of the set.
 *
 * http://www.w3.org/ns/hydra/core#offset
 */
var offset = 'http://www.w3.org/ns/hydra/core#offset';
/**
 * operation
 *
 * An operation supported by the Hydra resource
 *
 * http://www.w3.org/ns/hydra/core#operation
 */
var operation = 'http://www.w3.org/ns/hydra/core#operation';
/**
 * page index
 *
 * Instructs to provide a specific page of the collection at a given index.
 *
 * http://www.w3.org/ns/hydra/core#pageIndex
 */
var pageIndex = 'http://www.w3.org/ns/hydra/core#pageIndex';
/**
 * page reference
 *
 * Instructs to provide a specific page reference of the collection.
 *
 * http://www.w3.org/ns/hydra/core#pageReference
 */
var pageReference = 'http://www.w3.org/ns/hydra/core#pageReference';
/**
 * possible status
 *
 * A status that might be returned by the Web API (other statuses should be expected and properly handled as well)
 *
 * http://www.w3.org/ns/hydra/core#possibleStatus
 */
var possibleStatus = 'http://www.w3.org/ns/hydra/core#possibleStatus';
/**
 * possible header value
 *
 * Possible value of the header.
 *
 * http://www.w3.org/ns/hydra/core#possibleValue
 */
var possibleValue = 'http://www.w3.org/ns/hydra/core#possibleValue';
/**
 * previous
 *
 * The resource preceding the current instance in an interlinked set of resources.
 *
 * http://www.w3.org/ns/hydra/core#previous
 */
var previous = 'http://www.w3.org/ns/hydra/core#previous';
/**
 * property
 *
 * A property
 *
 * http://www.w3.org/ns/hydra/core#property
 */
var property = 'http://www.w3.org/ns/hydra/core#property';
/**
 * readable
 *
 * True if the client can retrieve the property's value, false otherwise.
 *
 * http://www.w3.org/ns/hydra/core#readable
 */
var readable = 'http://www.w3.org/ns/hydra/core#readable';
/**
 * required
 *
 * True if the property is required, false otherwise.
 *
 * http://www.w3.org/ns/hydra/core#required
 */
var required = 'http://www.w3.org/ns/hydra/core#required';
/**
 * relative Uri resolution
 *
 *
 *
 * http://www.w3.org/ns/hydra/core#resolveRelativeUsing
 */
var resolveRelativeUsing = 'http://www.w3.org/ns/hydra/core#resolveRelativeUsing';
/**
 * returns
 *
 * The information returned by the Web API on success
 *
 * http://www.w3.org/ns/hydra/core#returns
 */
var returns = 'http://www.w3.org/ns/hydra/core#returns';
/**
 * returns header
 *
 * Name of the header returned by the operation.
 *
 * http://www.w3.org/ns/hydra/core#returnsHeader
 */
var returnsHeader = 'http://www.w3.org/ns/hydra/core#returnsHeader';
/**
 * search
 *
 * A IRI template that can be used to query a collection.
 *
 * http://www.w3.org/ns/hydra/core#search
 */
var search = 'http://www.w3.org/ns/hydra/core#search';
/**
 * status code
 *
 * The HTTP status code. Please note it may happen this value will be different to actual status code received.
 *
 * http://www.w3.org/ns/hydra/core#statusCode
 */
var statusCode = 'http://www.w3.org/ns/hydra/core#statusCode';
/**
 * supported classes
 *
 * A class known to be supported by the Web API
 *
 * http://www.w3.org/ns/hydra/core#supportedClass
 */
var supportedClass = 'http://www.w3.org/ns/hydra/core#supportedClass';
/**
 * supported operation
 *
 * An operation supported by instances of the specific Hydra class, or the target of the Hydra link, or IRI template.
 *
 * http://www.w3.org/ns/hydra/core#supportedOperation
 */
var supportedOperation = 'http://www.w3.org/ns/hydra/core#supportedOperation';
/**
 * supported properties
 *
 * The properties known to be supported by a Hydra class
 *
 * http://www.w3.org/ns/hydra/core#supportedProperty
 */
var supportedProperty = 'http://www.w3.org/ns/hydra/core#supportedProperty';
/**
 * template
 *
 * A templated string with placeholders. The literal's datatype indicates the template syntax; if not specified, hydra:Rfc6570Template is assumed.
 *
 * http://www.w3.org/ns/hydra/core#template
 */
var template = 'http://www.w3.org/ns/hydra/core#template';
/**
 * title
 *
 * A title, often used along with a description.
 *
 * http://www.w3.org/ns/hydra/core#title
 */
var title = 'http://www.w3.org/ns/hydra/core#title';
/**
 * total items
 *
 * The total number of items referenced by a collection.
 *
 * http://www.w3.org/ns/hydra/core#totalItems
 */
var totalItems = 'http://www.w3.org/ns/hydra/core#totalItems';
/**
 * variable
 *
 * An IRI template variable
 *
 * http://www.w3.org/ns/hydra/core#variable
 */
var variable = 'http://www.w3.org/ns/hydra/core#variable';
/**
 * variable representation
 *
 * The representation format to use when expanding the IRI template.
 *
 * http://www.w3.org/ns/hydra/core#variableRepresentation
 */
var variableRepresentation = 'http://www.w3.org/ns/hydra/core#variableRepresentation';
/**
 * view
 *
 * A specific view of a resource.
 *
 * http://www.w3.org/ns/hydra/core#view
 */
var view = 'http://www.w3.org/ns/hydra/core#view';
/**
 * writable
 *
 * True if the client can change the property's value, false otherwise.
 *
 * http://www.w3.org/ns/hydra/core#writable
 */
var writable = 'http://www.w3.org/ns/hydra/core#writable';

var hydraImport = /*#__PURE__*/Object.freeze({
	ApiDocumentation: ApiDocumentation,
	BaseUriSource: BaseUriSource,
	BasicRepresentation: BasicRepresentation,
	Class: Class,
	Collection: Collection,
	Error__workaround: Error__workaround,
	ExplicitRepresentation: ExplicitRepresentation,
	HeaderSpecification: HeaderSpecification,
	IriTemplate: IriTemplate,
	IriTemplateMapping: IriTemplateMapping,
	Link: Link,
	Operation: Operation,
	PartialCollectionView: PartialCollectionView,
	Resource: Resource,
	Rfc6570Template: Rfc6570Template,
	Status: Status,
	SupportedProperty: SupportedProperty,
	TemplatedLink: TemplatedLink,
	VariableRepresentation: VariableRepresentation,
	apiDocumentation: apiDocumentation,
	closedSet: closedSet,
	collection: collection,
	description: description,
	entrypoint: entrypoint,
	expects: expects,
	expectsHeader: expectsHeader,
	extension: extension,
	first: first,
	freetextQuery: freetextQuery,
	headerName: headerName,
	last: last,
	limit: limit,
	mapping: mapping,
	member: member,
	method: method,
	next: next,
	offset: offset,
	operation: operation,
	pageIndex: pageIndex,
	pageReference: pageReference,
	possibleStatus: possibleStatus,
	possibleValue: possibleValue,
	previous: previous,
	property: property,
	readable: readable,
	required: required,
	resolveRelativeUsing: resolveRelativeUsing,
	returns: returns,
	returnsHeader: returnsHeader,
	search: search,
	statusCode: statusCode,
	supportedClass: supportedClass,
	supportedOperation: supportedOperation,
	supportedProperty: supportedProperty,
	template: template,
	title: title,
	totalItems: totalItems,
	variable: variable,
	variableRepresentation: variableRepresentation,
	view: view,
	writable: writable
});

export { supportedProperty as $, ApiDocumentation as A, BaseUriSource as B, Class as C, pageReference as D, Error__workaround as E, possibleStatus as F, possibleValue as G, HeaderSpecification as H, IriTemplate as I, previous as J, property as K, Link as L, readable as M, required as N, Operation as O, PartialCollectionView as P, resolveRelativeUsing as Q, Resource as R, Status as S, TemplatedLink as T, returns as U, VariableRepresentation as V, returnsHeader as W, search as X, statusCode as Y, supportedClass as Z, supportedOperation as _, BasicRepresentation as a, template as a0, title as a1, totalItems as a2, variable as a3, variableRepresentation as a4, view as a5, writable as a6, hydraImport as a7, Collection as b, ExplicitRepresentation as c, IriTemplateMapping as d, Rfc6570Template as e, SupportedProperty as f, apiDocumentation as g, closedSet as h, collection as i, description as j, entrypoint as k, expects as l, expectsHeader as m, extension as n, first as o, freetextQuery as p, headerName as q, last as r, limit as s, mapping as t, member as u, method as v, next as w, offset as x, operation as y, pageIndex as z };
