// rdf-namespaces/dist/link-76b28027.js
'use strict';

/**
 * access
 *
 * An event in which a network access is made.
 *
 * http://www.w3.org/2007/ont/link#Access
 */
var Access = 'http://www.w3.org/2007/ont/link#Access';
/**
 * document
 *
 * An Information Resource, such as a web page, image,
 *         data file, etc.
 *
 * http://www.w3.org/2007/ont/link#Document
 */
var Document = 'http://www.w3.org/2007/ont/link#Document';
/**
 * mailbox
 *
 * An end point in the Internet Mail sytem.
 *             Anything which has a URI in the malto: scheme is a Mailbox.
 *
 * http://www.w3.org/2007/ont/link#Mailbox
 */
var Mailbox = 'http://www.w3.org/2007/ont/link#Mailbox';
/**
 *
 *
 * A step in a network protocol, such as a request
 *         or response message.
 *
 * http://www.w3.org/2007/ont/link#ProtocolEvent
 */
var ProtocolEvent = 'http://www.w3.org/2007/ont/link#ProtocolEvent';
/**
 * data document
 *
 * An Information Resource whose meaning which can directly mapped into the RDF model. For example a data file which can be reterived as RDF using HTTP.
 *
 * http://www.w3.org/2007/ont/link#RDFDocument
 */
var RDFDocument = 'http://www.w3.org/2007/ont/link#RDFDocument';
/**
 * access
 *
 * A network response to a request.
 *
 * http://www.w3.org/2007/ont/link#Response
 */
var Response = 'http://www.w3.org/2007/ont/link#Response';
/**
 * session
 *
 * A run of the application software in which various
 *         network acecssses etc happen.
 *
 * http://www.w3.org/2007/ont/link#Session
 */
var Session = 'http://www.w3.org/2007/ont/link#Session';
/**
 * is mentioned in
 *
 *
 *       This class is mentioned in the following document
 *
 *
 * http://www.w3.org/2007/ont/link#isMentionedIn
 */
var isMentionedIn = 'http://www.w3.org/2007/ont/link#isMentionedIn';
/**
 * mentions
 *
 * This document mentions the following class
 *
 * http://www.w3.org/2007/ont/link#mentionsClass
 */
var mentionsClass = 'http://www.w3.org/2007/ont/link#mentionsClass';
/**
 * supports protocol
 *
 * The application supports this network protocol.
 *
 * http://www.w3.org/2007/ont/link#protocol
 */
var protocol = 'http://www.w3.org/2007/ont/link#protocol';
/**
 * redirected request
 *
 *
 *
 * http://www.w3.org/2007/ont/link#redirectedRequest
 */
var redirectedRequest = 'http://www.w3.org/2007/ont/link#redirectedRequest';
/**
 * referrer
 *
 * A resource which contains a reference to the subject.
 *          That is, one of the statements in the object file
 *          has the subject URI.
 *          This property is stored by the tabulator library to explain
 *          why it looked up a given URI. This can help
 *          trace an error when it was in fact a mis-spelling.
 *
 * http://www.w3.org/2007/ont/link#requestedBy
 */
var requestedBy = 'http://www.w3.org/2007/ont/link#requestedBy';
/**
 * requested URI
 *
 *
 *
 * http://www.w3.org/2007/ont/link#requestedURI
 */
var requestedURI = 'http://www.w3.org/2007/ont/link#requestedURI';
/**
 * response
 *
 *
 *
 * http://www.w3.org/2007/ont/link#response
 */
var response = 'http://www.w3.org/2007/ont/link#response';
/**
 * session
 *
 *
 *
 * http://www.w3.org/2007/ont/link#session
 */
var session = 'http://www.w3.org/2007/ont/link#session';
/**
 * uses source
 *
 * A source file accessed in this session.
 *
 * http://www.w3.org/2007/ont/link#source
 */
var source = 'http://www.w3.org/2007/ont/link#source';
/**
 * status
 *
 * The status of a network access.
 *
 * http://www.w3.org/2007/ont/link#status
 */
var status = 'http://www.w3.org/2007/ont/link#status';
/**
 * examples and test data
 *
 * Soemthing which can be used as an example for testing purposes
 *
 * http://www.w3.org/2007/ont/link#testDataset
 */
var testDataset = 'http://www.w3.org/2007/ont/link#testDataset';
/**
 * URI
 *
 * The URI requested in this access.
 *         This is NOT the relationship betwen a URI and a thing t identifiers
 *
 * http://www.w3.org/2007/ont/link#uri
 */
var uri = 'http://www.w3.org/2007/ont/link#uri';

var linkImport = /*#__PURE__*/Object.freeze({
	Access: Access,
	Document: Document,
	Mailbox: Mailbox,
	ProtocolEvent: ProtocolEvent,
	RDFDocument: RDFDocument,
	Response: Response,
	Session: Session,
	isMentionedIn: isMentionedIn,
	mentionsClass: mentionsClass,
	protocol: protocol,
	redirectedRequest: redirectedRequest,
	requestedBy: requestedBy,
	requestedURI: requestedURI,
	response: response,
	session: session,
	source: source,
	status: status,
	testDataset: testDataset,
	uri: uri
});

exports.Access = Access;
exports.Document = Document;
exports.Mailbox = Mailbox;
exports.ProtocolEvent = ProtocolEvent;
exports.RDFDocument = RDFDocument;
exports.Response = Response;
exports.Session = Session;
exports.isMentionedIn = isMentionedIn;
exports.linkImport = linkImport;
exports.mentionsClass = mentionsClass;
exports.protocol = protocol;
exports.redirectedRequest = redirectedRequest;
exports.requestedBy = requestedBy;
exports.requestedURI = requestedURI;
exports.response = response;
exports.session = session;
exports.source = source;
exports.status = status;
exports.testDataset = testDataset;
exports.uri = uri;
