// rdf-namespaces/dist/qu-52da41a2.js
/**
 * account
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/qif#Account
 */
var Account = 'http://www.w3.org/2000/10/swap/pim/qif#Account';
/**
 *
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/qif#Bank
 */
var Bank = 'http://www.w3.org/2000/10/swap/pim/qif#Bank';
/**
 * category
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/qif#Cat
 */
var Cat = 'http://www.w3.org/2000/10/swap/pim/qif#Cat';
/**
 * classified
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/qif#Classified
 */
var Classified = 'http://www.w3.org/2000/10/swap/pim/qif#Classified';
/**
 * receipt
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/qif#Receipt
 */
var Receipt = 'http://www.w3.org/2000/10/swap/pim/qif#Receipt';
/**
 * split transaction
 *
 * A transaction which has to be split into parts for classification.
 *
 * http://www.w3.org/2000/10/swap/pim/qif#SplitTransaction
 */
var SplitTransaction = 'http://www.w3.org/2000/10/swap/pim/qif#SplitTransaction';
/**
 * supporting document
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/qif#SupportingDocument
 */
var SupportingDocument = 'http://www.w3.org/2000/10/swap/pim/qif#SupportingDocument';
/**
 * transaction
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/qif#Transaction
 */
var Transaction = 'http://www.w3.org/2000/10/swap/pim/qif#Transaction';
/**
 * unclassified
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/qif#Unclassified
 */
var Unclassified = 'http://www.w3.org/2000/10/swap/pim/qif#Unclassified';
/**
 * unclassified income
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/qif#UnclassifiedIncome
 */
var UnclassifiedIncome = 'http://www.w3.org/2000/10/swap/pim/qif#UnclassifiedIncome';
/**
 * unclassified outgoing
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/qif#UnclassifiedOutgoing
 */
var UnclassifiedOutgoing = 'http://www.w3.org/2000/10/swap/pim/qif#UnclassifiedOutgoing';
/**
 * statement
 *
 * The document which is the source of data for this transaction.
 *
 * http://www.w3.org/2000/10/swap/pim/qif#accordingTo
 */
var accordingTo = 'http://www.w3.org/2000/10/swap/pim/qif#accordingTo';
/**
 * amount
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/qif#amount
 */
var amount = 'http://www.w3.org/2000/10/swap/pim/qif#amount';
/**
 * category
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/qif#category
 */
var category = 'http://www.w3.org/2000/10/swap/pim/qif#category';
/**
 * date
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/qif#date
 */
var date = 'http://www.w3.org/2000/10/swap/pim/qif#date';
/**
 * payee
 *
 *
 *
 * http://www.w3.org/2000/10/swap/pim/qif#payee
 */
var payee = 'http://www.w3.org/2000/10/swap/pim/qif#payee';
/**
 * part
 *
 * A transaction representing a part of this transaction.
 *     The sum of the amounts of the parts must equal the amount of this transaction.
 *
 * http://www.w3.org/2000/10/swap/pim/qif#splitPart
 */
var splitPart = 'http://www.w3.org/2000/10/swap/pim/qif#splitPart';
/**
 * part
 *
 * A reverse transaction, putting this transaction back in the Split account, for redistribution as parts.
 *     The amount of the reverse transaction must be the negation of this transaction.
 *
 * http://www.w3.org/2000/10/swap/pim/qif#splitReverse
 */
var splitReverse = 'http://www.w3.org/2000/10/swap/pim/qif#splitReverse';
/**
 * account
 *
 * The account to which money flows when the amount is positive.
 *
 * http://www.w3.org/2000/10/swap/pim/qif#toAccount
 */
var toAccount = 'http://www.w3.org/2000/10/swap/pim/qif#toAccount';

var quImport = /*#__PURE__*/Object.freeze({
	Account: Account,
	Bank: Bank,
	Cat: Cat,
	Classified: Classified,
	Receipt: Receipt,
	SplitTransaction: SplitTransaction,
	SupportingDocument: SupportingDocument,
	Transaction: Transaction,
	Unclassified: Unclassified,
	UnclassifiedIncome: UnclassifiedIncome,
	UnclassifiedOutgoing: UnclassifiedOutgoing,
	accordingTo: accordingTo,
	amount: amount,
	category: category,
	date: date,
	payee: payee,
	splitPart: splitPart,
	splitReverse: splitReverse,
	toAccount: toAccount
});

export { Account as A, Bank as B, Cat as C, Receipt as R, SplitTransaction as S, Transaction as T, Unclassified as U, Classified as a, SupportingDocument as b, UnclassifiedIncome as c, UnclassifiedOutgoing as d, accordingTo as e, amount as f, category as g, date as h, splitReverse as i, payee as p, quImport as q, splitPart as s, toAccount as t };
