// rdf-namespaces/dist/schema.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var schema$1 = require('./schema-d523d8f2.js');



exports.AMRadioChannel = schema$1.AMRadioChannel;
exports.APIReference = schema$1.APIReference;
exports.AboutPage = schema$1.AboutPage;
exports.AcceptAction = schema$1.AcceptAction;
exports.Accommodation = schema$1.Accommodation;
exports.AccountingService = schema$1.AccountingService;
exports.AchieveAction = schema$1.AchieveAction;
exports.Action = schema$1.Action;
exports.ActionAccessSpecification = schema$1.ActionAccessSpecification;
exports.ActionStatusType = schema$1.ActionStatusType;
exports.ActivateAction = schema$1.ActivateAction;
exports.AddAction = schema$1.AddAction;
exports.AdministrativeArea = schema$1.AdministrativeArea;
exports.AdultEntertainment = schema$1.AdultEntertainment;
exports.AdultOrientedEnumeration = schema$1.AdultOrientedEnumeration;
exports.AdvertiserContentArticle = schema$1.AdvertiserContentArticle;
exports.AggregateOffer = schema$1.AggregateOffer;
exports.AggregateRating = schema$1.AggregateRating;
exports.AgreeAction = schema$1.AgreeAction;
exports.Airline = schema$1.Airline;
exports.Airport = schema$1.Airport;
exports.AlignmentObject = schema$1.AlignmentObject;
exports.AllocateAction = schema$1.AllocateAction;
exports.AmpStory = schema$1.AmpStory;
exports.AmusementPark = schema$1.AmusementPark;
exports.AnalysisNewsArticle = schema$1.AnalysisNewsArticle;
exports.AnatomicalStructure = schema$1.AnatomicalStructure;
exports.AnatomicalSystem = schema$1.AnatomicalSystem;
exports.AnimalShelter = schema$1.AnimalShelter;
exports.Answer = schema$1.Answer;
exports.Apartment = schema$1.Apartment;
exports.ApartmentComplex = schema$1.ApartmentComplex;
exports.AppendAction = schema$1.AppendAction;
exports.ApplyAction = schema$1.ApplyAction;
exports.ApprovedIndication = schema$1.ApprovedIndication;
exports.Aquarium = schema$1.Aquarium;
exports.ArchiveComponent = schema$1.ArchiveComponent;
exports.ArchiveOrganization = schema$1.ArchiveOrganization;
exports.ArriveAction = schema$1.ArriveAction;
exports.ArtGallery = schema$1.ArtGallery;
exports.Artery = schema$1.Artery;
exports.Article = schema$1.Article;
exports.AskAction = schema$1.AskAction;
exports.AskPublicNewsArticle = schema$1.AskPublicNewsArticle;
exports.AssessAction = schema$1.AssessAction;
exports.AssignAction = schema$1.AssignAction;
exports.Atlas = schema$1.Atlas;
exports.Attorney = schema$1.Attorney;
exports.Audience = schema$1.Audience;
exports.AudioObject = schema$1.AudioObject;
exports.AudioObjectSnapshot = schema$1.AudioObjectSnapshot;
exports.Audiobook = schema$1.Audiobook;
exports.AuthorizeAction = schema$1.AuthorizeAction;
exports.AutoBodyShop = schema$1.AutoBodyShop;
exports.AutoDealer = schema$1.AutoDealer;
exports.AutoPartsStore = schema$1.AutoPartsStore;
exports.AutoRental = schema$1.AutoRental;
exports.AutoRepair = schema$1.AutoRepair;
exports.AutoWash = schema$1.AutoWash;
exports.AutomatedTeller = schema$1.AutomatedTeller;
exports.AutomotiveBusiness = schema$1.AutomotiveBusiness;
exports.BackgroundNewsArticle = schema$1.BackgroundNewsArticle;
exports.Bakery = schema$1.Bakery;
exports.BankAccount = schema$1.BankAccount;
exports.BankOrCreditUnion = schema$1.BankOrCreditUnion;
exports.BarOrPub = schema$1.BarOrPub;
exports.Barcode = schema$1.Barcode;
exports.Beach = schema$1.Beach;
exports.BeautySalon = schema$1.BeautySalon;
exports.BedAndBreakfast = schema$1.BedAndBreakfast;
exports.BedDetails = schema$1.BedDetails;
exports.BedType = schema$1.BedType;
exports.BefriendAction = schema$1.BefriendAction;
exports.BikeStore = schema$1.BikeStore;
exports.BioChemEntity = schema$1.BioChemEntity;
exports.Blog = schema$1.Blog;
exports.BlogPosting = schema$1.BlogPosting;
exports.BloodTest = schema$1.BloodTest;
exports.BoardingPolicyType = schema$1.BoardingPolicyType;
exports.BoatReservation = schema$1.BoatReservation;
exports.BoatTerminal = schema$1.BoatTerminal;
exports.BoatTrip = schema$1.BoatTrip;
exports.BodyMeasurementTypeEnumeration = schema$1.BodyMeasurementTypeEnumeration;
exports.BodyOfWater = schema$1.BodyOfWater;
exports.Bone = schema$1.Bone;
exports.Book = schema$1.Book;
exports.BookFormatType = schema$1.BookFormatType;
exports.BookSeries = schema$1.BookSeries;
exports.BookStore = schema$1.BookStore;
exports.BookmarkAction = schema$1.BookmarkAction;
exports.Boolean__workaround = schema$1.Boolean__workaround;
exports.BorrowAction = schema$1.BorrowAction;
exports.BowlingAlley = schema$1.BowlingAlley;
exports.BrainStructure = schema$1.BrainStructure;
exports.Brand = schema$1.Brand;
exports.BreadcrumbList = schema$1.BreadcrumbList;
exports.Brewery = schema$1.Brewery;
exports.Bridge = schema$1.Bridge;
exports.BroadcastChannel = schema$1.BroadcastChannel;
exports.BroadcastEvent = schema$1.BroadcastEvent;
exports.BroadcastFrequencySpecification = schema$1.BroadcastFrequencySpecification;
exports.BroadcastService = schema$1.BroadcastService;
exports.BrokerageAccount = schema$1.BrokerageAccount;
exports.BuddhistTemple = schema$1.BuddhistTemple;
exports.BusOrCoach = schema$1.BusOrCoach;
exports.BusReservation = schema$1.BusReservation;
exports.BusStation = schema$1.BusStation;
exports.BusStop = schema$1.BusStop;
exports.BusTrip = schema$1.BusTrip;
exports.BusinessAudience = schema$1.BusinessAudience;
exports.BusinessEntityType = schema$1.BusinessEntityType;
exports.BusinessEvent = schema$1.BusinessEvent;
exports.BusinessFunction = schema$1.BusinessFunction;
exports.BuyAction = schema$1.BuyAction;
exports.CDCPMDRecord = schema$1.CDCPMDRecord;
exports.CableOrSatelliteService = schema$1.CableOrSatelliteService;
exports.CafeOrCoffeeShop = schema$1.CafeOrCoffeeShop;
exports.Campground = schema$1.Campground;
exports.CampingPitch = schema$1.CampingPitch;
exports.Canal = schema$1.Canal;
exports.CancelAction = schema$1.CancelAction;
exports.Car = schema$1.Car;
exports.CarUsageType = schema$1.CarUsageType;
exports.Casino = schema$1.Casino;
exports.CategoryCode = schema$1.CategoryCode;
exports.CategoryCodeSet = schema$1.CategoryCodeSet;
exports.CatholicChurch = schema$1.CatholicChurch;
exports.Cemetery = schema$1.Cemetery;
exports.Chapter = schema$1.Chapter;
exports.CheckAction = schema$1.CheckAction;
exports.CheckInAction = schema$1.CheckInAction;
exports.CheckOutAction = schema$1.CheckOutAction;
exports.CheckoutPage = schema$1.CheckoutPage;
exports.ChemicalSubstance = schema$1.ChemicalSubstance;
exports.ChildCare = schema$1.ChildCare;
exports.ChildrensEvent = schema$1.ChildrensEvent;
exports.ChooseAction = schema$1.ChooseAction;
exports.Church = schema$1.Church;
exports.City = schema$1.City;
exports.CityHall = schema$1.CityHall;
exports.CivicStructure = schema$1.CivicStructure;
exports.Claim = schema$1.Claim;
exports.ClaimReview = schema$1.ClaimReview;
exports.Class = schema$1.Class;
exports.Clip = schema$1.Clip;
exports.ClothingStore = schema$1.ClothingStore;
exports.Code = schema$1.Code;
exports.Collection = schema$1.Collection;
exports.CollectionPage = schema$1.CollectionPage;
exports.CollegeOrUniversity = schema$1.CollegeOrUniversity;
exports.ComedyClub = schema$1.ComedyClub;
exports.ComedyEvent = schema$1.ComedyEvent;
exports.ComicCoverArt = schema$1.ComicCoverArt;
exports.ComicIssue = schema$1.ComicIssue;
exports.ComicSeries = schema$1.ComicSeries;
exports.ComicStory = schema$1.ComicStory;
exports.Comment = schema$1.Comment;
exports.CommentAction = schema$1.CommentAction;
exports.CommunicateAction = schema$1.CommunicateAction;
exports.CompleteDataFeed = schema$1.CompleteDataFeed;
exports.CompoundPriceSpecification = schema$1.CompoundPriceSpecification;
exports.ComputerLanguage = schema$1.ComputerLanguage;
exports.ComputerStore = schema$1.ComputerStore;
exports.ConfirmAction = schema$1.ConfirmAction;
exports.Consortium = schema$1.Consortium;
exports.ConstraintNode = schema$1.ConstraintNode;
exports.ConsumeAction = schema$1.ConsumeAction;
exports.ContactPage = schema$1.ContactPage;
exports.ContactPoint = schema$1.ContactPoint;
exports.ContactPointOption = schema$1.ContactPointOption;
exports.Continent = schema$1.Continent;
exports.ControlAction = schema$1.ControlAction;
exports.ConvenienceStore = schema$1.ConvenienceStore;
exports.Conversation = schema$1.Conversation;
exports.CookAction = schema$1.CookAction;
exports.Corporation = schema$1.Corporation;
exports.CorrectionComment = schema$1.CorrectionComment;
exports.Country = schema$1.Country;
exports.Course = schema$1.Course;
exports.CourseInstance = schema$1.CourseInstance;
exports.Courthouse = schema$1.Courthouse;
exports.CoverArt = schema$1.CoverArt;
exports.CovidTestingFacility = schema$1.CovidTestingFacility;
exports.CreateAction = schema$1.CreateAction;
exports.CreativeWork = schema$1.CreativeWork;
exports.CreativeWorkSeason = schema$1.CreativeWorkSeason;
exports.CreativeWorkSeries = schema$1.CreativeWorkSeries;
exports.CreditCard = schema$1.CreditCard;
exports.Crematorium = schema$1.Crematorium;
exports.CriticReview = schema$1.CriticReview;
exports.CssSelectorType = schema$1.CssSelectorType;
exports.CurrencyConversionService = schema$1.CurrencyConversionService;
exports.DDxElement = schema$1.DDxElement;
exports.DanceEvent = schema$1.DanceEvent;
exports.DanceGroup = schema$1.DanceGroup;
exports.DataCatalog = schema$1.DataCatalog;
exports.DataDownload = schema$1.DataDownload;
exports.DataFeed = schema$1.DataFeed;
exports.DataFeedItem = schema$1.DataFeedItem;
exports.DataType = schema$1.DataType;
exports.Dataset = schema$1.Dataset;
exports.DateTime = schema$1.DateTime;
exports.Date__workaround = schema$1.Date__workaround;
exports.DatedMoneySpecification = schema$1.DatedMoneySpecification;
exports.DayOfWeek = schema$1.DayOfWeek;
exports.DaySpa = schema$1.DaySpa;
exports.DeactivateAction = schema$1.DeactivateAction;
exports.DefenceEstablishment = schema$1.DefenceEstablishment;
exports.DefinedRegion = schema$1.DefinedRegion;
exports.DefinedTerm = schema$1.DefinedTerm;
exports.DefinedTermSet = schema$1.DefinedTermSet;
exports.DeleteAction = schema$1.DeleteAction;
exports.DeliveryChargeSpecification = schema$1.DeliveryChargeSpecification;
exports.DeliveryEvent = schema$1.DeliveryEvent;
exports.DeliveryMethod = schema$1.DeliveryMethod;
exports.DeliveryTimeSettings = schema$1.DeliveryTimeSettings;
exports.Demand = schema$1.Demand;
exports.Dentist = schema$1.Dentist;
exports.DepartAction = schema$1.DepartAction;
exports.DepartmentStore = schema$1.DepartmentStore;
exports.DepositAccount = schema$1.DepositAccount;
exports.DiagnosticLab = schema$1.DiagnosticLab;
exports.DiagnosticProcedure = schema$1.DiagnosticProcedure;
exports.Diet = schema$1.Diet;
exports.DietarySupplement = schema$1.DietarySupplement;
exports.DigitalDocument = schema$1.DigitalDocument;
exports.DigitalDocumentPermission = schema$1.DigitalDocumentPermission;
exports.DigitalDocumentPermissionType = schema$1.DigitalDocumentPermissionType;
exports.DigitalPlatformEnumeration = schema$1.DigitalPlatformEnumeration;
exports.DisagreeAction = schema$1.DisagreeAction;
exports.DiscoverAction = schema$1.DiscoverAction;
exports.DiscussionForumPosting = schema$1.DiscussionForumPosting;
exports.DislikeAction = schema$1.DislikeAction;
exports.Distance = schema$1.Distance;
exports.Distillery = schema$1.Distillery;
exports.DonateAction = schema$1.DonateAction;
exports.DoseSchedule = schema$1.DoseSchedule;
exports.DownloadAction = schema$1.DownloadAction;
exports.DrawAction = schema$1.DrawAction;
exports.Drawing = schema$1.Drawing;
exports.DrinkAction = schema$1.DrinkAction;
exports.DriveWheelConfigurationValue = schema$1.DriveWheelConfigurationValue;
exports.Drug = schema$1.Drug;
exports.DrugClass = schema$1.DrugClass;
exports.DrugCost = schema$1.DrugCost;
exports.DrugCostCategory = schema$1.DrugCostCategory;
exports.DrugLegalStatus = schema$1.DrugLegalStatus;
exports.DrugPregnancyCategory = schema$1.DrugPregnancyCategory;
exports.DrugPrescriptionStatus = schema$1.DrugPrescriptionStatus;
exports.DrugStrength = schema$1.DrugStrength;
exports.DryCleaningOrLaundry = schema$1.DryCleaningOrLaundry;
exports.Duration = schema$1.Duration;
exports.EUEnergyEfficiencyEnumeration = schema$1.EUEnergyEfficiencyEnumeration;
exports.EatAction = schema$1.EatAction;
exports.EducationEvent = schema$1.EducationEvent;
exports.EducationalAudience = schema$1.EducationalAudience;
exports.EducationalOccupationalCredential = schema$1.EducationalOccupationalCredential;
exports.EducationalOccupationalProgram = schema$1.EducationalOccupationalProgram;
exports.EducationalOrganization = schema$1.EducationalOrganization;
exports.Electrician = schema$1.Electrician;
exports.ElectronicsStore = schema$1.ElectronicsStore;
exports.ElementarySchool = schema$1.ElementarySchool;
exports.EmailMessage = schema$1.EmailMessage;
exports.Embassy = schema$1.Embassy;
exports.EmergencyService = schema$1.EmergencyService;
exports.EmployeeRole = schema$1.EmployeeRole;
exports.EmployerAggregateRating = schema$1.EmployerAggregateRating;
exports.EmployerReview = schema$1.EmployerReview;
exports.EmploymentAgency = schema$1.EmploymentAgency;
exports.EndorseAction = schema$1.EndorseAction;
exports.EndorsementRating = schema$1.EndorsementRating;
exports.Energy = schema$1.Energy;
exports.EnergyConsumptionDetails = schema$1.EnergyConsumptionDetails;
exports.EnergyEfficiencyEnumeration = schema$1.EnergyEfficiencyEnumeration;
exports.EnergyStarEnergyEfficiencyEnumeration = schema$1.EnergyStarEnergyEfficiencyEnumeration;
exports.EngineSpecification = schema$1.EngineSpecification;
exports.EntertainmentBusiness = schema$1.EntertainmentBusiness;
exports.EntryPoint = schema$1.EntryPoint;
exports.Enumeration = schema$1.Enumeration;
exports.Episode = schema$1.Episode;
exports.Event = schema$1.Event;
exports.EventAttendanceModeEnumeration = schema$1.EventAttendanceModeEnumeration;
exports.EventReservation = schema$1.EventReservation;
exports.EventSeries = schema$1.EventSeries;
exports.EventStatusType = schema$1.EventStatusType;
exports.EventVenue = schema$1.EventVenue;
exports.ExchangeRateSpecification = schema$1.ExchangeRateSpecification;
exports.ExerciseAction = schema$1.ExerciseAction;
exports.ExerciseGym = schema$1.ExerciseGym;
exports.ExercisePlan = schema$1.ExercisePlan;
exports.ExhibitionEvent = schema$1.ExhibitionEvent;
exports.FAQPage = schema$1.FAQPage;
exports.FMRadioChannel = schema$1.FMRadioChannel;
exports.FastFoodRestaurant = schema$1.FastFoodRestaurant;
exports.Festival = schema$1.Festival;
exports.FilmAction = schema$1.FilmAction;
exports.FinancialProduct = schema$1.FinancialProduct;
exports.FinancialService = schema$1.FinancialService;
exports.FindAction = schema$1.FindAction;
exports.FireStation = schema$1.FireStation;
exports.Flight = schema$1.Flight;
exports.FlightReservation = schema$1.FlightReservation;
exports.Float = schema$1.Float;
exports.FloorPlan = schema$1.FloorPlan;
exports.Florist = schema$1.Florist;
exports.FollowAction = schema$1.FollowAction;
exports.FoodEstablishment = schema$1.FoodEstablishment;
exports.FoodEstablishmentReservation = schema$1.FoodEstablishmentReservation;
exports.FoodEvent = schema$1.FoodEvent;
exports.FoodService = schema$1.FoodService;
exports.FundingAgency = schema$1.FundingAgency;
exports.FundingScheme = schema$1.FundingScheme;
exports.FurnitureStore = schema$1.FurnitureStore;
exports.Game = schema$1.Game;
exports.GameAvailabilityEnumeration = schema$1.GameAvailabilityEnumeration;
exports.GamePlayMode = schema$1.GamePlayMode;
exports.GameServer = schema$1.GameServer;
exports.GameServerStatus = schema$1.GameServerStatus;
exports.GardenStore = schema$1.GardenStore;
exports.GasStation = schema$1.GasStation;
exports.GatedResidenceCommunity = schema$1.GatedResidenceCommunity;
exports.GenderType = schema$1.GenderType;
exports.Gene = schema$1.Gene;
exports.GeneralContractor = schema$1.GeneralContractor;
exports.GeoCircle = schema$1.GeoCircle;
exports.GeoCoordinates = schema$1.GeoCoordinates;
exports.GeoShape = schema$1.GeoShape;
exports.GeospatialGeometry = schema$1.GeospatialGeometry;
exports.GiveAction = schema$1.GiveAction;
exports.GolfCourse = schema$1.GolfCourse;
exports.GovernmentBenefitsType = schema$1.GovernmentBenefitsType;
exports.GovernmentBuilding = schema$1.GovernmentBuilding;
exports.GovernmentOffice = schema$1.GovernmentOffice;
exports.GovernmentOrganization = schema$1.GovernmentOrganization;
exports.GovernmentPermit = schema$1.GovernmentPermit;
exports.GovernmentService = schema$1.GovernmentService;
exports.Grant = schema$1.Grant;
exports.GroceryStore = schema$1.GroceryStore;
exports.Guide = schema$1.Guide;
exports.HVACBusiness = schema$1.HVACBusiness;
exports.Hackathon = schema$1.Hackathon;
exports.HairSalon = schema$1.HairSalon;
exports.HardwareStore = schema$1.HardwareStore;
exports.HealthAndBeautyBusiness = schema$1.HealthAndBeautyBusiness;
exports.HealthAspectEnumeration = schema$1.HealthAspectEnumeration;
exports.HealthClub = schema$1.HealthClub;
exports.HealthInsurancePlan = schema$1.HealthInsurancePlan;
exports.HealthPlanCostSharingSpecification = schema$1.HealthPlanCostSharingSpecification;
exports.HealthPlanFormulary = schema$1.HealthPlanFormulary;
exports.HealthPlanNetwork = schema$1.HealthPlanNetwork;
exports.HealthTopicContent = schema$1.HealthTopicContent;
exports.HighSchool = schema$1.HighSchool;
exports.HinduTemple = schema$1.HinduTemple;
exports.HobbyShop = schema$1.HobbyShop;
exports.HomeAndConstructionBusiness = schema$1.HomeAndConstructionBusiness;
exports.HomeGoodsStore = schema$1.HomeGoodsStore;
exports.Hospital = schema$1.Hospital;
exports.Hostel = schema$1.Hostel;
exports.Hotel = schema$1.Hotel;
exports.HotelRoom = schema$1.HotelRoom;
exports.House = schema$1.House;
exports.HousePainter = schema$1.HousePainter;
exports.HowTo = schema$1.HowTo;
exports.HowToDirection = schema$1.HowToDirection;
exports.HowToItem = schema$1.HowToItem;
exports.HowToSection = schema$1.HowToSection;
exports.HowToStep = schema$1.HowToStep;
exports.HowToSupply = schema$1.HowToSupply;
exports.HowToTip = schema$1.HowToTip;
exports.HowToTool = schema$1.HowToTool;
exports.HyperToc = schema$1.HyperToc;
exports.HyperTocEntry = schema$1.HyperTocEntry;
exports.IceCreamShop = schema$1.IceCreamShop;
exports.IgnoreAction = schema$1.IgnoreAction;
exports.ImageGallery = schema$1.ImageGallery;
exports.ImageObject = schema$1.ImageObject;
exports.ImageObjectSnapshot = schema$1.ImageObjectSnapshot;
exports.ImagingTest = schema$1.ImagingTest;
exports.IndividualProduct = schema$1.IndividualProduct;
exports.InfectiousAgentClass = schema$1.InfectiousAgentClass;
exports.InfectiousDisease = schema$1.InfectiousDisease;
exports.InformAction = schema$1.InformAction;
exports.InsertAction = schema$1.InsertAction;
exports.InstallAction = schema$1.InstallAction;
exports.InsuranceAgency = schema$1.InsuranceAgency;
exports.Intangible = schema$1.Intangible;
exports.Integer = schema$1.Integer;
exports.InteractAction = schema$1.InteractAction;
exports.InteractionCounter = schema$1.InteractionCounter;
exports.InternetCafe = schema$1.InternetCafe;
exports.InvestmentFund = schema$1.InvestmentFund;
exports.InvestmentOrDeposit = schema$1.InvestmentOrDeposit;
exports.InviteAction = schema$1.InviteAction;
exports.Invoice = schema$1.Invoice;
exports.ItemAvailability = schema$1.ItemAvailability;
exports.ItemList = schema$1.ItemList;
exports.ItemListOrderType = schema$1.ItemListOrderType;
exports.ItemPage = schema$1.ItemPage;
exports.JewelryStore = schema$1.JewelryStore;
exports.JobPosting = schema$1.JobPosting;
exports.JoinAction = schema$1.JoinAction;
exports.Joint = schema$1.Joint;
exports.LakeBodyOfWater = schema$1.LakeBodyOfWater;
exports.Landform = schema$1.Landform;
exports.LandmarksOrHistoricalBuildings = schema$1.LandmarksOrHistoricalBuildings;
exports.Language = schema$1.Language;
exports.LearningResource = schema$1.LearningResource;
exports.LeaveAction = schema$1.LeaveAction;
exports.LegalForceStatus = schema$1.LegalForceStatus;
exports.LegalService = schema$1.LegalService;
exports.LegalValueLevel = schema$1.LegalValueLevel;
exports.Legislation = schema$1.Legislation;
exports.LegislationObject = schema$1.LegislationObject;
exports.LegislativeBuilding = schema$1.LegislativeBuilding;
exports.LendAction = schema$1.LendAction;
exports.Library = schema$1.Library;
exports.LibrarySystem = schema$1.LibrarySystem;
exports.LifestyleModification = schema$1.LifestyleModification;
exports.Ligament = schema$1.Ligament;
exports.LikeAction = schema$1.LikeAction;
exports.LinkRole = schema$1.LinkRole;
exports.LiquorStore = schema$1.LiquorStore;
exports.ListItem = schema$1.ListItem;
exports.ListenAction = schema$1.ListenAction;
exports.LiteraryEvent = schema$1.LiteraryEvent;
exports.LiveBlogPosting = schema$1.LiveBlogPosting;
exports.LoanOrCredit = schema$1.LoanOrCredit;
exports.LocalBusiness = schema$1.LocalBusiness;
exports.LocationFeatureSpecification = schema$1.LocationFeatureSpecification;
exports.Locksmith = schema$1.Locksmith;
exports.LodgingBusiness = schema$1.LodgingBusiness;
exports.LodgingReservation = schema$1.LodgingReservation;
exports.LoseAction = schema$1.LoseAction;
exports.LymphaticVessel = schema$1.LymphaticVessel;
exports.Manuscript = schema$1.Manuscript;
exports.MapCategoryType = schema$1.MapCategoryType;
exports.Map__workaround = schema$1.Map__workaround;
exports.MarryAction = schema$1.MarryAction;
exports.Mass = schema$1.Mass;
exports.MathSolver = schema$1.MathSolver;
exports.MaximumDoseSchedule = schema$1.MaximumDoseSchedule;
exports.MeasurementMethodEnum = schema$1.MeasurementMethodEnum;
exports.MeasurementTypeEnumeration = schema$1.MeasurementTypeEnumeration;
exports.MediaGallery = schema$1.MediaGallery;
exports.MediaManipulationRatingEnumeration = schema$1.MediaManipulationRatingEnumeration;
exports.MediaObject = schema$1.MediaObject;
exports.MediaReview = schema$1.MediaReview;
exports.MediaReviewItem = schema$1.MediaReviewItem;
exports.MediaSubscription = schema$1.MediaSubscription;
exports.MedicalAudience = schema$1.MedicalAudience;
exports.MedicalAudienceType = schema$1.MedicalAudienceType;
exports.MedicalBusiness = schema$1.MedicalBusiness;
exports.MedicalCause = schema$1.MedicalCause;
exports.MedicalClinic = schema$1.MedicalClinic;
exports.MedicalCode = schema$1.MedicalCode;
exports.MedicalCondition = schema$1.MedicalCondition;
exports.MedicalConditionStage = schema$1.MedicalConditionStage;
exports.MedicalContraindication = schema$1.MedicalContraindication;
exports.MedicalDevice = schema$1.MedicalDevice;
exports.MedicalDevicePurpose = schema$1.MedicalDevicePurpose;
exports.MedicalEntity = schema$1.MedicalEntity;
exports.MedicalEnumeration = schema$1.MedicalEnumeration;
exports.MedicalEvidenceLevel = schema$1.MedicalEvidenceLevel;
exports.MedicalGuideline = schema$1.MedicalGuideline;
exports.MedicalGuidelineContraindication = schema$1.MedicalGuidelineContraindication;
exports.MedicalGuidelineRecommendation = schema$1.MedicalGuidelineRecommendation;
exports.MedicalImagingTechnique = schema$1.MedicalImagingTechnique;
exports.MedicalIndication = schema$1.MedicalIndication;
exports.MedicalIntangible = schema$1.MedicalIntangible;
exports.MedicalObservationalStudy = schema$1.MedicalObservationalStudy;
exports.MedicalObservationalStudyDesign = schema$1.MedicalObservationalStudyDesign;
exports.MedicalOrganization = schema$1.MedicalOrganization;
exports.MedicalProcedure = schema$1.MedicalProcedure;
exports.MedicalProcedureType = schema$1.MedicalProcedureType;
exports.MedicalRiskCalculator = schema$1.MedicalRiskCalculator;
exports.MedicalRiskEstimator = schema$1.MedicalRiskEstimator;
exports.MedicalRiskFactor = schema$1.MedicalRiskFactor;
exports.MedicalRiskScore = schema$1.MedicalRiskScore;
exports.MedicalScholarlyArticle = schema$1.MedicalScholarlyArticle;
exports.MedicalSign = schema$1.MedicalSign;
exports.MedicalSignOrSymptom = schema$1.MedicalSignOrSymptom;
exports.MedicalSpecialty = schema$1.MedicalSpecialty;
exports.MedicalStudy = schema$1.MedicalStudy;
exports.MedicalStudyStatus = schema$1.MedicalStudyStatus;
exports.MedicalSymptom = schema$1.MedicalSymptom;
exports.MedicalTest = schema$1.MedicalTest;
exports.MedicalTestPanel = schema$1.MedicalTestPanel;
exports.MedicalTherapy = schema$1.MedicalTherapy;
exports.MedicalTrial = schema$1.MedicalTrial;
exports.MedicalTrialDesign = schema$1.MedicalTrialDesign;
exports.MedicalWebPage = schema$1.MedicalWebPage;
exports.MedicineSystem = schema$1.MedicineSystem;
exports.MeetingRoom = schema$1.MeetingRoom;
exports.MensClothingStore = schema$1.MensClothingStore;
exports.Menu = schema$1.Menu;
exports.MenuItem = schema$1.MenuItem;
exports.MenuSection = schema$1.MenuSection;
exports.MerchantReturnEnumeration = schema$1.MerchantReturnEnumeration;
exports.MerchantReturnPolicy = schema$1.MerchantReturnPolicy;
exports.MerchantReturnPolicySeasonalOverride = schema$1.MerchantReturnPolicySeasonalOverride;
exports.Message = schema$1.Message;
exports.MiddleSchool = schema$1.MiddleSchool;
exports.MobileApplication = schema$1.MobileApplication;
exports.MobilePhoneStore = schema$1.MobilePhoneStore;
exports.MolecularEntity = schema$1.MolecularEntity;
exports.MonetaryAmount = schema$1.MonetaryAmount;
exports.MonetaryAmountDistribution = schema$1.MonetaryAmountDistribution;
exports.MonetaryGrant = schema$1.MonetaryGrant;
exports.MoneyTransfer = schema$1.MoneyTransfer;
exports.MortgageLoan = schema$1.MortgageLoan;
exports.Mosque = schema$1.Mosque;
exports.Motel = schema$1.Motel;
exports.Motorcycle = schema$1.Motorcycle;
exports.MotorcycleDealer = schema$1.MotorcycleDealer;
exports.MotorcycleRepair = schema$1.MotorcycleRepair;
exports.MotorizedBicycle = schema$1.MotorizedBicycle;
exports.Mountain = schema$1.Mountain;
exports.MoveAction = schema$1.MoveAction;
exports.Movie = schema$1.Movie;
exports.MovieClip = schema$1.MovieClip;
exports.MovieRentalStore = schema$1.MovieRentalStore;
exports.MovieSeries = schema$1.MovieSeries;
exports.MovieTheater = schema$1.MovieTheater;
exports.MovingCompany = schema$1.MovingCompany;
exports.Muscle = schema$1.Muscle;
exports.Museum = schema$1.Museum;
exports.MusicAlbum = schema$1.MusicAlbum;
exports.MusicAlbumProductionType = schema$1.MusicAlbumProductionType;
exports.MusicAlbumReleaseType = schema$1.MusicAlbumReleaseType;
exports.MusicComposition = schema$1.MusicComposition;
exports.MusicEvent = schema$1.MusicEvent;
exports.MusicGroup = schema$1.MusicGroup;
exports.MusicPlaylist = schema$1.MusicPlaylist;
exports.MusicRecording = schema$1.MusicRecording;
exports.MusicRelease = schema$1.MusicRelease;
exports.MusicReleaseFormatType = schema$1.MusicReleaseFormatType;
exports.MusicStore = schema$1.MusicStore;
exports.MusicVenue = schema$1.MusicVenue;
exports.MusicVideoObject = schema$1.MusicVideoObject;
exports.NGO = schema$1.NGO;
exports.NLNonprofitType = schema$1.NLNonprofitType;
exports.NailSalon = schema$1.NailSalon;
exports.Nerve = schema$1.Nerve;
exports.NewsArticle = schema$1.NewsArticle;
exports.NewsMediaOrganization = schema$1.NewsMediaOrganization;
exports.Newspaper = schema$1.Newspaper;
exports.NightClub = schema$1.NightClub;
exports.NonprofitType = schema$1.NonprofitType;
exports.Notary = schema$1.Notary;
exports.NoteDigitalDocument = schema$1.NoteDigitalDocument;
exports.Number__workaround = schema$1.Number__workaround;
exports.NutritionInformation = schema$1.NutritionInformation;
exports.Observation = schema$1.Observation;
exports.Occupation = schema$1.Occupation;
exports.OccupationalExperienceRequirements = schema$1.OccupationalExperienceRequirements;
exports.OccupationalTherapy = schema$1.OccupationalTherapy;
exports.OceanBodyOfWater = schema$1.OceanBodyOfWater;
exports.Offer = schema$1.Offer;
exports.OfferCatalog = schema$1.OfferCatalog;
exports.OfferForLease = schema$1.OfferForLease;
exports.OfferForPurchase = schema$1.OfferForPurchase;
exports.OfferItemCondition = schema$1.OfferItemCondition;
exports.OfferShippingDetails = schema$1.OfferShippingDetails;
exports.OfficeEquipmentStore = schema$1.OfficeEquipmentStore;
exports.OnDemandEvent = schema$1.OnDemandEvent;
exports.OnlineBusiness = schema$1.OnlineBusiness;
exports.OnlineStore = schema$1.OnlineStore;
exports.OpeningHoursSpecification = schema$1.OpeningHoursSpecification;
exports.OpinionNewsArticle = schema$1.OpinionNewsArticle;
exports.Optician = schema$1.Optician;
exports.Order = schema$1.Order;
exports.OrderAction = schema$1.OrderAction;
exports.OrderItem = schema$1.OrderItem;
exports.OrderStatus = schema$1.OrderStatus;
exports.Organization = schema$1.Organization;
exports.OrganizationRole = schema$1.OrganizationRole;
exports.OrganizeAction = schema$1.OrganizeAction;
exports.OutletStore = schema$1.OutletStore;
exports.OwnershipInfo = schema$1.OwnershipInfo;
exports.PaintAction = schema$1.PaintAction;
exports.Painting = schema$1.Painting;
exports.PalliativeProcedure = schema$1.PalliativeProcedure;
exports.ParcelDelivery = schema$1.ParcelDelivery;
exports.ParentAudience = schema$1.ParentAudience;
exports.Park = schema$1.Park;
exports.ParkingFacility = schema$1.ParkingFacility;
exports.PathologyTest = schema$1.PathologyTest;
exports.Patient = schema$1.Patient;
exports.PawnShop = schema$1.PawnShop;
exports.PayAction = schema$1.PayAction;
exports.PaymentCard = schema$1.PaymentCard;
exports.PaymentChargeSpecification = schema$1.PaymentChargeSpecification;
exports.PaymentMethod = schema$1.PaymentMethod;
exports.PaymentService = schema$1.PaymentService;
exports.PaymentStatusType = schema$1.PaymentStatusType;
exports.PeopleAudience = schema$1.PeopleAudience;
exports.PerformAction = schema$1.PerformAction;
exports.PerformanceRole = schema$1.PerformanceRole;
exports.PerformingArtsTheater = schema$1.PerformingArtsTheater;
exports.PerformingGroup = schema$1.PerformingGroup;
exports.Periodical = schema$1.Periodical;
exports.Permit = schema$1.Permit;
exports.Person = schema$1.Person;
exports.PetStore = schema$1.PetStore;
exports.Pharmacy = schema$1.Pharmacy;
exports.Photograph = schema$1.Photograph;
exports.PhotographAction = schema$1.PhotographAction;
exports.PhysicalActivity = schema$1.PhysicalActivity;
exports.PhysicalActivityCategory = schema$1.PhysicalActivityCategory;
exports.PhysicalExam = schema$1.PhysicalExam;
exports.PhysicalTherapy = schema$1.PhysicalTherapy;
exports.Physician = schema$1.Physician;
exports.Place = schema$1.Place;
exports.PlaceOfWorship = schema$1.PlaceOfWorship;
exports.PlanAction = schema$1.PlanAction;
exports.Play = schema$1.Play;
exports.PlayAction = schema$1.PlayAction;
exports.PlayGameAction = schema$1.PlayGameAction;
exports.Playground = schema$1.Playground;
exports.Plumber = schema$1.Plumber;
exports.PodcastEpisode = schema$1.PodcastEpisode;
exports.PodcastSeason = schema$1.PodcastSeason;
exports.PodcastSeries = schema$1.PodcastSeries;
exports.PoliceStation = schema$1.PoliceStation;
exports.PoliticalParty = schema$1.PoliticalParty;
exports.Pond = schema$1.Pond;
exports.PostOffice = schema$1.PostOffice;
exports.PostalAddress = schema$1.PostalAddress;
exports.PostalCodeRangeSpecification = schema$1.PostalCodeRangeSpecification;
exports.Poster = schema$1.Poster;
exports.PreOrderAction = schema$1.PreOrderAction;
exports.PrependAction = schema$1.PrependAction;
exports.Preschool = schema$1.Preschool;
exports.PresentationDigitalDocument = schema$1.PresentationDigitalDocument;
exports.PreventionIndication = schema$1.PreventionIndication;
exports.PriceComponentTypeEnumeration = schema$1.PriceComponentTypeEnumeration;
exports.PriceSpecification = schema$1.PriceSpecification;
exports.PriceTypeEnumeration = schema$1.PriceTypeEnumeration;
exports.Product = schema$1.Product;
exports.ProductCollection = schema$1.ProductCollection;
exports.ProductGroup = schema$1.ProductGroup;
exports.ProductModel = schema$1.ProductModel;
exports.ProductReturnEnumeration = schema$1.ProductReturnEnumeration;
exports.ProductReturnPolicy = schema$1.ProductReturnPolicy;
exports.ProfessionalService = schema$1.ProfessionalService;
exports.ProfilePage = schema$1.ProfilePage;
exports.ProgramMembership = schema$1.ProgramMembership;
exports.Project = schema$1.Project;
exports.PronounceableText = schema$1.PronounceableText;
exports.Property = schema$1.Property;
exports.PropertyValue = schema$1.PropertyValue;
exports.PropertyValueSpecification = schema$1.PropertyValueSpecification;
exports.Protein = schema$1.Protein;
exports.PsychologicalTreatment = schema$1.PsychologicalTreatment;
exports.PublicSwimmingPool = schema$1.PublicSwimmingPool;
exports.PublicToilet = schema$1.PublicToilet;
exports.PublicationEvent = schema$1.PublicationEvent;
exports.PublicationIssue = schema$1.PublicationIssue;
exports.PublicationVolume = schema$1.PublicationVolume;
exports.QAPage = schema$1.QAPage;
exports.QualitativeValue = schema$1.QualitativeValue;
exports.QuantitativeValue = schema$1.QuantitativeValue;
exports.QuantitativeValueDistribution = schema$1.QuantitativeValueDistribution;
exports.Quantity = schema$1.Quantity;
exports.Question = schema$1.Question;
exports.Quiz = schema$1.Quiz;
exports.Quotation = schema$1.Quotation;
exports.QuoteAction = schema$1.QuoteAction;
exports.RVPark = schema$1.RVPark;
exports.RadiationTherapy = schema$1.RadiationTherapy;
exports.RadioBroadcastService = schema$1.RadioBroadcastService;
exports.RadioChannel = schema$1.RadioChannel;
exports.RadioClip = schema$1.RadioClip;
exports.RadioEpisode = schema$1.RadioEpisode;
exports.RadioSeason = schema$1.RadioSeason;
exports.RadioSeries = schema$1.RadioSeries;
exports.RadioStation = schema$1.RadioStation;
exports.Rating = schema$1.Rating;
exports.ReactAction = schema$1.ReactAction;
exports.ReadAction = schema$1.ReadAction;
exports.RealEstateAgent = schema$1.RealEstateAgent;
exports.RealEstateListing = schema$1.RealEstateListing;
exports.ReceiveAction = schema$1.ReceiveAction;
exports.Recipe = schema$1.Recipe;
exports.Recommendation = schema$1.Recommendation;
exports.RecommendedDoseSchedule = schema$1.RecommendedDoseSchedule;
exports.RecyclingCenter = schema$1.RecyclingCenter;
exports.RefundTypeEnumeration = schema$1.RefundTypeEnumeration;
exports.RegisterAction = schema$1.RegisterAction;
exports.RejectAction = schema$1.RejectAction;
exports.RentAction = schema$1.RentAction;
exports.RentalCarReservation = schema$1.RentalCarReservation;
exports.RepaymentSpecification = schema$1.RepaymentSpecification;
exports.ReplaceAction = schema$1.ReplaceAction;
exports.ReplyAction = schema$1.ReplyAction;
exports.Report = schema$1.Report;
exports.ReportageNewsArticle = schema$1.ReportageNewsArticle;
exports.ReportedDoseSchedule = schema$1.ReportedDoseSchedule;
exports.ResearchOrganization = schema$1.ResearchOrganization;
exports.ResearchProject = schema$1.ResearchProject;
exports.Researcher = schema$1.Researcher;
exports.Reservation = schema$1.Reservation;
exports.ReservationPackage = schema$1.ReservationPackage;
exports.ReservationStatusType = schema$1.ReservationStatusType;
exports.ReserveAction = schema$1.ReserveAction;
exports.Reservoir = schema$1.Reservoir;
exports.Residence = schema$1.Residence;
exports.Resort = schema$1.Resort;
exports.Restaurant = schema$1.Restaurant;
exports.RestrictedDiet = schema$1.RestrictedDiet;
exports.ResumeAction = schema$1.ResumeAction;
exports.ReturnAction = schema$1.ReturnAction;
exports.ReturnFeesEnumeration = schema$1.ReturnFeesEnumeration;
exports.ReturnLabelSourceEnumeration = schema$1.ReturnLabelSourceEnumeration;
exports.ReturnMethodEnumeration = schema$1.ReturnMethodEnumeration;
exports.Review = schema$1.Review;
exports.ReviewAction = schema$1.ReviewAction;
exports.ReviewNewsArticle = schema$1.ReviewNewsArticle;
exports.RiverBodyOfWater = schema$1.RiverBodyOfWater;
exports.Role = schema$1.Role;
exports.RoofingContractor = schema$1.RoofingContractor;
exports.Room = schema$1.Room;
exports.RsvpAction = schema$1.RsvpAction;
exports.RsvpResponseType = schema$1.RsvpResponseType;
exports.SaleEvent = schema$1.SaleEvent;
exports.SatiricalArticle = schema$1.SatiricalArticle;
exports.Schedule = schema$1.Schedule;
exports.ScheduleAction = schema$1.ScheduleAction;
exports.ScholarlyArticle = schema$1.ScholarlyArticle;
exports.School = schema$1.School;
exports.SchoolDistrict = schema$1.SchoolDistrict;
exports.ScreeningEvent = schema$1.ScreeningEvent;
exports.Sculpture = schema$1.Sculpture;
exports.SeaBodyOfWater = schema$1.SeaBodyOfWater;
exports.SearchAction = schema$1.SearchAction;
exports.SearchRescueOrganization = schema$1.SearchRescueOrganization;
exports.SearchResultsPage = schema$1.SearchResultsPage;
exports.Season = schema$1.Season;
exports.Seat = schema$1.Seat;
exports.SeekToAction = schema$1.SeekToAction;
exports.SelfStorage = schema$1.SelfStorage;
exports.SellAction = schema$1.SellAction;
exports.SendAction = schema$1.SendAction;
exports.Series = schema$1.Series;
exports.Service = schema$1.Service;
exports.ServiceChannel = schema$1.ServiceChannel;
exports.ShareAction = schema$1.ShareAction;
exports.SheetMusic = schema$1.SheetMusic;
exports.ShippingDeliveryTime = schema$1.ShippingDeliveryTime;
exports.ShippingRateSettings = schema$1.ShippingRateSettings;
exports.ShoeStore = schema$1.ShoeStore;
exports.ShoppingCenter = schema$1.ShoppingCenter;
exports.ShortStory = schema$1.ShortStory;
exports.SingleFamilyResidence = schema$1.SingleFamilyResidence;
exports.SiteNavigationElement = schema$1.SiteNavigationElement;
exports.SizeGroupEnumeration = schema$1.SizeGroupEnumeration;
exports.SizeSpecification = schema$1.SizeSpecification;
exports.SizeSystemEnumeration = schema$1.SizeSystemEnumeration;
exports.SkiResort = schema$1.SkiResort;
exports.SocialEvent = schema$1.SocialEvent;
exports.SocialMediaPosting = schema$1.SocialMediaPosting;
exports.SoftwareApplication = schema$1.SoftwareApplication;
exports.SoftwareSourceCode = schema$1.SoftwareSourceCode;
exports.SolveMathAction = schema$1.SolveMathAction;
exports.SomeProducts = schema$1.SomeProducts;
exports.SpeakableSpecification = schema$1.SpeakableSpecification;
exports.SpecialAnnouncement = schema$1.SpecialAnnouncement;
exports.Specialty = schema$1.Specialty;
exports.SportingGoodsStore = schema$1.SportingGoodsStore;
exports.SportsActivityLocation = schema$1.SportsActivityLocation;
exports.SportsClub = schema$1.SportsClub;
exports.SportsEvent = schema$1.SportsEvent;
exports.SportsOrganization = schema$1.SportsOrganization;
exports.SportsTeam = schema$1.SportsTeam;
exports.SpreadsheetDigitalDocument = schema$1.SpreadsheetDigitalDocument;
exports.StadiumOrArena = schema$1.StadiumOrArena;
exports.State = schema$1.State;
exports.Statement = schema$1.Statement;
exports.StatisticalPopulation = schema$1.StatisticalPopulation;
exports.StatisticalVariable = schema$1.StatisticalVariable;
exports.StatusEnumeration = schema$1.StatusEnumeration;
exports.SteeringPositionValue = schema$1.SteeringPositionValue;
exports.Store = schema$1.Store;
exports.StructuredValue = schema$1.StructuredValue;
exports.StupidType = schema$1.StupidType;
exports.SubscribeAction = schema$1.SubscribeAction;
exports.Substance = schema$1.Substance;
exports.SubwayStation = schema$1.SubwayStation;
exports.Suite = schema$1.Suite;
exports.SuperficialAnatomy = schema$1.SuperficialAnatomy;
exports.SurgicalProcedure = schema$1.SurgicalProcedure;
exports.SuspendAction = schema$1.SuspendAction;
exports.Syllabus = schema$1.Syllabus;
exports.Synagogue = schema$1.Synagogue;
exports.TVClip = schema$1.TVClip;
exports.TVEpisode = schema$1.TVEpisode;
exports.TVSeason = schema$1.TVSeason;
exports.TVSeries = schema$1.TVSeries;
exports.Table = schema$1.Table;
exports.TakeAction = schema$1.TakeAction;
exports.TattooParlor = schema$1.TattooParlor;
exports.Taxi = schema$1.Taxi;
exports.TaxiReservation = schema$1.TaxiReservation;
exports.TaxiService = schema$1.TaxiService;
exports.TaxiStand = schema$1.TaxiStand;
exports.Taxon = schema$1.Taxon;
exports.TechArticle = schema$1.TechArticle;
exports.TelevisionChannel = schema$1.TelevisionChannel;
exports.TelevisionStation = schema$1.TelevisionStation;
exports.TennisComplex = schema$1.TennisComplex;
exports.Text = schema$1.Text;
exports.TextDigitalDocument = schema$1.TextDigitalDocument;
exports.TextObject = schema$1.TextObject;
exports.TheaterEvent = schema$1.TheaterEvent;
exports.TheaterGroup = schema$1.TheaterGroup;
exports.TherapeuticProcedure = schema$1.TherapeuticProcedure;
exports.Thesis = schema$1.Thesis;
exports.Thing = schema$1.Thing;
exports.Ticket = schema$1.Ticket;
exports.TieAction = schema$1.TieAction;
exports.Time = schema$1.Time;
exports.TipAction = schema$1.TipAction;
exports.TireShop = schema$1.TireShop;
exports.TouristAttraction = schema$1.TouristAttraction;
exports.TouristDestination = schema$1.TouristDestination;
exports.TouristInformationCenter = schema$1.TouristInformationCenter;
exports.TouristTrip = schema$1.TouristTrip;
exports.ToyStore = schema$1.ToyStore;
exports.TrackAction = schema$1.TrackAction;
exports.TradeAction = schema$1.TradeAction;
exports.TrainReservation = schema$1.TrainReservation;
exports.TrainStation = schema$1.TrainStation;
exports.TrainTrip = schema$1.TrainTrip;
exports.TransferAction = schema$1.TransferAction;
exports.TravelAction = schema$1.TravelAction;
exports.TravelAgency = schema$1.TravelAgency;
exports.TreatmentIndication = schema$1.TreatmentIndication;
exports.Trip = schema$1.Trip;
exports.TypeAndQuantityNode = schema$1.TypeAndQuantityNode;
exports.UKNonprofitType = schema$1.UKNonprofitType;
exports.URL = schema$1.URL;
exports.USNonprofitType = schema$1.USNonprofitType;
exports.UnRegisterAction = schema$1.UnRegisterAction;
exports.UnitPriceSpecification = schema$1.UnitPriceSpecification;
exports.UpdateAction = schema$1.UpdateAction;
exports.UseAction = schema$1.UseAction;
exports.UserBlocks = schema$1.UserBlocks;
exports.UserCheckins = schema$1.UserCheckins;
exports.UserComments = schema$1.UserComments;
exports.UserDownloads = schema$1.UserDownloads;
exports.UserInteraction = schema$1.UserInteraction;
exports.UserLikes = schema$1.UserLikes;
exports.UserPageVisits = schema$1.UserPageVisits;
exports.UserPlays = schema$1.UserPlays;
exports.UserPlusOnes = schema$1.UserPlusOnes;
exports.UserReview = schema$1.UserReview;
exports.UserTweets = schema$1.UserTweets;
exports.VacationRental = schema$1.VacationRental;
exports.Vehicle = schema$1.Vehicle;
exports.Vein = schema$1.Vein;
exports.Vessel = schema$1.Vessel;
exports.VeterinaryCare = schema$1.VeterinaryCare;
exports.VideoGallery = schema$1.VideoGallery;
exports.VideoGame = schema$1.VideoGame;
exports.VideoGameClip = schema$1.VideoGameClip;
exports.VideoGameSeries = schema$1.VideoGameSeries;
exports.VideoObject = schema$1.VideoObject;
exports.VideoObjectSnapshot = schema$1.VideoObjectSnapshot;
exports.ViewAction = schema$1.ViewAction;
exports.VirtualLocation = schema$1.VirtualLocation;
exports.VisualArtsEvent = schema$1.VisualArtsEvent;
exports.VisualArtwork = schema$1.VisualArtwork;
exports.VitalSign = schema$1.VitalSign;
exports.Volcano = schema$1.Volcano;
exports.VoteAction = schema$1.VoteAction;
exports.WPAdBlock = schema$1.WPAdBlock;
exports.WPFooter = schema$1.WPFooter;
exports.WPHeader = schema$1.WPHeader;
exports.WPSideBar = schema$1.WPSideBar;
exports.WantAction = schema$1.WantAction;
exports.WarrantyPromise = schema$1.WarrantyPromise;
exports.WarrantyScope = schema$1.WarrantyScope;
exports.WatchAction = schema$1.WatchAction;
exports.Waterfall = schema$1.Waterfall;
exports.WearAction = schema$1.WearAction;
exports.WearableMeasurementTypeEnumeration = schema$1.WearableMeasurementTypeEnumeration;
exports.WearableSizeGroupEnumeration = schema$1.WearableSizeGroupEnumeration;
exports.WearableSizeSystemEnumeration = schema$1.WearableSizeSystemEnumeration;
exports.WebAPI = schema$1.WebAPI;
exports.WebApplication = schema$1.WebApplication;
exports.WebContent = schema$1.WebContent;
exports.WebPage = schema$1.WebPage;
exports.WebPageElement = schema$1.WebPageElement;
exports.WebSite = schema$1.WebSite;
exports.WholesaleStore = schema$1.WholesaleStore;
exports.WinAction = schema$1.WinAction;
exports.Winery = schema$1.Winery;
exports.WorkBasedProgram = schema$1.WorkBasedProgram;
exports.WorkersUnion = schema$1.WorkersUnion;
exports.WriteAction = schema$1.WriteAction;
exports.XPathType = schema$1.XPathType;
exports.Zoo = schema$1.Zoo;
exports.about = schema$1.about;
exports.abridged = schema$1.abridged;
exports.abstract = schema$1.abstract;
exports.accelerationTime = schema$1.accelerationTime;
exports.acceptedAnswer = schema$1.acceptedAnswer;
exports.acceptedOffer = schema$1.acceptedOffer;
exports.acceptedPaymentMethod = schema$1.acceptedPaymentMethod;
exports.acceptsReservations = schema$1.acceptsReservations;
exports.accessCode = schema$1.accessCode;
exports.accessMode = schema$1.accessMode;
exports.accessModeSufficient = schema$1.accessModeSufficient;
exports.accessibilityAPI = schema$1.accessibilityAPI;
exports.accessibilityControl = schema$1.accessibilityControl;
exports.accessibilityFeature = schema$1.accessibilityFeature;
exports.accessibilityHazard = schema$1.accessibilityHazard;
exports.accessibilitySummary = schema$1.accessibilitySummary;
exports.accommodationCategory = schema$1.accommodationCategory;
exports.accommodationFloorPlan = schema$1.accommodationFloorPlan;
exports.accountId = schema$1.accountId;
exports.accountMinimumInflow = schema$1.accountMinimumInflow;
exports.accountOverdraftLimit = schema$1.accountOverdraftLimit;
exports.accountablePerson = schema$1.accountablePerson;
exports.acquireLicensePage = schema$1.acquireLicensePage;
exports.acquiredFrom = schema$1.acquiredFrom;
exports.acrissCode = schema$1.acrissCode;
exports.actionAccessibilityRequirement = schema$1.actionAccessibilityRequirement;
exports.actionApplication = schema$1.actionApplication;
exports.actionOption = schema$1.actionOption;
exports.actionPlatform = schema$1.actionPlatform;
exports.actionStatus = schema$1.actionStatus;
exports.actionableFeedbackPolicy = schema$1.actionableFeedbackPolicy;
exports.activeIngredient = schema$1.activeIngredient;
exports.activityDuration = schema$1.activityDuration;
exports.activityFrequency = schema$1.activityFrequency;
exports.actor = schema$1.actor;
exports.actors = schema$1.actors;
exports.addOn = schema$1.addOn;
exports.additionalName = schema$1.additionalName;
exports.additionalNumberOfGuests = schema$1.additionalNumberOfGuests;
exports.additionalProperty = schema$1.additionalProperty;
exports.additionalType = schema$1.additionalType;
exports.additionalVariable = schema$1.additionalVariable;
exports.address = schema$1.address;
exports.addressCountry = schema$1.addressCountry;
exports.addressLocality = schema$1.addressLocality;
exports.addressRegion = schema$1.addressRegion;
exports.administrationRoute = schema$1.administrationRoute;
exports.advanceBookingRequirement = schema$1.advanceBookingRequirement;
exports.adverseOutcome = schema$1.adverseOutcome;
exports.affectedBy = schema$1.affectedBy;
exports.affiliation = schema$1.affiliation;
exports.afterMedia = schema$1.afterMedia;
exports.agent = schema$1.agent;
exports.aggregateRating = schema$1.aggregateRating;
exports.aircraft = schema$1.aircraft;
exports.album = schema$1.album;
exports.albumProductionType = schema$1.albumProductionType;
exports.albumRelease = schema$1.albumRelease;
exports.albumReleaseType = schema$1.albumReleaseType;
exports.albums = schema$1.albums;
exports.alcoholWarning = schema$1.alcoholWarning;
exports.algorithm = schema$1.algorithm;
exports.alignmentType = schema$1.alignmentType;
exports.alternateName = schema$1.alternateName;
exports.alternativeHeadline = schema$1.alternativeHeadline;
exports.alternativeOf = schema$1.alternativeOf;
exports.alumni = schema$1.alumni;
exports.alumniOf = schema$1.alumniOf;
exports.amenityFeature = schema$1.amenityFeature;
exports.amount = schema$1.amount;
exports.amountOfThisGood = schema$1.amountOfThisGood;
exports.announcementLocation = schema$1.announcementLocation;
exports.annualPercentageRate = schema$1.annualPercentageRate;
exports.answerCount = schema$1.answerCount;
exports.answerExplanation = schema$1.answerExplanation;
exports.antagonist = schema$1.antagonist;
exports.appearance = schema$1.appearance;
exports.applicableCountry = schema$1.applicableCountry;
exports.applicableLocation = schema$1.applicableLocation;
exports.applicantLocationRequirements = schema$1.applicantLocationRequirements;
exports.application = schema$1.application;
exports.applicationCategory = schema$1.applicationCategory;
exports.applicationContact = schema$1.applicationContact;
exports.applicationDeadline = schema$1.applicationDeadline;
exports.applicationStartDate = schema$1.applicationStartDate;
exports.applicationSubCategory = schema$1.applicationSubCategory;
exports.applicationSuite = schema$1.applicationSuite;
exports.appliesToDeliveryMethod = schema$1.appliesToDeliveryMethod;
exports.appliesToPaymentMethod = schema$1.appliesToPaymentMethod;
exports.archiveHeld = schema$1.archiveHeld;
exports.archivedAt = schema$1.archivedAt;
exports.area = schema$1.area;
exports.areaServed = schema$1.areaServed;
exports.arrivalAirport = schema$1.arrivalAirport;
exports.arrivalBoatTerminal = schema$1.arrivalBoatTerminal;
exports.arrivalBusStop = schema$1.arrivalBusStop;
exports.arrivalGate = schema$1.arrivalGate;
exports.arrivalPlatform = schema$1.arrivalPlatform;
exports.arrivalStation = schema$1.arrivalStation;
exports.arrivalTerminal = schema$1.arrivalTerminal;
exports.arrivalTime = schema$1.arrivalTime;
exports.artEdition = schema$1.artEdition;
exports.artMedium = schema$1.artMedium;
exports.arterialBranch = schema$1.arterialBranch;
exports.artform = schema$1.artform;
exports.articleBody = schema$1.articleBody;
exports.articleSection = schema$1.articleSection;
exports.artist = schema$1.artist;
exports.artworkSurface = schema$1.artworkSurface;
exports.asin = schema$1.asin;
exports.aspect = schema$1.aspect;
exports.assembly = schema$1.assembly;
exports.assemblyVersion = schema$1.assemblyVersion;
exports.assesses = schema$1.assesses;
exports.associatedAnatomy = schema$1.associatedAnatomy;
exports.associatedArticle = schema$1.associatedArticle;
exports.associatedClaimReview = schema$1.associatedClaimReview;
exports.associatedDisease = schema$1.associatedDisease;
exports.associatedMedia = schema$1.associatedMedia;
exports.associatedMediaReview = schema$1.associatedMediaReview;
exports.associatedPathophysiology = schema$1.associatedPathophysiology;
exports.associatedReview = schema$1.associatedReview;
exports.athlete = schema$1.athlete;
exports.attendee = schema$1.attendee;
exports.attendees = schema$1.attendees;
exports.audience = schema$1.audience;
exports.audienceType = schema$1.audienceType;
exports.audio = schema$1.audio;
exports.authenticator = schema$1.authenticator;
exports.author = schema$1.author;
exports.availability = schema$1.availability;
exports.availabilityEnds = schema$1.availabilityEnds;
exports.availabilityStarts = schema$1.availabilityStarts;
exports.availableAtOrFrom = schema$1.availableAtOrFrom;
exports.availableChannel = schema$1.availableChannel;
exports.availableDeliveryMethod = schema$1.availableDeliveryMethod;
exports.availableFrom = schema$1.availableFrom;
exports.availableIn = schema$1.availableIn;
exports.availableLanguage = schema$1.availableLanguage;
exports.availableOnDevice = schema$1.availableOnDevice;
exports.availableService = schema$1.availableService;
exports.availableStrength = schema$1.availableStrength;
exports.availableTest = schema$1.availableTest;
exports.availableThrough = schema$1.availableThrough;
exports.award = schema$1.award;
exports.awards = schema$1.awards;
exports.awayTeam = schema$1.awayTeam;
exports.backstory = schema$1.backstory;
exports.bankAccountType = schema$1.bankAccountType;
exports.baseSalary = schema$1.baseSalary;
exports.bccRecipient = schema$1.bccRecipient;
exports.bed = schema$1.bed;
exports.beforeMedia = schema$1.beforeMedia;
exports.beneficiaryBank = schema$1.beneficiaryBank;
exports.benefits = schema$1.benefits;
exports.benefitsSummaryUrl = schema$1.benefitsSummaryUrl;
exports.bestRating = schema$1.bestRating;
exports.billingAddress = schema$1.billingAddress;
exports.billingDuration = schema$1.billingDuration;
exports.billingIncrement = schema$1.billingIncrement;
exports.billingPeriod = schema$1.billingPeriod;
exports.billingStart = schema$1.billingStart;
exports.bioChemInteraction = schema$1.bioChemInteraction;
exports.bioChemSimilarity = schema$1.bioChemSimilarity;
exports.biologicalRole = schema$1.biologicalRole;
exports.biomechnicalClass = schema$1.biomechnicalClass;
exports.birthDate = schema$1.birthDate;
exports.birthPlace = schema$1.birthPlace;
exports.bitrate = schema$1.bitrate;
exports.blogPost = schema$1.blogPost;
exports.blogPosts = schema$1.blogPosts;
exports.bloodSupply = schema$1.bloodSupply;
exports.boardingGroup = schema$1.boardingGroup;
exports.boardingPolicy = schema$1.boardingPolicy;
exports.bodyLocation = schema$1.bodyLocation;
exports.bodyType = schema$1.bodyType;
exports.bookEdition = schema$1.bookEdition;
exports.bookFormat = schema$1.bookFormat;
exports.bookingAgent = schema$1.bookingAgent;
exports.bookingTime = schema$1.bookingTime;
exports.borrower = schema$1.borrower;
exports.box = schema$1.box;
exports.branch = schema$1.branch;
exports.branchCode = schema$1.branchCode;
exports.branchOf = schema$1.branchOf;
exports.brand = schema$1.brand;
exports.breadcrumb = schema$1.breadcrumb;
exports.breastfeedingWarning = schema$1.breastfeedingWarning;
exports.broadcastAffiliateOf = schema$1.broadcastAffiliateOf;
exports.broadcastChannelId = schema$1.broadcastChannelId;
exports.broadcastDisplayName = schema$1.broadcastDisplayName;
exports.broadcastFrequency = schema$1.broadcastFrequency;
exports.broadcastFrequencyValue = schema$1.broadcastFrequencyValue;
exports.broadcastOfEvent = schema$1.broadcastOfEvent;
exports.broadcastServiceTier = schema$1.broadcastServiceTier;
exports.broadcastSignalModulation = schema$1.broadcastSignalModulation;
exports.broadcastSubChannel = schema$1.broadcastSubChannel;
exports.broadcastTimezone = schema$1.broadcastTimezone;
exports.broadcaster = schema$1.broadcaster;
exports.broker = schema$1.broker;
exports.browserRequirements = schema$1.browserRequirements;
exports.busName = schema$1.busName;
exports.busNumber = schema$1.busNumber;
exports.businessDays = schema$1.businessDays;
exports.businessFunction = schema$1.businessFunction;
exports.buyer = schema$1.buyer;
exports.byArtist = schema$1.byArtist;
exports.byDay = schema$1.byDay;
exports.byMonth = schema$1.byMonth;
exports.byMonthDay = schema$1.byMonthDay;
exports.byMonthWeek = schema$1.byMonthWeek;
exports.callSign = schema$1.callSign;
exports.calories = schema$1.calories;
exports.candidate = schema$1.candidate;
exports.caption = schema$1.caption;
exports.carbohydrateContent = schema$1.carbohydrateContent;
exports.cargoVolume = schema$1.cargoVolume;
exports.carrier = schema$1.carrier;
exports.carrierRequirements = schema$1.carrierRequirements;
exports.cashBack = schema$1.cashBack;
exports.catalog = schema$1.catalog;
exports.catalogNumber = schema$1.catalogNumber;
exports.category = schema$1.category;
exports.causeOf = schema$1.causeOf;
exports.ccRecipient = schema$1.ccRecipient;
exports.character = schema$1.character;
exports.characterAttribute = schema$1.characterAttribute;
exports.characterName = schema$1.characterName;
exports.cheatCode = schema$1.cheatCode;
exports.checkinTime = schema$1.checkinTime;
exports.checkoutPageURLTemplate = schema$1.checkoutPageURLTemplate;
exports.checkoutTime = schema$1.checkoutTime;
exports.chemicalComposition = schema$1.chemicalComposition;
exports.chemicalRole = schema$1.chemicalRole;
exports.childMaxAge = schema$1.childMaxAge;
exports.childMinAge = schema$1.childMinAge;
exports.childTaxon = schema$1.childTaxon;
exports.children = schema$1.children;
exports.cholesterolContent = schema$1.cholesterolContent;
exports.circle = schema$1.circle;
exports.citation = schema$1.citation;
exports.claimInterpreter = schema$1.claimInterpreter;
exports.claimReviewed = schema$1.claimReviewed;
exports.clincalPharmacology = schema$1.clincalPharmacology;
exports.clinicalPharmacology = schema$1.clinicalPharmacology;
exports.clipNumber = schema$1.clipNumber;
exports.closes = schema$1.closes;
exports.coach = schema$1.coach;
exports.code = schema$1.code;
exports.codeRepository = schema$1.codeRepository;
exports.codeSampleType = schema$1.codeSampleType;
exports.codeValue = schema$1.codeValue;
exports.codingSystem = schema$1.codingSystem;
exports.colleague = schema$1.colleague;
exports.colleagues = schema$1.colleagues;
exports.collection = schema$1.collection;
exports.collectionSize = schema$1.collectionSize;
exports.color = schema$1.color;
exports.colorist = schema$1.colorist;
exports.comment = schema$1.comment;
exports.commentCount = schema$1.commentCount;
exports.commentText = schema$1.commentText;
exports.commentTime = schema$1.commentTime;
exports.competencyRequired = schema$1.competencyRequired;
exports.competitor = schema$1.competitor;
exports.composer = schema$1.composer;
exports.comprisedOf = schema$1.comprisedOf;
exports.conditionsOfAccess = schema$1.conditionsOfAccess;
exports.confirmationNumber = schema$1.confirmationNumber;
exports.connectedTo = schema$1.connectedTo;
exports.constraintProperty = schema$1.constraintProperty;
exports.contactOption = schema$1.contactOption;
exports.contactPoint = schema$1.contactPoint;
exports.contactPoints = schema$1.contactPoints;
exports.contactType = schema$1.contactType;
exports.contactlessPayment = schema$1.contactlessPayment;
exports.containedIn = schema$1.containedIn;
exports.containedInPlace = schema$1.containedInPlace;
exports.containsPlace = schema$1.containsPlace;
exports.containsSeason = schema$1.containsSeason;
exports.contentLocation = schema$1.contentLocation;
exports.contentRating = schema$1.contentRating;
exports.contentReferenceTime = schema$1.contentReferenceTime;
exports.contentSize = schema$1.contentSize;
exports.contentType = schema$1.contentType;
exports.contentUrl = schema$1.contentUrl;
exports.contraindication = schema$1.contraindication;
exports.contributor = schema$1.contributor;
exports.cookTime = schema$1.cookTime;
exports.cookingMethod = schema$1.cookingMethod;
exports.copyrightHolder = schema$1.copyrightHolder;
exports.copyrightNotice = schema$1.copyrightNotice;
exports.copyrightYear = schema$1.copyrightYear;
exports.correction = schema$1.correction;
exports.correctionsPolicy = schema$1.correctionsPolicy;
exports.costCategory = schema$1.costCategory;
exports.costCurrency = schema$1.costCurrency;
exports.costOrigin = schema$1.costOrigin;
exports.costPerUnit = schema$1.costPerUnit;
exports.countriesNotSupported = schema$1.countriesNotSupported;
exports.countriesSupported = schema$1.countriesSupported;
exports.countryOfAssembly = schema$1.countryOfAssembly;
exports.countryOfLastProcessing = schema$1.countryOfLastProcessing;
exports.countryOfOrigin = schema$1.countryOfOrigin;
exports.course = schema$1.course;
exports.courseCode = schema$1.courseCode;
exports.courseMode = schema$1.courseMode;
exports.coursePrerequisites = schema$1.coursePrerequisites;
exports.courseSchedule = schema$1.courseSchedule;
exports.courseWorkload = schema$1.courseWorkload;
exports.coverageEndTime = schema$1.coverageEndTime;
exports.coverageStartTime = schema$1.coverageStartTime;
exports.creativeWorkStatus = schema$1.creativeWorkStatus;
exports.creator = schema$1.creator;
exports.credentialCategory = schema$1.credentialCategory;
exports.creditText = schema$1.creditText;
exports.creditedTo = schema$1.creditedTo;
exports.cssSelector = schema$1.cssSelector;
exports.currenciesAccepted = schema$1.currenciesAccepted;
exports.currency = schema$1.currency;
exports.currentExchangeRate = schema$1.currentExchangeRate;
exports.customer = schema$1.customer;
exports.customerRemorseReturnFees = schema$1.customerRemorseReturnFees;
exports.customerRemorseReturnLabelSource = schema$1.customerRemorseReturnLabelSource;
exports.customerRemorseReturnShippingFeesAmount = schema$1.customerRemorseReturnShippingFeesAmount;
exports.cutoffTime = schema$1.cutoffTime;
exports.cvdCollectionDate = schema$1.cvdCollectionDate;
exports.cvdFacilityCounty = schema$1.cvdFacilityCounty;
exports.cvdFacilityId = schema$1.cvdFacilityId;
exports.cvdNumBeds = schema$1.cvdNumBeds;
exports.cvdNumBedsOcc = schema$1.cvdNumBedsOcc;
exports.cvdNumC19Died = schema$1.cvdNumC19Died;
exports.cvdNumC19HOPats = schema$1.cvdNumC19HOPats;
exports.cvdNumC19HospPats = schema$1.cvdNumC19HospPats;
exports.cvdNumC19MechVentPats = schema$1.cvdNumC19MechVentPats;
exports.cvdNumC19OFMechVentPats = schema$1.cvdNumC19OFMechVentPats;
exports.cvdNumC19OverflowPats = schema$1.cvdNumC19OverflowPats;
exports.cvdNumICUBeds = schema$1.cvdNumICUBeds;
exports.cvdNumICUBedsOcc = schema$1.cvdNumICUBedsOcc;
exports.cvdNumTotBeds = schema$1.cvdNumTotBeds;
exports.cvdNumVent = schema$1.cvdNumVent;
exports.cvdNumVentUse = schema$1.cvdNumVentUse;
exports.dataFeedElement = schema$1.dataFeedElement;
exports.dataset = schema$1.dataset;
exports.datasetTimeInterval = schema$1.datasetTimeInterval;
exports.dateCreated = schema$1.dateCreated;
exports.dateDeleted = schema$1.dateDeleted;
exports.dateIssued = schema$1.dateIssued;
exports.dateModified = schema$1.dateModified;
exports.datePosted = schema$1.datePosted;
exports.datePublished = schema$1.datePublished;
exports.dateRead = schema$1.dateRead;
exports.dateReceived = schema$1.dateReceived;
exports.dateSent = schema$1.dateSent;
exports.dateVehicleFirstRegistered = schema$1.dateVehicleFirstRegistered;
exports.dateline = schema$1.dateline;
exports.dayOfWeek = schema$1.dayOfWeek;
exports.deathDate = schema$1.deathDate;
exports.deathPlace = schema$1.deathPlace;
exports.defaultValue = schema$1.defaultValue;
exports.deliveryAddress = schema$1.deliveryAddress;
exports.deliveryLeadTime = schema$1.deliveryLeadTime;
exports.deliveryMethod = schema$1.deliveryMethod;
exports.deliveryStatus = schema$1.deliveryStatus;
exports.deliveryTime = schema$1.deliveryTime;
exports.department = schema$1.department;
exports.departureAirport = schema$1.departureAirport;
exports.departureBoatTerminal = schema$1.departureBoatTerminal;
exports.departureBusStop = schema$1.departureBusStop;
exports.departureGate = schema$1.departureGate;
exports.departurePlatform = schema$1.departurePlatform;
exports.departureStation = schema$1.departureStation;
exports.departureTerminal = schema$1.departureTerminal;
exports.departureTime = schema$1.departureTime;
exports.dependencies = schema$1.dependencies;
exports.depth = schema$1.depth;
exports.description = schema$1.description;
exports.device = schema$1.device;
exports.diagnosis = schema$1.diagnosis;
exports.diagram = schema$1.diagram;
exports.diet = schema$1.diet;
exports.dietFeatures = schema$1.dietFeatures;
exports.differentialDiagnosis = schema$1.differentialDiagnosis;
exports.directApply = schema$1.directApply;
exports.director = schema$1.director;
exports.directors = schema$1.directors;
exports.disambiguatingDescription = schema$1.disambiguatingDescription;
exports.discount = schema$1.discount;
exports.discountCode = schema$1.discountCode;
exports.discountCurrency = schema$1.discountCurrency;
exports.discusses = schema$1.discusses;
exports.discussionUrl = schema$1.discussionUrl;
exports.diseasePreventionInfo = schema$1.diseasePreventionInfo;
exports.diseaseSpreadStatistics = schema$1.diseaseSpreadStatistics;
exports.dissolutionDate = schema$1.dissolutionDate;
exports.distance = schema$1.distance;
exports.distinguishingSign = schema$1.distinguishingSign;
exports.distribution = schema$1.distribution;
exports.diversityPolicy = schema$1.diversityPolicy;
exports.diversityStaffingReport = schema$1.diversityStaffingReport;
exports.documentation = schema$1.documentation;
exports.doesNotShip = schema$1.doesNotShip;
exports.domainIncludes = schema$1.domainIncludes;
exports.domiciledMortgage = schema$1.domiciledMortgage;
exports.doorTime = schema$1.doorTime;
exports.dosageForm = schema$1.dosageForm;
exports.doseSchedule = schema$1.doseSchedule;
exports.doseUnit = schema$1.doseUnit;
exports.doseValue = schema$1.doseValue;
exports.downPayment = schema$1.downPayment;
exports.downloadUrl = schema$1.downloadUrl;
exports.downvoteCount = schema$1.downvoteCount;
exports.drainsTo = schema$1.drainsTo;
exports.driveWheelConfiguration = schema$1.driveWheelConfiguration;
exports.dropoffLocation = schema$1.dropoffLocation;
exports.dropoffTime = schema$1.dropoffTime;
exports.drug = schema$1.drug;
exports.drugClass = schema$1.drugClass;
exports.drugUnit = schema$1.drugUnit;
exports.duns = schema$1.duns;
exports.duplicateTherapy = schema$1.duplicateTherapy;
exports.duration = schema$1.duration;
exports.durationOfWarranty = schema$1.durationOfWarranty;
exports.duringMedia = schema$1.duringMedia;
exports.earlyPrepaymentPenalty = schema$1.earlyPrepaymentPenalty;
exports.editEIDR = schema$1.editEIDR;
exports.editor = schema$1.editor;
exports.eduQuestionType = schema$1.eduQuestionType;
exports.educationRequirements = schema$1.educationRequirements;
exports.educationalAlignment = schema$1.educationalAlignment;
exports.educationalCredentialAwarded = schema$1.educationalCredentialAwarded;
exports.educationalFramework = schema$1.educationalFramework;
exports.educationalLevel = schema$1.educationalLevel;
exports.educationalProgramMode = schema$1.educationalProgramMode;
exports.educationalRole = schema$1.educationalRole;
exports.educationalUse = schema$1.educationalUse;
exports.elevation = schema$1.elevation;
exports.eligibilityToWorkRequirement = schema$1.eligibilityToWorkRequirement;
exports.eligibleCustomerType = schema$1.eligibleCustomerType;
exports.eligibleDuration = schema$1.eligibleDuration;
exports.eligibleQuantity = schema$1.eligibleQuantity;
exports.eligibleRegion = schema$1.eligibleRegion;
exports.eligibleTransactionVolume = schema$1.eligibleTransactionVolume;
exports.email = schema$1.email;
exports.embedUrl = schema$1.embedUrl;
exports.embeddedTextCaption = schema$1.embeddedTextCaption;
exports.emissionsCO2 = schema$1.emissionsCO2;
exports.employee = schema$1.employee;
exports.employees = schema$1.employees;
exports.employerOverview = schema$1.employerOverview;
exports.employmentType = schema$1.employmentType;
exports.employmentUnit = schema$1.employmentUnit;
exports.encodesBioChemEntity = schema$1.encodesBioChemEntity;
exports.encodesCreativeWork = schema$1.encodesCreativeWork;
exports.encoding = schema$1.encoding;
exports.encodingFormat = schema$1.encodingFormat;
exports.encodingType = schema$1.encodingType;
exports.encodings = schema$1.encodings;
exports.endDate = schema$1.endDate;
exports.endOffset = schema$1.endOffset;
exports.endTime = schema$1.endTime;
exports.endorsee = schema$1.endorsee;
exports.endorsers = schema$1.endorsers;
exports.energyEfficiencyScaleMax = schema$1.energyEfficiencyScaleMax;
exports.energyEfficiencyScaleMin = schema$1.energyEfficiencyScaleMin;
exports.engineDisplacement = schema$1.engineDisplacement;
exports.enginePower = schema$1.enginePower;
exports.engineType = schema$1.engineType;
exports.entertainmentBusiness = schema$1.entertainmentBusiness;
exports.epidemiology = schema$1.epidemiology;
exports.episode = schema$1.episode;
exports.episodeNumber = schema$1.episodeNumber;
exports.episodes = schema$1.episodes;
exports.equal = schema$1.equal;
exports.error = schema$1.error;
exports.estimatedCost = schema$1.estimatedCost;
exports.estimatedFlightDuration = schema$1.estimatedFlightDuration;
exports.estimatedSalary = schema$1.estimatedSalary;
exports.estimatesRiskOf = schema$1.estimatesRiskOf;
exports.ethicsPolicy = schema$1.ethicsPolicy;
exports.event = schema$1.event;
exports.eventAttendanceMode = schema$1.eventAttendanceMode;
exports.eventSchedule = schema$1.eventSchedule;
exports.eventStatus = schema$1.eventStatus;
exports.events = schema$1.events;
exports.evidenceLevel = schema$1.evidenceLevel;
exports.evidenceOrigin = schema$1.evidenceOrigin;
exports.exampleOfWork = schema$1.exampleOfWork;
exports.exceptDate = schema$1.exceptDate;
exports.exchangeRateSpread = schema$1.exchangeRateSpread;
exports.executableLibraryName = schema$1.executableLibraryName;
exports.exerciseCourse = schema$1.exerciseCourse;
exports.exercisePlan = schema$1.exercisePlan;
exports.exerciseRelatedDiet = schema$1.exerciseRelatedDiet;
exports.exerciseType = schema$1.exerciseType;
exports.exifData = schema$1.exifData;
exports.expectedArrivalFrom = schema$1.expectedArrivalFrom;
exports.expectedArrivalUntil = schema$1.expectedArrivalUntil;
exports.expectedPrognosis = schema$1.expectedPrognosis;
exports.expectsAcceptanceOf = schema$1.expectsAcceptanceOf;
exports.experienceInPlaceOfEducation = schema$1.experienceInPlaceOfEducation;
exports.experienceRequirements = schema$1.experienceRequirements;
exports.expertConsiderations = schema$1.expertConsiderations;
exports.expires = schema$1.expires;
exports.expressedIn = schema$1.expressedIn;
exports.familyName = schema$1.familyName;
exports.fatContent = schema$1.fatContent;
exports.faxNumber = schema$1.faxNumber;
exports.featureList = schema$1.featureList;
exports.feesAndCommissionsSpecification = schema$1.feesAndCommissionsSpecification;
exports.fiberContent = schema$1.fiberContent;
exports.fileFormat = schema$1.fileFormat;
exports.fileSize = schema$1.fileSize;
exports.financialAidEligible = schema$1.financialAidEligible;
exports.firstAppearance = schema$1.firstAppearance;
exports.firstPerformance = schema$1.firstPerformance;
exports.flightDistance = schema$1.flightDistance;
exports.flightNumber = schema$1.flightNumber;
exports.floorLevel = schema$1.floorLevel;
exports.floorLimit = schema$1.floorLimit;
exports.floorSize = schema$1.floorSize;
exports.followee = schema$1.followee;
exports.follows = schema$1.follows;
exports.followup = schema$1.followup;
exports.foodEstablishment = schema$1.foodEstablishment;
exports.foodEvent = schema$1.foodEvent;
exports.foodWarning = schema$1.foodWarning;
exports.founder = schema$1.founder;
exports.founders = schema$1.founders;
exports.foundingDate = schema$1.foundingDate;
exports.foundingLocation = schema$1.foundingLocation;
exports.free = schema$1.free;
exports.freeShippingThreshold = schema$1.freeShippingThreshold;
exports.frequency = schema$1.frequency;
exports.fromLocation = schema$1.fromLocation;
exports.fuelCapacity = schema$1.fuelCapacity;
exports.fuelConsumption = schema$1.fuelConsumption;
exports.fuelEfficiency = schema$1.fuelEfficiency;
exports.fuelType = schema$1.fuelType;
exports.functionalClass = schema$1.functionalClass;
exports.fundedItem = schema$1.fundedItem;
exports.funder = schema$1.funder;
exports.funding = schema$1.funding;
exports.game = schema$1.game;
exports.gameAvailabilityType = schema$1.gameAvailabilityType;
exports.gameEdition = schema$1.gameEdition;
exports.gameItem = schema$1.gameItem;
exports.gameLocation = schema$1.gameLocation;
exports.gamePlatform = schema$1.gamePlatform;
exports.gameServer = schema$1.gameServer;
exports.gameTip = schema$1.gameTip;
exports.gender = schema$1.gender;
exports.genre = schema$1.genre;
exports.geo = schema$1.geo;
exports.geoContains = schema$1.geoContains;
exports.geoCoveredBy = schema$1.geoCoveredBy;
exports.geoCovers = schema$1.geoCovers;
exports.geoCrosses = schema$1.geoCrosses;
exports.geoDisjoint = schema$1.geoDisjoint;
exports.geoEquals = schema$1.geoEquals;
exports.geoIntersects = schema$1.geoIntersects;
exports.geoMidpoint = schema$1.geoMidpoint;
exports.geoOverlaps = schema$1.geoOverlaps;
exports.geoRadius = schema$1.geoRadius;
exports.geoTouches = schema$1.geoTouches;
exports.geoWithin = schema$1.geoWithin;
exports.geographicArea = schema$1.geographicArea;
exports.gettingTestedInfo = schema$1.gettingTestedInfo;
exports.givenName = schema$1.givenName;
exports.globalLocationNumber = schema$1.globalLocationNumber;
exports.governmentBenefitsInfo = schema$1.governmentBenefitsInfo;
exports.gracePeriod = schema$1.gracePeriod;
exports.grantee = schema$1.grantee;
exports.greater = schema$1.greater;
exports.greaterOrEqual = schema$1.greaterOrEqual;
exports.gtin = schema$1.gtin;
exports.gtin12 = schema$1.gtin12;
exports.gtin13 = schema$1.gtin13;
exports.gtin14 = schema$1.gtin14;
exports.gtin8 = schema$1.gtin8;
exports.guideline = schema$1.guideline;
exports.guidelineDate = schema$1.guidelineDate;
exports.guidelineSubject = schema$1.guidelineSubject;
exports.handlingTime = schema$1.handlingTime;
exports.hasAdultConsideration = schema$1.hasAdultConsideration;
exports.hasBioChemEntityPart = schema$1.hasBioChemEntityPart;
exports.hasBioPolymerSequence = schema$1.hasBioPolymerSequence;
exports.hasBroadcastChannel = schema$1.hasBroadcastChannel;
exports.hasCategoryCode = schema$1.hasCategoryCode;
exports.hasCourse = schema$1.hasCourse;
exports.hasCourseInstance = schema$1.hasCourseInstance;
exports.hasCredential = schema$1.hasCredential;
exports.hasDefinedTerm = schema$1.hasDefinedTerm;
exports.hasDeliveryMethod = schema$1.hasDeliveryMethod;
exports.hasDigitalDocumentPermission = schema$1.hasDigitalDocumentPermission;
exports.hasDriveThroughService = schema$1.hasDriveThroughService;
exports.hasEnergyConsumptionDetails = schema$1.hasEnergyConsumptionDetails;
exports.hasEnergyEfficiencyCategory = schema$1.hasEnergyEfficiencyCategory;
exports.hasHealthAspect = schema$1.hasHealthAspect;
exports.hasMap = schema$1.hasMap;
exports.hasMeasurement = schema$1.hasMeasurement;
exports.hasMenu = schema$1.hasMenu;
exports.hasMenuItem = schema$1.hasMenuItem;
exports.hasMenuSection = schema$1.hasMenuSection;
exports.hasMerchantReturnPolicy = schema$1.hasMerchantReturnPolicy;
exports.hasMolecularFunction = schema$1.hasMolecularFunction;
exports.hasOccupation = schema$1.hasOccupation;
exports.hasOfferCatalog = schema$1.hasOfferCatalog;
exports.hasPOS = schema$1.hasPOS;
exports.hasPart = schema$1.hasPart;
exports.hasProductReturnPolicy = schema$1.hasProductReturnPolicy;
exports.hasRepresentation = schema$1.hasRepresentation;
exports.hasVariant = schema$1.hasVariant;
exports.headline = schema$1.headline;
exports.healthCondition = schema$1.healthCondition;
exports.healthPlanCoinsuranceOption = schema$1.healthPlanCoinsuranceOption;
exports.healthPlanCoinsuranceRate = schema$1.healthPlanCoinsuranceRate;
exports.healthPlanCopay = schema$1.healthPlanCopay;
exports.healthPlanCopayOption = schema$1.healthPlanCopayOption;
exports.healthPlanCostSharing = schema$1.healthPlanCostSharing;
exports.healthPlanDrugOption = schema$1.healthPlanDrugOption;
exports.healthPlanDrugTier = schema$1.healthPlanDrugTier;
exports.healthPlanId = schema$1.healthPlanId;
exports.healthPlanMarketingUrl = schema$1.healthPlanMarketingUrl;
exports.healthPlanNetworkId = schema$1.healthPlanNetworkId;
exports.healthPlanNetworkTier = schema$1.healthPlanNetworkTier;
exports.healthPlanPharmacyCategory = schema$1.healthPlanPharmacyCategory;
exports.healthcareReportingData = schema$1.healthcareReportingData;
exports.height = schema$1.height;
exports.highPrice = schema$1.highPrice;
exports.hiringOrganization = schema$1.hiringOrganization;
exports.holdingArchive = schema$1.holdingArchive;
exports.homeLocation = schema$1.homeLocation;
exports.homeTeam = schema$1.homeTeam;
exports.honorificPrefix = schema$1.honorificPrefix;
exports.honorificSuffix = schema$1.honorificSuffix;
exports.hospitalAffiliation = schema$1.hospitalAffiliation;
exports.hostingOrganization = schema$1.hostingOrganization;
exports.hoursAvailable = schema$1.hoursAvailable;
exports.howPerformed = schema$1.howPerformed;
exports.httpMethod = schema$1.httpMethod;
exports.iataCode = schema$1.iataCode;
exports.icaoCode = schema$1.icaoCode;
exports.identifier = schema$1.identifier;
exports.identifyingExam = schema$1.identifyingExam;
exports.identifyingTest = schema$1.identifyingTest;
exports.illustrator = schema$1.illustrator;
exports.image = schema$1.image;
exports.imagingTechnique = schema$1.imagingTechnique;
exports.inAlbum = schema$1.inAlbum;
exports.inBroadcastLineup = schema$1.inBroadcastLineup;
exports.inChI = schema$1.inChI;
exports.inChIKey = schema$1.inChIKey;
exports.inCodeSet = schema$1.inCodeSet;
exports.inDefinedTermSet = schema$1.inDefinedTermSet;
exports.inLanguage = schema$1.inLanguage;
exports.inPlaylist = schema$1.inPlaylist;
exports.inProductGroupWithID = schema$1.inProductGroupWithID;
exports.inStoreReturnsOffered = schema$1.inStoreReturnsOffered;
exports.inSupportOf = schema$1.inSupportOf;
exports.incentiveCompensation = schema$1.incentiveCompensation;
exports.incentives = schema$1.incentives;
exports.includedComposition = schema$1.includedComposition;
exports.includedDataCatalog = schema$1.includedDataCatalog;
exports.includedInDataCatalog = schema$1.includedInDataCatalog;
exports.includedInHealthInsurancePlan = schema$1.includedInHealthInsurancePlan;
exports.includedRiskFactor = schema$1.includedRiskFactor;
exports.includesAttraction = schema$1.includesAttraction;
exports.includesHealthPlanFormulary = schema$1.includesHealthPlanFormulary;
exports.includesHealthPlanNetwork = schema$1.includesHealthPlanNetwork;
exports.includesObject = schema$1.includesObject;
exports.increasesRiskOf = schema$1.increasesRiskOf;
exports.industry = schema$1.industry;
exports.ineligibleRegion = schema$1.ineligibleRegion;
exports.infectiousAgent = schema$1.infectiousAgent;
exports.infectiousAgentClass = schema$1.infectiousAgentClass;
exports.ingredients = schema$1.ingredients;
exports.inker = schema$1.inker;
exports.insertion = schema$1.insertion;
exports.installUrl = schema$1.installUrl;
exports.instructor = schema$1.instructor;
exports.instrument = schema$1.instrument;
exports.intensity = schema$1.intensity;
exports.interactingDrug = schema$1.interactingDrug;
exports.interactionCount = schema$1.interactionCount;
exports.interactionService = schema$1.interactionService;
exports.interactionStatistic = schema$1.interactionStatistic;
exports.interactionType = schema$1.interactionType;
exports.interactivityType = schema$1.interactivityType;
exports.interestRate = schema$1.interestRate;
exports.interpretedAsClaim = schema$1.interpretedAsClaim;
exports.inventoryLevel = schema$1.inventoryLevel;
exports.inverseOf = schema$1.inverseOf;
exports.isAcceptingNewPatients = schema$1.isAcceptingNewPatients;
exports.isAccessibleForFree = schema$1.isAccessibleForFree;
exports.isAccessoryOrSparePartFor = schema$1.isAccessoryOrSparePartFor;
exports.isAvailableGenerically = schema$1.isAvailableGenerically;
exports.isBasedOn = schema$1.isBasedOn;
exports.isBasedOnUrl = schema$1.isBasedOnUrl;
exports.isConsumableFor = schema$1.isConsumableFor;
exports.isEncodedByBioChemEntity = schema$1.isEncodedByBioChemEntity;
exports.isFamilyFriendly = schema$1.isFamilyFriendly;
exports.isGift = schema$1.isGift;
exports.isInvolvedInBiologicalProcess = schema$1.isInvolvedInBiologicalProcess;
exports.isLiveBroadcast = schema$1.isLiveBroadcast;
exports.isLocatedInSubcellularLocation = schema$1.isLocatedInSubcellularLocation;
exports.isPartOf = schema$1.isPartOf;
exports.isPartOfBioChemEntity = schema$1.isPartOfBioChemEntity;
exports.isPlanForApartment = schema$1.isPlanForApartment;
exports.isProprietary = schema$1.isProprietary;
exports.isRelatedTo = schema$1.isRelatedTo;
exports.isResizable = schema$1.isResizable;
exports.isSimilarTo = schema$1.isSimilarTo;
exports.isUnlabelledFallback = schema$1.isUnlabelledFallback;
exports.isVariantOf = schema$1.isVariantOf;
exports.isbn = schema$1.isbn;
exports.isicV4 = schema$1.isicV4;
exports.iso6523Code = schema$1.iso6523Code;
exports.isrcCode = schema$1.isrcCode;
exports.issn = schema$1.issn;
exports.issueNumber = schema$1.issueNumber;
exports.issuedBy = schema$1.issuedBy;
exports.issuedThrough = schema$1.issuedThrough;
exports.iswcCode = schema$1.iswcCode;
exports.item = schema$1.item;
exports.itemCondition = schema$1.itemCondition;
exports.itemDefectReturnFees = schema$1.itemDefectReturnFees;
exports.itemDefectReturnLabelSource = schema$1.itemDefectReturnLabelSource;
exports.itemDefectReturnShippingFeesAmount = schema$1.itemDefectReturnShippingFeesAmount;
exports.itemListElement = schema$1.itemListElement;
exports.itemListOrder = schema$1.itemListOrder;
exports.itemLocation = schema$1.itemLocation;
exports.itemOffered = schema$1.itemOffered;
exports.itemReviewed = schema$1.itemReviewed;
exports.itemShipped = schema$1.itemShipped;
exports.itinerary = schema$1.itinerary;
exports.iupacName = schema$1.iupacName;
exports.jobBenefits = schema$1.jobBenefits;
exports.jobImmediateStart = schema$1.jobImmediateStart;
exports.jobLocation = schema$1.jobLocation;
exports.jobLocationType = schema$1.jobLocationType;
exports.jobStartDate = schema$1.jobStartDate;
exports.jobTitle = schema$1.jobTitle;
exports.jurisdiction = schema$1.jurisdiction;
exports.keywords = schema$1.keywords;
exports.knownVehicleDamages = schema$1.knownVehicleDamages;
exports.knows = schema$1.knows;
exports.knowsAbout = schema$1.knowsAbout;
exports.knowsLanguage = schema$1.knowsLanguage;
exports.labelDetails = schema$1.labelDetails;
exports.landlord = schema$1.landlord;
exports.language = schema$1.language;
exports.lastReviewed = schema$1.lastReviewed;
exports.latitude = schema$1.latitude;
exports.layoutImage = schema$1.layoutImage;
exports.learningResourceType = schema$1.learningResourceType;
exports.leaseLength = schema$1.leaseLength;
exports.legalName = schema$1.legalName;
exports.legalStatus = schema$1.legalStatus;
exports.legislationApplies = schema$1.legislationApplies;
exports.legislationChanges = schema$1.legislationChanges;
exports.legislationConsolidates = schema$1.legislationConsolidates;
exports.legislationDate = schema$1.legislationDate;
exports.legislationDateVersion = schema$1.legislationDateVersion;
exports.legislationIdentifier = schema$1.legislationIdentifier;
exports.legislationJurisdiction = schema$1.legislationJurisdiction;
exports.legislationLegalForce = schema$1.legislationLegalForce;
exports.legislationLegalValue = schema$1.legislationLegalValue;
exports.legislationPassedBy = schema$1.legislationPassedBy;
exports.legislationResponsible = schema$1.legislationResponsible;
exports.legislationTransposes = schema$1.legislationTransposes;
exports.legislationType = schema$1.legislationType;
exports.leiCode = schema$1.leiCode;
exports.lender = schema$1.lender;
exports.lesser = schema$1.lesser;
exports.lesserOrEqual = schema$1.lesserOrEqual;
exports.letterer = schema$1.letterer;
exports.license = schema$1.license;
exports.line = schema$1.line;
exports.linkRelationship = schema$1.linkRelationship;
exports.liveBlogUpdate = schema$1.liveBlogUpdate;
exports.loanMortgageMandateAmount = schema$1.loanMortgageMandateAmount;
exports.loanPaymentAmount = schema$1.loanPaymentAmount;
exports.loanPaymentFrequency = schema$1.loanPaymentFrequency;
exports.loanRepaymentForm = schema$1.loanRepaymentForm;
exports.loanTerm = schema$1.loanTerm;
exports.loanType = schema$1.loanType;
exports.location = schema$1.location;
exports.locationCreated = schema$1.locationCreated;
exports.lodgingUnitDescription = schema$1.lodgingUnitDescription;
exports.lodgingUnitType = schema$1.lodgingUnitType;
exports.logo = schema$1.logo;
exports.longitude = schema$1.longitude;
exports.loser = schema$1.loser;
exports.lowPrice = schema$1.lowPrice;
exports.lyricist = schema$1.lyricist;
exports.lyrics = schema$1.lyrics;
exports.mainContentOfPage = schema$1.mainContentOfPage;
exports.mainEntity = schema$1.mainEntity;
exports.mainEntityOfPage = schema$1.mainEntityOfPage;
exports.maintainer = schema$1.maintainer;
exports.makesOffer = schema$1.makesOffer;
exports.manufacturer = schema$1.manufacturer;
exports.map = schema$1.map;
exports.mapType = schema$1.mapType;
exports.maps = schema$1.maps;
exports.marginOfError = schema$1.marginOfError;
exports.masthead = schema$1.masthead;
exports.material = schema$1.material;
exports.materialExtent = schema$1.materialExtent;
exports.mathExpression = schema$1.mathExpression;
exports.maxPrice = schema$1.maxPrice;
exports.maxValue = schema$1.maxValue;
exports.maximumAttendeeCapacity = schema$1.maximumAttendeeCapacity;
exports.maximumEnrollment = schema$1.maximumEnrollment;
exports.maximumIntake = schema$1.maximumIntake;
exports.maximumPhysicalAttendeeCapacity = schema$1.maximumPhysicalAttendeeCapacity;
exports.maximumVirtualAttendeeCapacity = schema$1.maximumVirtualAttendeeCapacity;
exports.mealService = schema$1.mealService;
exports.measuredProperty = schema$1.measuredProperty;
exports.measurementDenominator = schema$1.measurementDenominator;
exports.measurementMethod = schema$1.measurementMethod;
exports.measurementQualifier = schema$1.measurementQualifier;
exports.measurementTechnique = schema$1.measurementTechnique;
exports.mechanismOfAction = schema$1.mechanismOfAction;
exports.mediaAuthenticityCategory = schema$1.mediaAuthenticityCategory;
exports.mediaItemAppearance = schema$1.mediaItemAppearance;
exports.median = schema$1.median;
exports.medicalAudience = schema$1.medicalAudience;
exports.medicalSpecialty = schema$1.medicalSpecialty;
exports.medicineSystem = schema$1.medicineSystem;
exports.meetsEmissionStandard = schema$1.meetsEmissionStandard;
exports.member = schema$1.member;
exports.memberOf = schema$1.memberOf;
exports.members = schema$1.members;
exports.membershipNumber = schema$1.membershipNumber;
exports.membershipPointsEarned = schema$1.membershipPointsEarned;
exports.memoryRequirements = schema$1.memoryRequirements;
exports.mentions = schema$1.mentions;
exports.menu = schema$1.menu;
exports.menuAddOn = schema$1.menuAddOn;
exports.merchant = schema$1.merchant;
exports.merchantReturnDays = schema$1.merchantReturnDays;
exports.merchantReturnLink = schema$1.merchantReturnLink;
exports.messageAttachment = schema$1.messageAttachment;
exports.mileageFromOdometer = schema$1.mileageFromOdometer;
exports.minPrice = schema$1.minPrice;
exports.minValue = schema$1.minValue;
exports.minimumPaymentDue = schema$1.minimumPaymentDue;
exports.missionCoveragePrioritiesPolicy = schema$1.missionCoveragePrioritiesPolicy;
exports.mobileUrl = schema$1.mobileUrl;
exports.model = schema$1.model;
exports.modelDate = schema$1.modelDate;
exports.modifiedTime = schema$1.modifiedTime;
exports.molecularFormula = schema$1.molecularFormula;
exports.molecularWeight = schema$1.molecularWeight;
exports.monoisotopicMolecularWeight = schema$1.monoisotopicMolecularWeight;
exports.monthlyMinimumRepaymentAmount = schema$1.monthlyMinimumRepaymentAmount;
exports.monthsOfExperience = schema$1.monthsOfExperience;
exports.mpn = schema$1.mpn;
exports.multipleValues = schema$1.multipleValues;
exports.muscleAction = schema$1.muscleAction;
exports.musicArrangement = schema$1.musicArrangement;
exports.musicBy = schema$1.musicBy;
exports.musicCompositionForm = schema$1.musicCompositionForm;
exports.musicGroupMember = schema$1.musicGroupMember;
exports.musicReleaseFormat = schema$1.musicReleaseFormat;
exports.musicalKey = schema$1.musicalKey;
exports.naics = schema$1.naics;
exports.name = schema$1.name;
exports.namedPosition = schema$1.namedPosition;
exports.nationality = schema$1.nationality;
exports.naturalProgression = schema$1.naturalProgression;
exports.negativeNotes = schema$1.negativeNotes;
exports.nerve = schema$1.nerve;
exports.nerveMotor = schema$1.nerveMotor;
exports.netWorth = schema$1.netWorth;
exports.newsUpdatesAndGuidelines = schema$1.newsUpdatesAndGuidelines;
exports.nextItem = schema$1.nextItem;
exports.noBylinesPolicy = schema$1.noBylinesPolicy;
exports.nonEqual = schema$1.nonEqual;
exports.nonProprietaryName = schema$1.nonProprietaryName;
exports.nonprofitStatus = schema$1.nonprofitStatus;
exports.normalRange = schema$1.normalRange;
exports.nsn = schema$1.nsn;
exports.numAdults = schema$1.numAdults;
exports.numChildren = schema$1.numChildren;
exports.numConstraints = schema$1.numConstraints;
exports.numTracks = schema$1.numTracks;
exports.numberOfAccommodationUnits = schema$1.numberOfAccommodationUnits;
exports.numberOfAirbags = schema$1.numberOfAirbags;
exports.numberOfAvailableAccommodationUnits = schema$1.numberOfAvailableAccommodationUnits;
exports.numberOfAxles = schema$1.numberOfAxles;
exports.numberOfBathroomsTotal = schema$1.numberOfBathroomsTotal;
exports.numberOfBedrooms = schema$1.numberOfBedrooms;
exports.numberOfBeds = schema$1.numberOfBeds;
exports.numberOfCredits = schema$1.numberOfCredits;
exports.numberOfDoors = schema$1.numberOfDoors;
exports.numberOfEmployees = schema$1.numberOfEmployees;
exports.numberOfEpisodes = schema$1.numberOfEpisodes;
exports.numberOfForwardGears = schema$1.numberOfForwardGears;
exports.numberOfFullBathrooms = schema$1.numberOfFullBathrooms;
exports.numberOfItems = schema$1.numberOfItems;
exports.numberOfLoanPayments = schema$1.numberOfLoanPayments;
exports.numberOfPages = schema$1.numberOfPages;
exports.numberOfPartialBathrooms = schema$1.numberOfPartialBathrooms;
exports.numberOfPlayers = schema$1.numberOfPlayers;
exports.numberOfPreviousOwners = schema$1.numberOfPreviousOwners;
exports.numberOfRooms = schema$1.numberOfRooms;
exports.numberOfSeasons = schema$1.numberOfSeasons;
exports.numberedPosition = schema$1.numberedPosition;
exports.nutrition = schema$1.nutrition;
exports.object = schema$1.object;
exports.observationAbout = schema$1.observationAbout;
exports.observationDate = schema$1.observationDate;
exports.observationPeriod = schema$1.observationPeriod;
exports.occupancy = schema$1.occupancy;
exports.occupationLocation = schema$1.occupationLocation;
exports.occupationalCategory = schema$1.occupationalCategory;
exports.occupationalCredentialAwarded = schema$1.occupationalCredentialAwarded;
exports.offerCount = schema$1.offerCount;
exports.offeredBy = schema$1.offeredBy;
exports.offers = schema$1.offers;
exports.offersPrescriptionByMail = schema$1.offersPrescriptionByMail;
exports.openingHours = schema$1.openingHours;
exports.openingHoursSpecification = schema$1.openingHoursSpecification;
exports.opens = schema$1.opens;
exports.operatingSystem = schema$1.operatingSystem;
exports.opponent = schema$1.opponent;
exports.option = schema$1.option;
exports.orderDate = schema$1.orderDate;
exports.orderDelivery = schema$1.orderDelivery;
exports.orderItemNumber = schema$1.orderItemNumber;
exports.orderItemStatus = schema$1.orderItemStatus;
exports.orderNumber = schema$1.orderNumber;
exports.orderQuantity = schema$1.orderQuantity;
exports.orderStatus = schema$1.orderStatus;
exports.orderedItem = schema$1.orderedItem;
exports.organizer = schema$1.organizer;
exports.originAddress = schema$1.originAddress;
exports.originalMediaContextDescription = schema$1.originalMediaContextDescription;
exports.originalMediaLink = schema$1.originalMediaLink;
exports.originatesFrom = schema$1.originatesFrom;
exports.overdosage = schema$1.overdosage;
exports.ownedFrom = schema$1.ownedFrom;
exports.ownedThrough = schema$1.ownedThrough;
exports.ownershipFundingInfo = schema$1.ownershipFundingInfo;
exports.owns = schema$1.owns;
exports.pageEnd = schema$1.pageEnd;
exports.pageStart = schema$1.pageStart;
exports.pagination = schema$1.pagination;
exports.parent = schema$1.parent;
exports.parentItem = schema$1.parentItem;
exports.parentOrganization = schema$1.parentOrganization;
exports.parentService = schema$1.parentService;
exports.parentTaxon = schema$1.parentTaxon;
exports.parents = schema$1.parents;
exports.partOfEpisode = schema$1.partOfEpisode;
exports.partOfInvoice = schema$1.partOfInvoice;
exports.partOfOrder = schema$1.partOfOrder;
exports.partOfSeason = schema$1.partOfSeason;
exports.partOfSeries = schema$1.partOfSeries;
exports.partOfSystem = schema$1.partOfSystem;
exports.partOfTVSeries = schema$1.partOfTVSeries;
exports.partOfTrip = schema$1.partOfTrip;
exports.participant = schema$1.participant;
exports.partySize = schema$1.partySize;
exports.passengerPriorityStatus = schema$1.passengerPriorityStatus;
exports.passengerSequenceNumber = schema$1.passengerSequenceNumber;
exports.pathophysiology = schema$1.pathophysiology;
exports.pattern = schema$1.pattern;
exports.payload = schema$1.payload;
exports.paymentAccepted = schema$1.paymentAccepted;
exports.paymentDue = schema$1.paymentDue;
exports.paymentDueDate = schema$1.paymentDueDate;
exports.paymentMethod = schema$1.paymentMethod;
exports.paymentMethodId = schema$1.paymentMethodId;
exports.paymentStatus = schema$1.paymentStatus;
exports.paymentUrl = schema$1.paymentUrl;
exports.penciler = schema$1.penciler;
exports.percentile10 = schema$1.percentile10;
exports.percentile25 = schema$1.percentile25;
exports.percentile75 = schema$1.percentile75;
exports.percentile90 = schema$1.percentile90;
exports.performTime = schema$1.performTime;
exports.performer = schema$1.performer;
exports.performerIn = schema$1.performerIn;
exports.performers = schema$1.performers;
exports.permissionType = schema$1.permissionType;
exports.permissions = schema$1.permissions;
exports.permitAudience = schema$1.permitAudience;
exports.permittedUsage = schema$1.permittedUsage;
exports.petsAllowed = schema$1.petsAllowed;
exports.phoneticText = schema$1.phoneticText;
exports.photo = schema$1.photo;
exports.photos = schema$1.photos;
exports.physicalRequirement = schema$1.physicalRequirement;
exports.physiologicalBenefits = schema$1.physiologicalBenefits;
exports.pickupLocation = schema$1.pickupLocation;
exports.pickupTime = schema$1.pickupTime;
exports.playMode = schema$1.playMode;
exports.playerType = schema$1.playerType;
exports.playersOnline = schema$1.playersOnline;
exports.polygon = schema$1.polygon;
exports.populationType = schema$1.populationType;
exports.position = schema$1.position;
exports.positiveNotes = schema$1.positiveNotes;
exports.possibleComplication = schema$1.possibleComplication;
exports.possibleTreatment = schema$1.possibleTreatment;
exports.postOfficeBoxNumber = schema$1.postOfficeBoxNumber;
exports.postOp = schema$1.postOp;
exports.postalCode = schema$1.postalCode;
exports.postalCodeBegin = schema$1.postalCodeBegin;
exports.postalCodeEnd = schema$1.postalCodeEnd;
exports.postalCodePrefix = schema$1.postalCodePrefix;
exports.postalCodeRange = schema$1.postalCodeRange;
exports.potentialAction = schema$1.potentialAction;
exports.potentialUse = schema$1.potentialUse;
exports.preOp = schema$1.preOp;
exports.predecessorOf = schema$1.predecessorOf;
exports.pregnancyCategory = schema$1.pregnancyCategory;
exports.pregnancyWarning = schema$1.pregnancyWarning;
exports.prepTime = schema$1.prepTime;
exports.preparation = schema$1.preparation;
exports.prescribingInfo = schema$1.prescribingInfo;
exports.prescriptionStatus = schema$1.prescriptionStatus;
exports.previousItem = schema$1.previousItem;
exports.previousStartDate = schema$1.previousStartDate;
exports.price = schema$1.price;
exports.priceComponent = schema$1.priceComponent;
exports.priceComponentType = schema$1.priceComponentType;
exports.priceCurrency = schema$1.priceCurrency;
exports.priceRange = schema$1.priceRange;
exports.priceSpecification = schema$1.priceSpecification;
exports.priceType = schema$1.priceType;
exports.priceValidUntil = schema$1.priceValidUntil;
exports.primaryImageOfPage = schema$1.primaryImageOfPage;
exports.primaryPrevention = schema$1.primaryPrevention;
exports.printColumn = schema$1.printColumn;
exports.printEdition = schema$1.printEdition;
exports.printPage = schema$1.printPage;
exports.printSection = schema$1.printSection;
exports.procedure = schema$1.procedure;
exports.procedureType = schema$1.procedureType;
exports.processingTime = schema$1.processingTime;
exports.processorRequirements = schema$1.processorRequirements;
exports.producer = schema$1.producer;
exports.produces = schema$1.produces;
exports.productGroupID = schema$1.productGroupID;
exports.productID = schema$1.productID;
exports.productReturnDays = schema$1.productReturnDays;
exports.productReturnLink = schema$1.productReturnLink;
exports.productSupported = schema$1.productSupported;
exports.productionCompany = schema$1.productionCompany;
exports.productionDate = schema$1.productionDate;
exports.proficiencyLevel = schema$1.proficiencyLevel;
exports.programMembershipUsed = schema$1.programMembershipUsed;
exports.programName = schema$1.programName;
exports.programPrerequisites = schema$1.programPrerequisites;
exports.programType = schema$1.programType;
exports.programmingLanguage = schema$1.programmingLanguage;
exports.programmingModel = schema$1.programmingModel;
exports.propertyID = schema$1.propertyID;
exports.proprietaryName = schema$1.proprietaryName;
exports.proteinContent = schema$1.proteinContent;
exports.provider = schema$1.provider;
exports.providerMobility = schema$1.providerMobility;
exports.providesBroadcastService = schema$1.providesBroadcastService;
exports.providesService = schema$1.providesService;
exports.publicAccess = schema$1.publicAccess;
exports.publicTransportClosuresInfo = schema$1.publicTransportClosuresInfo;
exports.publication = schema$1.publication;
exports.publicationType = schema$1.publicationType;
exports.publishedBy = schema$1.publishedBy;
exports.publishedOn = schema$1.publishedOn;
exports.publisher = schema$1.publisher;
exports.publisherImprint = schema$1.publisherImprint;
exports.publishingPrinciples = schema$1.publishingPrinciples;
exports.purchaseDate = schema$1.purchaseDate;
exports.qualifications = schema$1.qualifications;
exports.quarantineGuidelines = schema$1.quarantineGuidelines;
exports.query = schema$1.query;
exports.quest = schema$1.quest;
exports.question = schema$1.question;
exports.rangeIncludes = schema$1.rangeIncludes;
exports.ratingCount = schema$1.ratingCount;
exports.ratingExplanation = schema$1.ratingExplanation;
exports.ratingValue = schema$1.ratingValue;
exports.readBy = schema$1.readBy;
exports.readonlyValue = schema$1.readonlyValue;
exports.realEstateAgent = schema$1.realEstateAgent;
exports.recipe = schema$1.recipe;
exports.recipeCategory = schema$1.recipeCategory;
exports.recipeCuisine = schema$1.recipeCuisine;
exports.recipeIngredient = schema$1.recipeIngredient;
exports.recipeInstructions = schema$1.recipeInstructions;
exports.recipeYield = schema$1.recipeYield;
exports.recipient = schema$1.recipient;
exports.recognizedBy = schema$1.recognizedBy;
exports.recognizingAuthority = schema$1.recognizingAuthority;
exports.recommendationStrength = schema$1.recommendationStrength;
exports.recommendedIntake = schema$1.recommendedIntake;
exports.recordLabel = schema$1.recordLabel;
exports.recordedAs = schema$1.recordedAs;
exports.recordedAt = schema$1.recordedAt;
exports.recordedIn = schema$1.recordedIn;
exports.recordingOf = schema$1.recordingOf;
exports.recourseLoan = schema$1.recourseLoan;
exports.referenceQuantity = schema$1.referenceQuantity;
exports.referencesOrder = schema$1.referencesOrder;
exports.refundType = schema$1.refundType;
exports.regionDrained = schema$1.regionDrained;
exports.regionsAllowed = schema$1.regionsAllowed;
exports.relatedAnatomy = schema$1.relatedAnatomy;
exports.relatedCondition = schema$1.relatedCondition;
exports.relatedDrug = schema$1.relatedDrug;
exports.relatedLink = schema$1.relatedLink;
exports.relatedStructure = schema$1.relatedStructure;
exports.relatedTherapy = schema$1.relatedTherapy;
exports.relatedTo = schema$1.relatedTo;
exports.releaseDate = schema$1.releaseDate;
exports.releaseNotes = schema$1.releaseNotes;
exports.releaseOf = schema$1.releaseOf;
exports.releasedEvent = schema$1.releasedEvent;
exports.relevantOccupation = schema$1.relevantOccupation;
exports.relevantSpecialty = schema$1.relevantSpecialty;
exports.remainingAttendeeCapacity = schema$1.remainingAttendeeCapacity;
exports.renegotiableLoan = schema$1.renegotiableLoan;
exports.repeatCount = schema$1.repeatCount;
exports.repeatFrequency = schema$1.repeatFrequency;
exports.repetitions = schema$1.repetitions;
exports.replacee = schema$1.replacee;
exports.replacer = schema$1.replacer;
exports.replyToUrl = schema$1.replyToUrl;
exports.reportNumber = schema$1.reportNumber;
exports.representativeOfPage = schema$1.representativeOfPage;
exports.requiredCollateral = schema$1.requiredCollateral;
exports.requiredGender = schema$1.requiredGender;
exports.requiredMaxAge = schema$1.requiredMaxAge;
exports.requiredMinAge = schema$1.requiredMinAge;
exports.requiredQuantity = schema$1.requiredQuantity;
exports.requirements = schema$1.requirements;
exports.requiresSubscription = schema$1.requiresSubscription;
exports.reservationFor = schema$1.reservationFor;
exports.reservationId = schema$1.reservationId;
exports.reservationStatus = schema$1.reservationStatus;
exports.reservedTicket = schema$1.reservedTicket;
exports.responsibilities = schema$1.responsibilities;
exports.restPeriods = schema$1.restPeriods;
exports.restockingFee = schema$1.restockingFee;
exports.result = schema$1.result;
exports.resultComment = schema$1.resultComment;
exports.resultReview = schema$1.resultReview;
exports.returnFees = schema$1.returnFees;
exports.returnLabelSource = schema$1.returnLabelSource;
exports.returnMethod = schema$1.returnMethod;
exports.returnPolicyCategory = schema$1.returnPolicyCategory;
exports.returnPolicyCountry = schema$1.returnPolicyCountry;
exports.returnPolicySeasonalOverride = schema$1.returnPolicySeasonalOverride;
exports.returnShippingFeesAmount = schema$1.returnShippingFeesAmount;
exports.review = schema$1.review;
exports.reviewAspect = schema$1.reviewAspect;
exports.reviewBody = schema$1.reviewBody;
exports.reviewCount = schema$1.reviewCount;
exports.reviewRating = schema$1.reviewRating;
exports.reviewedBy = schema$1.reviewedBy;
exports.reviews = schema$1.reviews;
exports.riskFactor = schema$1.riskFactor;
exports.risks = schema$1.risks;
exports.roleName = schema$1.roleName;
exports.roofLoad = schema$1.roofLoad;
exports.rsvpResponse = schema$1.rsvpResponse;
exports.runsTo = schema$1.runsTo;
exports.runtime = schema$1.runtime;
exports.runtimePlatform = schema$1.runtimePlatform;
exports.rxcui = schema$1.rxcui;
exports.safetyConsideration = schema$1.safetyConsideration;
exports.salaryCurrency = schema$1.salaryCurrency;
exports.salaryUponCompletion = schema$1.salaryUponCompletion;
exports.sameAs = schema$1.sameAs;
exports.sampleType = schema$1.sampleType;
exports.saturatedFatContent = schema$1.saturatedFatContent;
exports.scheduleTimezone = schema$1.scheduleTimezone;
exports.scheduledPaymentDate = schema$1.scheduledPaymentDate;
exports.scheduledTime = schema$1.scheduledTime;
exports.schemaVersion = schema$1.schemaVersion;
exports.schoolClosuresInfo = schema$1.schoolClosuresInfo;
exports.screenCount = schema$1.screenCount;
exports.screenshot = schema$1.screenshot;
exports.sdDatePublished = schema$1.sdDatePublished;
exports.sdLicense = schema$1.sdLicense;
exports.sdPublisher = schema$1.sdPublisher;
exports.season = schema$1.season;
exports.seasonNumber = schema$1.seasonNumber;
exports.seasons = schema$1.seasons;
exports.seatNumber = schema$1.seatNumber;
exports.seatRow = schema$1.seatRow;
exports.seatSection = schema$1.seatSection;
exports.seatingCapacity = schema$1.seatingCapacity;
exports.seatingType = schema$1.seatingType;
exports.secondaryPrevention = schema$1.secondaryPrevention;
exports.securityClearanceRequirement = schema$1.securityClearanceRequirement;
exports.securityScreening = schema$1.securityScreening;
exports.seeks = schema$1.seeks;
exports.seller = schema$1.seller;
exports.sender = schema$1.sender;
exports.sensoryRequirement = schema$1.sensoryRequirement;
exports.sensoryUnit = schema$1.sensoryUnit;
exports.serialNumber = schema$1.serialNumber;
exports.seriousAdverseOutcome = schema$1.seriousAdverseOutcome;
exports.serverStatus = schema$1.serverStatus;
exports.servesCuisine = schema$1.servesCuisine;
exports.serviceArea = schema$1.serviceArea;
exports.serviceAudience = schema$1.serviceAudience;
exports.serviceLocation = schema$1.serviceLocation;
exports.serviceOperator = schema$1.serviceOperator;
exports.serviceOutput = schema$1.serviceOutput;
exports.servicePhone = schema$1.servicePhone;
exports.servicePostalAddress = schema$1.servicePostalAddress;
exports.serviceSmsNumber = schema$1.serviceSmsNumber;
exports.serviceType = schema$1.serviceType;
exports.serviceUrl = schema$1.serviceUrl;
exports.servingSize = schema$1.servingSize;
exports.sha256 = schema$1.sha256;
exports.sharedContent = schema$1.sharedContent;
exports.shippingDestination = schema$1.shippingDestination;
exports.shippingDetails = schema$1.shippingDetails;
exports.shippingLabel = schema$1.shippingLabel;
exports.shippingOrigin = schema$1.shippingOrigin;
exports.shippingRate = schema$1.shippingRate;
exports.shippingSettingsLink = schema$1.shippingSettingsLink;
exports.sibling = schema$1.sibling;
exports.siblings = schema$1.siblings;
exports.signDetected = schema$1.signDetected;
exports.signOrSymptom = schema$1.signOrSymptom;
exports.significance = schema$1.significance;
exports.significantLink = schema$1.significantLink;
exports.significantLinks = schema$1.significantLinks;
exports.size = schema$1.size;
exports.sizeGroup = schema$1.sizeGroup;
exports.sizeSystem = schema$1.sizeSystem;
exports.skills = schema$1.skills;
exports.sku = schema$1.sku;
exports.slogan = schema$1.slogan;
exports.smiles = schema$1.smiles;
exports.smokingAllowed = schema$1.smokingAllowed;
exports.sodiumContent = schema$1.sodiumContent;
exports.softwareAddOn = schema$1.softwareAddOn;
exports.softwareHelp = schema$1.softwareHelp;
exports.softwareRequirements = schema$1.softwareRequirements;
exports.softwareVersion = schema$1.softwareVersion;
exports.sourceOrganization = schema$1.sourceOrganization;
exports.sourcedFrom = schema$1.sourcedFrom;
exports.spatial = schema$1.spatial;
exports.spatialCoverage = schema$1.spatialCoverage;
exports.speakable = schema$1.speakable;
exports.specialCommitments = schema$1.specialCommitments;
exports.specialOpeningHoursSpecification = schema$1.specialOpeningHoursSpecification;
exports.specialty = schema$1.specialty;
exports.speechToTextMarkup = schema$1.speechToTextMarkup;
exports.speed = schema$1.speed;
exports.spokenByCharacter = schema$1.spokenByCharacter;
exports.sponsor = schema$1.sponsor;
exports.sport = schema$1.sport;
exports.sportsActivityLocation = schema$1.sportsActivityLocation;
exports.sportsEvent = schema$1.sportsEvent;
exports.sportsTeam = schema$1.sportsTeam;
exports.spouse = schema$1.spouse;
exports.stage = schema$1.stage;
exports.stageAsNumber = schema$1.stageAsNumber;
exports.starRating = schema$1.starRating;
exports.startDate = schema$1.startDate;
exports.startOffset = schema$1.startOffset;
exports.startTime = schema$1.startTime;
exports.statType = schema$1.statType;
exports.status = schema$1.status;
exports.steeringPosition = schema$1.steeringPosition;
exports.step = schema$1.step;
exports.stepValue = schema$1.stepValue;
exports.steps = schema$1.steps;
exports.storageRequirements = schema$1.storageRequirements;
exports.streetAddress = schema$1.streetAddress;
exports.strengthUnit = schema$1.strengthUnit;
exports.strengthValue = schema$1.strengthValue;
exports.structuralClass = schema$1.structuralClass;
exports.study = schema$1.study;
exports.studyDesign = schema$1.studyDesign;
exports.studyLocation = schema$1.studyLocation;
exports.studySubject = schema$1.studySubject;
exports.stupidProperty = schema$1.stupidProperty;
exports.subEvent = schema$1.subEvent;
exports.subEvents = schema$1.subEvents;
exports.subOrganization = schema$1.subOrganization;
exports.subReservation = schema$1.subReservation;
exports.subStageSuffix = schema$1.subStageSuffix;
exports.subStructure = schema$1.subStructure;
exports.subTest = schema$1.subTest;
exports.subTrip = schema$1.subTrip;
exports.subjectOf = schema$1.subjectOf;
exports.subtitleLanguage = schema$1.subtitleLanguage;
exports.successorOf = schema$1.successorOf;
exports.sugarContent = schema$1.sugarContent;
exports.suggestedAge = schema$1.suggestedAge;
exports.suggestedAnswer = schema$1.suggestedAnswer;
exports.suggestedGender = schema$1.suggestedGender;
exports.suggestedMaxAge = schema$1.suggestedMaxAge;
exports.suggestedMeasurement = schema$1.suggestedMeasurement;
exports.suggestedMinAge = schema$1.suggestedMinAge;
exports.suitableForDiet = schema$1.suitableForDiet;
exports.superEvent = schema$1.superEvent;
exports.supersededBy = schema$1.supersededBy;
exports.supply = schema$1.supply;
exports.supplyTo = schema$1.supplyTo;
exports.supportingData = schema$1.supportingData;
exports.surface = schema$1.surface;
exports.syllabusSections = schema$1.syllabusSections;
exports.target = schema$1.target;
exports.targetCollection = schema$1.targetCollection;
exports.targetDescription = schema$1.targetDescription;
exports.targetName = schema$1.targetName;
exports.targetPlatform = schema$1.targetPlatform;
exports.targetPopulation = schema$1.targetPopulation;
exports.targetProduct = schema$1.targetProduct;
exports.targetUrl = schema$1.targetUrl;
exports.taxID = schema$1.taxID;
exports.taxonRank = schema$1.taxonRank;
exports.taxonomicRange = schema$1.taxonomicRange;
exports.teaches = schema$1.teaches;
exports.telephone = schema$1.telephone;
exports.temporal = schema$1.temporal;
exports.temporalCoverage = schema$1.temporalCoverage;
exports.termCode = schema$1.termCode;
exports.termDuration = schema$1.termDuration;
exports.termsOfService = schema$1.termsOfService;
exports.termsPerYear = schema$1.termsPerYear;
exports.text = schema$1.text;
exports.textValue = schema$1.textValue;
exports.thumbnail = schema$1.thumbnail;
exports.thumbnailUrl = schema$1.thumbnailUrl;
exports.tickerSymbol = schema$1.tickerSymbol;
exports.ticketNumber = schema$1.ticketNumber;
exports.ticketToken = schema$1.ticketToken;
exports.ticketedSeat = schema$1.ticketedSeat;
exports.timeOfDay = schema$1.timeOfDay;
exports.timeRequired = schema$1.timeRequired;
exports.timeToComplete = schema$1.timeToComplete;
exports.tissueSample = schema$1.tissueSample;
exports.title = schema$1.title;
exports.titleEIDR = schema$1.titleEIDR;
exports.toLocation = schema$1.toLocation;
exports.toRecipient = schema$1.toRecipient;
exports.tocContinuation = schema$1.tocContinuation;
exports.tocEntry = schema$1.tocEntry;
exports.tongueWeight = schema$1.tongueWeight;
exports.tool = schema$1.tool;
exports.torque = schema$1.torque;
exports.totalHistoricalEnrollment = schema$1.totalHistoricalEnrollment;
exports.totalJobOpenings = schema$1.totalJobOpenings;
exports.totalPaymentDue = schema$1.totalPaymentDue;
exports.totalPrice = schema$1.totalPrice;
exports.totalTime = schema$1.totalTime;
exports.tourBookingPage = schema$1.tourBookingPage;
exports.touristType = schema$1.touristType;
exports.track = schema$1.track;
exports.trackingNumber = schema$1.trackingNumber;
exports.trackingUrl = schema$1.trackingUrl;
exports.tracks = schema$1.tracks;
exports.trailer = schema$1.trailer;
exports.trailerWeight = schema$1.trailerWeight;
exports.trainName = schema$1.trainName;
exports.trainNumber = schema$1.trainNumber;
exports.trainingSalary = schema$1.trainingSalary;
exports.transFatContent = schema$1.transFatContent;
exports.transcript = schema$1.transcript;
exports.transitTime = schema$1.transitTime;
exports.transitTimeLabel = schema$1.transitTimeLabel;
exports.translationOfWork = schema$1.translationOfWork;
exports.translator = schema$1.translator;
exports.transmissionMethod = schema$1.transmissionMethod;
exports.travelBans = schema$1.travelBans;
exports.trialDesign = schema$1.trialDesign;
exports.tributary = schema$1.tributary;
exports.tripOrigin = schema$1.tripOrigin;
exports.typeOfBed = schema$1.typeOfBed;
exports.typeOfGood = schema$1.typeOfGood;
exports.typicalAgeRange = schema$1.typicalAgeRange;
exports.typicalCreditsPerTerm = schema$1.typicalCreditsPerTerm;
exports.typicalTest = schema$1.typicalTest;
exports.underName = schema$1.underName;
exports.unitCode = schema$1.unitCode;
exports.unitText = schema$1.unitText;
exports.unnamedSourcesPolicy = schema$1.unnamedSourcesPolicy;
exports.unsaturatedFatContent = schema$1.unsaturatedFatContent;
exports.uploadDate = schema$1.uploadDate;
exports.upvoteCount = schema$1.upvoteCount;
exports.url = schema$1.url;
exports.urlTemplate = schema$1.urlTemplate;
exports.usageInfo = schema$1.usageInfo;
exports.usedToDiagnose = schema$1.usedToDiagnose;
exports.userInteractionCount = schema$1.userInteractionCount;
exports.usesDevice = schema$1.usesDevice;
exports.usesHealthPlanIdStandard = schema$1.usesHealthPlanIdStandard;
exports.utterances = schema$1.utterances;
exports.validFor = schema$1.validFor;
exports.validFrom = schema$1.validFrom;
exports.validIn = schema$1.validIn;
exports.validThrough = schema$1.validThrough;
exports.validUntil = schema$1.validUntil;
exports.value = schema$1.value;
exports.valueAddedTaxIncluded = schema$1.valueAddedTaxIncluded;
exports.valueMaxLength = schema$1.valueMaxLength;
exports.valueMinLength = schema$1.valueMinLength;
exports.valueName = schema$1.valueName;
exports.valuePattern = schema$1.valuePattern;
exports.valueReference = schema$1.valueReference;
exports.valueRequired = schema$1.valueRequired;
exports.variableMeasured = schema$1.variableMeasured;
exports.variablesMeasured = schema$1.variablesMeasured;
exports.variantCover = schema$1.variantCover;
exports.variesBy = schema$1.variesBy;
exports.vatID = schema$1.vatID;
exports.vehicleConfiguration = schema$1.vehicleConfiguration;
exports.vehicleEngine = schema$1.vehicleEngine;
exports.vehicleIdentificationNumber = schema$1.vehicleIdentificationNumber;
exports.vehicleInteriorColor = schema$1.vehicleInteriorColor;
exports.vehicleInteriorType = schema$1.vehicleInteriorType;
exports.vehicleModelDate = schema$1.vehicleModelDate;
exports.vehicleSeatingCapacity = schema$1.vehicleSeatingCapacity;
exports.vehicleSpecialUsage = schema$1.vehicleSpecialUsage;
exports.vehicleTransmission = schema$1.vehicleTransmission;
exports.vendor = schema$1.vendor;
exports.verificationFactCheckingPolicy = schema$1.verificationFactCheckingPolicy;
exports.version = schema$1.version;
exports.video = schema$1.video;
exports.videoFormat = schema$1.videoFormat;
exports.videoFrameSize = schema$1.videoFrameSize;
exports.videoQuality = schema$1.videoQuality;
exports.volumeNumber = schema$1.volumeNumber;
exports.warning = schema$1.warning;
exports.warranty = schema$1.warranty;
exports.warrantyPromise = schema$1.warrantyPromise;
exports.warrantyScope = schema$1.warrantyScope;
exports.webCheckinTime = schema$1.webCheckinTime;
exports.webFeed = schema$1.webFeed;
exports.weight = schema$1.weight;
exports.weightTotal = schema$1.weightTotal;
exports.wheelbase = schema$1.wheelbase;
exports.width = schema$1.width;
exports.winner = schema$1.winner;
exports.wordCount = schema$1.wordCount;
exports.workExample = schema$1.workExample;
exports.workFeatured = schema$1.workFeatured;
exports.workHours = schema$1.workHours;
exports.workLocation = schema$1.workLocation;
exports.workPerformed = schema$1.workPerformed;
exports.workPresented = schema$1.workPresented;
exports.workTranslation = schema$1.workTranslation;
exports.workload = schema$1.workload;
exports.worksFor = schema$1.worksFor;
exports.worstRating = schema$1.worstRating;
exports.xpath = schema$1.xpath;
exports.yearBuilt = schema$1.yearBuilt;
exports.yearlyRevenue = schema$1.yearlyRevenue;
exports.yearsInOperation = schema$1.yearsInOperation;
exports.yield__workaround = schema$1.yield__workaround;
