// rdf-namespaces/dist/schema_https.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var schema_https$1 = require('./schema_https-a2331cb2.js');



exports.AMRadioChannel = schema_https$1.AMRadioChannel;
exports.APIReference = schema_https$1.APIReference;
exports.AboutPage = schema_https$1.AboutPage;
exports.AcceptAction = schema_https$1.AcceptAction;
exports.Accommodation = schema_https$1.Accommodation;
exports.AccountingService = schema_https$1.AccountingService;
exports.AchieveAction = schema_https$1.AchieveAction;
exports.Action = schema_https$1.Action;
exports.ActionAccessSpecification = schema_https$1.ActionAccessSpecification;
exports.ActionStatusType = schema_https$1.ActionStatusType;
exports.ActivateAction = schema_https$1.ActivateAction;
exports.AddAction = schema_https$1.AddAction;
exports.AdministrativeArea = schema_https$1.AdministrativeArea;
exports.AdultEntertainment = schema_https$1.AdultEntertainment;
exports.AdultOrientedEnumeration = schema_https$1.AdultOrientedEnumeration;
exports.AdvertiserContentArticle = schema_https$1.AdvertiserContentArticle;
exports.AggregateOffer = schema_https$1.AggregateOffer;
exports.AggregateRating = schema_https$1.AggregateRating;
exports.AgreeAction = schema_https$1.AgreeAction;
exports.Airline = schema_https$1.Airline;
exports.Airport = schema_https$1.Airport;
exports.AlignmentObject = schema_https$1.AlignmentObject;
exports.AllocateAction = schema_https$1.AllocateAction;
exports.AmpStory = schema_https$1.AmpStory;
exports.AmusementPark = schema_https$1.AmusementPark;
exports.AnalysisNewsArticle = schema_https$1.AnalysisNewsArticle;
exports.AnatomicalStructure = schema_https$1.AnatomicalStructure;
exports.AnatomicalSystem = schema_https$1.AnatomicalSystem;
exports.AnimalShelter = schema_https$1.AnimalShelter;
exports.Answer = schema_https$1.Answer;
exports.Apartment = schema_https$1.Apartment;
exports.ApartmentComplex = schema_https$1.ApartmentComplex;
exports.AppendAction = schema_https$1.AppendAction;
exports.ApplyAction = schema_https$1.ApplyAction;
exports.ApprovedIndication = schema_https$1.ApprovedIndication;
exports.Aquarium = schema_https$1.Aquarium;
exports.ArchiveComponent = schema_https$1.ArchiveComponent;
exports.ArchiveOrganization = schema_https$1.ArchiveOrganization;
exports.ArriveAction = schema_https$1.ArriveAction;
exports.ArtGallery = schema_https$1.ArtGallery;
exports.Artery = schema_https$1.Artery;
exports.Article = schema_https$1.Article;
exports.AskAction = schema_https$1.AskAction;
exports.AskPublicNewsArticle = schema_https$1.AskPublicNewsArticle;
exports.AssessAction = schema_https$1.AssessAction;
exports.AssignAction = schema_https$1.AssignAction;
exports.Atlas = schema_https$1.Atlas;
exports.Attorney = schema_https$1.Attorney;
exports.Audience = schema_https$1.Audience;
exports.AudioObject = schema_https$1.AudioObject;
exports.AudioObjectSnapshot = schema_https$1.AudioObjectSnapshot;
exports.Audiobook = schema_https$1.Audiobook;
exports.AuthorizeAction = schema_https$1.AuthorizeAction;
exports.AutoBodyShop = schema_https$1.AutoBodyShop;
exports.AutoDealer = schema_https$1.AutoDealer;
exports.AutoPartsStore = schema_https$1.AutoPartsStore;
exports.AutoRental = schema_https$1.AutoRental;
exports.AutoRepair = schema_https$1.AutoRepair;
exports.AutoWash = schema_https$1.AutoWash;
exports.AutomatedTeller = schema_https$1.AutomatedTeller;
exports.AutomotiveBusiness = schema_https$1.AutomotiveBusiness;
exports.BackgroundNewsArticle = schema_https$1.BackgroundNewsArticle;
exports.Bakery = schema_https$1.Bakery;
exports.BankAccount = schema_https$1.BankAccount;
exports.BankOrCreditUnion = schema_https$1.BankOrCreditUnion;
exports.BarOrPub = schema_https$1.BarOrPub;
exports.Barcode = schema_https$1.Barcode;
exports.Beach = schema_https$1.Beach;
exports.BeautySalon = schema_https$1.BeautySalon;
exports.BedAndBreakfast = schema_https$1.BedAndBreakfast;
exports.BedDetails = schema_https$1.BedDetails;
exports.BedType = schema_https$1.BedType;
exports.BefriendAction = schema_https$1.BefriendAction;
exports.BikeStore = schema_https$1.BikeStore;
exports.BioChemEntity = schema_https$1.BioChemEntity;
exports.Blog = schema_https$1.Blog;
exports.BlogPosting = schema_https$1.BlogPosting;
exports.BloodTest = schema_https$1.BloodTest;
exports.BoardingPolicyType = schema_https$1.BoardingPolicyType;
exports.BoatReservation = schema_https$1.BoatReservation;
exports.BoatTerminal = schema_https$1.BoatTerminal;
exports.BoatTrip = schema_https$1.BoatTrip;
exports.BodyMeasurementTypeEnumeration = schema_https$1.BodyMeasurementTypeEnumeration;
exports.BodyOfWater = schema_https$1.BodyOfWater;
exports.Bone = schema_https$1.Bone;
exports.Book = schema_https$1.Book;
exports.BookFormatType = schema_https$1.BookFormatType;
exports.BookSeries = schema_https$1.BookSeries;
exports.BookStore = schema_https$1.BookStore;
exports.BookmarkAction = schema_https$1.BookmarkAction;
exports.Boolean__workaround = schema_https$1.Boolean__workaround;
exports.BorrowAction = schema_https$1.BorrowAction;
exports.BowlingAlley = schema_https$1.BowlingAlley;
exports.BrainStructure = schema_https$1.BrainStructure;
exports.Brand = schema_https$1.Brand;
exports.BreadcrumbList = schema_https$1.BreadcrumbList;
exports.Brewery = schema_https$1.Brewery;
exports.Bridge = schema_https$1.Bridge;
exports.BroadcastChannel = schema_https$1.BroadcastChannel;
exports.BroadcastEvent = schema_https$1.BroadcastEvent;
exports.BroadcastFrequencySpecification = schema_https$1.BroadcastFrequencySpecification;
exports.BroadcastService = schema_https$1.BroadcastService;
exports.BrokerageAccount = schema_https$1.BrokerageAccount;
exports.BuddhistTemple = schema_https$1.BuddhistTemple;
exports.BusOrCoach = schema_https$1.BusOrCoach;
exports.BusReservation = schema_https$1.BusReservation;
exports.BusStation = schema_https$1.BusStation;
exports.BusStop = schema_https$1.BusStop;
exports.BusTrip = schema_https$1.BusTrip;
exports.BusinessAudience = schema_https$1.BusinessAudience;
exports.BusinessEntityType = schema_https$1.BusinessEntityType;
exports.BusinessEvent = schema_https$1.BusinessEvent;
exports.BusinessFunction = schema_https$1.BusinessFunction;
exports.BuyAction = schema_https$1.BuyAction;
exports.CDCPMDRecord = schema_https$1.CDCPMDRecord;
exports.CableOrSatelliteService = schema_https$1.CableOrSatelliteService;
exports.CafeOrCoffeeShop = schema_https$1.CafeOrCoffeeShop;
exports.Campground = schema_https$1.Campground;
exports.CampingPitch = schema_https$1.CampingPitch;
exports.Canal = schema_https$1.Canal;
exports.CancelAction = schema_https$1.CancelAction;
exports.Car = schema_https$1.Car;
exports.CarUsageType = schema_https$1.CarUsageType;
exports.Casino = schema_https$1.Casino;
exports.CategoryCode = schema_https$1.CategoryCode;
exports.CategoryCodeSet = schema_https$1.CategoryCodeSet;
exports.CatholicChurch = schema_https$1.CatholicChurch;
exports.Cemetery = schema_https$1.Cemetery;
exports.Chapter = schema_https$1.Chapter;
exports.CheckAction = schema_https$1.CheckAction;
exports.CheckInAction = schema_https$1.CheckInAction;
exports.CheckOutAction = schema_https$1.CheckOutAction;
exports.CheckoutPage = schema_https$1.CheckoutPage;
exports.ChemicalSubstance = schema_https$1.ChemicalSubstance;
exports.ChildCare = schema_https$1.ChildCare;
exports.ChildrensEvent = schema_https$1.ChildrensEvent;
exports.ChooseAction = schema_https$1.ChooseAction;
exports.Church = schema_https$1.Church;
exports.City = schema_https$1.City;
exports.CityHall = schema_https$1.CityHall;
exports.CivicStructure = schema_https$1.CivicStructure;
exports.Claim = schema_https$1.Claim;
exports.ClaimReview = schema_https$1.ClaimReview;
exports.Class = schema_https$1.Class;
exports.Clip = schema_https$1.Clip;
exports.ClothingStore = schema_https$1.ClothingStore;
exports.Code = schema_https$1.Code;
exports.Collection = schema_https$1.Collection;
exports.CollectionPage = schema_https$1.CollectionPage;
exports.CollegeOrUniversity = schema_https$1.CollegeOrUniversity;
exports.ComedyClub = schema_https$1.ComedyClub;
exports.ComedyEvent = schema_https$1.ComedyEvent;
exports.ComicCoverArt = schema_https$1.ComicCoverArt;
exports.ComicIssue = schema_https$1.ComicIssue;
exports.ComicSeries = schema_https$1.ComicSeries;
exports.ComicStory = schema_https$1.ComicStory;
exports.Comment = schema_https$1.Comment;
exports.CommentAction = schema_https$1.CommentAction;
exports.CommunicateAction = schema_https$1.CommunicateAction;
exports.CompleteDataFeed = schema_https$1.CompleteDataFeed;
exports.CompoundPriceSpecification = schema_https$1.CompoundPriceSpecification;
exports.ComputerLanguage = schema_https$1.ComputerLanguage;
exports.ComputerStore = schema_https$1.ComputerStore;
exports.ConfirmAction = schema_https$1.ConfirmAction;
exports.Consortium = schema_https$1.Consortium;
exports.ConstraintNode = schema_https$1.ConstraintNode;
exports.ConsumeAction = schema_https$1.ConsumeAction;
exports.ContactPage = schema_https$1.ContactPage;
exports.ContactPoint = schema_https$1.ContactPoint;
exports.ContactPointOption = schema_https$1.ContactPointOption;
exports.Continent = schema_https$1.Continent;
exports.ControlAction = schema_https$1.ControlAction;
exports.ConvenienceStore = schema_https$1.ConvenienceStore;
exports.Conversation = schema_https$1.Conversation;
exports.CookAction = schema_https$1.CookAction;
exports.Corporation = schema_https$1.Corporation;
exports.CorrectionComment = schema_https$1.CorrectionComment;
exports.Country = schema_https$1.Country;
exports.Course = schema_https$1.Course;
exports.CourseInstance = schema_https$1.CourseInstance;
exports.Courthouse = schema_https$1.Courthouse;
exports.CoverArt = schema_https$1.CoverArt;
exports.CovidTestingFacility = schema_https$1.CovidTestingFacility;
exports.CreateAction = schema_https$1.CreateAction;
exports.CreativeWork = schema_https$1.CreativeWork;
exports.CreativeWorkSeason = schema_https$1.CreativeWorkSeason;
exports.CreativeWorkSeries = schema_https$1.CreativeWorkSeries;
exports.CreditCard = schema_https$1.CreditCard;
exports.Crematorium = schema_https$1.Crematorium;
exports.CriticReview = schema_https$1.CriticReview;
exports.CssSelectorType = schema_https$1.CssSelectorType;
exports.CurrencyConversionService = schema_https$1.CurrencyConversionService;
exports.DDxElement = schema_https$1.DDxElement;
exports.DanceEvent = schema_https$1.DanceEvent;
exports.DanceGroup = schema_https$1.DanceGroup;
exports.DataCatalog = schema_https$1.DataCatalog;
exports.DataDownload = schema_https$1.DataDownload;
exports.DataFeed = schema_https$1.DataFeed;
exports.DataFeedItem = schema_https$1.DataFeedItem;
exports.DataType = schema_https$1.DataType;
exports.Dataset = schema_https$1.Dataset;
exports.DateTime = schema_https$1.DateTime;
exports.Date__workaround = schema_https$1.Date__workaround;
exports.DatedMoneySpecification = schema_https$1.DatedMoneySpecification;
exports.DayOfWeek = schema_https$1.DayOfWeek;
exports.DaySpa = schema_https$1.DaySpa;
exports.DeactivateAction = schema_https$1.DeactivateAction;
exports.DefenceEstablishment = schema_https$1.DefenceEstablishment;
exports.DefinedRegion = schema_https$1.DefinedRegion;
exports.DefinedTerm = schema_https$1.DefinedTerm;
exports.DefinedTermSet = schema_https$1.DefinedTermSet;
exports.DeleteAction = schema_https$1.DeleteAction;
exports.DeliveryChargeSpecification = schema_https$1.DeliveryChargeSpecification;
exports.DeliveryEvent = schema_https$1.DeliveryEvent;
exports.DeliveryMethod = schema_https$1.DeliveryMethod;
exports.DeliveryTimeSettings = schema_https$1.DeliveryTimeSettings;
exports.Demand = schema_https$1.Demand;
exports.Dentist = schema_https$1.Dentist;
exports.DepartAction = schema_https$1.DepartAction;
exports.DepartmentStore = schema_https$1.DepartmentStore;
exports.DepositAccount = schema_https$1.DepositAccount;
exports.DiagnosticLab = schema_https$1.DiagnosticLab;
exports.DiagnosticProcedure = schema_https$1.DiagnosticProcedure;
exports.Diet = schema_https$1.Diet;
exports.DietarySupplement = schema_https$1.DietarySupplement;
exports.DigitalDocument = schema_https$1.DigitalDocument;
exports.DigitalDocumentPermission = schema_https$1.DigitalDocumentPermission;
exports.DigitalDocumentPermissionType = schema_https$1.DigitalDocumentPermissionType;
exports.DigitalPlatformEnumeration = schema_https$1.DigitalPlatformEnumeration;
exports.DisagreeAction = schema_https$1.DisagreeAction;
exports.DiscoverAction = schema_https$1.DiscoverAction;
exports.DiscussionForumPosting = schema_https$1.DiscussionForumPosting;
exports.DislikeAction = schema_https$1.DislikeAction;
exports.Distance = schema_https$1.Distance;
exports.Distillery = schema_https$1.Distillery;
exports.DonateAction = schema_https$1.DonateAction;
exports.DoseSchedule = schema_https$1.DoseSchedule;
exports.DownloadAction = schema_https$1.DownloadAction;
exports.DrawAction = schema_https$1.DrawAction;
exports.Drawing = schema_https$1.Drawing;
exports.DrinkAction = schema_https$1.DrinkAction;
exports.DriveWheelConfigurationValue = schema_https$1.DriveWheelConfigurationValue;
exports.Drug = schema_https$1.Drug;
exports.DrugClass = schema_https$1.DrugClass;
exports.DrugCost = schema_https$1.DrugCost;
exports.DrugCostCategory = schema_https$1.DrugCostCategory;
exports.DrugLegalStatus = schema_https$1.DrugLegalStatus;
exports.DrugPregnancyCategory = schema_https$1.DrugPregnancyCategory;
exports.DrugPrescriptionStatus = schema_https$1.DrugPrescriptionStatus;
exports.DrugStrength = schema_https$1.DrugStrength;
exports.DryCleaningOrLaundry = schema_https$1.DryCleaningOrLaundry;
exports.Duration = schema_https$1.Duration;
exports.EUEnergyEfficiencyEnumeration = schema_https$1.EUEnergyEfficiencyEnumeration;
exports.EatAction = schema_https$1.EatAction;
exports.EducationEvent = schema_https$1.EducationEvent;
exports.EducationalAudience = schema_https$1.EducationalAudience;
exports.EducationalOccupationalCredential = schema_https$1.EducationalOccupationalCredential;
exports.EducationalOccupationalProgram = schema_https$1.EducationalOccupationalProgram;
exports.EducationalOrganization = schema_https$1.EducationalOrganization;
exports.Electrician = schema_https$1.Electrician;
exports.ElectronicsStore = schema_https$1.ElectronicsStore;
exports.ElementarySchool = schema_https$1.ElementarySchool;
exports.EmailMessage = schema_https$1.EmailMessage;
exports.Embassy = schema_https$1.Embassy;
exports.EmergencyService = schema_https$1.EmergencyService;
exports.EmployeeRole = schema_https$1.EmployeeRole;
exports.EmployerAggregateRating = schema_https$1.EmployerAggregateRating;
exports.EmployerReview = schema_https$1.EmployerReview;
exports.EmploymentAgency = schema_https$1.EmploymentAgency;
exports.EndorseAction = schema_https$1.EndorseAction;
exports.EndorsementRating = schema_https$1.EndorsementRating;
exports.Energy = schema_https$1.Energy;
exports.EnergyConsumptionDetails = schema_https$1.EnergyConsumptionDetails;
exports.EnergyEfficiencyEnumeration = schema_https$1.EnergyEfficiencyEnumeration;
exports.EnergyStarEnergyEfficiencyEnumeration = schema_https$1.EnergyStarEnergyEfficiencyEnumeration;
exports.EngineSpecification = schema_https$1.EngineSpecification;
exports.EntertainmentBusiness = schema_https$1.EntertainmentBusiness;
exports.EntryPoint = schema_https$1.EntryPoint;
exports.Enumeration = schema_https$1.Enumeration;
exports.Episode = schema_https$1.Episode;
exports.Event = schema_https$1.Event;
exports.EventAttendanceModeEnumeration = schema_https$1.EventAttendanceModeEnumeration;
exports.EventReservation = schema_https$1.EventReservation;
exports.EventSeries = schema_https$1.EventSeries;
exports.EventStatusType = schema_https$1.EventStatusType;
exports.EventVenue = schema_https$1.EventVenue;
exports.ExchangeRateSpecification = schema_https$1.ExchangeRateSpecification;
exports.ExerciseAction = schema_https$1.ExerciseAction;
exports.ExerciseGym = schema_https$1.ExerciseGym;
exports.ExercisePlan = schema_https$1.ExercisePlan;
exports.ExhibitionEvent = schema_https$1.ExhibitionEvent;
exports.FAQPage = schema_https$1.FAQPage;
exports.FMRadioChannel = schema_https$1.FMRadioChannel;
exports.FastFoodRestaurant = schema_https$1.FastFoodRestaurant;
exports.Festival = schema_https$1.Festival;
exports.FilmAction = schema_https$1.FilmAction;
exports.FinancialProduct = schema_https$1.FinancialProduct;
exports.FinancialService = schema_https$1.FinancialService;
exports.FindAction = schema_https$1.FindAction;
exports.FireStation = schema_https$1.FireStation;
exports.Flight = schema_https$1.Flight;
exports.FlightReservation = schema_https$1.FlightReservation;
exports.Float = schema_https$1.Float;
exports.FloorPlan = schema_https$1.FloorPlan;
exports.Florist = schema_https$1.Florist;
exports.FollowAction = schema_https$1.FollowAction;
exports.FoodEstablishment = schema_https$1.FoodEstablishment;
exports.FoodEstablishmentReservation = schema_https$1.FoodEstablishmentReservation;
exports.FoodEvent = schema_https$1.FoodEvent;
exports.FoodService = schema_https$1.FoodService;
exports.FundingAgency = schema_https$1.FundingAgency;
exports.FundingScheme = schema_https$1.FundingScheme;
exports.FurnitureStore = schema_https$1.FurnitureStore;
exports.Game = schema_https$1.Game;
exports.GameAvailabilityEnumeration = schema_https$1.GameAvailabilityEnumeration;
exports.GamePlayMode = schema_https$1.GamePlayMode;
exports.GameServer = schema_https$1.GameServer;
exports.GameServerStatus = schema_https$1.GameServerStatus;
exports.GardenStore = schema_https$1.GardenStore;
exports.GasStation = schema_https$1.GasStation;
exports.GatedResidenceCommunity = schema_https$1.GatedResidenceCommunity;
exports.GenderType = schema_https$1.GenderType;
exports.Gene = schema_https$1.Gene;
exports.GeneralContractor = schema_https$1.GeneralContractor;
exports.GeoCircle = schema_https$1.GeoCircle;
exports.GeoCoordinates = schema_https$1.GeoCoordinates;
exports.GeoShape = schema_https$1.GeoShape;
exports.GeospatialGeometry = schema_https$1.GeospatialGeometry;
exports.GiveAction = schema_https$1.GiveAction;
exports.GolfCourse = schema_https$1.GolfCourse;
exports.GovernmentBenefitsType = schema_https$1.GovernmentBenefitsType;
exports.GovernmentBuilding = schema_https$1.GovernmentBuilding;
exports.GovernmentOffice = schema_https$1.GovernmentOffice;
exports.GovernmentOrganization = schema_https$1.GovernmentOrganization;
exports.GovernmentPermit = schema_https$1.GovernmentPermit;
exports.GovernmentService = schema_https$1.GovernmentService;
exports.Grant = schema_https$1.Grant;
exports.GroceryStore = schema_https$1.GroceryStore;
exports.Guide = schema_https$1.Guide;
exports.HVACBusiness = schema_https$1.HVACBusiness;
exports.Hackathon = schema_https$1.Hackathon;
exports.HairSalon = schema_https$1.HairSalon;
exports.HardwareStore = schema_https$1.HardwareStore;
exports.HealthAndBeautyBusiness = schema_https$1.HealthAndBeautyBusiness;
exports.HealthAspectEnumeration = schema_https$1.HealthAspectEnumeration;
exports.HealthClub = schema_https$1.HealthClub;
exports.HealthInsurancePlan = schema_https$1.HealthInsurancePlan;
exports.HealthPlanCostSharingSpecification = schema_https$1.HealthPlanCostSharingSpecification;
exports.HealthPlanFormulary = schema_https$1.HealthPlanFormulary;
exports.HealthPlanNetwork = schema_https$1.HealthPlanNetwork;
exports.HealthTopicContent = schema_https$1.HealthTopicContent;
exports.HighSchool = schema_https$1.HighSchool;
exports.HinduTemple = schema_https$1.HinduTemple;
exports.HobbyShop = schema_https$1.HobbyShop;
exports.HomeAndConstructionBusiness = schema_https$1.HomeAndConstructionBusiness;
exports.HomeGoodsStore = schema_https$1.HomeGoodsStore;
exports.Hospital = schema_https$1.Hospital;
exports.Hostel = schema_https$1.Hostel;
exports.Hotel = schema_https$1.Hotel;
exports.HotelRoom = schema_https$1.HotelRoom;
exports.House = schema_https$1.House;
exports.HousePainter = schema_https$1.HousePainter;
exports.HowTo = schema_https$1.HowTo;
exports.HowToDirection = schema_https$1.HowToDirection;
exports.HowToItem = schema_https$1.HowToItem;
exports.HowToSection = schema_https$1.HowToSection;
exports.HowToStep = schema_https$1.HowToStep;
exports.HowToSupply = schema_https$1.HowToSupply;
exports.HowToTip = schema_https$1.HowToTip;
exports.HowToTool = schema_https$1.HowToTool;
exports.HyperToc = schema_https$1.HyperToc;
exports.HyperTocEntry = schema_https$1.HyperTocEntry;
exports.IceCreamShop = schema_https$1.IceCreamShop;
exports.IgnoreAction = schema_https$1.IgnoreAction;
exports.ImageGallery = schema_https$1.ImageGallery;
exports.ImageObject = schema_https$1.ImageObject;
exports.ImageObjectSnapshot = schema_https$1.ImageObjectSnapshot;
exports.ImagingTest = schema_https$1.ImagingTest;
exports.IndividualProduct = schema_https$1.IndividualProduct;
exports.InfectiousAgentClass = schema_https$1.InfectiousAgentClass;
exports.InfectiousDisease = schema_https$1.InfectiousDisease;
exports.InformAction = schema_https$1.InformAction;
exports.InsertAction = schema_https$1.InsertAction;
exports.InstallAction = schema_https$1.InstallAction;
exports.InsuranceAgency = schema_https$1.InsuranceAgency;
exports.Intangible = schema_https$1.Intangible;
exports.Integer = schema_https$1.Integer;
exports.InteractAction = schema_https$1.InteractAction;
exports.InteractionCounter = schema_https$1.InteractionCounter;
exports.InternetCafe = schema_https$1.InternetCafe;
exports.InvestmentFund = schema_https$1.InvestmentFund;
exports.InvestmentOrDeposit = schema_https$1.InvestmentOrDeposit;
exports.InviteAction = schema_https$1.InviteAction;
exports.Invoice = schema_https$1.Invoice;
exports.ItemAvailability = schema_https$1.ItemAvailability;
exports.ItemList = schema_https$1.ItemList;
exports.ItemListOrderType = schema_https$1.ItemListOrderType;
exports.ItemPage = schema_https$1.ItemPage;
exports.JewelryStore = schema_https$1.JewelryStore;
exports.JobPosting = schema_https$1.JobPosting;
exports.JoinAction = schema_https$1.JoinAction;
exports.Joint = schema_https$1.Joint;
exports.LakeBodyOfWater = schema_https$1.LakeBodyOfWater;
exports.Landform = schema_https$1.Landform;
exports.LandmarksOrHistoricalBuildings = schema_https$1.LandmarksOrHistoricalBuildings;
exports.Language = schema_https$1.Language;
exports.LearningResource = schema_https$1.LearningResource;
exports.LeaveAction = schema_https$1.LeaveAction;
exports.LegalForceStatus = schema_https$1.LegalForceStatus;
exports.LegalService = schema_https$1.LegalService;
exports.LegalValueLevel = schema_https$1.LegalValueLevel;
exports.Legislation = schema_https$1.Legislation;
exports.LegislationObject = schema_https$1.LegislationObject;
exports.LegislativeBuilding = schema_https$1.LegislativeBuilding;
exports.LendAction = schema_https$1.LendAction;
exports.Library = schema_https$1.Library;
exports.LibrarySystem = schema_https$1.LibrarySystem;
exports.LifestyleModification = schema_https$1.LifestyleModification;
exports.Ligament = schema_https$1.Ligament;
exports.LikeAction = schema_https$1.LikeAction;
exports.LinkRole = schema_https$1.LinkRole;
exports.LiquorStore = schema_https$1.LiquorStore;
exports.ListItem = schema_https$1.ListItem;
exports.ListenAction = schema_https$1.ListenAction;
exports.LiteraryEvent = schema_https$1.LiteraryEvent;
exports.LiveBlogPosting = schema_https$1.LiveBlogPosting;
exports.LoanOrCredit = schema_https$1.LoanOrCredit;
exports.LocalBusiness = schema_https$1.LocalBusiness;
exports.LocationFeatureSpecification = schema_https$1.LocationFeatureSpecification;
exports.Locksmith = schema_https$1.Locksmith;
exports.LodgingBusiness = schema_https$1.LodgingBusiness;
exports.LodgingReservation = schema_https$1.LodgingReservation;
exports.LoseAction = schema_https$1.LoseAction;
exports.LymphaticVessel = schema_https$1.LymphaticVessel;
exports.Manuscript = schema_https$1.Manuscript;
exports.MapCategoryType = schema_https$1.MapCategoryType;
exports.Map__workaround = schema_https$1.Map__workaround;
exports.MarryAction = schema_https$1.MarryAction;
exports.Mass = schema_https$1.Mass;
exports.MathSolver = schema_https$1.MathSolver;
exports.MaximumDoseSchedule = schema_https$1.MaximumDoseSchedule;
exports.MeasurementMethodEnum = schema_https$1.MeasurementMethodEnum;
exports.MeasurementTypeEnumeration = schema_https$1.MeasurementTypeEnumeration;
exports.MediaGallery = schema_https$1.MediaGallery;
exports.MediaManipulationRatingEnumeration = schema_https$1.MediaManipulationRatingEnumeration;
exports.MediaObject = schema_https$1.MediaObject;
exports.MediaReview = schema_https$1.MediaReview;
exports.MediaReviewItem = schema_https$1.MediaReviewItem;
exports.MediaSubscription = schema_https$1.MediaSubscription;
exports.MedicalAudience = schema_https$1.MedicalAudience;
exports.MedicalAudienceType = schema_https$1.MedicalAudienceType;
exports.MedicalBusiness = schema_https$1.MedicalBusiness;
exports.MedicalCause = schema_https$1.MedicalCause;
exports.MedicalClinic = schema_https$1.MedicalClinic;
exports.MedicalCode = schema_https$1.MedicalCode;
exports.MedicalCondition = schema_https$1.MedicalCondition;
exports.MedicalConditionStage = schema_https$1.MedicalConditionStage;
exports.MedicalContraindication = schema_https$1.MedicalContraindication;
exports.MedicalDevice = schema_https$1.MedicalDevice;
exports.MedicalDevicePurpose = schema_https$1.MedicalDevicePurpose;
exports.MedicalEntity = schema_https$1.MedicalEntity;
exports.MedicalEnumeration = schema_https$1.MedicalEnumeration;
exports.MedicalEvidenceLevel = schema_https$1.MedicalEvidenceLevel;
exports.MedicalGuideline = schema_https$1.MedicalGuideline;
exports.MedicalGuidelineContraindication = schema_https$1.MedicalGuidelineContraindication;
exports.MedicalGuidelineRecommendation = schema_https$1.MedicalGuidelineRecommendation;
exports.MedicalImagingTechnique = schema_https$1.MedicalImagingTechnique;
exports.MedicalIndication = schema_https$1.MedicalIndication;
exports.MedicalIntangible = schema_https$1.MedicalIntangible;
exports.MedicalObservationalStudy = schema_https$1.MedicalObservationalStudy;
exports.MedicalObservationalStudyDesign = schema_https$1.MedicalObservationalStudyDesign;
exports.MedicalOrganization = schema_https$1.MedicalOrganization;
exports.MedicalProcedure = schema_https$1.MedicalProcedure;
exports.MedicalProcedureType = schema_https$1.MedicalProcedureType;
exports.MedicalRiskCalculator = schema_https$1.MedicalRiskCalculator;
exports.MedicalRiskEstimator = schema_https$1.MedicalRiskEstimator;
exports.MedicalRiskFactor = schema_https$1.MedicalRiskFactor;
exports.MedicalRiskScore = schema_https$1.MedicalRiskScore;
exports.MedicalScholarlyArticle = schema_https$1.MedicalScholarlyArticle;
exports.MedicalSign = schema_https$1.MedicalSign;
exports.MedicalSignOrSymptom = schema_https$1.MedicalSignOrSymptom;
exports.MedicalSpecialty = schema_https$1.MedicalSpecialty;
exports.MedicalStudy = schema_https$1.MedicalStudy;
exports.MedicalStudyStatus = schema_https$1.MedicalStudyStatus;
exports.MedicalSymptom = schema_https$1.MedicalSymptom;
exports.MedicalTest = schema_https$1.MedicalTest;
exports.MedicalTestPanel = schema_https$1.MedicalTestPanel;
exports.MedicalTherapy = schema_https$1.MedicalTherapy;
exports.MedicalTrial = schema_https$1.MedicalTrial;
exports.MedicalTrialDesign = schema_https$1.MedicalTrialDesign;
exports.MedicalWebPage = schema_https$1.MedicalWebPage;
exports.MedicineSystem = schema_https$1.MedicineSystem;
exports.MeetingRoom = schema_https$1.MeetingRoom;
exports.MensClothingStore = schema_https$1.MensClothingStore;
exports.Menu = schema_https$1.Menu;
exports.MenuItem = schema_https$1.MenuItem;
exports.MenuSection = schema_https$1.MenuSection;
exports.MerchantReturnEnumeration = schema_https$1.MerchantReturnEnumeration;
exports.MerchantReturnPolicy = schema_https$1.MerchantReturnPolicy;
exports.MerchantReturnPolicySeasonalOverride = schema_https$1.MerchantReturnPolicySeasonalOverride;
exports.Message = schema_https$1.Message;
exports.MiddleSchool = schema_https$1.MiddleSchool;
exports.MobileApplication = schema_https$1.MobileApplication;
exports.MobilePhoneStore = schema_https$1.MobilePhoneStore;
exports.MolecularEntity = schema_https$1.MolecularEntity;
exports.MonetaryAmount = schema_https$1.MonetaryAmount;
exports.MonetaryAmountDistribution = schema_https$1.MonetaryAmountDistribution;
exports.MonetaryGrant = schema_https$1.MonetaryGrant;
exports.MoneyTransfer = schema_https$1.MoneyTransfer;
exports.MortgageLoan = schema_https$1.MortgageLoan;
exports.Mosque = schema_https$1.Mosque;
exports.Motel = schema_https$1.Motel;
exports.Motorcycle = schema_https$1.Motorcycle;
exports.MotorcycleDealer = schema_https$1.MotorcycleDealer;
exports.MotorcycleRepair = schema_https$1.MotorcycleRepair;
exports.MotorizedBicycle = schema_https$1.MotorizedBicycle;
exports.Mountain = schema_https$1.Mountain;
exports.MoveAction = schema_https$1.MoveAction;
exports.Movie = schema_https$1.Movie;
exports.MovieClip = schema_https$1.MovieClip;
exports.MovieRentalStore = schema_https$1.MovieRentalStore;
exports.MovieSeries = schema_https$1.MovieSeries;
exports.MovieTheater = schema_https$1.MovieTheater;
exports.MovingCompany = schema_https$1.MovingCompany;
exports.Muscle = schema_https$1.Muscle;
exports.Museum = schema_https$1.Museum;
exports.MusicAlbum = schema_https$1.MusicAlbum;
exports.MusicAlbumProductionType = schema_https$1.MusicAlbumProductionType;
exports.MusicAlbumReleaseType = schema_https$1.MusicAlbumReleaseType;
exports.MusicComposition = schema_https$1.MusicComposition;
exports.MusicEvent = schema_https$1.MusicEvent;
exports.MusicGroup = schema_https$1.MusicGroup;
exports.MusicPlaylist = schema_https$1.MusicPlaylist;
exports.MusicRecording = schema_https$1.MusicRecording;
exports.MusicRelease = schema_https$1.MusicRelease;
exports.MusicReleaseFormatType = schema_https$1.MusicReleaseFormatType;
exports.MusicStore = schema_https$1.MusicStore;
exports.MusicVenue = schema_https$1.MusicVenue;
exports.MusicVideoObject = schema_https$1.MusicVideoObject;
exports.NGO = schema_https$1.NGO;
exports.NLNonprofitType = schema_https$1.NLNonprofitType;
exports.NailSalon = schema_https$1.NailSalon;
exports.Nerve = schema_https$1.Nerve;
exports.NewsArticle = schema_https$1.NewsArticle;
exports.NewsMediaOrganization = schema_https$1.NewsMediaOrganization;
exports.Newspaper = schema_https$1.Newspaper;
exports.NightClub = schema_https$1.NightClub;
exports.NonprofitType = schema_https$1.NonprofitType;
exports.Notary = schema_https$1.Notary;
exports.NoteDigitalDocument = schema_https$1.NoteDigitalDocument;
exports.Number__workaround = schema_https$1.Number__workaround;
exports.NutritionInformation = schema_https$1.NutritionInformation;
exports.Observation = schema_https$1.Observation;
exports.Occupation = schema_https$1.Occupation;
exports.OccupationalExperienceRequirements = schema_https$1.OccupationalExperienceRequirements;
exports.OccupationalTherapy = schema_https$1.OccupationalTherapy;
exports.OceanBodyOfWater = schema_https$1.OceanBodyOfWater;
exports.Offer = schema_https$1.Offer;
exports.OfferCatalog = schema_https$1.OfferCatalog;
exports.OfferForLease = schema_https$1.OfferForLease;
exports.OfferForPurchase = schema_https$1.OfferForPurchase;
exports.OfferItemCondition = schema_https$1.OfferItemCondition;
exports.OfferShippingDetails = schema_https$1.OfferShippingDetails;
exports.OfficeEquipmentStore = schema_https$1.OfficeEquipmentStore;
exports.OnDemandEvent = schema_https$1.OnDemandEvent;
exports.OnlineBusiness = schema_https$1.OnlineBusiness;
exports.OnlineStore = schema_https$1.OnlineStore;
exports.OpeningHoursSpecification = schema_https$1.OpeningHoursSpecification;
exports.OpinionNewsArticle = schema_https$1.OpinionNewsArticle;
exports.Optician = schema_https$1.Optician;
exports.Order = schema_https$1.Order;
exports.OrderAction = schema_https$1.OrderAction;
exports.OrderItem = schema_https$1.OrderItem;
exports.OrderStatus = schema_https$1.OrderStatus;
exports.Organization = schema_https$1.Organization;
exports.OrganizationRole = schema_https$1.OrganizationRole;
exports.OrganizeAction = schema_https$1.OrganizeAction;
exports.OutletStore = schema_https$1.OutletStore;
exports.OwnershipInfo = schema_https$1.OwnershipInfo;
exports.PaintAction = schema_https$1.PaintAction;
exports.Painting = schema_https$1.Painting;
exports.PalliativeProcedure = schema_https$1.PalliativeProcedure;
exports.ParcelDelivery = schema_https$1.ParcelDelivery;
exports.ParentAudience = schema_https$1.ParentAudience;
exports.Park = schema_https$1.Park;
exports.ParkingFacility = schema_https$1.ParkingFacility;
exports.PathologyTest = schema_https$1.PathologyTest;
exports.Patient = schema_https$1.Patient;
exports.PawnShop = schema_https$1.PawnShop;
exports.PayAction = schema_https$1.PayAction;
exports.PaymentCard = schema_https$1.PaymentCard;
exports.PaymentChargeSpecification = schema_https$1.PaymentChargeSpecification;
exports.PaymentMethod = schema_https$1.PaymentMethod;
exports.PaymentService = schema_https$1.PaymentService;
exports.PaymentStatusType = schema_https$1.PaymentStatusType;
exports.PeopleAudience = schema_https$1.PeopleAudience;
exports.PerformAction = schema_https$1.PerformAction;
exports.PerformanceRole = schema_https$1.PerformanceRole;
exports.PerformingArtsTheater = schema_https$1.PerformingArtsTheater;
exports.PerformingGroup = schema_https$1.PerformingGroup;
exports.Periodical = schema_https$1.Periodical;
exports.Permit = schema_https$1.Permit;
exports.Person = schema_https$1.Person;
exports.PetStore = schema_https$1.PetStore;
exports.Pharmacy = schema_https$1.Pharmacy;
exports.Photograph = schema_https$1.Photograph;
exports.PhotographAction = schema_https$1.PhotographAction;
exports.PhysicalActivity = schema_https$1.PhysicalActivity;
exports.PhysicalActivityCategory = schema_https$1.PhysicalActivityCategory;
exports.PhysicalExam = schema_https$1.PhysicalExam;
exports.PhysicalTherapy = schema_https$1.PhysicalTherapy;
exports.Physician = schema_https$1.Physician;
exports.Place = schema_https$1.Place;
exports.PlaceOfWorship = schema_https$1.PlaceOfWorship;
exports.PlanAction = schema_https$1.PlanAction;
exports.Play = schema_https$1.Play;
exports.PlayAction = schema_https$1.PlayAction;
exports.PlayGameAction = schema_https$1.PlayGameAction;
exports.Playground = schema_https$1.Playground;
exports.Plumber = schema_https$1.Plumber;
exports.PodcastEpisode = schema_https$1.PodcastEpisode;
exports.PodcastSeason = schema_https$1.PodcastSeason;
exports.PodcastSeries = schema_https$1.PodcastSeries;
exports.PoliceStation = schema_https$1.PoliceStation;
exports.PoliticalParty = schema_https$1.PoliticalParty;
exports.Pond = schema_https$1.Pond;
exports.PostOffice = schema_https$1.PostOffice;
exports.PostalAddress = schema_https$1.PostalAddress;
exports.PostalCodeRangeSpecification = schema_https$1.PostalCodeRangeSpecification;
exports.Poster = schema_https$1.Poster;
exports.PreOrderAction = schema_https$1.PreOrderAction;
exports.PrependAction = schema_https$1.PrependAction;
exports.Preschool = schema_https$1.Preschool;
exports.PresentationDigitalDocument = schema_https$1.PresentationDigitalDocument;
exports.PreventionIndication = schema_https$1.PreventionIndication;
exports.PriceComponentTypeEnumeration = schema_https$1.PriceComponentTypeEnumeration;
exports.PriceSpecification = schema_https$1.PriceSpecification;
exports.PriceTypeEnumeration = schema_https$1.PriceTypeEnumeration;
exports.Product = schema_https$1.Product;
exports.ProductCollection = schema_https$1.ProductCollection;
exports.ProductGroup = schema_https$1.ProductGroup;
exports.ProductModel = schema_https$1.ProductModel;
exports.ProductReturnEnumeration = schema_https$1.ProductReturnEnumeration;
exports.ProductReturnPolicy = schema_https$1.ProductReturnPolicy;
exports.ProfessionalService = schema_https$1.ProfessionalService;
exports.ProfilePage = schema_https$1.ProfilePage;
exports.ProgramMembership = schema_https$1.ProgramMembership;
exports.Project = schema_https$1.Project;
exports.PronounceableText = schema_https$1.PronounceableText;
exports.Property = schema_https$1.Property;
exports.PropertyValue = schema_https$1.PropertyValue;
exports.PropertyValueSpecification = schema_https$1.PropertyValueSpecification;
exports.Protein = schema_https$1.Protein;
exports.PsychologicalTreatment = schema_https$1.PsychologicalTreatment;
exports.PublicSwimmingPool = schema_https$1.PublicSwimmingPool;
exports.PublicToilet = schema_https$1.PublicToilet;
exports.PublicationEvent = schema_https$1.PublicationEvent;
exports.PublicationIssue = schema_https$1.PublicationIssue;
exports.PublicationVolume = schema_https$1.PublicationVolume;
exports.QAPage = schema_https$1.QAPage;
exports.QualitativeValue = schema_https$1.QualitativeValue;
exports.QuantitativeValue = schema_https$1.QuantitativeValue;
exports.QuantitativeValueDistribution = schema_https$1.QuantitativeValueDistribution;
exports.Quantity = schema_https$1.Quantity;
exports.Question = schema_https$1.Question;
exports.Quiz = schema_https$1.Quiz;
exports.Quotation = schema_https$1.Quotation;
exports.QuoteAction = schema_https$1.QuoteAction;
exports.RVPark = schema_https$1.RVPark;
exports.RadiationTherapy = schema_https$1.RadiationTherapy;
exports.RadioBroadcastService = schema_https$1.RadioBroadcastService;
exports.RadioChannel = schema_https$1.RadioChannel;
exports.RadioClip = schema_https$1.RadioClip;
exports.RadioEpisode = schema_https$1.RadioEpisode;
exports.RadioSeason = schema_https$1.RadioSeason;
exports.RadioSeries = schema_https$1.RadioSeries;
exports.RadioStation = schema_https$1.RadioStation;
exports.Rating = schema_https$1.Rating;
exports.ReactAction = schema_https$1.ReactAction;
exports.ReadAction = schema_https$1.ReadAction;
exports.RealEstateAgent = schema_https$1.RealEstateAgent;
exports.RealEstateListing = schema_https$1.RealEstateListing;
exports.ReceiveAction = schema_https$1.ReceiveAction;
exports.Recipe = schema_https$1.Recipe;
exports.Recommendation = schema_https$1.Recommendation;
exports.RecommendedDoseSchedule = schema_https$1.RecommendedDoseSchedule;
exports.RecyclingCenter = schema_https$1.RecyclingCenter;
exports.RefundTypeEnumeration = schema_https$1.RefundTypeEnumeration;
exports.RegisterAction = schema_https$1.RegisterAction;
exports.RejectAction = schema_https$1.RejectAction;
exports.RentAction = schema_https$1.RentAction;
exports.RentalCarReservation = schema_https$1.RentalCarReservation;
exports.RepaymentSpecification = schema_https$1.RepaymentSpecification;
exports.ReplaceAction = schema_https$1.ReplaceAction;
exports.ReplyAction = schema_https$1.ReplyAction;
exports.Report = schema_https$1.Report;
exports.ReportageNewsArticle = schema_https$1.ReportageNewsArticle;
exports.ReportedDoseSchedule = schema_https$1.ReportedDoseSchedule;
exports.ResearchOrganization = schema_https$1.ResearchOrganization;
exports.ResearchProject = schema_https$1.ResearchProject;
exports.Researcher = schema_https$1.Researcher;
exports.Reservation = schema_https$1.Reservation;
exports.ReservationPackage = schema_https$1.ReservationPackage;
exports.ReservationStatusType = schema_https$1.ReservationStatusType;
exports.ReserveAction = schema_https$1.ReserveAction;
exports.Reservoir = schema_https$1.Reservoir;
exports.Residence = schema_https$1.Residence;
exports.Resort = schema_https$1.Resort;
exports.Restaurant = schema_https$1.Restaurant;
exports.RestrictedDiet = schema_https$1.RestrictedDiet;
exports.ResumeAction = schema_https$1.ResumeAction;
exports.ReturnAction = schema_https$1.ReturnAction;
exports.ReturnFeesEnumeration = schema_https$1.ReturnFeesEnumeration;
exports.ReturnLabelSourceEnumeration = schema_https$1.ReturnLabelSourceEnumeration;
exports.ReturnMethodEnumeration = schema_https$1.ReturnMethodEnumeration;
exports.Review = schema_https$1.Review;
exports.ReviewAction = schema_https$1.ReviewAction;
exports.ReviewNewsArticle = schema_https$1.ReviewNewsArticle;
exports.RiverBodyOfWater = schema_https$1.RiverBodyOfWater;
exports.Role = schema_https$1.Role;
exports.RoofingContractor = schema_https$1.RoofingContractor;
exports.Room = schema_https$1.Room;
exports.RsvpAction = schema_https$1.RsvpAction;
exports.RsvpResponseType = schema_https$1.RsvpResponseType;
exports.SaleEvent = schema_https$1.SaleEvent;
exports.SatiricalArticle = schema_https$1.SatiricalArticle;
exports.Schedule = schema_https$1.Schedule;
exports.ScheduleAction = schema_https$1.ScheduleAction;
exports.ScholarlyArticle = schema_https$1.ScholarlyArticle;
exports.School = schema_https$1.School;
exports.SchoolDistrict = schema_https$1.SchoolDistrict;
exports.ScreeningEvent = schema_https$1.ScreeningEvent;
exports.Sculpture = schema_https$1.Sculpture;
exports.SeaBodyOfWater = schema_https$1.SeaBodyOfWater;
exports.SearchAction = schema_https$1.SearchAction;
exports.SearchRescueOrganization = schema_https$1.SearchRescueOrganization;
exports.SearchResultsPage = schema_https$1.SearchResultsPage;
exports.Season = schema_https$1.Season;
exports.Seat = schema_https$1.Seat;
exports.SeekToAction = schema_https$1.SeekToAction;
exports.SelfStorage = schema_https$1.SelfStorage;
exports.SellAction = schema_https$1.SellAction;
exports.SendAction = schema_https$1.SendAction;
exports.Series = schema_https$1.Series;
exports.Service = schema_https$1.Service;
exports.ServiceChannel = schema_https$1.ServiceChannel;
exports.ShareAction = schema_https$1.ShareAction;
exports.SheetMusic = schema_https$1.SheetMusic;
exports.ShippingDeliveryTime = schema_https$1.ShippingDeliveryTime;
exports.ShippingRateSettings = schema_https$1.ShippingRateSettings;
exports.ShoeStore = schema_https$1.ShoeStore;
exports.ShoppingCenter = schema_https$1.ShoppingCenter;
exports.ShortStory = schema_https$1.ShortStory;
exports.SingleFamilyResidence = schema_https$1.SingleFamilyResidence;
exports.SiteNavigationElement = schema_https$1.SiteNavigationElement;
exports.SizeGroupEnumeration = schema_https$1.SizeGroupEnumeration;
exports.SizeSpecification = schema_https$1.SizeSpecification;
exports.SizeSystemEnumeration = schema_https$1.SizeSystemEnumeration;
exports.SkiResort = schema_https$1.SkiResort;
exports.SocialEvent = schema_https$1.SocialEvent;
exports.SocialMediaPosting = schema_https$1.SocialMediaPosting;
exports.SoftwareApplication = schema_https$1.SoftwareApplication;
exports.SoftwareSourceCode = schema_https$1.SoftwareSourceCode;
exports.SolveMathAction = schema_https$1.SolveMathAction;
exports.SomeProducts = schema_https$1.SomeProducts;
exports.SpeakableSpecification = schema_https$1.SpeakableSpecification;
exports.SpecialAnnouncement = schema_https$1.SpecialAnnouncement;
exports.Specialty = schema_https$1.Specialty;
exports.SportingGoodsStore = schema_https$1.SportingGoodsStore;
exports.SportsActivityLocation = schema_https$1.SportsActivityLocation;
exports.SportsClub = schema_https$1.SportsClub;
exports.SportsEvent = schema_https$1.SportsEvent;
exports.SportsOrganization = schema_https$1.SportsOrganization;
exports.SportsTeam = schema_https$1.SportsTeam;
exports.SpreadsheetDigitalDocument = schema_https$1.SpreadsheetDigitalDocument;
exports.StadiumOrArena = schema_https$1.StadiumOrArena;
exports.State = schema_https$1.State;
exports.Statement = schema_https$1.Statement;
exports.StatisticalPopulation = schema_https$1.StatisticalPopulation;
exports.StatisticalVariable = schema_https$1.StatisticalVariable;
exports.StatusEnumeration = schema_https$1.StatusEnumeration;
exports.SteeringPositionValue = schema_https$1.SteeringPositionValue;
exports.Store = schema_https$1.Store;
exports.StructuredValue = schema_https$1.StructuredValue;
exports.StupidType = schema_https$1.StupidType;
exports.SubscribeAction = schema_https$1.SubscribeAction;
exports.Substance = schema_https$1.Substance;
exports.SubwayStation = schema_https$1.SubwayStation;
exports.Suite = schema_https$1.Suite;
exports.SuperficialAnatomy = schema_https$1.SuperficialAnatomy;
exports.SurgicalProcedure = schema_https$1.SurgicalProcedure;
exports.SuspendAction = schema_https$1.SuspendAction;
exports.Syllabus = schema_https$1.Syllabus;
exports.Synagogue = schema_https$1.Synagogue;
exports.TVClip = schema_https$1.TVClip;
exports.TVEpisode = schema_https$1.TVEpisode;
exports.TVSeason = schema_https$1.TVSeason;
exports.TVSeries = schema_https$1.TVSeries;
exports.Table = schema_https$1.Table;
exports.TakeAction = schema_https$1.TakeAction;
exports.TattooParlor = schema_https$1.TattooParlor;
exports.Taxi = schema_https$1.Taxi;
exports.TaxiReservation = schema_https$1.TaxiReservation;
exports.TaxiService = schema_https$1.TaxiService;
exports.TaxiStand = schema_https$1.TaxiStand;
exports.Taxon = schema_https$1.Taxon;
exports.TechArticle = schema_https$1.TechArticle;
exports.TelevisionChannel = schema_https$1.TelevisionChannel;
exports.TelevisionStation = schema_https$1.TelevisionStation;
exports.TennisComplex = schema_https$1.TennisComplex;
exports.Text = schema_https$1.Text;
exports.TextDigitalDocument = schema_https$1.TextDigitalDocument;
exports.TextObject = schema_https$1.TextObject;
exports.TheaterEvent = schema_https$1.TheaterEvent;
exports.TheaterGroup = schema_https$1.TheaterGroup;
exports.TherapeuticProcedure = schema_https$1.TherapeuticProcedure;
exports.Thesis = schema_https$1.Thesis;
exports.Thing = schema_https$1.Thing;
exports.Ticket = schema_https$1.Ticket;
exports.TieAction = schema_https$1.TieAction;
exports.Time = schema_https$1.Time;
exports.TipAction = schema_https$1.TipAction;
exports.TireShop = schema_https$1.TireShop;
exports.TouristAttraction = schema_https$1.TouristAttraction;
exports.TouristDestination = schema_https$1.TouristDestination;
exports.TouristInformationCenter = schema_https$1.TouristInformationCenter;
exports.TouristTrip = schema_https$1.TouristTrip;
exports.ToyStore = schema_https$1.ToyStore;
exports.TrackAction = schema_https$1.TrackAction;
exports.TradeAction = schema_https$1.TradeAction;
exports.TrainReservation = schema_https$1.TrainReservation;
exports.TrainStation = schema_https$1.TrainStation;
exports.TrainTrip = schema_https$1.TrainTrip;
exports.TransferAction = schema_https$1.TransferAction;
exports.TravelAction = schema_https$1.TravelAction;
exports.TravelAgency = schema_https$1.TravelAgency;
exports.TreatmentIndication = schema_https$1.TreatmentIndication;
exports.Trip = schema_https$1.Trip;
exports.TypeAndQuantityNode = schema_https$1.TypeAndQuantityNode;
exports.UKNonprofitType = schema_https$1.UKNonprofitType;
exports.URL = schema_https$1.URL;
exports.USNonprofitType = schema_https$1.USNonprofitType;
exports.UnRegisterAction = schema_https$1.UnRegisterAction;
exports.UnitPriceSpecification = schema_https$1.UnitPriceSpecification;
exports.UpdateAction = schema_https$1.UpdateAction;
exports.UseAction = schema_https$1.UseAction;
exports.UserBlocks = schema_https$1.UserBlocks;
exports.UserCheckins = schema_https$1.UserCheckins;
exports.UserComments = schema_https$1.UserComments;
exports.UserDownloads = schema_https$1.UserDownloads;
exports.UserInteraction = schema_https$1.UserInteraction;
exports.UserLikes = schema_https$1.UserLikes;
exports.UserPageVisits = schema_https$1.UserPageVisits;
exports.UserPlays = schema_https$1.UserPlays;
exports.UserPlusOnes = schema_https$1.UserPlusOnes;
exports.UserReview = schema_https$1.UserReview;
exports.UserTweets = schema_https$1.UserTweets;
exports.VacationRental = schema_https$1.VacationRental;
exports.Vehicle = schema_https$1.Vehicle;
exports.Vein = schema_https$1.Vein;
exports.Vessel = schema_https$1.Vessel;
exports.VeterinaryCare = schema_https$1.VeterinaryCare;
exports.VideoGallery = schema_https$1.VideoGallery;
exports.VideoGame = schema_https$1.VideoGame;
exports.VideoGameClip = schema_https$1.VideoGameClip;
exports.VideoGameSeries = schema_https$1.VideoGameSeries;
exports.VideoObject = schema_https$1.VideoObject;
exports.VideoObjectSnapshot = schema_https$1.VideoObjectSnapshot;
exports.ViewAction = schema_https$1.ViewAction;
exports.VirtualLocation = schema_https$1.VirtualLocation;
exports.VisualArtsEvent = schema_https$1.VisualArtsEvent;
exports.VisualArtwork = schema_https$1.VisualArtwork;
exports.VitalSign = schema_https$1.VitalSign;
exports.Volcano = schema_https$1.Volcano;
exports.VoteAction = schema_https$1.VoteAction;
exports.WPAdBlock = schema_https$1.WPAdBlock;
exports.WPFooter = schema_https$1.WPFooter;
exports.WPHeader = schema_https$1.WPHeader;
exports.WPSideBar = schema_https$1.WPSideBar;
exports.WantAction = schema_https$1.WantAction;
exports.WarrantyPromise = schema_https$1.WarrantyPromise;
exports.WarrantyScope = schema_https$1.WarrantyScope;
exports.WatchAction = schema_https$1.WatchAction;
exports.Waterfall = schema_https$1.Waterfall;
exports.WearAction = schema_https$1.WearAction;
exports.WearableMeasurementTypeEnumeration = schema_https$1.WearableMeasurementTypeEnumeration;
exports.WearableSizeGroupEnumeration = schema_https$1.WearableSizeGroupEnumeration;
exports.WearableSizeSystemEnumeration = schema_https$1.WearableSizeSystemEnumeration;
exports.WebAPI = schema_https$1.WebAPI;
exports.WebApplication = schema_https$1.WebApplication;
exports.WebContent = schema_https$1.WebContent;
exports.WebPage = schema_https$1.WebPage;
exports.WebPageElement = schema_https$1.WebPageElement;
exports.WebSite = schema_https$1.WebSite;
exports.WholesaleStore = schema_https$1.WholesaleStore;
exports.WinAction = schema_https$1.WinAction;
exports.Winery = schema_https$1.Winery;
exports.WorkBasedProgram = schema_https$1.WorkBasedProgram;
exports.WorkersUnion = schema_https$1.WorkersUnion;
exports.WriteAction = schema_https$1.WriteAction;
exports.XPathType = schema_https$1.XPathType;
exports.Zoo = schema_https$1.Zoo;
exports.about = schema_https$1.about;
exports.abridged = schema_https$1.abridged;
exports.abstract = schema_https$1.abstract;
exports.accelerationTime = schema_https$1.accelerationTime;
exports.acceptedAnswer = schema_https$1.acceptedAnswer;
exports.acceptedOffer = schema_https$1.acceptedOffer;
exports.acceptedPaymentMethod = schema_https$1.acceptedPaymentMethod;
exports.acceptsReservations = schema_https$1.acceptsReservations;
exports.accessCode = schema_https$1.accessCode;
exports.accessMode = schema_https$1.accessMode;
exports.accessModeSufficient = schema_https$1.accessModeSufficient;
exports.accessibilityAPI = schema_https$1.accessibilityAPI;
exports.accessibilityControl = schema_https$1.accessibilityControl;
exports.accessibilityFeature = schema_https$1.accessibilityFeature;
exports.accessibilityHazard = schema_https$1.accessibilityHazard;
exports.accessibilitySummary = schema_https$1.accessibilitySummary;
exports.accommodationCategory = schema_https$1.accommodationCategory;
exports.accommodationFloorPlan = schema_https$1.accommodationFloorPlan;
exports.accountId = schema_https$1.accountId;
exports.accountMinimumInflow = schema_https$1.accountMinimumInflow;
exports.accountOverdraftLimit = schema_https$1.accountOverdraftLimit;
exports.accountablePerson = schema_https$1.accountablePerson;
exports.acquireLicensePage = schema_https$1.acquireLicensePage;
exports.acquiredFrom = schema_https$1.acquiredFrom;
exports.acrissCode = schema_https$1.acrissCode;
exports.actionAccessibilityRequirement = schema_https$1.actionAccessibilityRequirement;
exports.actionApplication = schema_https$1.actionApplication;
exports.actionOption = schema_https$1.actionOption;
exports.actionPlatform = schema_https$1.actionPlatform;
exports.actionStatus = schema_https$1.actionStatus;
exports.actionableFeedbackPolicy = schema_https$1.actionableFeedbackPolicy;
exports.activeIngredient = schema_https$1.activeIngredient;
exports.activityDuration = schema_https$1.activityDuration;
exports.activityFrequency = schema_https$1.activityFrequency;
exports.actor = schema_https$1.actor;
exports.actors = schema_https$1.actors;
exports.addOn = schema_https$1.addOn;
exports.additionalName = schema_https$1.additionalName;
exports.additionalNumberOfGuests = schema_https$1.additionalNumberOfGuests;
exports.additionalProperty = schema_https$1.additionalProperty;
exports.additionalType = schema_https$1.additionalType;
exports.additionalVariable = schema_https$1.additionalVariable;
exports.address = schema_https$1.address;
exports.addressCountry = schema_https$1.addressCountry;
exports.addressLocality = schema_https$1.addressLocality;
exports.addressRegion = schema_https$1.addressRegion;
exports.administrationRoute = schema_https$1.administrationRoute;
exports.advanceBookingRequirement = schema_https$1.advanceBookingRequirement;
exports.adverseOutcome = schema_https$1.adverseOutcome;
exports.affectedBy = schema_https$1.affectedBy;
exports.affiliation = schema_https$1.affiliation;
exports.afterMedia = schema_https$1.afterMedia;
exports.agent = schema_https$1.agent;
exports.aggregateRating = schema_https$1.aggregateRating;
exports.aircraft = schema_https$1.aircraft;
exports.album = schema_https$1.album;
exports.albumProductionType = schema_https$1.albumProductionType;
exports.albumRelease = schema_https$1.albumRelease;
exports.albumReleaseType = schema_https$1.albumReleaseType;
exports.albums = schema_https$1.albums;
exports.alcoholWarning = schema_https$1.alcoholWarning;
exports.algorithm = schema_https$1.algorithm;
exports.alignmentType = schema_https$1.alignmentType;
exports.alternateName = schema_https$1.alternateName;
exports.alternativeHeadline = schema_https$1.alternativeHeadline;
exports.alternativeOf = schema_https$1.alternativeOf;
exports.alumni = schema_https$1.alumni;
exports.alumniOf = schema_https$1.alumniOf;
exports.amenityFeature = schema_https$1.amenityFeature;
exports.amount = schema_https$1.amount;
exports.amountOfThisGood = schema_https$1.amountOfThisGood;
exports.announcementLocation = schema_https$1.announcementLocation;
exports.annualPercentageRate = schema_https$1.annualPercentageRate;
exports.answerCount = schema_https$1.answerCount;
exports.answerExplanation = schema_https$1.answerExplanation;
exports.antagonist = schema_https$1.antagonist;
exports.appearance = schema_https$1.appearance;
exports.applicableCountry = schema_https$1.applicableCountry;
exports.applicableLocation = schema_https$1.applicableLocation;
exports.applicantLocationRequirements = schema_https$1.applicantLocationRequirements;
exports.application = schema_https$1.application;
exports.applicationCategory = schema_https$1.applicationCategory;
exports.applicationContact = schema_https$1.applicationContact;
exports.applicationDeadline = schema_https$1.applicationDeadline;
exports.applicationStartDate = schema_https$1.applicationStartDate;
exports.applicationSubCategory = schema_https$1.applicationSubCategory;
exports.applicationSuite = schema_https$1.applicationSuite;
exports.appliesToDeliveryMethod = schema_https$1.appliesToDeliveryMethod;
exports.appliesToPaymentMethod = schema_https$1.appliesToPaymentMethod;
exports.archiveHeld = schema_https$1.archiveHeld;
exports.archivedAt = schema_https$1.archivedAt;
exports.area = schema_https$1.area;
exports.areaServed = schema_https$1.areaServed;
exports.arrivalAirport = schema_https$1.arrivalAirport;
exports.arrivalBoatTerminal = schema_https$1.arrivalBoatTerminal;
exports.arrivalBusStop = schema_https$1.arrivalBusStop;
exports.arrivalGate = schema_https$1.arrivalGate;
exports.arrivalPlatform = schema_https$1.arrivalPlatform;
exports.arrivalStation = schema_https$1.arrivalStation;
exports.arrivalTerminal = schema_https$1.arrivalTerminal;
exports.arrivalTime = schema_https$1.arrivalTime;
exports.artEdition = schema_https$1.artEdition;
exports.artMedium = schema_https$1.artMedium;
exports.arterialBranch = schema_https$1.arterialBranch;
exports.artform = schema_https$1.artform;
exports.articleBody = schema_https$1.articleBody;
exports.articleSection = schema_https$1.articleSection;
exports.artist = schema_https$1.artist;
exports.artworkSurface = schema_https$1.artworkSurface;
exports.asin = schema_https$1.asin;
exports.aspect = schema_https$1.aspect;
exports.assembly = schema_https$1.assembly;
exports.assemblyVersion = schema_https$1.assemblyVersion;
exports.assesses = schema_https$1.assesses;
exports.associatedAnatomy = schema_https$1.associatedAnatomy;
exports.associatedArticle = schema_https$1.associatedArticle;
exports.associatedClaimReview = schema_https$1.associatedClaimReview;
exports.associatedDisease = schema_https$1.associatedDisease;
exports.associatedMedia = schema_https$1.associatedMedia;
exports.associatedMediaReview = schema_https$1.associatedMediaReview;
exports.associatedPathophysiology = schema_https$1.associatedPathophysiology;
exports.associatedReview = schema_https$1.associatedReview;
exports.athlete = schema_https$1.athlete;
exports.attendee = schema_https$1.attendee;
exports.attendees = schema_https$1.attendees;
exports.audience = schema_https$1.audience;
exports.audienceType = schema_https$1.audienceType;
exports.audio = schema_https$1.audio;
exports.authenticator = schema_https$1.authenticator;
exports.author = schema_https$1.author;
exports.availability = schema_https$1.availability;
exports.availabilityEnds = schema_https$1.availabilityEnds;
exports.availabilityStarts = schema_https$1.availabilityStarts;
exports.availableAtOrFrom = schema_https$1.availableAtOrFrom;
exports.availableChannel = schema_https$1.availableChannel;
exports.availableDeliveryMethod = schema_https$1.availableDeliveryMethod;
exports.availableFrom = schema_https$1.availableFrom;
exports.availableIn = schema_https$1.availableIn;
exports.availableLanguage = schema_https$1.availableLanguage;
exports.availableOnDevice = schema_https$1.availableOnDevice;
exports.availableService = schema_https$1.availableService;
exports.availableStrength = schema_https$1.availableStrength;
exports.availableTest = schema_https$1.availableTest;
exports.availableThrough = schema_https$1.availableThrough;
exports.award = schema_https$1.award;
exports.awards = schema_https$1.awards;
exports.awayTeam = schema_https$1.awayTeam;
exports.backstory = schema_https$1.backstory;
exports.bankAccountType = schema_https$1.bankAccountType;
exports.baseSalary = schema_https$1.baseSalary;
exports.bccRecipient = schema_https$1.bccRecipient;
exports.bed = schema_https$1.bed;
exports.beforeMedia = schema_https$1.beforeMedia;
exports.beneficiaryBank = schema_https$1.beneficiaryBank;
exports.benefits = schema_https$1.benefits;
exports.benefitsSummaryUrl = schema_https$1.benefitsSummaryUrl;
exports.bestRating = schema_https$1.bestRating;
exports.billingAddress = schema_https$1.billingAddress;
exports.billingDuration = schema_https$1.billingDuration;
exports.billingIncrement = schema_https$1.billingIncrement;
exports.billingPeriod = schema_https$1.billingPeriod;
exports.billingStart = schema_https$1.billingStart;
exports.bioChemInteraction = schema_https$1.bioChemInteraction;
exports.bioChemSimilarity = schema_https$1.bioChemSimilarity;
exports.biologicalRole = schema_https$1.biologicalRole;
exports.biomechnicalClass = schema_https$1.biomechnicalClass;
exports.birthDate = schema_https$1.birthDate;
exports.birthPlace = schema_https$1.birthPlace;
exports.bitrate = schema_https$1.bitrate;
exports.blogPost = schema_https$1.blogPost;
exports.blogPosts = schema_https$1.blogPosts;
exports.bloodSupply = schema_https$1.bloodSupply;
exports.boardingGroup = schema_https$1.boardingGroup;
exports.boardingPolicy = schema_https$1.boardingPolicy;
exports.bodyLocation = schema_https$1.bodyLocation;
exports.bodyType = schema_https$1.bodyType;
exports.bookEdition = schema_https$1.bookEdition;
exports.bookFormat = schema_https$1.bookFormat;
exports.bookingAgent = schema_https$1.bookingAgent;
exports.bookingTime = schema_https$1.bookingTime;
exports.borrower = schema_https$1.borrower;
exports.box = schema_https$1.box;
exports.branch = schema_https$1.branch;
exports.branchCode = schema_https$1.branchCode;
exports.branchOf = schema_https$1.branchOf;
exports.brand = schema_https$1.brand;
exports.breadcrumb = schema_https$1.breadcrumb;
exports.breastfeedingWarning = schema_https$1.breastfeedingWarning;
exports.broadcastAffiliateOf = schema_https$1.broadcastAffiliateOf;
exports.broadcastChannelId = schema_https$1.broadcastChannelId;
exports.broadcastDisplayName = schema_https$1.broadcastDisplayName;
exports.broadcastFrequency = schema_https$1.broadcastFrequency;
exports.broadcastFrequencyValue = schema_https$1.broadcastFrequencyValue;
exports.broadcastOfEvent = schema_https$1.broadcastOfEvent;
exports.broadcastServiceTier = schema_https$1.broadcastServiceTier;
exports.broadcastSignalModulation = schema_https$1.broadcastSignalModulation;
exports.broadcastSubChannel = schema_https$1.broadcastSubChannel;
exports.broadcastTimezone = schema_https$1.broadcastTimezone;
exports.broadcaster = schema_https$1.broadcaster;
exports.broker = schema_https$1.broker;
exports.browserRequirements = schema_https$1.browserRequirements;
exports.busName = schema_https$1.busName;
exports.busNumber = schema_https$1.busNumber;
exports.businessDays = schema_https$1.businessDays;
exports.businessFunction = schema_https$1.businessFunction;
exports.buyer = schema_https$1.buyer;
exports.byArtist = schema_https$1.byArtist;
exports.byDay = schema_https$1.byDay;
exports.byMonth = schema_https$1.byMonth;
exports.byMonthDay = schema_https$1.byMonthDay;
exports.byMonthWeek = schema_https$1.byMonthWeek;
exports.callSign = schema_https$1.callSign;
exports.calories = schema_https$1.calories;
exports.candidate = schema_https$1.candidate;
exports.caption = schema_https$1.caption;
exports.carbohydrateContent = schema_https$1.carbohydrateContent;
exports.cargoVolume = schema_https$1.cargoVolume;
exports.carrier = schema_https$1.carrier;
exports.carrierRequirements = schema_https$1.carrierRequirements;
exports.cashBack = schema_https$1.cashBack;
exports.catalog = schema_https$1.catalog;
exports.catalogNumber = schema_https$1.catalogNumber;
exports.category = schema_https$1.category;
exports.causeOf = schema_https$1.causeOf;
exports.ccRecipient = schema_https$1.ccRecipient;
exports.character = schema_https$1.character;
exports.characterAttribute = schema_https$1.characterAttribute;
exports.characterName = schema_https$1.characterName;
exports.cheatCode = schema_https$1.cheatCode;
exports.checkinTime = schema_https$1.checkinTime;
exports.checkoutPageURLTemplate = schema_https$1.checkoutPageURLTemplate;
exports.checkoutTime = schema_https$1.checkoutTime;
exports.chemicalComposition = schema_https$1.chemicalComposition;
exports.chemicalRole = schema_https$1.chemicalRole;
exports.childMaxAge = schema_https$1.childMaxAge;
exports.childMinAge = schema_https$1.childMinAge;
exports.childTaxon = schema_https$1.childTaxon;
exports.children = schema_https$1.children;
exports.cholesterolContent = schema_https$1.cholesterolContent;
exports.circle = schema_https$1.circle;
exports.citation = schema_https$1.citation;
exports.claimInterpreter = schema_https$1.claimInterpreter;
exports.claimReviewed = schema_https$1.claimReviewed;
exports.clincalPharmacology = schema_https$1.clincalPharmacology;
exports.clinicalPharmacology = schema_https$1.clinicalPharmacology;
exports.clipNumber = schema_https$1.clipNumber;
exports.closes = schema_https$1.closes;
exports.coach = schema_https$1.coach;
exports.code = schema_https$1.code;
exports.codeRepository = schema_https$1.codeRepository;
exports.codeSampleType = schema_https$1.codeSampleType;
exports.codeValue = schema_https$1.codeValue;
exports.codingSystem = schema_https$1.codingSystem;
exports.colleague = schema_https$1.colleague;
exports.colleagues = schema_https$1.colleagues;
exports.collection = schema_https$1.collection;
exports.collectionSize = schema_https$1.collectionSize;
exports.color = schema_https$1.color;
exports.colorist = schema_https$1.colorist;
exports.comment = schema_https$1.comment;
exports.commentCount = schema_https$1.commentCount;
exports.commentText = schema_https$1.commentText;
exports.commentTime = schema_https$1.commentTime;
exports.competencyRequired = schema_https$1.competencyRequired;
exports.competitor = schema_https$1.competitor;
exports.composer = schema_https$1.composer;
exports.comprisedOf = schema_https$1.comprisedOf;
exports.conditionsOfAccess = schema_https$1.conditionsOfAccess;
exports.confirmationNumber = schema_https$1.confirmationNumber;
exports.connectedTo = schema_https$1.connectedTo;
exports.constraintProperty = schema_https$1.constraintProperty;
exports.contactOption = schema_https$1.contactOption;
exports.contactPoint = schema_https$1.contactPoint;
exports.contactPoints = schema_https$1.contactPoints;
exports.contactType = schema_https$1.contactType;
exports.contactlessPayment = schema_https$1.contactlessPayment;
exports.containedIn = schema_https$1.containedIn;
exports.containedInPlace = schema_https$1.containedInPlace;
exports.containsPlace = schema_https$1.containsPlace;
exports.containsSeason = schema_https$1.containsSeason;
exports.contentLocation = schema_https$1.contentLocation;
exports.contentRating = schema_https$1.contentRating;
exports.contentReferenceTime = schema_https$1.contentReferenceTime;
exports.contentSize = schema_https$1.contentSize;
exports.contentType = schema_https$1.contentType;
exports.contentUrl = schema_https$1.contentUrl;
exports.contraindication = schema_https$1.contraindication;
exports.contributor = schema_https$1.contributor;
exports.cookTime = schema_https$1.cookTime;
exports.cookingMethod = schema_https$1.cookingMethod;
exports.copyrightHolder = schema_https$1.copyrightHolder;
exports.copyrightNotice = schema_https$1.copyrightNotice;
exports.copyrightYear = schema_https$1.copyrightYear;
exports.correction = schema_https$1.correction;
exports.correctionsPolicy = schema_https$1.correctionsPolicy;
exports.costCategory = schema_https$1.costCategory;
exports.costCurrency = schema_https$1.costCurrency;
exports.costOrigin = schema_https$1.costOrigin;
exports.costPerUnit = schema_https$1.costPerUnit;
exports.countriesNotSupported = schema_https$1.countriesNotSupported;
exports.countriesSupported = schema_https$1.countriesSupported;
exports.countryOfAssembly = schema_https$1.countryOfAssembly;
exports.countryOfLastProcessing = schema_https$1.countryOfLastProcessing;
exports.countryOfOrigin = schema_https$1.countryOfOrigin;
exports.course = schema_https$1.course;
exports.courseCode = schema_https$1.courseCode;
exports.courseMode = schema_https$1.courseMode;
exports.coursePrerequisites = schema_https$1.coursePrerequisites;
exports.courseSchedule = schema_https$1.courseSchedule;
exports.courseWorkload = schema_https$1.courseWorkload;
exports.coverageEndTime = schema_https$1.coverageEndTime;
exports.coverageStartTime = schema_https$1.coverageStartTime;
exports.creativeWorkStatus = schema_https$1.creativeWorkStatus;
exports.creator = schema_https$1.creator;
exports.credentialCategory = schema_https$1.credentialCategory;
exports.creditText = schema_https$1.creditText;
exports.creditedTo = schema_https$1.creditedTo;
exports.cssSelector = schema_https$1.cssSelector;
exports.currenciesAccepted = schema_https$1.currenciesAccepted;
exports.currency = schema_https$1.currency;
exports.currentExchangeRate = schema_https$1.currentExchangeRate;
exports.customer = schema_https$1.customer;
exports.customerRemorseReturnFees = schema_https$1.customerRemorseReturnFees;
exports.customerRemorseReturnLabelSource = schema_https$1.customerRemorseReturnLabelSource;
exports.customerRemorseReturnShippingFeesAmount = schema_https$1.customerRemorseReturnShippingFeesAmount;
exports.cutoffTime = schema_https$1.cutoffTime;
exports.cvdCollectionDate = schema_https$1.cvdCollectionDate;
exports.cvdFacilityCounty = schema_https$1.cvdFacilityCounty;
exports.cvdFacilityId = schema_https$1.cvdFacilityId;
exports.cvdNumBeds = schema_https$1.cvdNumBeds;
exports.cvdNumBedsOcc = schema_https$1.cvdNumBedsOcc;
exports.cvdNumC19Died = schema_https$1.cvdNumC19Died;
exports.cvdNumC19HOPats = schema_https$1.cvdNumC19HOPats;
exports.cvdNumC19HospPats = schema_https$1.cvdNumC19HospPats;
exports.cvdNumC19MechVentPats = schema_https$1.cvdNumC19MechVentPats;
exports.cvdNumC19OFMechVentPats = schema_https$1.cvdNumC19OFMechVentPats;
exports.cvdNumC19OverflowPats = schema_https$1.cvdNumC19OverflowPats;
exports.cvdNumICUBeds = schema_https$1.cvdNumICUBeds;
exports.cvdNumICUBedsOcc = schema_https$1.cvdNumICUBedsOcc;
exports.cvdNumTotBeds = schema_https$1.cvdNumTotBeds;
exports.cvdNumVent = schema_https$1.cvdNumVent;
exports.cvdNumVentUse = schema_https$1.cvdNumVentUse;
exports.dataFeedElement = schema_https$1.dataFeedElement;
exports.dataset = schema_https$1.dataset;
exports.datasetTimeInterval = schema_https$1.datasetTimeInterval;
exports.dateCreated = schema_https$1.dateCreated;
exports.dateDeleted = schema_https$1.dateDeleted;
exports.dateIssued = schema_https$1.dateIssued;
exports.dateModified = schema_https$1.dateModified;
exports.datePosted = schema_https$1.datePosted;
exports.datePublished = schema_https$1.datePublished;
exports.dateRead = schema_https$1.dateRead;
exports.dateReceived = schema_https$1.dateReceived;
exports.dateSent = schema_https$1.dateSent;
exports.dateVehicleFirstRegistered = schema_https$1.dateVehicleFirstRegistered;
exports.dateline = schema_https$1.dateline;
exports.dayOfWeek = schema_https$1.dayOfWeek;
exports.deathDate = schema_https$1.deathDate;
exports.deathPlace = schema_https$1.deathPlace;
exports.defaultValue = schema_https$1.defaultValue;
exports.deliveryAddress = schema_https$1.deliveryAddress;
exports.deliveryLeadTime = schema_https$1.deliveryLeadTime;
exports.deliveryMethod = schema_https$1.deliveryMethod;
exports.deliveryStatus = schema_https$1.deliveryStatus;
exports.deliveryTime = schema_https$1.deliveryTime;
exports.department = schema_https$1.department;
exports.departureAirport = schema_https$1.departureAirport;
exports.departureBoatTerminal = schema_https$1.departureBoatTerminal;
exports.departureBusStop = schema_https$1.departureBusStop;
exports.departureGate = schema_https$1.departureGate;
exports.departurePlatform = schema_https$1.departurePlatform;
exports.departureStation = schema_https$1.departureStation;
exports.departureTerminal = schema_https$1.departureTerminal;
exports.departureTime = schema_https$1.departureTime;
exports.dependencies = schema_https$1.dependencies;
exports.depth = schema_https$1.depth;
exports.description = schema_https$1.description;
exports.device = schema_https$1.device;
exports.diagnosis = schema_https$1.diagnosis;
exports.diagram = schema_https$1.diagram;
exports.diet = schema_https$1.diet;
exports.dietFeatures = schema_https$1.dietFeatures;
exports.differentialDiagnosis = schema_https$1.differentialDiagnosis;
exports.directApply = schema_https$1.directApply;
exports.director = schema_https$1.director;
exports.directors = schema_https$1.directors;
exports.disambiguatingDescription = schema_https$1.disambiguatingDescription;
exports.discount = schema_https$1.discount;
exports.discountCode = schema_https$1.discountCode;
exports.discountCurrency = schema_https$1.discountCurrency;
exports.discusses = schema_https$1.discusses;
exports.discussionUrl = schema_https$1.discussionUrl;
exports.diseasePreventionInfo = schema_https$1.diseasePreventionInfo;
exports.diseaseSpreadStatistics = schema_https$1.diseaseSpreadStatistics;
exports.dissolutionDate = schema_https$1.dissolutionDate;
exports.distance = schema_https$1.distance;
exports.distinguishingSign = schema_https$1.distinguishingSign;
exports.distribution = schema_https$1.distribution;
exports.diversityPolicy = schema_https$1.diversityPolicy;
exports.diversityStaffingReport = schema_https$1.diversityStaffingReport;
exports.documentation = schema_https$1.documentation;
exports.doesNotShip = schema_https$1.doesNotShip;
exports.domainIncludes = schema_https$1.domainIncludes;
exports.domiciledMortgage = schema_https$1.domiciledMortgage;
exports.doorTime = schema_https$1.doorTime;
exports.dosageForm = schema_https$1.dosageForm;
exports.doseSchedule = schema_https$1.doseSchedule;
exports.doseUnit = schema_https$1.doseUnit;
exports.doseValue = schema_https$1.doseValue;
exports.downPayment = schema_https$1.downPayment;
exports.downloadUrl = schema_https$1.downloadUrl;
exports.downvoteCount = schema_https$1.downvoteCount;
exports.drainsTo = schema_https$1.drainsTo;
exports.driveWheelConfiguration = schema_https$1.driveWheelConfiguration;
exports.dropoffLocation = schema_https$1.dropoffLocation;
exports.dropoffTime = schema_https$1.dropoffTime;
exports.drug = schema_https$1.drug;
exports.drugClass = schema_https$1.drugClass;
exports.drugUnit = schema_https$1.drugUnit;
exports.duns = schema_https$1.duns;
exports.duplicateTherapy = schema_https$1.duplicateTherapy;
exports.duration = schema_https$1.duration;
exports.durationOfWarranty = schema_https$1.durationOfWarranty;
exports.duringMedia = schema_https$1.duringMedia;
exports.earlyPrepaymentPenalty = schema_https$1.earlyPrepaymentPenalty;
exports.editEIDR = schema_https$1.editEIDR;
exports.editor = schema_https$1.editor;
exports.eduQuestionType = schema_https$1.eduQuestionType;
exports.educationRequirements = schema_https$1.educationRequirements;
exports.educationalAlignment = schema_https$1.educationalAlignment;
exports.educationalCredentialAwarded = schema_https$1.educationalCredentialAwarded;
exports.educationalFramework = schema_https$1.educationalFramework;
exports.educationalLevel = schema_https$1.educationalLevel;
exports.educationalProgramMode = schema_https$1.educationalProgramMode;
exports.educationalRole = schema_https$1.educationalRole;
exports.educationalUse = schema_https$1.educationalUse;
exports.elevation = schema_https$1.elevation;
exports.eligibilityToWorkRequirement = schema_https$1.eligibilityToWorkRequirement;
exports.eligibleCustomerType = schema_https$1.eligibleCustomerType;
exports.eligibleDuration = schema_https$1.eligibleDuration;
exports.eligibleQuantity = schema_https$1.eligibleQuantity;
exports.eligibleRegion = schema_https$1.eligibleRegion;
exports.eligibleTransactionVolume = schema_https$1.eligibleTransactionVolume;
exports.email = schema_https$1.email;
exports.embedUrl = schema_https$1.embedUrl;
exports.embeddedTextCaption = schema_https$1.embeddedTextCaption;
exports.emissionsCO2 = schema_https$1.emissionsCO2;
exports.employee = schema_https$1.employee;
exports.employees = schema_https$1.employees;
exports.employerOverview = schema_https$1.employerOverview;
exports.employmentType = schema_https$1.employmentType;
exports.employmentUnit = schema_https$1.employmentUnit;
exports.encodesBioChemEntity = schema_https$1.encodesBioChemEntity;
exports.encodesCreativeWork = schema_https$1.encodesCreativeWork;
exports.encoding = schema_https$1.encoding;
exports.encodingFormat = schema_https$1.encodingFormat;
exports.encodingType = schema_https$1.encodingType;
exports.encodings = schema_https$1.encodings;
exports.endDate = schema_https$1.endDate;
exports.endOffset = schema_https$1.endOffset;
exports.endTime = schema_https$1.endTime;
exports.endorsee = schema_https$1.endorsee;
exports.endorsers = schema_https$1.endorsers;
exports.energyEfficiencyScaleMax = schema_https$1.energyEfficiencyScaleMax;
exports.energyEfficiencyScaleMin = schema_https$1.energyEfficiencyScaleMin;
exports.engineDisplacement = schema_https$1.engineDisplacement;
exports.enginePower = schema_https$1.enginePower;
exports.engineType = schema_https$1.engineType;
exports.entertainmentBusiness = schema_https$1.entertainmentBusiness;
exports.epidemiology = schema_https$1.epidemiology;
exports.episode = schema_https$1.episode;
exports.episodeNumber = schema_https$1.episodeNumber;
exports.episodes = schema_https$1.episodes;
exports.equal = schema_https$1.equal;
exports.error = schema_https$1.error;
exports.estimatedCost = schema_https$1.estimatedCost;
exports.estimatedFlightDuration = schema_https$1.estimatedFlightDuration;
exports.estimatedSalary = schema_https$1.estimatedSalary;
exports.estimatesRiskOf = schema_https$1.estimatesRiskOf;
exports.ethicsPolicy = schema_https$1.ethicsPolicy;
exports.event = schema_https$1.event;
exports.eventAttendanceMode = schema_https$1.eventAttendanceMode;
exports.eventSchedule = schema_https$1.eventSchedule;
exports.eventStatus = schema_https$1.eventStatus;
exports.events = schema_https$1.events;
exports.evidenceLevel = schema_https$1.evidenceLevel;
exports.evidenceOrigin = schema_https$1.evidenceOrigin;
exports.exampleOfWork = schema_https$1.exampleOfWork;
exports.exceptDate = schema_https$1.exceptDate;
exports.exchangeRateSpread = schema_https$1.exchangeRateSpread;
exports.executableLibraryName = schema_https$1.executableLibraryName;
exports.exerciseCourse = schema_https$1.exerciseCourse;
exports.exercisePlan = schema_https$1.exercisePlan;
exports.exerciseRelatedDiet = schema_https$1.exerciseRelatedDiet;
exports.exerciseType = schema_https$1.exerciseType;
exports.exifData = schema_https$1.exifData;
exports.expectedArrivalFrom = schema_https$1.expectedArrivalFrom;
exports.expectedArrivalUntil = schema_https$1.expectedArrivalUntil;
exports.expectedPrognosis = schema_https$1.expectedPrognosis;
exports.expectsAcceptanceOf = schema_https$1.expectsAcceptanceOf;
exports.experienceInPlaceOfEducation = schema_https$1.experienceInPlaceOfEducation;
exports.experienceRequirements = schema_https$1.experienceRequirements;
exports.expertConsiderations = schema_https$1.expertConsiderations;
exports.expires = schema_https$1.expires;
exports.expressedIn = schema_https$1.expressedIn;
exports.familyName = schema_https$1.familyName;
exports.fatContent = schema_https$1.fatContent;
exports.faxNumber = schema_https$1.faxNumber;
exports.featureList = schema_https$1.featureList;
exports.feesAndCommissionsSpecification = schema_https$1.feesAndCommissionsSpecification;
exports.fiberContent = schema_https$1.fiberContent;
exports.fileFormat = schema_https$1.fileFormat;
exports.fileSize = schema_https$1.fileSize;
exports.financialAidEligible = schema_https$1.financialAidEligible;
exports.firstAppearance = schema_https$1.firstAppearance;
exports.firstPerformance = schema_https$1.firstPerformance;
exports.flightDistance = schema_https$1.flightDistance;
exports.flightNumber = schema_https$1.flightNumber;
exports.floorLevel = schema_https$1.floorLevel;
exports.floorLimit = schema_https$1.floorLimit;
exports.floorSize = schema_https$1.floorSize;
exports.followee = schema_https$1.followee;
exports.follows = schema_https$1.follows;
exports.followup = schema_https$1.followup;
exports.foodEstablishment = schema_https$1.foodEstablishment;
exports.foodEvent = schema_https$1.foodEvent;
exports.foodWarning = schema_https$1.foodWarning;
exports.founder = schema_https$1.founder;
exports.founders = schema_https$1.founders;
exports.foundingDate = schema_https$1.foundingDate;
exports.foundingLocation = schema_https$1.foundingLocation;
exports.free = schema_https$1.free;
exports.freeShippingThreshold = schema_https$1.freeShippingThreshold;
exports.frequency = schema_https$1.frequency;
exports.fromLocation = schema_https$1.fromLocation;
exports.fuelCapacity = schema_https$1.fuelCapacity;
exports.fuelConsumption = schema_https$1.fuelConsumption;
exports.fuelEfficiency = schema_https$1.fuelEfficiency;
exports.fuelType = schema_https$1.fuelType;
exports.functionalClass = schema_https$1.functionalClass;
exports.fundedItem = schema_https$1.fundedItem;
exports.funder = schema_https$1.funder;
exports.funding = schema_https$1.funding;
exports.game = schema_https$1.game;
exports.gameAvailabilityType = schema_https$1.gameAvailabilityType;
exports.gameEdition = schema_https$1.gameEdition;
exports.gameItem = schema_https$1.gameItem;
exports.gameLocation = schema_https$1.gameLocation;
exports.gamePlatform = schema_https$1.gamePlatform;
exports.gameServer = schema_https$1.gameServer;
exports.gameTip = schema_https$1.gameTip;
exports.gender = schema_https$1.gender;
exports.genre = schema_https$1.genre;
exports.geo = schema_https$1.geo;
exports.geoContains = schema_https$1.geoContains;
exports.geoCoveredBy = schema_https$1.geoCoveredBy;
exports.geoCovers = schema_https$1.geoCovers;
exports.geoCrosses = schema_https$1.geoCrosses;
exports.geoDisjoint = schema_https$1.geoDisjoint;
exports.geoEquals = schema_https$1.geoEquals;
exports.geoIntersects = schema_https$1.geoIntersects;
exports.geoMidpoint = schema_https$1.geoMidpoint;
exports.geoOverlaps = schema_https$1.geoOverlaps;
exports.geoRadius = schema_https$1.geoRadius;
exports.geoTouches = schema_https$1.geoTouches;
exports.geoWithin = schema_https$1.geoWithin;
exports.geographicArea = schema_https$1.geographicArea;
exports.gettingTestedInfo = schema_https$1.gettingTestedInfo;
exports.givenName = schema_https$1.givenName;
exports.globalLocationNumber = schema_https$1.globalLocationNumber;
exports.governmentBenefitsInfo = schema_https$1.governmentBenefitsInfo;
exports.gracePeriod = schema_https$1.gracePeriod;
exports.grantee = schema_https$1.grantee;
exports.greater = schema_https$1.greater;
exports.greaterOrEqual = schema_https$1.greaterOrEqual;
exports.gtin = schema_https$1.gtin;
exports.gtin12 = schema_https$1.gtin12;
exports.gtin13 = schema_https$1.gtin13;
exports.gtin14 = schema_https$1.gtin14;
exports.gtin8 = schema_https$1.gtin8;
exports.guideline = schema_https$1.guideline;
exports.guidelineDate = schema_https$1.guidelineDate;
exports.guidelineSubject = schema_https$1.guidelineSubject;
exports.handlingTime = schema_https$1.handlingTime;
exports.hasAdultConsideration = schema_https$1.hasAdultConsideration;
exports.hasBioChemEntityPart = schema_https$1.hasBioChemEntityPart;
exports.hasBioPolymerSequence = schema_https$1.hasBioPolymerSequence;
exports.hasBroadcastChannel = schema_https$1.hasBroadcastChannel;
exports.hasCategoryCode = schema_https$1.hasCategoryCode;
exports.hasCourse = schema_https$1.hasCourse;
exports.hasCourseInstance = schema_https$1.hasCourseInstance;
exports.hasCredential = schema_https$1.hasCredential;
exports.hasDefinedTerm = schema_https$1.hasDefinedTerm;
exports.hasDeliveryMethod = schema_https$1.hasDeliveryMethod;
exports.hasDigitalDocumentPermission = schema_https$1.hasDigitalDocumentPermission;
exports.hasDriveThroughService = schema_https$1.hasDriveThroughService;
exports.hasEnergyConsumptionDetails = schema_https$1.hasEnergyConsumptionDetails;
exports.hasEnergyEfficiencyCategory = schema_https$1.hasEnergyEfficiencyCategory;
exports.hasHealthAspect = schema_https$1.hasHealthAspect;
exports.hasMap = schema_https$1.hasMap;
exports.hasMeasurement = schema_https$1.hasMeasurement;
exports.hasMenu = schema_https$1.hasMenu;
exports.hasMenuItem = schema_https$1.hasMenuItem;
exports.hasMenuSection = schema_https$1.hasMenuSection;
exports.hasMerchantReturnPolicy = schema_https$1.hasMerchantReturnPolicy;
exports.hasMolecularFunction = schema_https$1.hasMolecularFunction;
exports.hasOccupation = schema_https$1.hasOccupation;
exports.hasOfferCatalog = schema_https$1.hasOfferCatalog;
exports.hasPOS = schema_https$1.hasPOS;
exports.hasPart = schema_https$1.hasPart;
exports.hasProductReturnPolicy = schema_https$1.hasProductReturnPolicy;
exports.hasRepresentation = schema_https$1.hasRepresentation;
exports.hasVariant = schema_https$1.hasVariant;
exports.headline = schema_https$1.headline;
exports.healthCondition = schema_https$1.healthCondition;
exports.healthPlanCoinsuranceOption = schema_https$1.healthPlanCoinsuranceOption;
exports.healthPlanCoinsuranceRate = schema_https$1.healthPlanCoinsuranceRate;
exports.healthPlanCopay = schema_https$1.healthPlanCopay;
exports.healthPlanCopayOption = schema_https$1.healthPlanCopayOption;
exports.healthPlanCostSharing = schema_https$1.healthPlanCostSharing;
exports.healthPlanDrugOption = schema_https$1.healthPlanDrugOption;
exports.healthPlanDrugTier = schema_https$1.healthPlanDrugTier;
exports.healthPlanId = schema_https$1.healthPlanId;
exports.healthPlanMarketingUrl = schema_https$1.healthPlanMarketingUrl;
exports.healthPlanNetworkId = schema_https$1.healthPlanNetworkId;
exports.healthPlanNetworkTier = schema_https$1.healthPlanNetworkTier;
exports.healthPlanPharmacyCategory = schema_https$1.healthPlanPharmacyCategory;
exports.healthcareReportingData = schema_https$1.healthcareReportingData;
exports.height = schema_https$1.height;
exports.highPrice = schema_https$1.highPrice;
exports.hiringOrganization = schema_https$1.hiringOrganization;
exports.holdingArchive = schema_https$1.holdingArchive;
exports.homeLocation = schema_https$1.homeLocation;
exports.homeTeam = schema_https$1.homeTeam;
exports.honorificPrefix = schema_https$1.honorificPrefix;
exports.honorificSuffix = schema_https$1.honorificSuffix;
exports.hospitalAffiliation = schema_https$1.hospitalAffiliation;
exports.hostingOrganization = schema_https$1.hostingOrganization;
exports.hoursAvailable = schema_https$1.hoursAvailable;
exports.howPerformed = schema_https$1.howPerformed;
exports.httpMethod = schema_https$1.httpMethod;
exports.iataCode = schema_https$1.iataCode;
exports.icaoCode = schema_https$1.icaoCode;
exports.identifier = schema_https$1.identifier;
exports.identifyingExam = schema_https$1.identifyingExam;
exports.identifyingTest = schema_https$1.identifyingTest;
exports.illustrator = schema_https$1.illustrator;
exports.image = schema_https$1.image;
exports.imagingTechnique = schema_https$1.imagingTechnique;
exports.inAlbum = schema_https$1.inAlbum;
exports.inBroadcastLineup = schema_https$1.inBroadcastLineup;
exports.inChI = schema_https$1.inChI;
exports.inChIKey = schema_https$1.inChIKey;
exports.inCodeSet = schema_https$1.inCodeSet;
exports.inDefinedTermSet = schema_https$1.inDefinedTermSet;
exports.inLanguage = schema_https$1.inLanguage;
exports.inPlaylist = schema_https$1.inPlaylist;
exports.inProductGroupWithID = schema_https$1.inProductGroupWithID;
exports.inStoreReturnsOffered = schema_https$1.inStoreReturnsOffered;
exports.inSupportOf = schema_https$1.inSupportOf;
exports.incentiveCompensation = schema_https$1.incentiveCompensation;
exports.incentives = schema_https$1.incentives;
exports.includedComposition = schema_https$1.includedComposition;
exports.includedDataCatalog = schema_https$1.includedDataCatalog;
exports.includedInDataCatalog = schema_https$1.includedInDataCatalog;
exports.includedInHealthInsurancePlan = schema_https$1.includedInHealthInsurancePlan;
exports.includedRiskFactor = schema_https$1.includedRiskFactor;
exports.includesAttraction = schema_https$1.includesAttraction;
exports.includesHealthPlanFormulary = schema_https$1.includesHealthPlanFormulary;
exports.includesHealthPlanNetwork = schema_https$1.includesHealthPlanNetwork;
exports.includesObject = schema_https$1.includesObject;
exports.increasesRiskOf = schema_https$1.increasesRiskOf;
exports.industry = schema_https$1.industry;
exports.ineligibleRegion = schema_https$1.ineligibleRegion;
exports.infectiousAgent = schema_https$1.infectiousAgent;
exports.infectiousAgentClass = schema_https$1.infectiousAgentClass;
exports.ingredients = schema_https$1.ingredients;
exports.inker = schema_https$1.inker;
exports.insertion = schema_https$1.insertion;
exports.installUrl = schema_https$1.installUrl;
exports.instructor = schema_https$1.instructor;
exports.instrument = schema_https$1.instrument;
exports.intensity = schema_https$1.intensity;
exports.interactingDrug = schema_https$1.interactingDrug;
exports.interactionCount = schema_https$1.interactionCount;
exports.interactionService = schema_https$1.interactionService;
exports.interactionStatistic = schema_https$1.interactionStatistic;
exports.interactionType = schema_https$1.interactionType;
exports.interactivityType = schema_https$1.interactivityType;
exports.interestRate = schema_https$1.interestRate;
exports.interpretedAsClaim = schema_https$1.interpretedAsClaim;
exports.inventoryLevel = schema_https$1.inventoryLevel;
exports.inverseOf = schema_https$1.inverseOf;
exports.isAcceptingNewPatients = schema_https$1.isAcceptingNewPatients;
exports.isAccessibleForFree = schema_https$1.isAccessibleForFree;
exports.isAccessoryOrSparePartFor = schema_https$1.isAccessoryOrSparePartFor;
exports.isAvailableGenerically = schema_https$1.isAvailableGenerically;
exports.isBasedOn = schema_https$1.isBasedOn;
exports.isBasedOnUrl = schema_https$1.isBasedOnUrl;
exports.isConsumableFor = schema_https$1.isConsumableFor;
exports.isEncodedByBioChemEntity = schema_https$1.isEncodedByBioChemEntity;
exports.isFamilyFriendly = schema_https$1.isFamilyFriendly;
exports.isGift = schema_https$1.isGift;
exports.isInvolvedInBiologicalProcess = schema_https$1.isInvolvedInBiologicalProcess;
exports.isLiveBroadcast = schema_https$1.isLiveBroadcast;
exports.isLocatedInSubcellularLocation = schema_https$1.isLocatedInSubcellularLocation;
exports.isPartOf = schema_https$1.isPartOf;
exports.isPartOfBioChemEntity = schema_https$1.isPartOfBioChemEntity;
exports.isPlanForApartment = schema_https$1.isPlanForApartment;
exports.isProprietary = schema_https$1.isProprietary;
exports.isRelatedTo = schema_https$1.isRelatedTo;
exports.isResizable = schema_https$1.isResizable;
exports.isSimilarTo = schema_https$1.isSimilarTo;
exports.isUnlabelledFallback = schema_https$1.isUnlabelledFallback;
exports.isVariantOf = schema_https$1.isVariantOf;
exports.isbn = schema_https$1.isbn;
exports.isicV4 = schema_https$1.isicV4;
exports.iso6523Code = schema_https$1.iso6523Code;
exports.isrcCode = schema_https$1.isrcCode;
exports.issn = schema_https$1.issn;
exports.issueNumber = schema_https$1.issueNumber;
exports.issuedBy = schema_https$1.issuedBy;
exports.issuedThrough = schema_https$1.issuedThrough;
exports.iswcCode = schema_https$1.iswcCode;
exports.item = schema_https$1.item;
exports.itemCondition = schema_https$1.itemCondition;
exports.itemDefectReturnFees = schema_https$1.itemDefectReturnFees;
exports.itemDefectReturnLabelSource = schema_https$1.itemDefectReturnLabelSource;
exports.itemDefectReturnShippingFeesAmount = schema_https$1.itemDefectReturnShippingFeesAmount;
exports.itemListElement = schema_https$1.itemListElement;
exports.itemListOrder = schema_https$1.itemListOrder;
exports.itemLocation = schema_https$1.itemLocation;
exports.itemOffered = schema_https$1.itemOffered;
exports.itemReviewed = schema_https$1.itemReviewed;
exports.itemShipped = schema_https$1.itemShipped;
exports.itinerary = schema_https$1.itinerary;
exports.iupacName = schema_https$1.iupacName;
exports.jobBenefits = schema_https$1.jobBenefits;
exports.jobImmediateStart = schema_https$1.jobImmediateStart;
exports.jobLocation = schema_https$1.jobLocation;
exports.jobLocationType = schema_https$1.jobLocationType;
exports.jobStartDate = schema_https$1.jobStartDate;
exports.jobTitle = schema_https$1.jobTitle;
exports.jurisdiction = schema_https$1.jurisdiction;
exports.keywords = schema_https$1.keywords;
exports.knownVehicleDamages = schema_https$1.knownVehicleDamages;
exports.knows = schema_https$1.knows;
exports.knowsAbout = schema_https$1.knowsAbout;
exports.knowsLanguage = schema_https$1.knowsLanguage;
exports.labelDetails = schema_https$1.labelDetails;
exports.landlord = schema_https$1.landlord;
exports.language = schema_https$1.language;
exports.lastReviewed = schema_https$1.lastReviewed;
exports.latitude = schema_https$1.latitude;
exports.layoutImage = schema_https$1.layoutImage;
exports.learningResourceType = schema_https$1.learningResourceType;
exports.leaseLength = schema_https$1.leaseLength;
exports.legalName = schema_https$1.legalName;
exports.legalStatus = schema_https$1.legalStatus;
exports.legislationApplies = schema_https$1.legislationApplies;
exports.legislationChanges = schema_https$1.legislationChanges;
exports.legislationConsolidates = schema_https$1.legislationConsolidates;
exports.legislationDate = schema_https$1.legislationDate;
exports.legislationDateVersion = schema_https$1.legislationDateVersion;
exports.legislationIdentifier = schema_https$1.legislationIdentifier;
exports.legislationJurisdiction = schema_https$1.legislationJurisdiction;
exports.legislationLegalForce = schema_https$1.legislationLegalForce;
exports.legislationLegalValue = schema_https$1.legislationLegalValue;
exports.legislationPassedBy = schema_https$1.legislationPassedBy;
exports.legislationResponsible = schema_https$1.legislationResponsible;
exports.legislationTransposes = schema_https$1.legislationTransposes;
exports.legislationType = schema_https$1.legislationType;
exports.leiCode = schema_https$1.leiCode;
exports.lender = schema_https$1.lender;
exports.lesser = schema_https$1.lesser;
exports.lesserOrEqual = schema_https$1.lesserOrEqual;
exports.letterer = schema_https$1.letterer;
exports.license = schema_https$1.license;
exports.line = schema_https$1.line;
exports.linkRelationship = schema_https$1.linkRelationship;
exports.liveBlogUpdate = schema_https$1.liveBlogUpdate;
exports.loanMortgageMandateAmount = schema_https$1.loanMortgageMandateAmount;
exports.loanPaymentAmount = schema_https$1.loanPaymentAmount;
exports.loanPaymentFrequency = schema_https$1.loanPaymentFrequency;
exports.loanRepaymentForm = schema_https$1.loanRepaymentForm;
exports.loanTerm = schema_https$1.loanTerm;
exports.loanType = schema_https$1.loanType;
exports.location = schema_https$1.location;
exports.locationCreated = schema_https$1.locationCreated;
exports.lodgingUnitDescription = schema_https$1.lodgingUnitDescription;
exports.lodgingUnitType = schema_https$1.lodgingUnitType;
exports.logo = schema_https$1.logo;
exports.longitude = schema_https$1.longitude;
exports.loser = schema_https$1.loser;
exports.lowPrice = schema_https$1.lowPrice;
exports.lyricist = schema_https$1.lyricist;
exports.lyrics = schema_https$1.lyrics;
exports.mainContentOfPage = schema_https$1.mainContentOfPage;
exports.mainEntity = schema_https$1.mainEntity;
exports.mainEntityOfPage = schema_https$1.mainEntityOfPage;
exports.maintainer = schema_https$1.maintainer;
exports.makesOffer = schema_https$1.makesOffer;
exports.manufacturer = schema_https$1.manufacturer;
exports.map = schema_https$1.map;
exports.mapType = schema_https$1.mapType;
exports.maps = schema_https$1.maps;
exports.marginOfError = schema_https$1.marginOfError;
exports.masthead = schema_https$1.masthead;
exports.material = schema_https$1.material;
exports.materialExtent = schema_https$1.materialExtent;
exports.mathExpression = schema_https$1.mathExpression;
exports.maxPrice = schema_https$1.maxPrice;
exports.maxValue = schema_https$1.maxValue;
exports.maximumAttendeeCapacity = schema_https$1.maximumAttendeeCapacity;
exports.maximumEnrollment = schema_https$1.maximumEnrollment;
exports.maximumIntake = schema_https$1.maximumIntake;
exports.maximumPhysicalAttendeeCapacity = schema_https$1.maximumPhysicalAttendeeCapacity;
exports.maximumVirtualAttendeeCapacity = schema_https$1.maximumVirtualAttendeeCapacity;
exports.mealService = schema_https$1.mealService;
exports.measuredProperty = schema_https$1.measuredProperty;
exports.measurementDenominator = schema_https$1.measurementDenominator;
exports.measurementMethod = schema_https$1.measurementMethod;
exports.measurementQualifier = schema_https$1.measurementQualifier;
exports.measurementTechnique = schema_https$1.measurementTechnique;
exports.mechanismOfAction = schema_https$1.mechanismOfAction;
exports.mediaAuthenticityCategory = schema_https$1.mediaAuthenticityCategory;
exports.mediaItemAppearance = schema_https$1.mediaItemAppearance;
exports.median = schema_https$1.median;
exports.medicalAudience = schema_https$1.medicalAudience;
exports.medicalSpecialty = schema_https$1.medicalSpecialty;
exports.medicineSystem = schema_https$1.medicineSystem;
exports.meetsEmissionStandard = schema_https$1.meetsEmissionStandard;
exports.member = schema_https$1.member;
exports.memberOf = schema_https$1.memberOf;
exports.members = schema_https$1.members;
exports.membershipNumber = schema_https$1.membershipNumber;
exports.membershipPointsEarned = schema_https$1.membershipPointsEarned;
exports.memoryRequirements = schema_https$1.memoryRequirements;
exports.mentions = schema_https$1.mentions;
exports.menu = schema_https$1.menu;
exports.menuAddOn = schema_https$1.menuAddOn;
exports.merchant = schema_https$1.merchant;
exports.merchantReturnDays = schema_https$1.merchantReturnDays;
exports.merchantReturnLink = schema_https$1.merchantReturnLink;
exports.messageAttachment = schema_https$1.messageAttachment;
exports.mileageFromOdometer = schema_https$1.mileageFromOdometer;
exports.minPrice = schema_https$1.minPrice;
exports.minValue = schema_https$1.minValue;
exports.minimumPaymentDue = schema_https$1.minimumPaymentDue;
exports.missionCoveragePrioritiesPolicy = schema_https$1.missionCoveragePrioritiesPolicy;
exports.mobileUrl = schema_https$1.mobileUrl;
exports.model = schema_https$1.model;
exports.modelDate = schema_https$1.modelDate;
exports.modifiedTime = schema_https$1.modifiedTime;
exports.molecularFormula = schema_https$1.molecularFormula;
exports.molecularWeight = schema_https$1.molecularWeight;
exports.monoisotopicMolecularWeight = schema_https$1.monoisotopicMolecularWeight;
exports.monthlyMinimumRepaymentAmount = schema_https$1.monthlyMinimumRepaymentAmount;
exports.monthsOfExperience = schema_https$1.monthsOfExperience;
exports.mpn = schema_https$1.mpn;
exports.multipleValues = schema_https$1.multipleValues;
exports.muscleAction = schema_https$1.muscleAction;
exports.musicArrangement = schema_https$1.musicArrangement;
exports.musicBy = schema_https$1.musicBy;
exports.musicCompositionForm = schema_https$1.musicCompositionForm;
exports.musicGroupMember = schema_https$1.musicGroupMember;
exports.musicReleaseFormat = schema_https$1.musicReleaseFormat;
exports.musicalKey = schema_https$1.musicalKey;
exports.naics = schema_https$1.naics;
exports.name = schema_https$1.name;
exports.namedPosition = schema_https$1.namedPosition;
exports.nationality = schema_https$1.nationality;
exports.naturalProgression = schema_https$1.naturalProgression;
exports.negativeNotes = schema_https$1.negativeNotes;
exports.nerve = schema_https$1.nerve;
exports.nerveMotor = schema_https$1.nerveMotor;
exports.netWorth = schema_https$1.netWorth;
exports.newsUpdatesAndGuidelines = schema_https$1.newsUpdatesAndGuidelines;
exports.nextItem = schema_https$1.nextItem;
exports.noBylinesPolicy = schema_https$1.noBylinesPolicy;
exports.nonEqual = schema_https$1.nonEqual;
exports.nonProprietaryName = schema_https$1.nonProprietaryName;
exports.nonprofitStatus = schema_https$1.nonprofitStatus;
exports.normalRange = schema_https$1.normalRange;
exports.nsn = schema_https$1.nsn;
exports.numAdults = schema_https$1.numAdults;
exports.numChildren = schema_https$1.numChildren;
exports.numConstraints = schema_https$1.numConstraints;
exports.numTracks = schema_https$1.numTracks;
exports.numberOfAccommodationUnits = schema_https$1.numberOfAccommodationUnits;
exports.numberOfAirbags = schema_https$1.numberOfAirbags;
exports.numberOfAvailableAccommodationUnits = schema_https$1.numberOfAvailableAccommodationUnits;
exports.numberOfAxles = schema_https$1.numberOfAxles;
exports.numberOfBathroomsTotal = schema_https$1.numberOfBathroomsTotal;
exports.numberOfBedrooms = schema_https$1.numberOfBedrooms;
exports.numberOfBeds = schema_https$1.numberOfBeds;
exports.numberOfCredits = schema_https$1.numberOfCredits;
exports.numberOfDoors = schema_https$1.numberOfDoors;
exports.numberOfEmployees = schema_https$1.numberOfEmployees;
exports.numberOfEpisodes = schema_https$1.numberOfEpisodes;
exports.numberOfForwardGears = schema_https$1.numberOfForwardGears;
exports.numberOfFullBathrooms = schema_https$1.numberOfFullBathrooms;
exports.numberOfItems = schema_https$1.numberOfItems;
exports.numberOfLoanPayments = schema_https$1.numberOfLoanPayments;
exports.numberOfPages = schema_https$1.numberOfPages;
exports.numberOfPartialBathrooms = schema_https$1.numberOfPartialBathrooms;
exports.numberOfPlayers = schema_https$1.numberOfPlayers;
exports.numberOfPreviousOwners = schema_https$1.numberOfPreviousOwners;
exports.numberOfRooms = schema_https$1.numberOfRooms;
exports.numberOfSeasons = schema_https$1.numberOfSeasons;
exports.numberedPosition = schema_https$1.numberedPosition;
exports.nutrition = schema_https$1.nutrition;
exports.object = schema_https$1.object;
exports.observationAbout = schema_https$1.observationAbout;
exports.observationDate = schema_https$1.observationDate;
exports.observationPeriod = schema_https$1.observationPeriod;
exports.occupancy = schema_https$1.occupancy;
exports.occupationLocation = schema_https$1.occupationLocation;
exports.occupationalCategory = schema_https$1.occupationalCategory;
exports.occupationalCredentialAwarded = schema_https$1.occupationalCredentialAwarded;
exports.offerCount = schema_https$1.offerCount;
exports.offeredBy = schema_https$1.offeredBy;
exports.offers = schema_https$1.offers;
exports.offersPrescriptionByMail = schema_https$1.offersPrescriptionByMail;
exports.openingHours = schema_https$1.openingHours;
exports.openingHoursSpecification = schema_https$1.openingHoursSpecification;
exports.opens = schema_https$1.opens;
exports.operatingSystem = schema_https$1.operatingSystem;
exports.opponent = schema_https$1.opponent;
exports.option = schema_https$1.option;
exports.orderDate = schema_https$1.orderDate;
exports.orderDelivery = schema_https$1.orderDelivery;
exports.orderItemNumber = schema_https$1.orderItemNumber;
exports.orderItemStatus = schema_https$1.orderItemStatus;
exports.orderNumber = schema_https$1.orderNumber;
exports.orderQuantity = schema_https$1.orderQuantity;
exports.orderStatus = schema_https$1.orderStatus;
exports.orderedItem = schema_https$1.orderedItem;
exports.organizer = schema_https$1.organizer;
exports.originAddress = schema_https$1.originAddress;
exports.originalMediaContextDescription = schema_https$1.originalMediaContextDescription;
exports.originalMediaLink = schema_https$1.originalMediaLink;
exports.originatesFrom = schema_https$1.originatesFrom;
exports.overdosage = schema_https$1.overdosage;
exports.ownedFrom = schema_https$1.ownedFrom;
exports.ownedThrough = schema_https$1.ownedThrough;
exports.ownershipFundingInfo = schema_https$1.ownershipFundingInfo;
exports.owns = schema_https$1.owns;
exports.pageEnd = schema_https$1.pageEnd;
exports.pageStart = schema_https$1.pageStart;
exports.pagination = schema_https$1.pagination;
exports.parent = schema_https$1.parent;
exports.parentItem = schema_https$1.parentItem;
exports.parentOrganization = schema_https$1.parentOrganization;
exports.parentService = schema_https$1.parentService;
exports.parentTaxon = schema_https$1.parentTaxon;
exports.parents = schema_https$1.parents;
exports.partOfEpisode = schema_https$1.partOfEpisode;
exports.partOfInvoice = schema_https$1.partOfInvoice;
exports.partOfOrder = schema_https$1.partOfOrder;
exports.partOfSeason = schema_https$1.partOfSeason;
exports.partOfSeries = schema_https$1.partOfSeries;
exports.partOfSystem = schema_https$1.partOfSystem;
exports.partOfTVSeries = schema_https$1.partOfTVSeries;
exports.partOfTrip = schema_https$1.partOfTrip;
exports.participant = schema_https$1.participant;
exports.partySize = schema_https$1.partySize;
exports.passengerPriorityStatus = schema_https$1.passengerPriorityStatus;
exports.passengerSequenceNumber = schema_https$1.passengerSequenceNumber;
exports.pathophysiology = schema_https$1.pathophysiology;
exports.pattern = schema_https$1.pattern;
exports.payload = schema_https$1.payload;
exports.paymentAccepted = schema_https$1.paymentAccepted;
exports.paymentDue = schema_https$1.paymentDue;
exports.paymentDueDate = schema_https$1.paymentDueDate;
exports.paymentMethod = schema_https$1.paymentMethod;
exports.paymentMethodId = schema_https$1.paymentMethodId;
exports.paymentStatus = schema_https$1.paymentStatus;
exports.paymentUrl = schema_https$1.paymentUrl;
exports.penciler = schema_https$1.penciler;
exports.percentile10 = schema_https$1.percentile10;
exports.percentile25 = schema_https$1.percentile25;
exports.percentile75 = schema_https$1.percentile75;
exports.percentile90 = schema_https$1.percentile90;
exports.performTime = schema_https$1.performTime;
exports.performer = schema_https$1.performer;
exports.performerIn = schema_https$1.performerIn;
exports.performers = schema_https$1.performers;
exports.permissionType = schema_https$1.permissionType;
exports.permissions = schema_https$1.permissions;
exports.permitAudience = schema_https$1.permitAudience;
exports.permittedUsage = schema_https$1.permittedUsage;
exports.petsAllowed = schema_https$1.petsAllowed;
exports.phoneticText = schema_https$1.phoneticText;
exports.photo = schema_https$1.photo;
exports.photos = schema_https$1.photos;
exports.physicalRequirement = schema_https$1.physicalRequirement;
exports.physiologicalBenefits = schema_https$1.physiologicalBenefits;
exports.pickupLocation = schema_https$1.pickupLocation;
exports.pickupTime = schema_https$1.pickupTime;
exports.playMode = schema_https$1.playMode;
exports.playerType = schema_https$1.playerType;
exports.playersOnline = schema_https$1.playersOnline;
exports.polygon = schema_https$1.polygon;
exports.populationType = schema_https$1.populationType;
exports.position = schema_https$1.position;
exports.positiveNotes = schema_https$1.positiveNotes;
exports.possibleComplication = schema_https$1.possibleComplication;
exports.possibleTreatment = schema_https$1.possibleTreatment;
exports.postOfficeBoxNumber = schema_https$1.postOfficeBoxNumber;
exports.postOp = schema_https$1.postOp;
exports.postalCode = schema_https$1.postalCode;
exports.postalCodeBegin = schema_https$1.postalCodeBegin;
exports.postalCodeEnd = schema_https$1.postalCodeEnd;
exports.postalCodePrefix = schema_https$1.postalCodePrefix;
exports.postalCodeRange = schema_https$1.postalCodeRange;
exports.potentialAction = schema_https$1.potentialAction;
exports.potentialUse = schema_https$1.potentialUse;
exports.preOp = schema_https$1.preOp;
exports.predecessorOf = schema_https$1.predecessorOf;
exports.pregnancyCategory = schema_https$1.pregnancyCategory;
exports.pregnancyWarning = schema_https$1.pregnancyWarning;
exports.prepTime = schema_https$1.prepTime;
exports.preparation = schema_https$1.preparation;
exports.prescribingInfo = schema_https$1.prescribingInfo;
exports.prescriptionStatus = schema_https$1.prescriptionStatus;
exports.previousItem = schema_https$1.previousItem;
exports.previousStartDate = schema_https$1.previousStartDate;
exports.price = schema_https$1.price;
exports.priceComponent = schema_https$1.priceComponent;
exports.priceComponentType = schema_https$1.priceComponentType;
exports.priceCurrency = schema_https$1.priceCurrency;
exports.priceRange = schema_https$1.priceRange;
exports.priceSpecification = schema_https$1.priceSpecification;
exports.priceType = schema_https$1.priceType;
exports.priceValidUntil = schema_https$1.priceValidUntil;
exports.primaryImageOfPage = schema_https$1.primaryImageOfPage;
exports.primaryPrevention = schema_https$1.primaryPrevention;
exports.printColumn = schema_https$1.printColumn;
exports.printEdition = schema_https$1.printEdition;
exports.printPage = schema_https$1.printPage;
exports.printSection = schema_https$1.printSection;
exports.procedure = schema_https$1.procedure;
exports.procedureType = schema_https$1.procedureType;
exports.processingTime = schema_https$1.processingTime;
exports.processorRequirements = schema_https$1.processorRequirements;
exports.producer = schema_https$1.producer;
exports.produces = schema_https$1.produces;
exports.productGroupID = schema_https$1.productGroupID;
exports.productID = schema_https$1.productID;
exports.productReturnDays = schema_https$1.productReturnDays;
exports.productReturnLink = schema_https$1.productReturnLink;
exports.productSupported = schema_https$1.productSupported;
exports.productionCompany = schema_https$1.productionCompany;
exports.productionDate = schema_https$1.productionDate;
exports.proficiencyLevel = schema_https$1.proficiencyLevel;
exports.programMembershipUsed = schema_https$1.programMembershipUsed;
exports.programName = schema_https$1.programName;
exports.programPrerequisites = schema_https$1.programPrerequisites;
exports.programType = schema_https$1.programType;
exports.programmingLanguage = schema_https$1.programmingLanguage;
exports.programmingModel = schema_https$1.programmingModel;
exports.propertyID = schema_https$1.propertyID;
exports.proprietaryName = schema_https$1.proprietaryName;
exports.proteinContent = schema_https$1.proteinContent;
exports.provider = schema_https$1.provider;
exports.providerMobility = schema_https$1.providerMobility;
exports.providesBroadcastService = schema_https$1.providesBroadcastService;
exports.providesService = schema_https$1.providesService;
exports.publicAccess = schema_https$1.publicAccess;
exports.publicTransportClosuresInfo = schema_https$1.publicTransportClosuresInfo;
exports.publication = schema_https$1.publication;
exports.publicationType = schema_https$1.publicationType;
exports.publishedBy = schema_https$1.publishedBy;
exports.publishedOn = schema_https$1.publishedOn;
exports.publisher = schema_https$1.publisher;
exports.publisherImprint = schema_https$1.publisherImprint;
exports.publishingPrinciples = schema_https$1.publishingPrinciples;
exports.purchaseDate = schema_https$1.purchaseDate;
exports.qualifications = schema_https$1.qualifications;
exports.quarantineGuidelines = schema_https$1.quarantineGuidelines;
exports.query = schema_https$1.query;
exports.quest = schema_https$1.quest;
exports.question = schema_https$1.question;
exports.rangeIncludes = schema_https$1.rangeIncludes;
exports.ratingCount = schema_https$1.ratingCount;
exports.ratingExplanation = schema_https$1.ratingExplanation;
exports.ratingValue = schema_https$1.ratingValue;
exports.readBy = schema_https$1.readBy;
exports.readonlyValue = schema_https$1.readonlyValue;
exports.realEstateAgent = schema_https$1.realEstateAgent;
exports.recipe = schema_https$1.recipe;
exports.recipeCategory = schema_https$1.recipeCategory;
exports.recipeCuisine = schema_https$1.recipeCuisine;
exports.recipeIngredient = schema_https$1.recipeIngredient;
exports.recipeInstructions = schema_https$1.recipeInstructions;
exports.recipeYield = schema_https$1.recipeYield;
exports.recipient = schema_https$1.recipient;
exports.recognizedBy = schema_https$1.recognizedBy;
exports.recognizingAuthority = schema_https$1.recognizingAuthority;
exports.recommendationStrength = schema_https$1.recommendationStrength;
exports.recommendedIntake = schema_https$1.recommendedIntake;
exports.recordLabel = schema_https$1.recordLabel;
exports.recordedAs = schema_https$1.recordedAs;
exports.recordedAt = schema_https$1.recordedAt;
exports.recordedIn = schema_https$1.recordedIn;
exports.recordingOf = schema_https$1.recordingOf;
exports.recourseLoan = schema_https$1.recourseLoan;
exports.referenceQuantity = schema_https$1.referenceQuantity;
exports.referencesOrder = schema_https$1.referencesOrder;
exports.refundType = schema_https$1.refundType;
exports.regionDrained = schema_https$1.regionDrained;
exports.regionsAllowed = schema_https$1.regionsAllowed;
exports.relatedAnatomy = schema_https$1.relatedAnatomy;
exports.relatedCondition = schema_https$1.relatedCondition;
exports.relatedDrug = schema_https$1.relatedDrug;
exports.relatedLink = schema_https$1.relatedLink;
exports.relatedStructure = schema_https$1.relatedStructure;
exports.relatedTherapy = schema_https$1.relatedTherapy;
exports.relatedTo = schema_https$1.relatedTo;
exports.releaseDate = schema_https$1.releaseDate;
exports.releaseNotes = schema_https$1.releaseNotes;
exports.releaseOf = schema_https$1.releaseOf;
exports.releasedEvent = schema_https$1.releasedEvent;
exports.relevantOccupation = schema_https$1.relevantOccupation;
exports.relevantSpecialty = schema_https$1.relevantSpecialty;
exports.remainingAttendeeCapacity = schema_https$1.remainingAttendeeCapacity;
exports.renegotiableLoan = schema_https$1.renegotiableLoan;
exports.repeatCount = schema_https$1.repeatCount;
exports.repeatFrequency = schema_https$1.repeatFrequency;
exports.repetitions = schema_https$1.repetitions;
exports.replacee = schema_https$1.replacee;
exports.replacer = schema_https$1.replacer;
exports.replyToUrl = schema_https$1.replyToUrl;
exports.reportNumber = schema_https$1.reportNumber;
exports.representativeOfPage = schema_https$1.representativeOfPage;
exports.requiredCollateral = schema_https$1.requiredCollateral;
exports.requiredGender = schema_https$1.requiredGender;
exports.requiredMaxAge = schema_https$1.requiredMaxAge;
exports.requiredMinAge = schema_https$1.requiredMinAge;
exports.requiredQuantity = schema_https$1.requiredQuantity;
exports.requirements = schema_https$1.requirements;
exports.requiresSubscription = schema_https$1.requiresSubscription;
exports.reservationFor = schema_https$1.reservationFor;
exports.reservationId = schema_https$1.reservationId;
exports.reservationStatus = schema_https$1.reservationStatus;
exports.reservedTicket = schema_https$1.reservedTicket;
exports.responsibilities = schema_https$1.responsibilities;
exports.restPeriods = schema_https$1.restPeriods;
exports.restockingFee = schema_https$1.restockingFee;
exports.result = schema_https$1.result;
exports.resultComment = schema_https$1.resultComment;
exports.resultReview = schema_https$1.resultReview;
exports.returnFees = schema_https$1.returnFees;
exports.returnLabelSource = schema_https$1.returnLabelSource;
exports.returnMethod = schema_https$1.returnMethod;
exports.returnPolicyCategory = schema_https$1.returnPolicyCategory;
exports.returnPolicyCountry = schema_https$1.returnPolicyCountry;
exports.returnPolicySeasonalOverride = schema_https$1.returnPolicySeasonalOverride;
exports.returnShippingFeesAmount = schema_https$1.returnShippingFeesAmount;
exports.review = schema_https$1.review;
exports.reviewAspect = schema_https$1.reviewAspect;
exports.reviewBody = schema_https$1.reviewBody;
exports.reviewCount = schema_https$1.reviewCount;
exports.reviewRating = schema_https$1.reviewRating;
exports.reviewedBy = schema_https$1.reviewedBy;
exports.reviews = schema_https$1.reviews;
exports.riskFactor = schema_https$1.riskFactor;
exports.risks = schema_https$1.risks;
exports.roleName = schema_https$1.roleName;
exports.roofLoad = schema_https$1.roofLoad;
exports.rsvpResponse = schema_https$1.rsvpResponse;
exports.runsTo = schema_https$1.runsTo;
exports.runtime = schema_https$1.runtime;
exports.runtimePlatform = schema_https$1.runtimePlatform;
exports.rxcui = schema_https$1.rxcui;
exports.safetyConsideration = schema_https$1.safetyConsideration;
exports.salaryCurrency = schema_https$1.salaryCurrency;
exports.salaryUponCompletion = schema_https$1.salaryUponCompletion;
exports.sameAs = schema_https$1.sameAs;
exports.sampleType = schema_https$1.sampleType;
exports.saturatedFatContent = schema_https$1.saturatedFatContent;
exports.scheduleTimezone = schema_https$1.scheduleTimezone;
exports.scheduledPaymentDate = schema_https$1.scheduledPaymentDate;
exports.scheduledTime = schema_https$1.scheduledTime;
exports.schemaVersion = schema_https$1.schemaVersion;
exports.schoolClosuresInfo = schema_https$1.schoolClosuresInfo;
exports.screenCount = schema_https$1.screenCount;
exports.screenshot = schema_https$1.screenshot;
exports.sdDatePublished = schema_https$1.sdDatePublished;
exports.sdLicense = schema_https$1.sdLicense;
exports.sdPublisher = schema_https$1.sdPublisher;
exports.season = schema_https$1.season;
exports.seasonNumber = schema_https$1.seasonNumber;
exports.seasons = schema_https$1.seasons;
exports.seatNumber = schema_https$1.seatNumber;
exports.seatRow = schema_https$1.seatRow;
exports.seatSection = schema_https$1.seatSection;
exports.seatingCapacity = schema_https$1.seatingCapacity;
exports.seatingType = schema_https$1.seatingType;
exports.secondaryPrevention = schema_https$1.secondaryPrevention;
exports.securityClearanceRequirement = schema_https$1.securityClearanceRequirement;
exports.securityScreening = schema_https$1.securityScreening;
exports.seeks = schema_https$1.seeks;
exports.seller = schema_https$1.seller;
exports.sender = schema_https$1.sender;
exports.sensoryRequirement = schema_https$1.sensoryRequirement;
exports.sensoryUnit = schema_https$1.sensoryUnit;
exports.serialNumber = schema_https$1.serialNumber;
exports.seriousAdverseOutcome = schema_https$1.seriousAdverseOutcome;
exports.serverStatus = schema_https$1.serverStatus;
exports.servesCuisine = schema_https$1.servesCuisine;
exports.serviceArea = schema_https$1.serviceArea;
exports.serviceAudience = schema_https$1.serviceAudience;
exports.serviceLocation = schema_https$1.serviceLocation;
exports.serviceOperator = schema_https$1.serviceOperator;
exports.serviceOutput = schema_https$1.serviceOutput;
exports.servicePhone = schema_https$1.servicePhone;
exports.servicePostalAddress = schema_https$1.servicePostalAddress;
exports.serviceSmsNumber = schema_https$1.serviceSmsNumber;
exports.serviceType = schema_https$1.serviceType;
exports.serviceUrl = schema_https$1.serviceUrl;
exports.servingSize = schema_https$1.servingSize;
exports.sha256 = schema_https$1.sha256;
exports.sharedContent = schema_https$1.sharedContent;
exports.shippingDestination = schema_https$1.shippingDestination;
exports.shippingDetails = schema_https$1.shippingDetails;
exports.shippingLabel = schema_https$1.shippingLabel;
exports.shippingOrigin = schema_https$1.shippingOrigin;
exports.shippingRate = schema_https$1.shippingRate;
exports.shippingSettingsLink = schema_https$1.shippingSettingsLink;
exports.sibling = schema_https$1.sibling;
exports.siblings = schema_https$1.siblings;
exports.signDetected = schema_https$1.signDetected;
exports.signOrSymptom = schema_https$1.signOrSymptom;
exports.significance = schema_https$1.significance;
exports.significantLink = schema_https$1.significantLink;
exports.significantLinks = schema_https$1.significantLinks;
exports.size = schema_https$1.size;
exports.sizeGroup = schema_https$1.sizeGroup;
exports.sizeSystem = schema_https$1.sizeSystem;
exports.skills = schema_https$1.skills;
exports.sku = schema_https$1.sku;
exports.slogan = schema_https$1.slogan;
exports.smiles = schema_https$1.smiles;
exports.smokingAllowed = schema_https$1.smokingAllowed;
exports.sodiumContent = schema_https$1.sodiumContent;
exports.softwareAddOn = schema_https$1.softwareAddOn;
exports.softwareHelp = schema_https$1.softwareHelp;
exports.softwareRequirements = schema_https$1.softwareRequirements;
exports.softwareVersion = schema_https$1.softwareVersion;
exports.sourceOrganization = schema_https$1.sourceOrganization;
exports.sourcedFrom = schema_https$1.sourcedFrom;
exports.spatial = schema_https$1.spatial;
exports.spatialCoverage = schema_https$1.spatialCoverage;
exports.speakable = schema_https$1.speakable;
exports.specialCommitments = schema_https$1.specialCommitments;
exports.specialOpeningHoursSpecification = schema_https$1.specialOpeningHoursSpecification;
exports.specialty = schema_https$1.specialty;
exports.speechToTextMarkup = schema_https$1.speechToTextMarkup;
exports.speed = schema_https$1.speed;
exports.spokenByCharacter = schema_https$1.spokenByCharacter;
exports.sponsor = schema_https$1.sponsor;
exports.sport = schema_https$1.sport;
exports.sportsActivityLocation = schema_https$1.sportsActivityLocation;
exports.sportsEvent = schema_https$1.sportsEvent;
exports.sportsTeam = schema_https$1.sportsTeam;
exports.spouse = schema_https$1.spouse;
exports.stage = schema_https$1.stage;
exports.stageAsNumber = schema_https$1.stageAsNumber;
exports.starRating = schema_https$1.starRating;
exports.startDate = schema_https$1.startDate;
exports.startOffset = schema_https$1.startOffset;
exports.startTime = schema_https$1.startTime;
exports.statType = schema_https$1.statType;
exports.status = schema_https$1.status;
exports.steeringPosition = schema_https$1.steeringPosition;
exports.step = schema_https$1.step;
exports.stepValue = schema_https$1.stepValue;
exports.steps = schema_https$1.steps;
exports.storageRequirements = schema_https$1.storageRequirements;
exports.streetAddress = schema_https$1.streetAddress;
exports.strengthUnit = schema_https$1.strengthUnit;
exports.strengthValue = schema_https$1.strengthValue;
exports.structuralClass = schema_https$1.structuralClass;
exports.study = schema_https$1.study;
exports.studyDesign = schema_https$1.studyDesign;
exports.studyLocation = schema_https$1.studyLocation;
exports.studySubject = schema_https$1.studySubject;
exports.stupidProperty = schema_https$1.stupidProperty;
exports.subEvent = schema_https$1.subEvent;
exports.subEvents = schema_https$1.subEvents;
exports.subOrganization = schema_https$1.subOrganization;
exports.subReservation = schema_https$1.subReservation;
exports.subStageSuffix = schema_https$1.subStageSuffix;
exports.subStructure = schema_https$1.subStructure;
exports.subTest = schema_https$1.subTest;
exports.subTrip = schema_https$1.subTrip;
exports.subjectOf = schema_https$1.subjectOf;
exports.subtitleLanguage = schema_https$1.subtitleLanguage;
exports.successorOf = schema_https$1.successorOf;
exports.sugarContent = schema_https$1.sugarContent;
exports.suggestedAge = schema_https$1.suggestedAge;
exports.suggestedAnswer = schema_https$1.suggestedAnswer;
exports.suggestedGender = schema_https$1.suggestedGender;
exports.suggestedMaxAge = schema_https$1.suggestedMaxAge;
exports.suggestedMeasurement = schema_https$1.suggestedMeasurement;
exports.suggestedMinAge = schema_https$1.suggestedMinAge;
exports.suitableForDiet = schema_https$1.suitableForDiet;
exports.superEvent = schema_https$1.superEvent;
exports.supersededBy = schema_https$1.supersededBy;
exports.supply = schema_https$1.supply;
exports.supplyTo = schema_https$1.supplyTo;
exports.supportingData = schema_https$1.supportingData;
exports.surface = schema_https$1.surface;
exports.syllabusSections = schema_https$1.syllabusSections;
exports.target = schema_https$1.target;
exports.targetCollection = schema_https$1.targetCollection;
exports.targetDescription = schema_https$1.targetDescription;
exports.targetName = schema_https$1.targetName;
exports.targetPlatform = schema_https$1.targetPlatform;
exports.targetPopulation = schema_https$1.targetPopulation;
exports.targetProduct = schema_https$1.targetProduct;
exports.targetUrl = schema_https$1.targetUrl;
exports.taxID = schema_https$1.taxID;
exports.taxonRank = schema_https$1.taxonRank;
exports.taxonomicRange = schema_https$1.taxonomicRange;
exports.teaches = schema_https$1.teaches;
exports.telephone = schema_https$1.telephone;
exports.temporal = schema_https$1.temporal;
exports.temporalCoverage = schema_https$1.temporalCoverage;
exports.termCode = schema_https$1.termCode;
exports.termDuration = schema_https$1.termDuration;
exports.termsOfService = schema_https$1.termsOfService;
exports.termsPerYear = schema_https$1.termsPerYear;
exports.text = schema_https$1.text;
exports.textValue = schema_https$1.textValue;
exports.thumbnail = schema_https$1.thumbnail;
exports.thumbnailUrl = schema_https$1.thumbnailUrl;
exports.tickerSymbol = schema_https$1.tickerSymbol;
exports.ticketNumber = schema_https$1.ticketNumber;
exports.ticketToken = schema_https$1.ticketToken;
exports.ticketedSeat = schema_https$1.ticketedSeat;
exports.timeOfDay = schema_https$1.timeOfDay;
exports.timeRequired = schema_https$1.timeRequired;
exports.timeToComplete = schema_https$1.timeToComplete;
exports.tissueSample = schema_https$1.tissueSample;
exports.title = schema_https$1.title;
exports.titleEIDR = schema_https$1.titleEIDR;
exports.toLocation = schema_https$1.toLocation;
exports.toRecipient = schema_https$1.toRecipient;
exports.tocContinuation = schema_https$1.tocContinuation;
exports.tocEntry = schema_https$1.tocEntry;
exports.tongueWeight = schema_https$1.tongueWeight;
exports.tool = schema_https$1.tool;
exports.torque = schema_https$1.torque;
exports.totalHistoricalEnrollment = schema_https$1.totalHistoricalEnrollment;
exports.totalJobOpenings = schema_https$1.totalJobOpenings;
exports.totalPaymentDue = schema_https$1.totalPaymentDue;
exports.totalPrice = schema_https$1.totalPrice;
exports.totalTime = schema_https$1.totalTime;
exports.tourBookingPage = schema_https$1.tourBookingPage;
exports.touristType = schema_https$1.touristType;
exports.track = schema_https$1.track;
exports.trackingNumber = schema_https$1.trackingNumber;
exports.trackingUrl = schema_https$1.trackingUrl;
exports.tracks = schema_https$1.tracks;
exports.trailer = schema_https$1.trailer;
exports.trailerWeight = schema_https$1.trailerWeight;
exports.trainName = schema_https$1.trainName;
exports.trainNumber = schema_https$1.trainNumber;
exports.trainingSalary = schema_https$1.trainingSalary;
exports.transFatContent = schema_https$1.transFatContent;
exports.transcript = schema_https$1.transcript;
exports.transitTime = schema_https$1.transitTime;
exports.transitTimeLabel = schema_https$1.transitTimeLabel;
exports.translationOfWork = schema_https$1.translationOfWork;
exports.translator = schema_https$1.translator;
exports.transmissionMethod = schema_https$1.transmissionMethod;
exports.travelBans = schema_https$1.travelBans;
exports.trialDesign = schema_https$1.trialDesign;
exports.tributary = schema_https$1.tributary;
exports.tripOrigin = schema_https$1.tripOrigin;
exports.typeOfBed = schema_https$1.typeOfBed;
exports.typeOfGood = schema_https$1.typeOfGood;
exports.typicalAgeRange = schema_https$1.typicalAgeRange;
exports.typicalCreditsPerTerm = schema_https$1.typicalCreditsPerTerm;
exports.typicalTest = schema_https$1.typicalTest;
exports.underName = schema_https$1.underName;
exports.unitCode = schema_https$1.unitCode;
exports.unitText = schema_https$1.unitText;
exports.unnamedSourcesPolicy = schema_https$1.unnamedSourcesPolicy;
exports.unsaturatedFatContent = schema_https$1.unsaturatedFatContent;
exports.uploadDate = schema_https$1.uploadDate;
exports.upvoteCount = schema_https$1.upvoteCount;
exports.url = schema_https$1.url;
exports.urlTemplate = schema_https$1.urlTemplate;
exports.usageInfo = schema_https$1.usageInfo;
exports.usedToDiagnose = schema_https$1.usedToDiagnose;
exports.userInteractionCount = schema_https$1.userInteractionCount;
exports.usesDevice = schema_https$1.usesDevice;
exports.usesHealthPlanIdStandard = schema_https$1.usesHealthPlanIdStandard;
exports.utterances = schema_https$1.utterances;
exports.validFor = schema_https$1.validFor;
exports.validFrom = schema_https$1.validFrom;
exports.validIn = schema_https$1.validIn;
exports.validThrough = schema_https$1.validThrough;
exports.validUntil = schema_https$1.validUntil;
exports.value = schema_https$1.value;
exports.valueAddedTaxIncluded = schema_https$1.valueAddedTaxIncluded;
exports.valueMaxLength = schema_https$1.valueMaxLength;
exports.valueMinLength = schema_https$1.valueMinLength;
exports.valueName = schema_https$1.valueName;
exports.valuePattern = schema_https$1.valuePattern;
exports.valueReference = schema_https$1.valueReference;
exports.valueRequired = schema_https$1.valueRequired;
exports.variableMeasured = schema_https$1.variableMeasured;
exports.variablesMeasured = schema_https$1.variablesMeasured;
exports.variantCover = schema_https$1.variantCover;
exports.variesBy = schema_https$1.variesBy;
exports.vatID = schema_https$1.vatID;
exports.vehicleConfiguration = schema_https$1.vehicleConfiguration;
exports.vehicleEngine = schema_https$1.vehicleEngine;
exports.vehicleIdentificationNumber = schema_https$1.vehicleIdentificationNumber;
exports.vehicleInteriorColor = schema_https$1.vehicleInteriorColor;
exports.vehicleInteriorType = schema_https$1.vehicleInteriorType;
exports.vehicleModelDate = schema_https$1.vehicleModelDate;
exports.vehicleSeatingCapacity = schema_https$1.vehicleSeatingCapacity;
exports.vehicleSpecialUsage = schema_https$1.vehicleSpecialUsage;
exports.vehicleTransmission = schema_https$1.vehicleTransmission;
exports.vendor = schema_https$1.vendor;
exports.verificationFactCheckingPolicy = schema_https$1.verificationFactCheckingPolicy;
exports.version = schema_https$1.version;
exports.video = schema_https$1.video;
exports.videoFormat = schema_https$1.videoFormat;
exports.videoFrameSize = schema_https$1.videoFrameSize;
exports.videoQuality = schema_https$1.videoQuality;
exports.volumeNumber = schema_https$1.volumeNumber;
exports.warning = schema_https$1.warning;
exports.warranty = schema_https$1.warranty;
exports.warrantyPromise = schema_https$1.warrantyPromise;
exports.warrantyScope = schema_https$1.warrantyScope;
exports.webCheckinTime = schema_https$1.webCheckinTime;
exports.webFeed = schema_https$1.webFeed;
exports.weight = schema_https$1.weight;
exports.weightTotal = schema_https$1.weightTotal;
exports.wheelbase = schema_https$1.wheelbase;
exports.width = schema_https$1.width;
exports.winner = schema_https$1.winner;
exports.wordCount = schema_https$1.wordCount;
exports.workExample = schema_https$1.workExample;
exports.workFeatured = schema_https$1.workFeatured;
exports.workHours = schema_https$1.workHours;
exports.workLocation = schema_https$1.workLocation;
exports.workPerformed = schema_https$1.workPerformed;
exports.workPresented = schema_https$1.workPresented;
exports.workTranslation = schema_https$1.workTranslation;
exports.workload = schema_https$1.workload;
exports.worksFor = schema_https$1.worksFor;
exports.worstRating = schema_https$1.worstRating;
exports.xpath = schema_https$1.xpath;
exports.yearBuilt = schema_https$1.yearBuilt;
exports.yearlyRevenue = schema_https$1.yearlyRevenue;
exports.yearsInOperation = schema_https$1.yearsInOperation;
exports.yield__workaround = schema_https$1.yield__workaround;
