// rdf-namespaces/dist/sec-cc3c5111.js
'use strict';

/**
 * Digital proof
 *
 *
 *
 * https://w3id.org/security#Proof
 */
var Proof = 'https://w3id.org/security#Proof';
/**
 * An RDF Graph for a digital proof
 *
 *
 *
 * https://w3id.org/security#ProofGraph
 */
var ProofGraph = 'https://w3id.org/security#ProofGraph';
/**
 * Verification method
 *
 *
 *
 * https://w3id.org/security#VerificationMethod
 */
var VerificationMethod = 'https://w3id.org/security#VerificationMethod';
/**
 * A Data Integrity Proof
 *
 *
 *
 * https://w3id.org/security#DataIntegrityProof
 */
var DataIntegrityProof = 'https://w3id.org/security#DataIntegrityProof';
/**
 * Multikey Verification Method
 *
 *
 *
 * https://w3id.org/security#Multikey
 */
var Multikey = 'https://w3id.org/security#Multikey';
/**
 * JSON Web Key Verification Method
 *
 *
 *
 * https://w3id.org/security#JsonWebKey
 */
var JsonWebKey = 'https://w3id.org/security#JsonWebKey';
/**
 * ED2559 Verification Key, 2020 version
 *
 *
 *
 * https://w3id.org/security#Ed25519VerificationKey2020
 */
var Ed25519VerificationKey2020 = 'https://w3id.org/security#Ed25519VerificationKey2020';
/**
 * Ed25519 Signature Suite, 2020 version
 *
 *
 *
 * https://w3id.org/security#Ed25519Signature2020
 */
var Ed25519Signature2020 = 'https://w3id.org/security#Ed25519Signature2020';
/**
 * Processing error
 *
 *
 *
 * https://w3id.org/security#ProcessingError
 */
var ProcessingError = 'https://w3id.org/security#ProcessingError';
/**
 * Cryptographic key (deprecated)
 *
 *
 *
 * https://w3id.org/security#Key
 */
var Key = 'https://w3id.org/security#Key';
/**
 * ecdsa-sep256k1, 2019 version (deprecated)
 *
 *
 *
 * https://w3id.org/security#EcdsaSecp256k1Signature2019
 */
var EcdsaSecp256k1Signature2019 = 'https://w3id.org/security#EcdsaSecp256k1Signature2019';
/**
 * ecdsa-sep256k1, 2020 version (deprecated)
 *
 *
 *
 * https://w3id.org/security#EcdsaSecp256k1Signature2020
 */
var EcdsaSecp256k1Signature2020 = 'https://w3id.org/security#EcdsaSecp256k1Signature2020';
/**
 * ecdsa-secp256k1 verification key, 2019 version (deprecated)
 *
 *
 *
 * https://w3id.org/security#EcdsaSecp256k1VerificationKey2019
 */
var EcdsaSecp256k1VerificationKey2019 = 'https://w3id.org/security#EcdsaSecp256k1VerificationKey2019';
/**
 * ecdsa-secp256k1 recovery signature, 2020 version (deprecated)
 *
 *
 *
 * https://w3id.org/security#EcdsaSecp256k1RecoverySignature2020
 */
var EcdsaSecp256k1RecoverySignature2020 = 'https://w3id.org/security#EcdsaSecp256k1RecoverySignature2020';
/**
 * ecdsa-secp256k1 recovery method, 2020 version (deprecated)
 *
 *
 *
 * https://w3id.org/security#EcdsaSecp256k1RecoveryMethod2020
 */
var EcdsaSecp256k1RecoveryMethod2020 = 'https://w3id.org/security#EcdsaSecp256k1RecoveryMethod2020';
/**
 * Merkle Proof (deprecated)
 *
 *
 *
 * https://w3id.org/security#MerkleProof2019
 */
var MerkleProof2019 = 'https://w3id.org/security#MerkleProof2019';
/**
 * X25519 Key Agreement Key, 2019 version (deprecated)
 *
 *
 *
 * https://w3id.org/security#X25519KeyAgreementKey2019
 */
var X25519KeyAgreementKey2019 = 'https://w3id.org/security#X25519KeyAgreementKey2019';
/**
 * ED2559 Verification Key, 2018 version (deprecated)
 *
 *
 *
 * https://w3id.org/security#Ed25519VerificationKey2018
 */
var Ed25519VerificationKey2018 = 'https://w3id.org/security#Ed25519VerificationKey2018';
/**
 * JSON Web Key, 2020 version (deprecated)
 *
 *
 *
 * https://w3id.org/security#JsonWebKey2020
 */
var JsonWebKey2020 = 'https://w3id.org/security#JsonWebKey2020';
/**
 * JSON Web Signature, 2020 version (deprecated)
 *
 *
 *
 * https://w3id.org/security#JsonWebSignature2020
 */
var JsonWebSignature2020 = 'https://w3id.org/security#JsonWebSignature2020';
/**
 * BBS Signature, 2020 version (deprecated)
 *
 *
 *
 * https://w3id.org/security#BbsBlsSignature2020
 */
var BbsBlsSignature2020 = 'https://w3id.org/security#BbsBlsSignature2020';
/**
 * BBS Signature Proof, 2020 version (deprecated)
 *
 *
 *
 * https://w3id.org/security#BbsBlsSignatureProof2020
 */
var BbsBlsSignatureProof2020 = 'https://w3id.org/security#BbsBlsSignatureProof2020';
/**
 * BLS 12381 G1 Signature Key, 2020 version (deprecated)
 *
 *
 *
 * https://w3id.org/security#Bls12381G1Key2020
 */
var Bls12381G1Key2020 = 'https://w3id.org/security#Bls12381G1Key2020';
/**
 * BLS 12381 G2 Signature Key, 2020 version (deprecated)
 *
 *
 *
 * https://w3id.org/security#Bls12381G2Key2020
 */
var Bls12381G2Key2020 = 'https://w3id.org/security#Bls12381G2Key2020';
/**
 * Blockchain account ID (deprecated)
 *
 *
 *
 * https://w3id.org/security#blockchainAccountId
 */
var blockchainAccountId = 'https://w3id.org/security#blockchainAccountId';
/**
 * Ethereum address (deprecated)
 *
 *
 *
 * https://w3id.org/security#ethereumAddress
 */
var ethereumAddress = 'https://w3id.org/security#ethereumAddress';
/**
 * Base58-encoded Public Key (deprecated)
 *
 *
 *
 * https://w3id.org/security#publicKeyBase58
 */
var publicKeyBase58 = 'https://w3id.org/security#publicKeyBase58';
/**
 * Public key PEM (deprecated)
 *
 *
 *
 * https://w3id.org/security#publicKeyPem
 */
var publicKeyPem = 'https://w3id.org/security#publicKeyPem';
/**
 * Hex-encoded version of public Key (deprecated)
 *
 *
 *
 * https://w3id.org/security#publicKeyHex
 */
var publicKeyHex = 'https://w3id.org/security#publicKeyHex';
/**
 * Json Web Signature (deprecated)
 *
 *
 *
 * https://w3id.org/security#jws
 */
var jws = 'https://w3id.org/security#jws';

var secImport = /*#__PURE__*/Object.freeze({
	Proof: Proof,
	ProofGraph: ProofGraph,
	VerificationMethod: VerificationMethod,
	DataIntegrityProof: DataIntegrityProof,
	Multikey: Multikey,
	JsonWebKey: JsonWebKey,
	Ed25519VerificationKey2020: Ed25519VerificationKey2020,
	Ed25519Signature2020: Ed25519Signature2020,
	ProcessingError: ProcessingError,
	Key: Key,
	EcdsaSecp256k1Signature2019: EcdsaSecp256k1Signature2019,
	EcdsaSecp256k1Signature2020: EcdsaSecp256k1Signature2020,
	EcdsaSecp256k1VerificationKey2019: EcdsaSecp256k1VerificationKey2019,
	EcdsaSecp256k1RecoverySignature2020: EcdsaSecp256k1RecoverySignature2020,
	EcdsaSecp256k1RecoveryMethod2020: EcdsaSecp256k1RecoveryMethod2020,
	MerkleProof2019: MerkleProof2019,
	X25519KeyAgreementKey2019: X25519KeyAgreementKey2019,
	Ed25519VerificationKey2018: Ed25519VerificationKey2018,
	JsonWebKey2020: JsonWebKey2020,
	JsonWebSignature2020: JsonWebSignature2020,
	BbsBlsSignature2020: BbsBlsSignature2020,
	BbsBlsSignatureProof2020: BbsBlsSignatureProof2020,
	Bls12381G1Key2020: Bls12381G1Key2020,
	Bls12381G2Key2020: Bls12381G2Key2020,
	blockchainAccountId: blockchainAccountId,
	ethereumAddress: ethereumAddress,
	publicKeyBase58: publicKeyBase58,
	publicKeyPem: publicKeyPem,
	publicKeyHex: publicKeyHex,
	jws: jws
});

exports.BbsBlsSignature2020 = BbsBlsSignature2020;
exports.BbsBlsSignatureProof2020 = BbsBlsSignatureProof2020;
exports.Bls12381G1Key2020 = Bls12381G1Key2020;
exports.Bls12381G2Key2020 = Bls12381G2Key2020;
exports.DataIntegrityProof = DataIntegrityProof;
exports.EcdsaSecp256k1RecoveryMethod2020 = EcdsaSecp256k1RecoveryMethod2020;
exports.EcdsaSecp256k1RecoverySignature2020 = EcdsaSecp256k1RecoverySignature2020;
exports.EcdsaSecp256k1Signature2019 = EcdsaSecp256k1Signature2019;
exports.EcdsaSecp256k1Signature2020 = EcdsaSecp256k1Signature2020;
exports.EcdsaSecp256k1VerificationKey2019 = EcdsaSecp256k1VerificationKey2019;
exports.Ed25519Signature2020 = Ed25519Signature2020;
exports.Ed25519VerificationKey2018 = Ed25519VerificationKey2018;
exports.Ed25519VerificationKey2020 = Ed25519VerificationKey2020;
exports.JsonWebKey = JsonWebKey;
exports.JsonWebKey2020 = JsonWebKey2020;
exports.JsonWebSignature2020 = JsonWebSignature2020;
exports.Key = Key;
exports.MerkleProof2019 = MerkleProof2019;
exports.Multikey = Multikey;
exports.ProcessingError = ProcessingError;
exports.Proof = Proof;
exports.ProofGraph = ProofGraph;
exports.VerificationMethod = VerificationMethod;
exports.X25519KeyAgreementKey2019 = X25519KeyAgreementKey2019;
exports.blockchainAccountId = blockchainAccountId;
exports.ethereumAddress = ethereumAddress;
exports.jws = jws;
exports.publicKeyBase58 = publicKeyBase58;
exports.publicKeyHex = publicKeyHex;
exports.publicKeyPem = publicKeyPem;
exports.secImport = secImport;
