// rdf-namespaces/dist/sec.js
'use strict';

Object.defineProperty(exports, '__esModule', { value: true });

var sec$1 = require('./sec-cc3c5111.js');



exports.BbsBlsSignature2020 = sec$1.BbsBlsSignature2020;
exports.BbsBlsSignatureProof2020 = sec$1.BbsBlsSignatureProof2020;
exports.Bls12381G1Key2020 = sec$1.Bls12381G1Key2020;
exports.Bls12381G2Key2020 = sec$1.Bls12381G2Key2020;
exports.DataIntegrityProof = sec$1.DataIntegrityProof;
exports.EcdsaSecp256k1RecoveryMethod2020 = sec$1.EcdsaSecp256k1RecoveryMethod2020;
exports.EcdsaSecp256k1RecoverySignature2020 = sec$1.EcdsaSecp256k1RecoverySignature2020;
exports.EcdsaSecp256k1Signature2019 = sec$1.EcdsaSecp256k1Signature2019;
exports.EcdsaSecp256k1Signature2020 = sec$1.EcdsaSecp256k1Signature2020;
exports.EcdsaSecp256k1VerificationKey2019 = sec$1.EcdsaSecp256k1VerificationKey2019;
exports.Ed25519Signature2020 = sec$1.Ed25519Signature2020;
exports.Ed25519VerificationKey2018 = sec$1.Ed25519VerificationKey2018;
exports.Ed25519VerificationKey2020 = sec$1.Ed25519VerificationKey2020;
exports.JsonWebKey = sec$1.JsonWebKey;
exports.JsonWebKey2020 = sec$1.JsonWebKey2020;
exports.JsonWebSignature2020 = sec$1.JsonWebSignature2020;
exports.Key = sec$1.Key;
exports.MerkleProof2019 = sec$1.MerkleProof2019;
exports.Multikey = sec$1.Multikey;
exports.ProcessingError = sec$1.ProcessingError;
exports.Proof = sec$1.Proof;
exports.ProofGraph = sec$1.ProofGraph;
exports.VerificationMethod = sec$1.VerificationMethod;
exports.X25519KeyAgreementKey2019 = sec$1.X25519KeyAgreementKey2019;
exports.blockchainAccountId = sec$1.blockchainAccountId;
exports.ethereumAddress = sec$1.ethereumAddress;
exports.jws = sec$1.jws;
exports.publicKeyBase58 = sec$1.publicKeyBase58;
exports.publicKeyHex = sec$1.publicKeyHex;
exports.publicKeyPem = sec$1.publicKeyPem;
