// rdf-namespaces/dist/sec.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * Digital proof
 * 
 * 
 *
 * https://w3id.org/security#Proof
 */
export const Proof: Class = 'https://w3id.org/security#Proof';

/**
 * An RDF Graph for a digital proof
 * 
 * 
 *
 * https://w3id.org/security#ProofGraph
 */
export const ProofGraph: Class = 'https://w3id.org/security#ProofGraph';

/**
 * Verification method
 * 
 * 
 *
 * https://w3id.org/security#VerificationMethod
 */
export const VerificationMethod: Class = 'https://w3id.org/security#VerificationMethod';

/**
 * A Data Integrity Proof
 * 
 * 
 *
 * https://w3id.org/security#DataIntegrityProof
 */
export const DataIntegrityProof: Class = 'https://w3id.org/security#DataIntegrityProof';

/**
 * Multikey Verification Method
 * 
 * 
 *
 * https://w3id.org/security#Multikey
 */
export const Multikey: Class = 'https://w3id.org/security#Multikey';

/**
 * JSON Web Key Verification Method
 * 
 * 
 *
 * https://w3id.org/security#JsonWebKey
 */
export const JsonWebKey: Class = 'https://w3id.org/security#JsonWebKey';

/**
 * ED2559 Verification Key, 2020 version
 * 
 * 
 *
 * https://w3id.org/security#Ed25519VerificationKey2020
 */
export const Ed25519VerificationKey2020: Class = 'https://w3id.org/security#Ed25519VerificationKey2020';

/**
 * Ed25519 Signature Suite, 2020 version
 * 
 * 
 *
 * https://w3id.org/security#Ed25519Signature2020
 */
export const Ed25519Signature2020: Class = 'https://w3id.org/security#Ed25519Signature2020';

/**
 * Processing error
 * 
 * 
 *
 * https://w3id.org/security#ProcessingError
 */
export const ProcessingError: Class = 'https://w3id.org/security#ProcessingError';

/**
 * Cryptographic key (deprecated)
 * 
 * 
 *
 * https://w3id.org/security#Key
 */
export const Key: Class = 'https://w3id.org/security#Key';

/**
 * ecdsa-sep256k1, 2019 version (deprecated)
 * 
 * 
 *
 * https://w3id.org/security#EcdsaSecp256k1Signature2019
 */
export const EcdsaSecp256k1Signature2019: Class = 'https://w3id.org/security#EcdsaSecp256k1Signature2019';

/**
 * ecdsa-sep256k1, 2020 version (deprecated)
 * 
 * 
 *
 * https://w3id.org/security#EcdsaSecp256k1Signature2020
 */
export const EcdsaSecp256k1Signature2020: Class = 'https://w3id.org/security#EcdsaSecp256k1Signature2020';

/**
 * ecdsa-secp256k1 verification key, 2019 version (deprecated)
 * 
 * 
 *
 * https://w3id.org/security#EcdsaSecp256k1VerificationKey2019
 */
export const EcdsaSecp256k1VerificationKey2019: Class = 'https://w3id.org/security#EcdsaSecp256k1VerificationKey2019';

/**
 * ecdsa-secp256k1 recovery signature, 2020 version (deprecated)
 * 
 * 
 *
 * https://w3id.org/security#EcdsaSecp256k1RecoverySignature2020
 */
export const EcdsaSecp256k1RecoverySignature2020: Class = 'https://w3id.org/security#EcdsaSecp256k1RecoverySignature2020';

/**
 * ecdsa-secp256k1 recovery method, 2020 version (deprecated)
 * 
 * 
 *
 * https://w3id.org/security#EcdsaSecp256k1RecoveryMethod2020
 */
export const EcdsaSecp256k1RecoveryMethod2020: Class = 'https://w3id.org/security#EcdsaSecp256k1RecoveryMethod2020';

/**
 * Merkle Proof (deprecated)
 * 
 * 
 *
 * https://w3id.org/security#MerkleProof2019
 */
export const MerkleProof2019: Class = 'https://w3id.org/security#MerkleProof2019';

/**
 * X25519 Key Agreement Key, 2019 version (deprecated)
 * 
 * 
 *
 * https://w3id.org/security#X25519KeyAgreementKey2019
 */
export const X25519KeyAgreementKey2019: Class = 'https://w3id.org/security#X25519KeyAgreementKey2019';

/**
 * ED2559 Verification Key, 2018 version (deprecated)
 * 
 * 
 *
 * https://w3id.org/security#Ed25519VerificationKey2018
 */
export const Ed25519VerificationKey2018: Class = 'https://w3id.org/security#Ed25519VerificationKey2018';

/**
 * JSON Web Key, 2020 version (deprecated)
 * 
 * 
 *
 * https://w3id.org/security#JsonWebKey2020
 */
export const JsonWebKey2020: Class = 'https://w3id.org/security#JsonWebKey2020';

/**
 * JSON Web Signature, 2020 version (deprecated)
 * 
 * 
 *
 * https://w3id.org/security#JsonWebSignature2020
 */
export const JsonWebSignature2020: Class = 'https://w3id.org/security#JsonWebSignature2020';

/**
 * BBS Signature, 2020 version (deprecated)
 * 
 * 
 *
 * https://w3id.org/security#BbsBlsSignature2020
 */
export const BbsBlsSignature2020: Class = 'https://w3id.org/security#BbsBlsSignature2020';

/**
 * BBS Signature Proof, 2020 version (deprecated)
 * 
 * 
 *
 * https://w3id.org/security#BbsBlsSignatureProof2020
 */
export const BbsBlsSignatureProof2020: Class = 'https://w3id.org/security#BbsBlsSignatureProof2020';

/**
 * BLS 12381 G1 Signature Key, 2020 version (deprecated)
 * 
 * 
 *
 * https://w3id.org/security#Bls12381G1Key2020
 */
export const Bls12381G1Key2020: Class = 'https://w3id.org/security#Bls12381G1Key2020';

/**
 * BLS 12381 G2 Signature Key, 2020 version (deprecated)
 * 
 * 
 *
 * https://w3id.org/security#Bls12381G2Key2020
 */
export const Bls12381G2Key2020: Class = 'https://w3id.org/security#Bls12381G2Key2020';

/**
 * Blockchain account ID (deprecated)
 * 
 * 
 *
 * https://w3id.org/security#blockchainAccountId
 */
export const blockchainAccountId: Property = 'https://w3id.org/security#blockchainAccountId';

/**
 * Ethereum address (deprecated)
 * 
 * 
 *
 * https://w3id.org/security#ethereumAddress
 */
export const ethereumAddress: Property = 'https://w3id.org/security#ethereumAddress';

/**
 * Base58-encoded Public Key (deprecated)
 * 
 * 
 *
 * https://w3id.org/security#publicKeyBase58
 */
export const publicKeyBase58: Property = 'https://w3id.org/security#publicKeyBase58';

/**
 * Public key PEM (deprecated)
 * 
 * 
 *
 * https://w3id.org/security#publicKeyPem
 */
export const publicKeyPem: Property = 'https://w3id.org/security#publicKeyPem';

/**
 * Hex-encoded version of public Key (deprecated)
 * 
 * 
 *
 * https://w3id.org/security#publicKeyHex
 */
export const publicKeyHex: Property = 'https://w3id.org/security#publicKeyHex';

/**
 * Json Web Signature (deprecated)
 * 
 * 
 *
 * https://w3id.org/security#jws
 */
export const jws: Property = 'https://w3id.org/security#jws';
