// rdf-namespaces/dist/sioc.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#Community
 */
export const Community: OwlClass = 'http://rdfs.org/sioc/ns#Community';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#Container
 */
export const Container: OwlClass = 'http://rdfs.org/sioc/ns#Container';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#Forum
 */
export const Forum: OwlClass = 'http://rdfs.org/sioc/ns#Forum';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#Item
 */
export const Item: OwlClass = 'http://rdfs.org/sioc/ns#Item';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#Post
 */
export const Post: OwlClass = 'http://rdfs.org/sioc/ns#Post';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#Role
 */
export const Role: OwlClass = 'http://rdfs.org/sioc/ns#Role';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#Space
 */
export const Space: OwlClass = 'http://rdfs.org/sioc/ns#Space';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#Site
 */
export const Site: OwlClass = 'http://rdfs.org/sioc/ns#Site';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#Thread
 */
export const Thread: OwlClass = 'http://rdfs.org/sioc/ns#Thread';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#UserAccount
 */
export const UserAccount: OwlClass = 'http://rdfs.org/sioc/ns#UserAccount';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#Usergroup
 */
export const Usergroup: OwlClass = 'http://rdfs.org/sioc/ns#Usergroup';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#about
 */
export const about: OwlObjectProperty = 'http://rdfs.org/sioc/ns#about';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#account_of
 */
export const account_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#account_of';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#addressed_to
 */
export const addressed_to: OwlObjectProperty = 'http://rdfs.org/sioc/ns#addressed_to';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#administrator_of
 */
export const administrator_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#administrator_of';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#attachment
 */
export const attachment: OwlObjectProperty = 'http://rdfs.org/sioc/ns#attachment';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#avatar
 */
export const avatar: OwlObjectProperty = 'http://rdfs.org/sioc/ns#avatar';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#container_of
 */
export const container_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#container_of';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#content
 */
export const content: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#content';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#creator_of
 */
export const creator_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#creator_of';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#delivered_at
 */
export const delivered_at: OwlObjectProperty = 'http://rdfs.org/sioc/ns#delivered_at';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#discussion_of
 */
export const discussion_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#discussion_of';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#email
 */
export const email: OwlObjectProperty = 'http://rdfs.org/sioc/ns#email';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#email_sha1
 */
export const email_sha1: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#email_sha1';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#embeds_knowledge
 */
export const embeds_knowledge: OwlObjectProperty = 'http://rdfs.org/sioc/ns#embeds_knowledge';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#feed
 */
export const feed: OwlObjectProperty = 'http://rdfs.org/sioc/ns#feed';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#follows
 */
export const follows: OwlObjectProperty = 'http://rdfs.org/sioc/ns#follows';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#function_of
 */
export const function_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#function_of';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#generator
 */
export const generator: OwlObjectProperty = 'http://rdfs.org/sioc/ns#generator';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#has_administrator
 */
export const has_administrator: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_administrator';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#has_container
 */
export const has_container: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_container';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#has_creator
 */
export const has_creator: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_creator';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#has_discussion
 */
export const has_discussion: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_discussion';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#has_function
 */
export const has_function: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_function';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#has_host
 */
export const has_host: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_host';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#has_member
 */
export const has_member: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_member';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#has_moderator
 */
export const has_moderator: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_moderator';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#has_modifier
 */
export const has_modifier: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_modifier';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#has_owner
 */
export const has_owner: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_owner';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#has_parent
 */
export const has_parent: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_parent';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#has_reply
 */
export const has_reply: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_reply';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#has_scope
 */
export const has_scope: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_scope';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#has_space
 */
export const has_space: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_space';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#has_subscriber
 */
export const has_subscriber: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_subscriber';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#has_usergroup
 */
export const has_usergroup: OwlObjectProperty = 'http://rdfs.org/sioc/ns#has_usergroup';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#host_of
 */
export const host_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#host_of';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#id
 */
export const id: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#id';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#ip_address
 */
export const ip_address: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#ip_address';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#last_activity_date
 */
export const last_activity_date: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#last_activity_date';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#last_item_date
 */
export const last_item_date: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#last_item_date';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#last_reply_date
 */
export const last_reply_date: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#last_reply_date';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#latest_version
 */
export const latest_version: OwlObjectProperty = 'http://rdfs.org/sioc/ns#latest_version';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#likes
 */
export const likes: OwlObjectProperty = 'http://rdfs.org/sioc/ns#likes';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#link
 */
export const link: OwlObjectProperty = 'http://rdfs.org/sioc/ns#link';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#links_to
 */
export const links_to: OwlObjectProperty = 'http://rdfs.org/sioc/ns#links_to';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#member_of
 */
export const member_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#member_of';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#mentions
 */
export const mentions: OwlObjectProperty = 'http://rdfs.org/sioc/ns#mentions';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#moderator_of
 */
export const moderator_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#moderator_of';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#modifier_of
 */
export const modifier_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#modifier_of';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#name
 */
export const name: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#name';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#next_by_date
 */
export const next_by_date: OwlObjectProperty = 'http://rdfs.org/sioc/ns#next_by_date';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#next_version
 */
export const next_version: OwlObjectProperty = 'http://rdfs.org/sioc/ns#next_version';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#note
 */
export const note: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#note';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#num_authors
 */
export const num_authors: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#num_authors';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#num_items
 */
export const num_items: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#num_items';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#num_replies
 */
export const num_replies: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#num_replies';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#num_threads
 */
export const num_threads: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#num_threads';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#num_views
 */
export const num_views: OwlDatatypeProperty = 'http://rdfs.org/sioc/ns#num_views';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#owner_of
 */
export const owner_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#owner_of';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#parent_of
 */
export const parent_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#parent_of';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#previous_by_date
 */
export const previous_by_date: OwlObjectProperty = 'http://rdfs.org/sioc/ns#previous_by_date';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#previous_version
 */
export const previous_version: OwlObjectProperty = 'http://rdfs.org/sioc/ns#previous_version';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#read_at
 */
export const read_at: OwlObjectProperty = 'http://rdfs.org/sioc/ns#read_at';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#related_to
 */
export const related_to: OwlObjectProperty = 'http://rdfs.org/sioc/ns#related_to';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#reply_of
 */
export const reply_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#reply_of';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#respond_to
 */
export const respond_to: OwlObjectProperty = 'http://rdfs.org/sioc/ns#respond_to';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#scope_of
 */
export const scope_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#scope_of';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#shared_by
 */
export const shared_by: OwlObjectProperty = 'http://rdfs.org/sioc/ns#shared_by';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#space_of
 */
export const space_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#space_of';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#subscriber_of
 */
export const subscriber_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#subscriber_of';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#topic
 */
export const topic: OwlObjectProperty = 'http://rdfs.org/sioc/ns#topic';

/**
 * 
 * 
 * 
 *
 * http://rdfs.org/sioc/ns#usergroup_of
 */
export const usergroup_of: OwlObjectProperty = 'http://rdfs.org/sioc/ns#usergroup_of';
