// rdf-namespaces/dist/skos.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#Concept
 */
export const Concept: OwlClass = 'http://www.w3.org/2004/02/skos/core#Concept';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#ConceptScheme
 */
export const ConceptScheme: OwlClass = 'http://www.w3.org/2004/02/skos/core#ConceptScheme';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#Collection
 */
export const Collection: OwlClass = 'http://www.w3.org/2004/02/skos/core#Collection';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#OrderedCollection
 */
export const OrderedCollection: OwlClass = 'http://www.w3.org/2004/02/skos/core#OrderedCollection';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#inScheme
 */
export const inScheme: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#inScheme';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#hasTopConcept
 */
export const hasTopConcept: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#hasTopConcept';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#topConceptOf
 */
export const topConceptOf: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#topConceptOf';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#notation
 */
export const notation: OwlDatatypeProperty = 'http://www.w3.org/2004/02/skos/core#notation';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#semanticRelation
 */
export const semanticRelation: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#semanticRelation';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#broader
 */
export const broader: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#broader';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#narrower
 */
export const narrower: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#narrower';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#related
 */
export const related: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#related';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#broaderTransitive
 */
export const broaderTransitive: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#broaderTransitive';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#narrowerTransitive
 */
export const narrowerTransitive: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#narrowerTransitive';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#member
 */
export const member: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#member';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#memberList
 */
export const memberList: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#memberList';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#mappingRelation
 */
export const mappingRelation: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#mappingRelation';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#broadMatch
 */
export const broadMatch: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#broadMatch';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#narrowMatch
 */
export const narrowMatch: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#narrowMatch';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#relatedMatch
 */
export const relatedMatch: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#relatedMatch';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#exactMatch
 */
export const exactMatch: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#exactMatch';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2004/02/skos/core#closeMatch
 */
export const closeMatch: OwlObjectProperty = 'http://www.w3.org/2004/02/skos/core#closeMatch';
