// rdf-namespaces/dist/skos-328791ae.js
'use strict';

/**
 *
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#Concept
 */
var Concept = 'http://www.w3.org/2004/02/skos/core#Concept';
/**
 *
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#ConceptScheme
 */
var ConceptScheme = 'http://www.w3.org/2004/02/skos/core#ConceptScheme';
/**
 *
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#Collection
 */
var Collection = 'http://www.w3.org/2004/02/skos/core#Collection';
/**
 *
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#OrderedCollection
 */
var OrderedCollection = 'http://www.w3.org/2004/02/skos/core#OrderedCollection';
/**
 *
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#inScheme
 */
var inScheme = 'http://www.w3.org/2004/02/skos/core#inScheme';
/**
 *
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#hasTopConcept
 */
var hasTopConcept = 'http://www.w3.org/2004/02/skos/core#hasTopConcept';
/**
 *
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#topConceptOf
 */
var topConceptOf = 'http://www.w3.org/2004/02/skos/core#topConceptOf';
/**
 *
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#notation
 */
var notation = 'http://www.w3.org/2004/02/skos/core#notation';
/**
 *
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#semanticRelation
 */
var semanticRelation = 'http://www.w3.org/2004/02/skos/core#semanticRelation';
/**
 *
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#broader
 */
var broader = 'http://www.w3.org/2004/02/skos/core#broader';
/**
 *
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#narrower
 */
var narrower = 'http://www.w3.org/2004/02/skos/core#narrower';
/**
 *
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#related
 */
var related = 'http://www.w3.org/2004/02/skos/core#related';
/**
 *
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#broaderTransitive
 */
var broaderTransitive = 'http://www.w3.org/2004/02/skos/core#broaderTransitive';
/**
 *
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#narrowerTransitive
 */
var narrowerTransitive = 'http://www.w3.org/2004/02/skos/core#narrowerTransitive';
/**
 *
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#member
 */
var member = 'http://www.w3.org/2004/02/skos/core#member';
/**
 *
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#memberList
 */
var memberList = 'http://www.w3.org/2004/02/skos/core#memberList';
/**
 *
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#mappingRelation
 */
var mappingRelation = 'http://www.w3.org/2004/02/skos/core#mappingRelation';
/**
 *
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#broadMatch
 */
var broadMatch = 'http://www.w3.org/2004/02/skos/core#broadMatch';
/**
 *
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#narrowMatch
 */
var narrowMatch = 'http://www.w3.org/2004/02/skos/core#narrowMatch';
/**
 *
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#relatedMatch
 */
var relatedMatch = 'http://www.w3.org/2004/02/skos/core#relatedMatch';
/**
 *
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#exactMatch
 */
var exactMatch = 'http://www.w3.org/2004/02/skos/core#exactMatch';
/**
 *
 *
 *
 *
 * http://www.w3.org/2004/02/skos/core#closeMatch
 */
var closeMatch = 'http://www.w3.org/2004/02/skos/core#closeMatch';

var skosImport = /*#__PURE__*/Object.freeze({
	Concept: Concept,
	ConceptScheme: ConceptScheme,
	Collection: Collection,
	OrderedCollection: OrderedCollection,
	inScheme: inScheme,
	hasTopConcept: hasTopConcept,
	topConceptOf: topConceptOf,
	notation: notation,
	semanticRelation: semanticRelation,
	broader: broader,
	narrower: narrower,
	related: related,
	broaderTransitive: broaderTransitive,
	narrowerTransitive: narrowerTransitive,
	member: member,
	memberList: memberList,
	mappingRelation: mappingRelation,
	broadMatch: broadMatch,
	narrowMatch: narrowMatch,
	relatedMatch: relatedMatch,
	exactMatch: exactMatch,
	closeMatch: closeMatch
});

exports.Collection = Collection;
exports.Concept = Concept;
exports.ConceptScheme = ConceptScheme;
exports.OrderedCollection = OrderedCollection;
exports.broadMatch = broadMatch;
exports.broader = broader;
exports.broaderTransitive = broaderTransitive;
exports.closeMatch = closeMatch;
exports.exactMatch = exactMatch;
exports.hasTopConcept = hasTopConcept;
exports.inScheme = inScheme;
exports.mappingRelation = mappingRelation;
exports.member = member;
exports.memberList = memberList;
exports.narrowMatch = narrowMatch;
exports.narrower = narrower;
exports.narrowerTransitive = narrowerTransitive;
exports.notation = notation;
exports.related = related;
exports.relatedMatch = relatedMatch;
exports.semanticRelation = semanticRelation;
exports.skosImport = skosImport;
exports.topConceptOf = topConceptOf;
