// rdf-namespaces/dist/solid.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#Account
 */
export const Account: Class = 'http://www.w3.org/ns/solid/terms#Account';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#Inbox
 */
export const Inbox: Class = 'http://www.w3.org/ns/solid/terms#Inbox';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#InsertDeletePatch
 */
export const InsertDeletePatch: Class = 'http://www.w3.org/ns/solid/terms#InsertDeletePatch';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#ListedDocument
 */
export const ListedDocument: Class = 'http://www.w3.org/ns/solid/terms#ListedDocument';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#Notification
 */
export const Notification: Class = 'http://www.w3.org/ns/solid/terms#Notification';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#Patch
 */
export const Patch: Class = 'http://www.w3.org/ns/solid/terms#Patch';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#Timeline
 */
export const Timeline: Class = 'http://www.w3.org/ns/solid/terms#Timeline';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#TypeIndex
 */
export const TypeIndex: Class = 'http://www.w3.org/ns/solid/terms#TypeIndex';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#TypeRegistration
 */
export const TypeRegistration: Class = 'http://www.w3.org/ns/solid/terms#TypeRegistration';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#UnlistedDocument
 */
export const UnlistedDocument: Class = 'http://www.w3.org/ns/solid/terms#UnlistedDocument';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#account
 */
export const account: Property = 'http://www.w3.org/ns/solid/terms#account';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#deletes
 */
export const deletes: Property = 'http://www.w3.org/ns/solid/terms#deletes';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#forClass
 */
export const forClass: Property = 'http://www.w3.org/ns/solid/terms#forClass';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#inbox
 */
export const inbox: Property = 'http://www.w3.org/ns/solid/terms#inbox';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#inserts
 */
export const inserts: Property = 'http://www.w3.org/ns/solid/terms#inserts';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#instance
 */
export const instance: Property = 'http://www.w3.org/ns/solid/terms#instance';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#instanceContainer
 */
export const instanceContainer: Property = 'http://www.w3.org/ns/solid/terms#instanceContainer';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#loginEndpoint
 */
export const loginEndpoint: Property = 'http://www.w3.org/ns/solid/terms#loginEndpoint';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#logoutEndpoint
 */
export const logoutEndpoint: Property = 'http://www.w3.org/ns/solid/terms#logoutEndpoint';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#notification
 */
export const notification: Property = 'http://www.w3.org/ns/solid/terms#notification';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#oidcIssuer
 */
export const oidcIssuer: Property = 'http://www.w3.org/ns/solid/terms#oidcIssuer';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#owner
 */
export const owner: Property = 'http://www.w3.org/ns/solid/terms#owner';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#patches
 */
export const patches: Property = 'http://www.w3.org/ns/solid/terms#patches';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#privateTypeIndex
 */
export const privateTypeIndex: Property = 'http://www.w3.org/ns/solid/terms#privateTypeIndex';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#publicTypeIndex
 */
export const publicTypeIndex: Property = 'http://www.w3.org/ns/solid/terms#publicTypeIndex';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#read
 */
export const read: Property = 'http://www.w3.org/ns/solid/terms#read';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#storageQuota
 */
export const storageQuota: Property = 'http://www.w3.org/ns/solid/terms#storageQuota';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#storageUsage
 */
export const storageUsage: Property = 'http://www.w3.org/ns/solid/terms#storageUsage';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#typeIndex
 */
export const typeIndex: Property = 'http://www.w3.org/ns/solid/terms#typeIndex';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#timeline
 */
export const timeline: Property = 'http://www.w3.org/ns/solid/terms#timeline';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/ns/solid/terms#where
 */
export const where: Property = 'http://www.w3.org/ns/solid/terms#where';
