// rdf-namespaces/dist/vcard.ts
type Property = string;
type Class = string;
type Datatype = string;
type OwlClass = string;
type OwlObjectProperty = string;
type OwlDatatypeProperty = string;
type HydraResource = string;
type HydraClass = string;
type HydraLink = string;
type HydraTemplatedLink = string;
type HydraVariableRepresentation = string;

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Acquaintance
 */
export const Acquaintance: OwlClass = 'http://www.w3.org/2006/vcard/ns#Acquaintance';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Agent
 */
export const Agent: OwlClass = 'http://www.w3.org/2006/vcard/ns#Agent';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#BBS
 */
export const BBS: OwlClass = 'http://www.w3.org/2006/vcard/ns#BBS';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Car
 */
export const Car: OwlClass = 'http://www.w3.org/2006/vcard/ns#Car';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Cell
 */
export const Cell: OwlClass = 'http://www.w3.org/2006/vcard/ns#Cell';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Child
 */
export const Child: OwlClass = 'http://www.w3.org/2006/vcard/ns#Child';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Colleague
 */
export const Colleague: OwlClass = 'http://www.w3.org/2006/vcard/ns#Colleague';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Contact
 */
export const Contact: OwlClass = 'http://www.w3.org/2006/vcard/ns#Contact';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Coresident
 */
export const Coresident: OwlClass = 'http://www.w3.org/2006/vcard/ns#Coresident';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Coworker
 */
export const Coworker: OwlClass = 'http://www.w3.org/2006/vcard/ns#Coworker';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Crush
 */
export const Crush: OwlClass = 'http://www.w3.org/2006/vcard/ns#Crush';

/**
 * 
 * 
 * null
 * Note that `Date` is a reserved Javascript keyword, and is therefore suffixed by `__workaround`.
 * For a list of reserved keywords, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Keywords.
 *
 * http://www.w3.org/2006/vcard/ns#Date
 */
export const Date__workaround: OwlClass = 'http://www.w3.org/2006/vcard/ns#Date';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Dom
 */
export const Dom: OwlClass = 'http://www.w3.org/2006/vcard/ns#Dom';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Emergency
 */
export const Emergency: OwlClass = 'http://www.w3.org/2006/vcard/ns#Emergency';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Fax
 */
export const Fax: OwlClass = 'http://www.w3.org/2006/vcard/ns#Fax';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Female
 */
export const Female: OwlClass = 'http://www.w3.org/2006/vcard/ns#Female';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Friend
 */
export const Friend: OwlClass = 'http://www.w3.org/2006/vcard/ns#Friend';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Home
 */
export const Home: OwlClass = 'http://www.w3.org/2006/vcard/ns#Home';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#ISDN
 */
export const ISDN: OwlClass = 'http://www.w3.org/2006/vcard/ns#ISDN';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Internet
 */
export const Internet: OwlClass = 'http://www.w3.org/2006/vcard/ns#Internet';

/**
 * 
 * 
 * null
 * Note that `Intl` is a reserved Javascript keyword, and is therefore suffixed by `__workaround`.
 * For a list of reserved keywords, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Keywords.
 *
 * http://www.w3.org/2006/vcard/ns#Intl
 */
export const Intl__workaround: OwlClass = 'http://www.w3.org/2006/vcard/ns#Intl';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Kin
 */
export const Kin: OwlClass = 'http://www.w3.org/2006/vcard/ns#Kin';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Label
 */
export const Label: OwlClass = 'http://www.w3.org/2006/vcard/ns#Label';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Male
 */
export const Male: OwlClass = 'http://www.w3.org/2006/vcard/ns#Male';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Me
 */
export const Me: OwlClass = 'http://www.w3.org/2006/vcard/ns#Me';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Met
 */
export const Met: OwlClass = 'http://www.w3.org/2006/vcard/ns#Met';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Modem
 */
export const Modem: OwlClass = 'http://www.w3.org/2006/vcard/ns#Modem';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Msg
 */
export const Msg: OwlClass = 'http://www.w3.org/2006/vcard/ns#Msg';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Muse
 */
export const Muse: OwlClass = 'http://www.w3.org/2006/vcard/ns#Muse';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Neighbor
 */
export const Neighbor: OwlClass = 'http://www.w3.org/2006/vcard/ns#Neighbor';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#None
 */
export const None: OwlClass = 'http://www.w3.org/2006/vcard/ns#None';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Other
 */
export const Other: OwlClass = 'http://www.w3.org/2006/vcard/ns#Other';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#PCS
 */
export const PCS: OwlClass = 'http://www.w3.org/2006/vcard/ns#PCS';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Pager
 */
export const Pager: OwlClass = 'http://www.w3.org/2006/vcard/ns#Pager';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Parcel
 */
export const Parcel: OwlClass = 'http://www.w3.org/2006/vcard/ns#Parcel';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Parent
 */
export const Parent: OwlClass = 'http://www.w3.org/2006/vcard/ns#Parent';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Postal
 */
export const Postal: OwlClass = 'http://www.w3.org/2006/vcard/ns#Postal';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Pref
 */
export const Pref: OwlClass = 'http://www.w3.org/2006/vcard/ns#Pref';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Sibling
 */
export const Sibling: OwlClass = 'http://www.w3.org/2006/vcard/ns#Sibling';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Spouse
 */
export const Spouse: OwlClass = 'http://www.w3.org/2006/vcard/ns#Spouse';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Sweetheart
 */
export const Sweetheart: OwlClass = 'http://www.w3.org/2006/vcard/ns#Sweetheart';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Tel
 */
export const Tel: OwlClass = 'http://www.w3.org/2006/vcard/ns#Tel';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Text
 */
export const Text: OwlClass = 'http://www.w3.org/2006/vcard/ns#Text';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#TextPhone
 */
export const TextPhone: OwlClass = 'http://www.w3.org/2006/vcard/ns#TextPhone';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Unknown
 */
export const Unknown: OwlClass = 'http://www.w3.org/2006/vcard/ns#Unknown';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Video
 */
export const Video: OwlClass = 'http://www.w3.org/2006/vcard/ns#Video';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Voice
 */
export const Voice: OwlClass = 'http://www.w3.org/2006/vcard/ns#Voice';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Work
 */
export const Work: OwlClass = 'http://www.w3.org/2006/vcard/ns#Work';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#X400
 */
export const X400: OwlClass = 'http://www.w3.org/2006/vcard/ns#X400';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#adr
 */
export const adr: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#adr';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#agent
 */
export const agent: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#agent';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#anniversary
 */
export const anniversary: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#anniversary';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#bday
 */
export const bday: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#bday';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#category
 */
export const category: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#category';

/**
 * 
 * 
 * null
 * Note that `class` is a reserved Javascript keyword, and is therefore suffixed by `__workaround`.
 * For a list of reserved keywords, see https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#Keywords.
 *
 * http://www.w3.org/2006/vcard/ns#class
 */
export const class__workaround: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#class';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#email
 */
export const email: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#email';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#geo
 */
export const geo: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#geo';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasAdditionalName
 */
export const hasAdditionalName: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasAdditionalName';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasCalendarBusy
 */
export const hasCalendarBusy: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasCalendarBusy';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasCalendarLink
 */
export const hasCalendarLink: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasCalendarLink';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasCalendarRequest
 */
export const hasCalendarRequest: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasCalendarRequest';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasCategory
 */
export const hasCategory: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasCategory';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasCountryName
 */
export const hasCountryName: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasCountryName';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasFN
 */
export const hasFN: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasFN';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasFamilyName
 */
export const hasFamilyName: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasFamilyName';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasGender
 */
export const hasGender: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasGender';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasGivenName
 */
export const hasGivenName: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasGivenName';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasHonorificPrefix
 */
export const hasHonorificPrefix: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasHonorificPrefix';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasHonorificSuffix
 */
export const hasHonorificSuffix: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasHonorificSuffix';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasInstantMessage
 */
export const hasInstantMessage: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasInstantMessage';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasLanguage
 */
export const hasLanguage: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasLanguage';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasLocality
 */
export const hasLocality: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasLocality';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasNickname
 */
export const hasNickname: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasNickname';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasNote
 */
export const hasNote: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasNote';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasOrganizationName
 */
export const hasOrganizationName: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasOrganizationName';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasOrganizationUnit
 */
export const hasOrganizationUnit: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasOrganizationUnit';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasPostalCode
 */
export const hasPostalCode: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasPostalCode';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasRegion
 */
export const hasRegion: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasRegion';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasRelated
 */
export const hasRelated: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasRelated';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasRole
 */
export const hasRole: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasRole';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasSource
 */
export const hasSource: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasSource';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasStreetAddress
 */
export const hasStreetAddress: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasStreetAddress';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasTitle
 */
export const hasTitle: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasTitle';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasUID
 */
export const hasUID: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasUID';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasValue
 */
export const hasValue: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasValue';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#label
 */
export const label: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#label';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#language
 */
export const language: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#language';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#latitude
 */
export const latitude: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#latitude';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#longitude
 */
export const longitude: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#longitude';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#mailer
 */
export const mailer: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#mailer';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#note
 */
export const note: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#note';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#org
 */
export const org: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#org';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#prodid
 */
export const prodid: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#prodid';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#rev
 */
export const rev: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#rev';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#role
 */
export const role: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#role';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#title
 */
export const title: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#title';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#tz
 */
export const tz: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#tz';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#value
 */
export const value: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#value';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Address
 */
export const Address: OwlClass = 'http://www.w3.org/2006/vcard/ns#Address';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Email
 */
export const Email: OwlClass = 'http://www.w3.org/2006/vcard/ns#Email';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Group
 */
export const Group: OwlClass = 'http://www.w3.org/2006/vcard/ns#Group';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Individual
 */
export const Individual: OwlClass = 'http://www.w3.org/2006/vcard/ns#Individual';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Name
 */
export const Name: OwlClass = 'http://www.w3.org/2006/vcard/ns#Name';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#VCard
 */
export const VCard: OwlClass = 'http://www.w3.org/2006/vcard/ns#VCard';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#fn
 */
export const fn: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#fn';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasAddress
 */
export const hasAddress: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasAddress';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasEmail
 */
export const hasEmail: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasEmail';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasGeo
 */
export const hasGeo: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasGeo';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasKey
 */
export const hasKey: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasKey';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasLogo
 */
export const hasLogo: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasLogo';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasName
 */
export const hasName: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasName';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasPhoto
 */
export const hasPhoto: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasPhoto';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasSound
 */
export const hasSound: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasSound';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasTelephone
 */
export const hasTelephone: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasTelephone';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasURL
 */
export const hasURL: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasURL';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#key
 */
export const key: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#key';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#logo
 */
export const logo: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#logo';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#n
 */
export const n: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#n';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#nickname
 */
export const nickname: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#nickname';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#photo
 */
export const photo: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#photo';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#sound
 */
export const sound: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#sound';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#tel
 */
export const tel: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#tel';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#url
 */
export const url: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#url';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Location
 */
export const Location: OwlClass = 'http://www.w3.org/2006/vcard/ns#Location';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#hasMember
 */
export const hasMember: OwlObjectProperty = 'http://www.w3.org/2006/vcard/ns#hasMember';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#locality
 */
export const locality: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#locality';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#region
 */
export const region: OwlDatatypeProperty = 'http://www.w3.org/2006/vcard/ns#region';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Organization
 */
export const Organization: OwlClass = 'http://www.w3.org/2006/vcard/ns#Organization';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Gender
 */
export const Gender: OwlClass = 'http://www.w3.org/2006/vcard/ns#Gender';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Kind
 */
export const Kind: OwlClass = 'http://www.w3.org/2006/vcard/ns#Kind';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#Type
 */
export const Type: OwlClass = 'http://www.w3.org/2006/vcard/ns#Type';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#TelephoneType
 */
export const TelephoneType: OwlClass = 'http://www.w3.org/2006/vcard/ns#TelephoneType';

/**
 * 
 * 
 * 
 *
 * http://www.w3.org/2006/vcard/ns#RelatedType
 */
export const RelatedType: OwlClass = 'http://www.w3.org/2006/vcard/ns#RelatedType';
