// @inrupt/solid-client-access-grants/dist/gConsent/constants.mjs
import { gc } from '../common/constants.mjs';

//
// Copyright Inrupt Inc.
//
// Permission is hereby granted, free of charge, to any person obtaining a copy
// of this software and associated documentation files (the "Software"), to deal in
// the Software without restriction, including without limitation the rights to use,
// copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
// Software, and to permit persons to whom the Software is furnished to do so,
// subject to the following conditions:
//
// The above copyright notice and this permission notice shall be included in
// all copies or substantial portions of the Software.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
//
const GC_CONSENT_STATUS_DENIED_ABBREV = "ConsentStatusDenied";
const GC_CONSENT_STATUS_EXPLICITLY_GIVEN_ABBREV = "ConsentStatusExplicitlyGiven";
const GC_CONSENT_STATUS_REQUESTED_ABBREV = "ConsentStatusRequested";
// TODO: Add dependency on generated vocabulary.
const PIM_STORAGE = "http://www.w3.org/ns/pim/space#storage";
// TODO: This is a temporary filler term, until we publish a definitive term for this.
const PREFERRED_CONSENT_MANAGEMENT_UI = "http://inrupt.com/ns/ess#ConsentManagementUI";
const CONTEXT_VC_W3C = "https://www.w3.org/2018/credentials/v1";
// This static context is used from the 2.1 version, instead of having a context
// specific to the deployment.
const CONTEXT_ESS_DEFAULT = "https://schema.inrupt.com/credentials/v1.jsonld";
// According to the [ESS documentation](https://docs.inrupt.com/ess/latest/services/service-vc/#ess-vc-service-endpoints),
// the JSON-LD context for ESS-issued VCs will match the following template.
const instanciateContextVcEssTemplate = (essVcDomain) => `https://${essVcDomain}/credentials/v1`;
const extraContext = [
    "https://w3id.org/security/data-integrity/v1",
    "https://w3id.org/vc-revocation-list-2020/v1",
    "https://w3id.org/vc/status-list/2021/v1",
    "https://w3id.org/security/suites/ed25519-2020/v1",
];
// A default context value is provided for mocking purpose accross the codebase.
const ACCESS_GRANT_CONTEXT_DEFAULT = [
    CONTEXT_VC_W3C,
    CONTEXT_ESS_DEFAULT,
    instanciateContextVcEssTemplate("vc.inrupt.com"),
];
[
    ...ACCESS_GRANT_CONTEXT_DEFAULT,
    ...extraContext,
];
// When issuing a VC using a given service,"https://schema.inrupt.com/credentials/v1.jsonld" be sure to set the context using the following.
const instanciateEssAccessGrantContext = (essVcDomain) => [
    CONTEXT_VC_W3C,
    CONTEXT_ESS_DEFAULT,
    instanciateContextVcEssTemplate(essVcDomain),
];
const CREDENTIAL_TYPE_ACCESS_REQUEST = "SolidAccessRequest";
const CREDENTIAL_TYPE_ACCESS_GRANT = "SolidAccessGrant";
const CREDENTIAL_TYPE_ACCESS_DENIAL = "SolidAccessDenial";
const CREDENTIAL_TYPE_BASE = "VerifiableCredential";
const PRESENTATION_TYPE_BASE = "VerifiablePresentation";
const ACCESS_CREDENTIAL_TYPE = new Set([
    CREDENTIAL_TYPE_ACCESS_REQUEST,
    CREDENTIAL_TYPE_ACCESS_GRANT,
    CREDENTIAL_TYPE_ACCESS_DENIAL,
    "vc:SolidAccessDenial",
]);
const ACCESS_GRANT_STATUS = Object.freeze(new Set([
    gc.ConsentStatusDenied.value,
    gc.ConsentStatusExplicitlyGiven.value,
    GC_CONSENT_STATUS_DENIED_ABBREV,
    "gc:ConsentStatusDenied",
    "Consent:StatusDenied",
    GC_CONSENT_STATUS_EXPLICITLY_GIVEN_ABBREV,
]));
const ACCESS_REQUEST_STATUS = Object.freeze(new Set([
    gc.ConsentStatusRequested.value,
    GC_CONSENT_STATUS_REQUESTED_ABBREV,
]));
const ACCESS_STATUS = Object.freeze(new Set([...ACCESS_GRANT_STATUS, ...ACCESS_REQUEST_STATUS]));

export { ACCESS_CREDENTIAL_TYPE, ACCESS_GRANT_CONTEXT_DEFAULT, ACCESS_GRANT_STATUS, ACCESS_REQUEST_STATUS, ACCESS_STATUS, CONTEXT_ESS_DEFAULT, CONTEXT_VC_W3C, CREDENTIAL_TYPE_ACCESS_DENIAL, CREDENTIAL_TYPE_ACCESS_GRANT, CREDENTIAL_TYPE_ACCESS_REQUEST, CREDENTIAL_TYPE_BASE, GC_CONSENT_STATUS_DENIED_ABBREV, GC_CONSENT_STATUS_EXPLICITLY_GIVEN_ABBREV, GC_CONSENT_STATUS_REQUESTED_ABBREV, PIM_STORAGE, PREFERRED_CONSENT_MANAGEMENT_UI, PRESENTATION_TYPE_BASE, instanciateEssAccessGrantContext };
