// @inrupt/solid-client-access-grants/dist/gConsent/index.mjs
export { getAccessApiEndpoint } from './discover/getAccessApiEndpoint.mjs';
export { getAccessManagementUi } from './discover/getAccessManagementUi.mjs';
export { redirectToAccessManagementUi } from './discover/redirectToAccessManagementUi.mjs';
export { cancelAccessRequest } from './request/cancelAccessRequest.mjs';
export { default as issueAccessRequest } from './request/issueAccessRequest.mjs';
export { getAccessGrantFromRedirectUrl } from './request/getAccessGrantFromRedirectUrl.mjs';
export { approveAccessRequest } from './manage/approveAccessRequest.mjs';
export { denyAccessRequest } from './manage/denyAccessRequest.mjs';
export { getAccessGrant } from './manage/getAccessGrant.mjs';
export { getAccessGrantAll } from './manage/getAccessGrantAll.mjs';
export { getAccessRequestFromRedirectUrl } from './manage/getAccessRequestFromRedirectUrl.mjs';
export { getAccessRequest } from './manage/getAccessRequest.mjs';
export { revokeAccessGrant } from './manage/revokeAccessGrant.mjs';
export { GRANT_VC_URL_PARAM_NAME, redirectToRequestor } from './manage/redirectToRequestor.mjs';

//
// Copyright Inrupt Inc.
//
// Permission is hereby granted, free of charge, to any person obtaining a copy
// of this software and associated documentation files (the "Software"), to deal in
// the Software without restriction, including without limitation the rights to use,
// copy, modify, merge, publish, distribute, sublicense, and/or sell copies of the
// Software, and to permit persons to whom the Software is furnished to do so,
// subject to the following conditions:
//
// The above copyright notice and this permission notice shall be included in
// all copies or substantial portions of the Software.
//
// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR IMPLIED,
// INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS FOR A
// PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
// HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN ACTION
// OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN CONNECTION WITH THE
// SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
//
