// @inrupt/solid-client-access-grants/dist/index.mjs
export { CredentialIsAccessGrantAny } from './type/AccessGrant.mjs';
import * as index from './gConsent/index.mjs';
export { index as gConsent };
export { isValidAccessGrant } from './common/verify/isValidAccessGrant.mjs';
import * as index$1 from './common/index.mjs';
export { index$1 as common };
export { fetchWithVc } from './fetch/index.mjs';
export { getFile, overwriteFile, saveFileInContainer } from './resource/file.mjs';
export { createContainerInContainer } from './resource/createContainerInContainer.mjs';
export { deleteFile } from './resource/deleteFile.mjs';
export { deleteSolidDataset } from './resource/deleteSolidDataset.mjs';
export { getSolidDataset } from './resource/getSolidDataset.mjs';
export { saveSolidDatasetAt } from './resource/saveSolidDatasetAt.mjs';
export { saveSolidDatasetInContainer } from './resource/saveSolidDatasetInContainer.mjs';
export { AccessGrantWrapper, getAccessModes, getConsent, getInbox, getInherit, getPurposes, getRequestor, getResourceOwner, getResources, getTypes } from './common/getters.mjs';
export { approveAccessRequest } from './gConsent/manage/approveAccessRequest.mjs';
export { cancelAccessRequest } from './gConsent/request/cancelAccessRequest.mjs';
export { denyAccessRequest } from './gConsent/manage/denyAccessRequest.mjs';
export { getAccessApiEndpoint } from './gConsent/discover/getAccessApiEndpoint.mjs';
export { getAccessGrant } from './gConsent/manage/getAccessGrant.mjs';
export { getAccessGrantAll } from './gConsent/manage/getAccessGrantAll.mjs';
export { getAccessGrantFromRedirectUrl } from './gConsent/request/getAccessGrantFromRedirectUrl.mjs';
export { getAccessRequest } from './gConsent/manage/getAccessRequest.mjs';
export { getAccessManagementUi } from './gConsent/discover/getAccessManagementUi.mjs';
export { getAccessRequestFromRedirectUrl } from './gConsent/manage/getAccessRequestFromRedirectUrl.mjs';
export { default as issueAccessRequest } from './gConsent/request/issueAccessRequest.mjs';
export { redirectToAccessManagementUi } from './gConsent/discover/redirectToAccessManagementUi.mjs';
export { GRANT_VC_URL_PARAM_NAME, redirectToRequestor } from './gConsent/manage/redirectToRequestor.mjs';
export { revokeAccessGrant } from './gConsent/manage/revokeAccessGrant.mjs';
export { getCredentialSubject, getExpirationDate, getId, getIssuanceDate, getIssuer } from '@inrupt/solid-client-vc';
